--- before_pass
+++ after_pass
@@ -3,14 +3,20 @@ extern Y {
     bit<32> get();
 }
 control c(out bit<32> x)(Y y) {
+    bit<32> tmp;
     apply {
-        x = y.get();
+        {
+            tmp = y.get();
+            x = tmp;
+        }
     }
 }
 control d(out bit<32> x) {
     @name("cinst") c(Y(32w16)) cinst_0;
     apply {
-        cinst_0.apply(x);
+        {
+            cinst_0.apply(x);
+        }
     }
 }
 control dproto(out bit<32> x);
