--- before_pass
+++ after_pass
@@ -2,17 +2,23 @@
 control E(out bit<1> b);
 control D(out bit<1> b) {
     apply {
-        b = 1w1;
+        {
+            b = 1w1;
+        }
     }
 }
 control F(out bit<1> b) {
     apply {
-        b = 1w0;
+        {
+            b = 1w0;
+        }
     }
 }
 control C(out bit<1> b)(E d) {
     apply {
-        d.apply(b);
+        {
+            d.apply(b);
+        }
     }
 }
 control Ingress(out bit<1> b) {
@@ -21,8 +27,12 @@ control Ingress(out bit<1> b) {
     @name("c0") C(d_0) c0_0;
     @name("c1") C(f_0) c1_0;
     apply {
-        c0_0.apply(b);
-        c1_0.apply(b);
+        {
+            c0_0.apply(b);
+        }
+        {
+            c1_0.apply(b);
+        }
     }
 }
 package top(E _e);
