--- before_pass
+++ after_pass
@@ -40,7 +40,7 @@ struct ing_to_egr {
 }
 parser ing_parse(packet_in buffer, out ing_in_headers parsed_hdr) {
     state start {
-        buffer.extract(parsed_hdr.ethernet);
+        buffer.extract<ethernet_t>(parsed_hdr.ethernet);
         transition accept;
     }
 }
@@ -53,12 +53,12 @@ control ingress(in ing_in_headers ihdr,
 }
 control ing_deparse(in ing_out_headers ohdr, packet_out b) {
     apply {
-        b.emit(ohdr.ethernet);
+        b.emit<ethernet_t>(ohdr.ethernet);
     }
 }
 parser egr_parse(packet_in buffer, out egr_in_headers parsed_hdr) {
     state start {
-        buffer.extract(parsed_hdr.ethernet);
+        buffer.extract<ethernet_t>(parsed_hdr.ethernet);
         transition accept;
     }
 }
@@ -70,11 +70,11 @@ control egress(in egr_in_headers ihdr, i
 }
 control egr_deparse(in egr_out_headers ohdr, packet_out b) {
     apply {
-        b.emit(ohdr.ethernet);
+        b.emit<ethernet_t>(ohdr.ethernet);
     }
 }
-Ingress(ing_parse(), ingress(), ing_deparse()) ig1;
+Ingress<ing_to_egr, ing_in_headers, ing_out_headers>(ing_parse(), ingress(), ing_deparse()) ig1;
 Ingress<ing_to_egr, ing_in_headers, ing_out_headers>(ing_parse(), ingress(), ing_deparse()) ig2;
-Egress(egr_parse(), egress(), egr_deparse()) eg1;
+Egress<ing_to_egr, egr_in_headers, egr_out_headers>(egr_parse(), egress(), egr_deparse()) eg1;
 Egress<ing_to_egr, egr_in_headers, egr_out_headers>(egr_parse(), egress(), egr_deparse()) eg2;
-Switch(ig1, eg1) main;
+Switch<ing_to_egr>(ig1, eg1) main;
