--- before_pass
+++ after_pass
@@ -68,7 +68,7 @@ parser Tcp_option_parser(packet_in b, ou
         transition start;
     }
     state parse_tcp_option_sack {
-        b.extract(vec.next.sack, (bit<32>)(8 * (b.lookahead<Tcp_option_sack_top>()).length - 16));
+        b.extract(vec.next.sack, (bit<32>)(8w8 * (b.lookahead<Tcp_option_sack_top>()).length - 8w16));
         transition start;
     }
 }
