--- before_pass
+++ after_pass
@@ -39,19 +39,19 @@ parser ParserImpl(packet_in packet, out
     state parse_ipv4 {
         packet.extract(hdr.ipv4);
         transition select(hdr.ipv4.version) {
-            4: check_ipv4_ihl;
+            4w4: check_ipv4_ihl;
             default: accept;
         }
     }
     state check_ipv4_ihl {
         transition select(hdr.ipv4.ihl) {
-            5: check_ipv4_frag_offset;
+            4w5: check_ipv4_frag_offset;
             default: accept;
         }
     }
     state check_ipv4_frag_offset {
         transition select(hdr.ipv4.fragOffset) {
-            0x1fff: check_ipv4_protocol;
+            13w0x1fff: check_ipv4_protocol;
             default: accept;
         }
     }
