--- before_pass
+++ after_pass
@@ -25,7 +25,6 @@ control EgressI(inout H hdr, inout M met
         }
         default_action = a_0();
     }
-    @name("do") IngressI() do_0;
     apply {
         bool hasReturned = false;
         switch (t_0.apply().action_run) {
@@ -35,7 +34,8 @@ control EgressI(inout H hdr, inout M met
                 }
             }
             default: {
-                do_0.apply(hdr, meta, std_meta);
+                {
+                }
             }
         }
     }
