--- before_pass
+++ after_pass
@@ -15,21 +15,21 @@ control IngressI(inout H hdr, inout M me
     }
 }
 control EgressI(inout H hdr, inout M meta, inout std_meta_t std_meta) {
-    @name("a") action a_0() {
+    @name("a") action a() {
     }
-    @name("t") table t {
+    @name("t") table t_0 {
         key = {
         }
         actions = {
-            a_0();
+            a();
         }
-        default_action = a_0();
+        default_action = a();
     }
     apply {
-        bool hasReturned_0 = false;
-        switch (t.apply().action_run) {
-            a_0: {
-                hasReturned_0 = true;
+        bool hasReturned = false;
+        switch (t_0.apply().action_run) {
+            a: {
+                hasReturned = true;
             }
             default: {
             }
