--- before_pass
+++ after_pass
@@ -22,12 +22,12 @@ parser MyEP(packet_in buffer, out EMPTY
 control MyIC(inout ethernet_t a, inout EMPTY b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("ap") ActionProfile(32w1024) ap_0;
-    @name("a1") action a1() {
+    @name("MyIC.ap") ActionProfile(32w1024) ap_0;
+    @name("MyIC.a1") action a1() {
     }
-    @name("a2") action a2() {
+    @name("MyIC.a2") action a2() {
     }
-    @name("tbl") table tbl_0 {
+    @name("MyIC.tbl") table tbl_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
