--- before_pass
+++ after_pass
@@ -4,16 +4,16 @@ control empty();
 package top(empty e);
 control Ing() {
     bool b;
-    bit<32> a = (bit<32>)32w2;
+    bit<32> a = 32w2;
     action cond() {
         b = true;
         if (b) {
-            a = (bit<32>)32w5;
+            a = 32w5;
         } else {
             if (b && a == 32w5) {
-                a = (bit<32>)32w10;
+                a = 32w10;
             } else {
-                a = (bit<32>)32w20;
+                a = 32w20;
             }
         }
     }
