--- before_pass
+++ after_pass
@@ -6,9 +6,7 @@ control Ing() {
     bool b_0;
     bit<32> a_0;
     @name("cond") action cond_0() {
-        {
-            b_0 = true;
-        }
+        b_0 = true;
         if (b_0) {
             a_0 = 32w5;
         } else if (b_0 && a_0 == 32w5) {
@@ -18,12 +16,8 @@ control Ing() {
         }
     }
     apply {
-        {
-            a_0 = 32w2;
-        }
-        {
-            cond_0();
-        }
+        a_0 = 32w2;
+        cond_0();
     }
 }
 top(Ing()) main;
