--- before_pass
+++ after_pass
@@ -3,20 +3,20 @@
 control empty();
 package top(empty e);
 control Ing() {
-    bool b;
-    bit<32> a = 32w2;
-    action cond() {
-        b = true;
-        if (b) {
-            a = 32w5;
-        } else if (b && a == 32w5) {
-            a = 32w10;
+    bool b_0;
+    bit<32> a_0 = 32w2;
+    @name("cond") action cond_0() {
+        b_0 = true;
+        if (b_0) {
+            a_0 = 32w5;
+        } else if (b_0 && a_0 == 32w5) {
+            a_0 = 32w10;
         } else {
-            a = 32w20;
+            a_0 = 32w20;
         }
     }
     apply {
-        cond();
+        cond_0();
     }
 }
 top(Ing()) main;
