--- before_pass
+++ after_pass
@@ -3,14 +3,14 @@
 control empty();
 package top(empty e);
 control Ing() {
-    bool b_0;
-    bit<32> a_0;
-    @name("cond") action cond_0() {
-        b_0 = true;
+    bool b;
+    bit<32> a;
+    @name("cond") action cond() {
+        b = true;
     }
     apply {
-        a_0 = 32w2;
-        cond_0();
+        a = 32w2;
+        cond();
     }
 }
 top(Ing()) main;
