--- before_pass
+++ after_pass
@@ -9,11 +9,11 @@ struct headers_t {
 struct metadata_t {
 }
 parser TestParser(packet_in b, out headers_t headers, inout metadata_t meta, inout standard_metadata_t standard_metadata) {
-    bit<32> test_f_0;
+    bit<32> test_f;
     state start {
         b.extract<test_header_t>(headers.test.next);
-        test_f_0 = headers.test.lastIndex << 1;
-        transition select(test_f_0) {
+        test_f = headers.test.lastIndex << 1;
+        transition select(test_f) {
             32w0: f;
             default: a;
         }
