--- before_pass
+++ after_pass
@@ -32,12 +32,12 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    action my_a(bit<32> v) {
+    @name("my_a") action my_a_0(bit<32> v) {
         h.h.f = v;
     }
     apply {
-        my_a(32w0);
-        my_a(32w1);
+        my_a_0(32w0);
+        my_a_0(32w1);
     }
 }
 V1Switch<Headers, Meta>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
