--- before_pass
+++ after_pass
@@ -138,25 +138,8 @@ parser FabricParser(packet_in packet, ou
         transition accept;
     }
 }
-control PacketIoIngress(inout parsed_headers_t hdr, inout fabric_metadata_t fabric_metadata, inout standard_metadata_t standard_metadata) {
-    apply {
-        if (hdr.packet_out.isValid()) {
-            standard_metadata.egress_spec = (PortId_t)(PortIdUInt_t)(bit<32>)hdr.packet_out.egress_port;
-            hdr.packet_out.setInvalid();
-            exit;
-        }
-    }
-}
-control PacketIoEgress(inout parsed_headers_t hdr, inout fabric_metadata_t fabric_metadata, inout standard_metadata_t standard_metadata) {
-    apply {
-        if (standard_metadata.egress_port == (PortId_t)9w192) {
-            hdr.packet_in.setValid();
-            hdr.packet_in.ingress_port = (PortIdInHeader_t)(bit<32>)(PortIdUInt_t)standard_metadata.ingress_port;
-        }
-    }
-}
-control Filtering(inout parsed_headers_t hdr, inout fabric_metadata_t fabric_metadata, inout standard_metadata_t standard_metadata) {
-    @name("t") table t_0 {
+control FabricIngress(inout parsed_headers_t hdr, inout fabric_metadata_t fabric_metadata, inout standard_metadata_t standard_metadata) {
+    @name("filtering.t") table filtering_t {
         key = {
             standard_metadata.ingress_port: exact @name("standard_metadata.ingress_port") ;
         }
@@ -167,47 +150,48 @@ control Filtering(inout parsed_headers_t
         }
         default_action = NoAction();
     }
-    apply {
-        t_0.apply();
-    }
-}
-control Forwarding(inout parsed_headers_t hdr, inout fabric_metadata_t fabric_metadata, inout standard_metadata_t standard_metadata) {
-    PortId_t mask_0;
-    @name("fwd") action fwd_0(PortId_t next_port) {
+    PortId_t forwarding_mask;
+    @name("forwarding.fwd") action forwarding_fwd(PortId_t next_port) {
         standard_metadata.egress_spec = next_port;
     }
-    @name("t") table t_1 {
+    @name("forwarding.t") table forwarding_t {
         key = {
             hdr.ipv4.dstAddr: exact @name("hdr.ipv4.dstAddr") ;
         }
         actions = {
             drop();
-            fwd_0();
+            forwarding_fwd();
             @defaultonly NoAction();
         }
         default_action = NoAction();
     }
     apply {
-        t_1.apply();
-        standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec + 9w1);
-        mask_0 = (PortId_t)9w0xf;
-        standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec & (PortIdUInt_t)mask_0);
-    }
-}
-control FabricIngress(inout parsed_headers_t hdr, inout fabric_metadata_t fabric_metadata, inout standard_metadata_t standard_metadata) {
-    @name("packet_io_ingress") PacketIoIngress() packet_io_ingress_0;
-    @name("filtering") Filtering() filtering_0;
-    @name("forwarding") Forwarding() forwarding_0;
-    apply {
-        packet_io_ingress_0.apply(hdr, fabric_metadata, standard_metadata);
-        filtering_0.apply(hdr, fabric_metadata, standard_metadata);
-        forwarding_0.apply(hdr, fabric_metadata, standard_metadata);
+        {
+            if (hdr.packet_out.isValid()) {
+                standard_metadata.egress_spec = (PortId_t)(PortIdUInt_t)(bit<32>)hdr.packet_out.egress_port;
+                hdr.packet_out.setInvalid();
+                exit;
+            }
+        }
+        {
+            filtering_t.apply();
+        }
+        {
+            forwarding_t.apply();
+            standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec + 9w1);
+            forwarding_mask = (PortId_t)9w0xf;
+            standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec & (PortIdUInt_t)forwarding_mask);
+        }
     }
 }
 control FabricEgress(inout parsed_headers_t hdr, inout fabric_metadata_t fabric_metadata, inout standard_metadata_t standard_metadata) {
-    @name("pkt_io_egress") PacketIoEgress() pkt_io_egress_0;
     apply {
-        pkt_io_egress_0.apply(hdr, fabric_metadata, standard_metadata);
+        {
+            if (standard_metadata.egress_port == (PortId_t)9w192) {
+                hdr.packet_in.setValid();
+                hdr.packet_in.ingress_port = (PortIdInHeader_t)(bit<32>)(PortIdUInt_t)standard_metadata.ingress_port;
+            }
+        }
     }
 }
 control FabricDeparser(packet_out packet, in parsed_headers_t hdr) {
