--- before_pass
+++ after_pass
@@ -174,7 +174,7 @@ control PacketIoEgress(inout parsed_head
 control Filtering(inout parsed_headers_t hdr, inout fabric_metadata_t fabric_metadata, inout standard_metadata_t standard_metadata) {
     table t {
         key = {
-            standard_metadata.ingress_port: exact;
+            standard_metadata.ingress_port: exact @name("standard_metadata.ingress_port") ;
         }
         actions = {
             drop();
@@ -193,7 +193,7 @@ control Forwarding(inout parsed_headers_
     }
     table t {
         key = {
-            hdr.ipv4.dstAddr: exact;
+            hdr.ipv4.dstAddr: exact @name("hdr.ipv4.dstAddr") ;
         }
         actions = {
             drop();
