--- before_pass
+++ after_pass
@@ -139,7 +139,7 @@ parser FabricParser(packet_in packet, ou
     }
 }
 control FabricIngress(inout parsed_headers_t hdr, inout fabric_metadata_t fabric_metadata, inout standard_metadata_t standard_metadata) {
-    @name("filtering.t") table filtering_t {
+    @name("filtering.t") table filtering_t_0 {
         key = {
             standard_metadata.ingress_port: exact @name("standard_metadata.ingress_port") ;
         }
@@ -150,17 +150,17 @@ control FabricIngress(inout parsed_heade
         }
         default_action = NoAction();
     }
-    PortId_t forwarding_mask;
-    @name("forwarding.fwd") action forwarding_fwd(PortId_t next_port) {
+    PortId_t forwarding_mask_0;
+    @name("forwarding.fwd") action forwarding_fwd_0(PortId_t next_port) {
         standard_metadata.egress_spec = next_port;
     }
-    @name("forwarding.t") table forwarding_t {
+    @name("forwarding.t") table forwarding_t_0 {
         key = {
             hdr.ipv4.dstAddr: exact @name("hdr.ipv4.dstAddr") ;
         }
         actions = {
             drop();
-            forwarding_fwd();
+            forwarding_fwd_0();
             @defaultonly NoAction();
         }
         default_action = NoAction();
@@ -171,11 +171,11 @@ control FabricIngress(inout parsed_heade
             hdr.packet_out.setInvalid();
             exit;
         }
-        filtering_t.apply();
-        forwarding_t.apply();
+        filtering_t_0.apply();
+        forwarding_t_0.apply();
         standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec + 9w1);
-        forwarding_mask = (PortId_t)9w0xf;
-        standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec & (PortIdUInt_t)forwarding_mask);
+        forwarding_mask_0 = (PortId_t)9w0xf;
+        standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec & (PortIdUInt_t)forwarding_mask_0);
     }
 }
 control FabricEgress(inout parsed_headers_t hdr, inout fabric_metadata_t fabric_metadata, inout standard_metadata_t standard_metadata) {
