--- before_pass
+++ after_pass
@@ -134,43 +134,43 @@ parser FabricParser(packet_in packet, ou
     }
 }
 control FabricIngress(inout parsed_headers_t hdr, inout fabric_metadata_t fabric_metadata, inout standard_metadata_t standard_metadata) {
-    @name(".drop") action drop_1() {
+    @name(".drop") action drop() {
         mark_to_drop();
     }
-    @name(".drop") action drop_2() {
+    @name(".drop") action drop_0() {
         mark_to_drop();
     }
-    @name(".nop") action nop_1() {
+    @name(".nop") action nop() {
     }
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name(".NoAction") action NoAction_2() {
+    @name(".NoAction") action NoAction_3() {
     }
-    @name("filtering.t") table filtering_t_0 {
+    @name("filtering.t") table filtering_t {
         key = {
             standard_metadata.ingress_port: exact @name("standard_metadata.ingress_port") ;
         }
         actions = {
-            drop_1();
-            nop_1();
-            @defaultonly NoAction_1();
+            drop();
+            nop();
+            @defaultonly NoAction_0();
         }
-        default_action = NoAction_1();
+        default_action = NoAction_0();
     }
-    PortId_t forwarding_mask_0;
-    @name("forwarding.fwd") action forwarding_fwd(PortId_t next_port) {
+    PortId_t forwarding_mask;
+    @name("forwarding.fwd") action forwarding_fwd_0(PortId_t next_port) {
         standard_metadata.egress_spec = next_port;
     }
-    @name("forwarding.t") table forwarding_t_0 {
+    @name("forwarding.t") table forwarding_t {
         key = {
             hdr.ipv4.dstAddr: exact @name("hdr.ipv4.dstAddr") ;
         }
         actions = {
-            drop_2();
-            forwarding_fwd();
-            @defaultonly NoAction_2();
+            drop_0();
+            forwarding_fwd_0();
+            @defaultonly NoAction_3();
         }
-        default_action = NoAction_2();
+        default_action = NoAction_3();
     }
     apply {
         if (hdr.packet_out.isValid()) {
@@ -178,11 +178,11 @@ control FabricIngress(inout parsed_heade
             hdr.packet_out.setInvalid();
             exit;
         }
-        filtering_t_0.apply();
-        forwarding_t_0.apply();
+        filtering_t.apply();
+        forwarding_t.apply();
         standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec + 9w1);
-        forwarding_mask_0 = (PortId_t)9w0xf;
-        standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec & (PortIdUInt_t)forwarding_mask_0);
+        forwarding_mask = (PortId_t)9w0xf;
+        standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec & (PortIdUInt_t)forwarding_mask);
     }
 }
 control FabricEgress(inout parsed_headers_t hdr, inout fabric_metadata_t fabric_metadata, inout standard_metadata_t standard_metadata) {
