--- before_pass
+++ after_pass
@@ -172,6 +172,7 @@ control Filtering(inout parsed_headers_t
     }
 }
 control Forwarding(inout parsed_headers_t hdr, inout fabric_metadata_t fabric_metadata, inout standard_metadata_t standard_metadata) {
+    PortId_t mask_0;
     @name("fwd") action fwd_0(PortId_t next_port) {
         standard_metadata.egress_spec = next_port;
     }
@@ -189,7 +190,7 @@ control Forwarding(inout parsed_headers_
     apply {
         t_1.apply();
         standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec + 9w1);
-        PortId_t mask_0 = (PortId_t)9w0xf;
+        mask_0 = (PortId_t)9w0xf;
         standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec & (PortIdUInt_t)mask_0);
     }
 }
