--- before_pass
+++ after_pass
@@ -204,8 +204,8 @@ control Forwarding(inout parsed_headers_
     }
     apply {
         t.apply();
-        PortIdUInt_t tmp2 = 1;
-        standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec + 1);
+        bit<9> tmp2 = (bit<9>)9w1;
+        standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec + 9w1);
         PortId_t mask = (PortId_t)(PortIdUInt_t)0xf;
         standard_metadata.egress_spec = (PortId_t)((PortIdUInt_t)standard_metadata.egress_spec & (PortIdUInt_t)mask);
     }
