--- before_pass
+++ after_pass
@@ -15,14 +15,14 @@ control IngressI(inout H hdr, inout M me
     }
     @name(".NoAction") action NoAction_3() {
     }
-    @name("drop") action drop() {
+    @name("IngressI.drop") action drop() {
         mark_to_drop(smeta);
     }
-    @name("drop") action drop_2() {
+    @name("IngressI.drop") action drop_2() {
         mark_to_drop(smeta);
     }
-    @name("as") action_selector(HashAlgorithm.identity, 32w1024, 32w10) as_0;
-    @name("indirect_ws") table indirect_ws_0 {
+    @name("IngressI.as") action_selector(HashAlgorithm.identity, 32w1024, 32w10) as_0;
+    @name("IngressI.indirect_ws") table indirect_ws_0 {
         key = {
             meta.hash1: selector @name("meta.hash1") ;
         }
@@ -33,7 +33,7 @@ control IngressI(inout H hdr, inout M me
         const default_action = NoAction_0();
         @name("ap_ws") implementation = as_0;
     }
-    @name("indirect_ws_1") table indirect_ws_2 {
+    @name("IngressI.indirect_ws_1") table indirect_ws_2 {
         key = {
             meta.hash1: selector @name("meta.hash1") ;
         }
