--- before_pass
+++ after_pass
@@ -19,7 +19,7 @@ control IngressI(inout H hdr, inout M me
     action_selector(HashAlgorithm.identity, 32w1024, 32w10) as;
     table indirect_ws {
         key = {
-            meta.hash1: selector;
+            meta.hash1: selector @name("meta.hash1") ;
         }
         actions = {
             drop();
@@ -30,7 +30,7 @@ control IngressI(inout H hdr, inout M me
     }
     table indirect_ws_1 {
         key = {
-            meta.hash1: selector;
+            meta.hash1: selector @name("meta.hash1") ;
         }
         actions = {
             drop();
