--- before_pass
+++ after_pass
@@ -11,21 +11,21 @@ header_union U {
 control ct(out bit<32> b);
 package top(ct _ct);
 control c(out bit<32> x) {
-    U u_0;
-    U[2] u2_0;
-    bool b_0;
+    U u;
+    U[2] u2;
+    bool b_1;
     apply {
-        b_0 = u_0.isValid();
-        u_0.h1.isValid();
-        x = u_0.h1.f + u_0.h2.g;
-        u_0.h1.setValid();
-        u_0.h1.f = 32w0;
-        x = x + u_0.h1.f;
-        u_0.h2.g = 32w0;
-        x = x + u_0.h2.g;
-        u2_0[0].h1.setValid();
-        u2_0[0].h1.f = 32w2;
-        x = x + u2_0[1].h2.g + u2_0[0].h1.f;
+        b_1 = u.isValid();
+        u.h1.isValid();
+        x = u.h1.f + u.h2.g;
+        u.h1.setValid();
+        u.h1.f = 32w0;
+        x = x + u.h1.f;
+        u.h2.g = 32w0;
+        x = x + u.h2.g;
+        u2[0].h1.setValid();
+        u2[0].h1.f = 32w2;
+        x = x + u2[1].h2.g + u2[0].h1.f;
     }
 }
 top(c()) main;
