--- before_pass
+++ after_pass
@@ -18,12 +18,12 @@ control c(out bit<32> x) {
         b = b || u.h1.isValid();
         x = u.h1.f + u.h2.g;
         u.h1.setValid();
-        u.h1.f = (bit<32>)32w0;
+        u.h1.f = 32w0;
         x = x + u.h1.f;
-        u.h2.g = (bit<32>)32w0;
+        u.h2.g = 32w0;
         x = x + u.h2.g;
         u2[0].h1.setValid();
-        u2[0].h1.f = (bit<32>)32w2;
+        u2[0].h1.f = 32w2;
         x = x + u2[1].h2.g + u2[0].h1.f;
     }
 }
