--- before_pass
+++ after_pass
@@ -133,11 +133,11 @@ control cDoOneOp(inout headers hdr, in b
     }
 }
 control cIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
-    cDoOneOp() do_one_op;
+    @name("do_one_op") cDoOneOp() do_one_op_0;
     apply {
-        do_one_op.apply(hdr, hdr.h1.op1);
-        do_one_op.apply(hdr, hdr.h1.op2);
-        do_one_op.apply(hdr, hdr.h1.op3);
+        do_one_op_0.apply(hdr, hdr.h1.op1);
+        do_one_op_0.apply(hdr, hdr.h1.op2);
+        do_one_op_0.apply(hdr, hdr.h1.op3);
         hdr.h1.h2_valid_bits = 8w0;
         if (hdr.h2[0].isValid()) {
             hdr.h1.h2_valid_bits[0:0] = 1w1;
