--- before_pass
+++ after_pass
@@ -4,7 +4,6 @@ error {
 }
 #include <core.p4>
 #include <v1model.p4>
-typedef bit<48> EthernetAddress;
 typedef bit<32> IPv4Address;
 header ethernet_t {
     bit<48> dstAddr;
@@ -50,7 +49,6 @@ struct mystruct1_t {
 struct metadata {
     mystruct1_t mystruct1;
 }
-typedef tuple<bit<4>, bit<4>, bit<8>, varbit<56>> myTuple1;
 parser parserI(packet_in pkt, out headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
     state start {
         pkt.extract<ethernet_t>(hdr.ethernet);
