--- before_pass
+++ after_pass
@@ -39,9 +39,11 @@ parser prs(packet_in p, out Headers_t he
     }
 }
 control pipe(inout Headers_t headers, out bool pass) {
+    @name(".NoAction") action NoAction_1() {
+    }
     IPv4Address address;
     bool pass_1;
-    @name("c1.Reject") action c1_Reject_0() {
+    @name("c1.Reject") action c1_Reject() {
         pass_1 = false;
     }
     @name("c1.Check_ip") table c1_Check_ip_0 {
@@ -49,11 +51,11 @@ control pipe(inout Headers_t headers, ou
             address: exact @name("address") ;
         }
         actions = {
-            c1_Reject_0();
-            NoAction();
+            c1_Reject();
+            NoAction_1();
         }
         implementation = hash_table(32w1024);
-        const default_action = NoAction();
+        const default_action = NoAction_1();
     }
     apply {
         bool hasReturned_0 = false;
