--- before_pass
+++ after_pass
@@ -39,34 +39,34 @@ parser prs(packet_in p, out Headers_t he
     }
 }
 control Check(in IPv4Address address, inout bool pass) {
-    action Reject() {
+    @name("Reject") action Reject_0() {
         pass = false;
     }
-    table Check_ip {
+    @name("Check_ip") table Check_ip_0 {
         key = {
             address: exact @name("address") ;
         }
         actions = {
-            Reject();
+            Reject_0();
             NoAction();
         }
         implementation = hash_table(32w1024);
         const default_action = NoAction();
     }
     apply {
-        Check_ip.apply();
+        Check_ip_0.apply();
     }
 }
 control pipe(inout Headers_t headers, out bool pass) {
-    Check() c1;
+    @name("c1") Check() c1_0;
     apply {
         pass = true;
         if (!headers.ipv4.isValid()) {
             pass = false;
             return;
         }
-        c1.apply(headers.ipv4.srcAddr, pass);
-        c1.apply(headers.ipv4.dstAddr, pass);
+        c1_0.apply(headers.ipv4.srcAddr, pass);
+        c1_0.apply(headers.ipv4.dstAddr, pass);
     }
 }
 ebpfFilter<Headers_t>(prs(), pipe()) main;
