--- before_pass
+++ after_pass
@@ -47,11 +47,11 @@ control Check(in IPv4Address address, in
             address: exact;
         }
         actions = {
-            Reject;
-            NoAction;
+            Reject();
+            NoAction();
         }
         implementation = hash_table(1024);
-        const default_action = NoAction;
+        const default_action = NoAction();
     }
     apply {
         Check_ip.apply();
