--- before_pass
+++ after_pass
@@ -15,7 +15,9 @@ struct Meta_t {
 }
 parser p(packet_in b, out Header_t h, inout Meta_t m, inout standard_metadata_t sm) {
     state start {
-        b.extract<hdr>(h.h);
+        {
+            b.extract<hdr>(h.h);
+        }
         transition accept;
     }
 }
@@ -33,12 +35,16 @@ control egress(inout Header_t h, inout M
 }
 control deparser(packet_out b, in Header_t h) {
     apply {
-        b.emit<hdr>(h.h);
+        {
+            b.emit<hdr>(h.h);
+        }
     }
 }
 control ingress(inout Header_t h, inout Meta_t m, inout standard_metadata_t standard_meta) {
     @name("set_error_idx") action set_error_idx_0(bit<8> idx) {
-        h.h.e = idx;
+        {
+            h.h.e = idx;
+        }
     }
     @name("t_exact") table t_exact_0 {
         key = {
@@ -53,7 +59,9 @@ control ingress(inout Header_t h, inout
         }
     }
     apply {
-        t_exact_0.apply();
+        {
+            t_exact_0.apply();
+        }
     }
 }
 V1Switch<Header_t, Meta_t>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
