--- before_pass
+++ after_pass
@@ -48,9 +48,9 @@ control ingress(inout Header_t h, inout
         actions = {
             set_error_idx();
         }
-        default_action = set_error_idx(0);
+        default_action = set_error_idx(8w0);
         const entries = {
-                        error.NoError : set_error_idx(1);
+                        error.NoError : set_error_idx(8w1);
         }
     }
     apply {
