--- before_pass
+++ after_pass
@@ -22,7 +22,7 @@ struct metadata {
 }
 parser MyParser(packet_in pkt, out headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
     state start {
-        pkt.extract(hdr.h1);
+        pkt.extract<test_header>(hdr.h1);
         verify(false, error.IPv4BadPacket);
         verify(false, error.IPv4HeaderTooShort);
         transition accept;
@@ -49,7 +49,7 @@ control MyComputeChecksum(inout headers
 }
 control MyDeparser(packet_out packet, in headers hdr) {
     apply {
-        packet.emit(hdr.h1);
+        packet.emit<test_header>(hdr.h1);
     }
 }
-V1Switch(MyParser(), MyVerifyChecksum(), MyIngress(), MyEgress(), MyComputeChecksum(), MyDeparser()) main;
+V1Switch<headers, metadata>(MyParser(), MyVerifyChecksum(), MyIngress(), MyEgress(), MyComputeChecksum(), MyDeparser()) main;
