--- before_pass
+++ after_pass
@@ -19,17 +19,13 @@ control c(inout bit<32> x) {
                 } else {
                     tmp = a_0;
                 }
-                {
-                    hasReturned = true;
-                    retval = tmp;
-                }
+                hasReturned = true;
+                retval = tmp;
                 tmp_0 = retval;
             }
             tmp_1 = a_1 + tmp_0;
-            {
-                hasReturned_0 = true;
-                retval_0 = tmp_1;
-            }
+            hasReturned_0 = true;
+            retval_0 = tmp_1;
             tmp_2 = retval_0;
         }
         x = tmp_2;
