--- before_pass
+++ after_pass
@@ -3,21 +3,15 @@ struct S {
 }
 control c(inout S data) {
     apply {
-        {
-            data.f = 32w0;
-        }
+        data.f = 32w0;
     }
 }
 control caller() {
     S data_0;
     @name("cinst") c() cinst_0;
     apply {
-        {
-            data_0.f = 32w0;
-        }
-        {
-            cinst_0.apply(data_0);
-        }
+        data_0.f = 32w0;
+        cinst_0.apply(data_0);
     }
 }
 control none();
