--- before_pass
+++ after_pass
@@ -31,15 +31,15 @@ control egress(inout headers hdr, inout
 }
 @name("namedmeter") direct_meter<bit<32>>(MeterType.packets) my_meter;
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    @name("_drop") action _drop_0() {
+    @name("_drop") action _drop() {
         mark_to_drop(standard_metadata);
     }
-    @name("_nop") action _nop_1() {
+    @name("_nop") action _nop() {
     }
-    @name("m_filter") table m_filter_0 {
+    @name("m_filter") table m_filter {
         actions = {
-            _drop_0();
-            _nop_1();
+            _drop();
+            _nop();
             NoAction();
         }
         key = {
@@ -48,17 +48,17 @@ control ingress(inout headers hdr, inout
         size = 16;
         default_action = NoAction();
     }
-    @name("m_action") action m_action(bit<9> meter_idx) {
+    @name("m_action") action m_action_0(bit<9> meter_idx) {
         standard_metadata.egress_spec = meter_idx;
         my_meter.read(meta.meta.meter_tag);
     }
-    @name("_nop") action _nop_2() {
+    @name("_nop") action _nop_0() {
         my_meter.read(meta.meta.meter_tag);
     }
-    @name("m_table") table m_table_0 {
+    @name("m_table") table m_table {
         actions = {
-            m_action();
-            _nop_2();
+            m_action_0();
+            _nop_0();
             NoAction();
         }
         key = {
@@ -69,8 +69,8 @@ control ingress(inout headers hdr, inout
         meters = my_meter;
     }
     apply {
-        m_table_0.apply();
-        m_filter_0.apply();
+        m_table.apply();
+        m_filter.apply();
     }
 }
 control DeparserImpl(packet_out packet, in headers hdr) {
