--- before_pass
+++ after_pass
@@ -31,50 +31,50 @@ control egress(inout headers hdr, inout
 }
 @name("namedmeter") direct_meter<bit<32>>(MeterType.packets) my_meter;
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name(".NoAction") action NoAction_2() {
+    @name(".NoAction") action NoAction_3() {
     }
-    @name("_drop") action _drop_0() {
+    @name("_drop") action _drop() {
         mark_to_drop(standard_metadata);
     }
-    @name("_nop") action _nop_1() {
+    @name("_nop") action _nop() {
     }
-    @name("m_filter") table m_filter {
+    @name("m_filter") table m_filter_0 {
         actions = {
-            _drop_0();
-            _nop_1();
-            NoAction_1();
+            _drop();
+            _nop();
+            NoAction_0();
         }
         key = {
             meta.meta.meter_tag: exact @name("meta.meta.meter_tag") ;
         }
         size = 16;
-        default_action = NoAction_1();
+        default_action = NoAction_0();
     }
-    @name("m_action") action m_action(bit<9> meter_idx) {
+    @name("m_action") action m_action_0(bit<9> meter_idx) {
         standard_metadata.egress_spec = meter_idx;
         my_meter.read(meta.meta.meter_tag);
     }
-    @name("_nop") action _nop_2() {
+    @name("_nop") action _nop_0() {
         my_meter.read(meta.meta.meter_tag);
     }
-    @name("m_table") table m_table {
+    @name("m_table") table m_table_0 {
         actions = {
-            m_action();
-            _nop_2();
-            NoAction_2();
+            m_action_0();
+            _nop_0();
+            NoAction_3();
         }
         key = {
             hdr.ethernet.srcAddr: exact @name("hdr.ethernet.srcAddr") ;
         }
         size = 16384;
-        default_action = NoAction_2();
+        default_action = NoAction_3();
         meters = my_meter;
     }
     apply {
-        m_table.apply();
-        m_filter.apply();
+        m_table_0.apply();
+        m_filter_0.apply();
     }
 }
 control DeparserImpl(packet_out packet, in headers hdr) {
