--- before_pass
+++ after_pass
@@ -1,10 +1,10 @@
 #include <core.p4>
 parser p(out bit<32> b) {
-    bit<32> a = 1;
+    bit<32> a = (bit<32>)32w1;
     state start {
-        b = (a == 0 ? 32w2 : 3);
-        b = b + 1;
-        b = (a > 0 ? (a > 1 ? b + 1 : b + 2) : b + 3);
+        b = (a == 32w0 ? 32w2 : 32w3);
+        b = b + 32w1;
+        b = (a > 32w0 ? (a > 32w1 ? b + 32w1 : b + 32w2) : b + 32w3);
         transition accept;
     }
 }
