--- before_pass
+++ after_pass
@@ -4,21 +4,15 @@ header Header {
 }
 parser p0(packet_in p, out Header h) {
     state start {
-        {
-            p.extract<Header>(h);
-        }
+        p.extract<Header>(h);
         transition accept;
     }
 }
 parser p1(packet_in p, out Header[2] h) {
     @name("p0inst") p0() p0inst_0;
     state start {
-        {
-            p0inst_0.apply(p, h[0]);
-        }
-        {
-            p0inst_0.apply(p, h[1]);
-        }
+        p0inst_0.apply(p, h[0]);
+        p0inst_0.apply(p, h[1]);
         transition accept;
     }
 }
