--- before_pass
+++ after_pass
@@ -37,20 +37,20 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Header_t h, inout Meta_t m, inout standard_metadata_t standard_meta) {
-    @name("set_error_idx") action set_error_idx_0(bit<8> idx) {
+    @name("set_error_idx") action set_error_idx(bit<8> idx) {
         h.h.e = idx;
     }
-    @name("t_exact") table t_exact {
+    @name("t_exact") table t_exact_0 {
         key = {
             m.my_err: exact @name("m.my_err") ;
         }
         actions = {
-            set_error_idx_0();
+            set_error_idx();
         }
-        default_action = set_error_idx_0(8w0);
+        default_action = set_error_idx(8w0);
     }
     apply {
-        t_exact.apply();
+        t_exact_0.apply();
     }
 }
 V1Switch<Header_t, Meta_t>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
