--- before_pass
+++ after_pass
@@ -39,10 +39,10 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Header_t h, inout Meta_t m, inout standard_metadata_t standard_meta) {
-    @name("a") action a() {
+    @name("a") action a_0() {
         standard_meta.egress_spec = 9w0;
     }
-    @name("a_with_control_params") action a_with_control_params(bit<9> x) {
+    @name("a_with_control_params") action a_with_control_params_0(bit<9> x) {
         standard_meta.egress_spec = x;
     }
     @name("t_exact") table t_exact {
@@ -50,13 +50,13 @@ control ingress(inout Header_t h, inout
             h.h.row.e: exact @name("h.h.row.e") ;
         }
         actions = {
-            a();
-            a_with_control_params();
+            a_0();
+            a_with_control_params_0();
         }
-        default_action = a();
+        default_action = a_0();
         const entries = {
-                        8w0x1 : a_with_control_params(9w1);
-                        8w0x2 : a_with_control_params(9w2);
+                        8w0x1 : a_with_control_params_0(9w1);
+                        8w0x2 : a_with_control_params_0(9w2);
         }
     }
     apply {
