--- before_pass
+++ after_pass
@@ -36,7 +36,7 @@ parser parserI(packet_in pkt, out header
     state start {
         pkt.extract(hdr.ethernet);
         transition select(hdr.ethernet.etherType) {
-            0x800: parse_ipv4;
+            16w0x800: parse_ipv4;
             default: accept;
         }
     }
@@ -48,7 +48,7 @@ parser parserI(packet_in pkt, out header
 control cIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
     action hash_drop_decision() {
         hash(meta.mystruct1.hash1, HashAlgorithm.crc16, 16w0, { hdr.ipv4.srcAddr, hdr.ipv4.dstAddr, hdr.ipv4.protocol }, 32w0xffff);
-        meta.mystruct1.hash_drop = meta.mystruct1.hash1 < 0x8000;
+        meta.mystruct1.hash_drop = meta.mystruct1.hash1 < 16w0x8000;
     }
     table guh {
         key = {
