--- before_pass
+++ after_pass
@@ -52,7 +52,7 @@ control cIngress(inout headers hdr, inou
     }
     table guh {
         key = {
-            hdr.ipv4.dstAddr: exact;
+            hdr.ipv4.dstAddr: exact @name("hdr.ipv4.dstAddr") ;
         }
         actions = {
             hash_drop_decision();
@@ -61,8 +61,8 @@ control cIngress(inout headers hdr, inou
     }
     table debug_table {
         key = {
-            meta.mystruct1.hash1    : exact;
-            meta.mystruct1.hash_drop: exact;
+            meta.mystruct1.hash1    : exact @name("meta.mystruct1.hash1") ;
+            meta.mystruct1.hash_drop: exact @name("meta.mystruct1.hash_drop") ;
         }
         actions = {
             NoAction();
