--- before_pass
+++ after_pass
@@ -47,7 +47,7 @@ parser parserI(packet_in pkt, out header
 }
 control cIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
     action hash_drop_decision() {
-        hash(meta.mystruct1.hash1, HashAlgorithm.crc16, (bit<16>)0, { hdr.ipv4.srcAddr, hdr.ipv4.dstAddr, hdr.ipv4.protocol }, (bit<32>)0xffff);
+        hash(meta.mystruct1.hash1, HashAlgorithm.crc16, 16w0, { hdr.ipv4.srcAddr, hdr.ipv4.dstAddr, hdr.ipv4.protocol }, 32w0xffff);
         meta.mystruct1.hash_drop = meta.mystruct1.hash1 < 0x8000;
     }
     table guh {
