--- before_pass
+++ after_pass
@@ -55,9 +55,9 @@ control cIngress(inout headers hdr, inou
             hdr.ipv4.dstAddr: exact;
         }
         actions = {
-            hash_drop_decision;
+            hash_drop_decision();
         }
-        default_action = hash_drop_decision;
+        default_action = hash_drop_decision();
     }
     table debug_table {
         key = {
@@ -65,8 +65,9 @@ control cIngress(inout headers hdr, inou
             meta.mystruct1.hash_drop: exact;
         }
         actions = {
-            NoAction;
+            NoAction();
         }
+        default_action = NoAction();
     }
     apply {
         if (hdr.ipv4.isValid()) {
