--- before_pass
+++ after_pass
@@ -60,7 +60,7 @@ control ingress(inout packet_t hdrs, ino
     }
     table test1 {
         key = {
-            hdrs.data.f1: ternary;
+            hdrs.data.f1: ternary @name("hdrs.data.f1") ;
         }
         actions = {
             setb1();
@@ -70,7 +70,7 @@ control ingress(inout packet_t hdrs, ino
     }
     table ex1 {
         key = {
-            hdrs.extra[0].h: ternary;
+            hdrs.extra[0].h: ternary @name("hdrs.extra[0].h") ;
         }
         actions = {
             setbyte(hdrs.extra[0].b1);
@@ -83,7 +83,7 @@ control ingress(inout packet_t hdrs, ino
     }
     table tbl1 {
         key = {
-            hdrs.data.f2: ternary;
+            hdrs.data.f2: ternary @name("hdrs.data.f2") ;
         }
         actions = {
             setbyte(hdrs.data.b2);
@@ -93,7 +93,7 @@ control ingress(inout packet_t hdrs, ino
     }
     table tbl2 {
         key = {
-            hdrs.data.f2: ternary;
+            hdrs.data.f2: ternary @name("hdrs.data.f2") ;
         }
         actions = {
             setbyte(hdrs.extra[1].b1);
@@ -103,7 +103,7 @@ control ingress(inout packet_t hdrs, ino
     }
     table tbl3 {
         key = {
-            hdrs.data.f2: ternary;
+            hdrs.data.f2: ternary @name("hdrs.data.f2") ;
         }
         actions = {
             setbyte(hdrs.extra[2].b2);
