--- before_pass
+++ after_pass
@@ -54,17 +54,17 @@ control ingress(inout packet_t hdrs, ino
     }
     @name("noop") action noop_8() {
     }
-    @name("setbyte") action setbyte(out bit<8> reg_0, bit<8> val) {
-        reg_0 = val;
+    @name("setbyte") action setbyte(out bit<8> reg, bit<8> val) {
+        reg = val;
     }
-    @name("setbyte") action setbyte_4(out bit<8> reg_0, bit<8> val) {
-        reg_0 = val;
+    @name("setbyte") action setbyte_4(out bit<8> reg_1, bit<8> val) {
+        reg_1 = val;
     }
-    @name("setbyte") action setbyte_5(out bit<8> reg_0, bit<8> val) {
-        reg_0 = val;
+    @name("setbyte") action setbyte_5(out bit<8> reg_2, bit<8> val) {
+        reg_2 = val;
     }
-    @name("setbyte") action setbyte_6(out bit<8> reg_0, bit<8> val) {
-        reg_0 = val;
+    @name("setbyte") action setbyte_6(out bit<8> reg_3, bit<8> val) {
+        reg_3 = val;
     }
     @name("act1") action act1(bit<8> val) {
         hdrs.extra[0].b1 = val;
