--- before_pass
+++ after_pass
@@ -18,15 +18,15 @@ parser p(packet_in b, out Headers h, ino
 }
 register<bit<8>>(32w256) r;
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
+    bit<8> x_0;
     apply {
-        bit<8> x_0;
         r.read(x_0, (bit<32>)h.myhdr.reg_idx_to_update);
         r.write((bit<32>)h.myhdr.reg_idx_to_update, 8w0x2a);
     }
 }
 control egress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
+    bit<8> tmp_0;
     apply {
-        bit<8> tmp_0;
         r.read(tmp_0, (bit<32>)h.myhdr.reg_idx_to_update);
         tmp_0 = tmp_0 + h.myhdr.value_to_add;
         r.write((bit<32>)h.myhdr.reg_idx_to_update, tmp_0);
