--- before_pass
+++ after_pass
@@ -10,7 +10,7 @@ control compute(inout hdr h) {
     }
     table t {
         key = {
-            h.a + 1: exact @name("e") ;
+            h.a + 32w1: exact @name("e") ;
         }
         actions = {
             a();
@@ -54,7 +54,7 @@ control ingress(inout Headers h, inout M
     compute() c;
     apply {
         c.apply(h.h);
-        sm.egress_spec = 0;
+        sm.egress_spec = (bit<9>)9w0;
     }
 }
 V1Switch(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
