--- before_pass
+++ after_pass
@@ -4,24 +4,6 @@ header hdr {
     bit<32> a;
     bit<32> b;
 }
-control compute(inout hdr h) {
-    @name("a") action a_0() {
-        h.b = h.a;
-    }
-    @name("t") table t_0 {
-        key = {
-            h.a + 32w1: exact @name("e") ;
-        }
-        actions = {
-            a_0();
-            NoAction();
-        }
-        default_action = NoAction();
-    }
-    apply {
-        t_0.apply();
-    }
-}
 struct Headers {
     hdr h;
 }
@@ -51,9 +33,23 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name("c") compute() c_0;
+    @name("c.a") action c_a() {
+        h.h.b = h.h.a;
+    }
+    @name("c.t") table c_t {
+        key = {
+            h.h.a + 32w1: exact @name("e") ;
+        }
+        actions = {
+            c_a();
+            NoAction();
+        }
+        default_action = NoAction();
+    }
     apply {
-        c_0.apply(h.h);
+        {
+            c_t.apply();
+        }
         sm.egress_spec = 9w0;
     }
 }
