--- before_pass
+++ after_pass
@@ -35,10 +35,10 @@ control deparser(packet_out b, in Header
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("c.a") action c_a_0() {
+    @name("ingress.c.a") action c_a_0() {
         h.h.b = h.h.a;
     }
-    @name("c.t") table c_t {
+    @name("ingress.c.t") table c_t {
         key = {
             h.h.a + 32w1: exact @name("e") ;
         }
