--- before_pass
+++ after_pass
@@ -42,11 +42,11 @@ parser prs(packet_in p, out Headers_t he
 control pipe(inout Headers_t headers, out bool pass) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("Reject") action Reject(IPv4Address add) {
+    @name("pipe.Reject") action Reject(IPv4Address add) {
         pass = false;
         headers.ipv4[0].srcAddr = add;
     }
-    @name("Check_src_ip") table Check_src_ip_0 {
+    @name("pipe.Check_src_ip") table Check_src_ip_0 {
         key = {
             headers.ipv4[0].srcAddr: exact @name("headers.ipv4[0].srcAddr") ;
         }
