--- before_pass
+++ after_pass
@@ -20,21 +20,21 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout EMPTY b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name("counter0") DirectCounter<bit<12>>(PSA_CounterType_t.PACKETS) counter0;
-    @name("tbl") table tbl {
+    @name("counter0") DirectCounter<bit<12>>(PSA_CounterType_t.PACKETS) counter0_0;
+    @name("tbl") table tbl_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
-            NoAction_1();
+            NoAction_0();
         }
-        psa_direct_counter = counter0;
-        default_action = NoAction_1();
+        psa_direct_counter = counter0_0;
+        default_action = NoAction_0();
     }
     apply {
-        tbl.apply();
+        tbl_0.apply();
     }
 }
 control MyEC(inout EMPTY a, inout EMPTY b, in psa_egress_input_metadata_t c, inout psa_egress_output_metadata_t d) {
