--- before_pass
+++ after_pass
@@ -33,14 +33,14 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name("c.add") action c_add_0(bit<32> data) {
+    @name("c.add") action c_add(bit<32> data) {
         h.h.b = h.h.a + data;
     }
     @name("c.t") table c_t_0 {
         actions = {
-            c_add_0();
+            c_add();
         }
-        const default_action = c_add_0(32w10);
+        const default_action = c_add(32w10);
     }
     apply {
         c_t_0.apply();
