--- before_pass
+++ after_pass
@@ -18,10 +18,10 @@ parser ParserImpl(packet_in packet, out
 control IngressImpl(inout parsed_headers_t hdr, inout metadata_t meta, inout standard_metadata_t standard_metadata) {
     apply {
         if (meta.foo == meta.bar) {
-            meta.foo._v = meta.foo._v + 1;
+            meta.foo._v = meta.foo._v + 9w1;
         }
         if (meta.foo == PSA_CPU_PORT) {
-            meta.foo._v = meta.foo._v + 1;
+            meta.foo._v = meta.foo._v + 9w1;
         }
     }
 }
