--- before_pass
+++ after_pass
@@ -49,10 +49,10 @@ control pipe(inout Headers_t headers, ou
     }
     table t {
         key = {
-            headers.ipv4.srcAddr    : exact;
-            headers.ipv4.dstAddr    : exact;
-            headers.ethernet.dstAddr: exact;
-            headers.ethernet.srcAddr: exact;
+            headers.ipv4.srcAddr    : exact @name("headers.ipv4.srcAddr") ;
+            headers.ipv4.dstAddr    : exact @name("headers.ipv4.dstAddr") ;
+            headers.ethernet.dstAddr: exact @name("headers.ethernet.dstAddr") ;
+            headers.ethernet.srcAddr: exact @name("headers.ethernet.srcAddr") ;
         }
         actions = {
             invalidate();
