--- before_pass
+++ after_pass
@@ -11,12 +11,14 @@ parser ParserImpl(packet_in packet, out
     }
 }
 control IngressImpl(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
+    @name(".NoAction") action NoAction_1() {
+    }
     bit<1> registerData;
     @name("testRegister") register<bit<1>>(32w1) testRegister;
-    @name("drop") action drop() {
+    @name("drop") action drop_0() {
         mark_to_drop(standard_metadata);
     }
-    @name("forward") action forward() {
+    @name("forward") action forward_0() {
         standard_metadata.egress_spec = 9w1;
     }
     @name("debug_table") table debug_table {
@@ -24,11 +26,11 @@ control IngressImpl(inout headers hdr, i
             meta.test: exact @name("meta.test") ;
         }
         actions = {
-            drop();
-            forward();
-            @defaultonly NoAction();
+            drop_0();
+            forward_0();
+            @defaultonly NoAction_1();
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     apply {
         testRegister.read(registerData, 32w0);
