--- before_pass
+++ after_pass
@@ -11,12 +11,12 @@ parser ParserImpl(packet_in packet, out
     }
 }
 control IngressImpl(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    register<bit<1>>(1) testRegister;
+    register<bit<1>>(32w1) testRegister;
     action drop() {
         mark_to_drop(standard_metadata);
     }
     action forward() {
-        standard_metadata.egress_spec = 1;
+        standard_metadata.egress_spec = (bit<9>)9w1;
     }
     table debug_table {
         key = {
@@ -31,7 +31,7 @@ control IngressImpl(inout headers hdr, i
     }
     apply {
         bit<1> registerData;
-        testRegister.read(registerData, 0);
+        testRegister.read(registerData, 32w0);
         meta.test = (bool)registerData;
         debug_table.apply();
     }
