--- before_pass
+++ after_pass
@@ -11,29 +11,29 @@ parser ParserImpl(packet_in packet, out
     }
 }
 control IngressImpl(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    register<bit<1>>(32w1) testRegister;
-    action drop() {
+    @name("testRegister") register<bit<1>>(32w1) testRegister_0;
+    @name("drop") action drop_0() {
         mark_to_drop(standard_metadata);
     }
-    action forward() {
+    @name("forward") action forward_0() {
         standard_metadata.egress_spec = 9w1;
     }
-    table debug_table {
+    @name("debug_table") table debug_table_0 {
         key = {
             meta.test: exact @name("meta.test") ;
         }
         actions = {
-            drop();
-            forward();
+            drop_0();
+            forward_0();
             @defaultonly NoAction();
         }
         default_action = NoAction();
     }
     apply {
-        bit<1> registerData;
-        testRegister.read(registerData, 32w0);
-        meta.test = (bool)registerData;
-        debug_table.apply();
+        bit<1> registerData_0;
+        testRegister_0.read(registerData_0, 32w0);
+        meta.test = (bool)registerData_0;
+        debug_table_0.apply();
     }
 }
 control VerifyChecksumImpl(inout headers hdr, inout metadata meta) {
