--- before_pass
+++ after_pass
@@ -21,9 +21,9 @@ parser MyEP(packet_in buffer, out EMPTY
 }
 control MyIC(inout ethernet_t a, inout EMPTY b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
     bit<10> data_0;
+    bit<10> tmp;
     @name("reg") Register<bit<10>, bit<10>>(32w1024) reg_0;
     @name("execute_register") action execute_register_0(bit<10> idx) {
-        bit<10> tmp;
         tmp = reg_0.read(idx);
         data_0 = tmp;
     }
