--- before_pass
+++ after_pass
@@ -10,11 +10,11 @@ struct m {
 }
 parser MyParser(packet_in b, out h hdrs, inout m meta, inout standard_metadata_t std) {
     state start {
-        meta.counter = 4;
+        meta.counter = (bit<3>)3w4;
         transition q0;
     }
     state q0 {
-        meta.counter = meta.counter - 1;
+        meta.counter = meta.counter - 3w1;
         transition accept;
     }
 }
