--- before_pass
+++ after_pass
@@ -1,15 +1,9 @@
 control c(inout bit<32> x) {
     apply {
-        {
-            x = 32w10;
-        }
+        x = 32w10;
         if (x == 32w10) {
-            {
-                x = x + 32w2;
-            }
-            {
-                x = x + 32w4294967290;
-            }
+            x = x + 32w2;
+            x = x + 32w4294967290;
         } else {
             x = x << 2;
         }
