--- before_pass
+++ after_pass
@@ -20,7 +20,7 @@ control MyVerifyChecksum(inout my_packet
 control MyIngress(inout my_packet p, inout my_metadata m, inout standard_metadata_t s) {
     apply {
         bit<32> x;
-        hash(x, HashAlgorithm.crc32, 32w0, { p.h.f ^ 32w0xffff }, 32w65536);
+        hash<bit<32>, bit<32>, tuple<bit<32>>, bit<32>>(x, HashAlgorithm.crc32, 32w0, { p.h.f ^ 32w0xffff }, 32w65536);
     }
 }
 control MyEgress(inout my_packet p, inout my_metadata m, inout standard_metadata_t s) {
@@ -35,4 +35,4 @@ control MyDeparser(packet_out b, in my_p
     apply {
     }
 }
-V1Switch(MyParser(), MyVerifyChecksum(), MyIngress(), MyEgress(), MyComputeChecksum(), MyDeparser()) main;
+V1Switch<my_packet, my_metadata>(MyParser(), MyVerifyChecksum(), MyIngress(), MyEgress(), MyComputeChecksum(), MyDeparser()) main;
