--- before_pass
+++ after_pass
@@ -20,7 +20,7 @@ parser p(packet_in b, out Headers h, ino
     state start {
         b.extract(h.h1);
         transition select(h.h1.a) {
-            0: getH1;
+            8w0: getH1;
             default: getH2;
         }
     }
@@ -56,7 +56,7 @@ control ingress(inout Headers h, inout M
         if (h.u.h2.isValid()) {
             h.u.h2.setInvalid();
             h.u.h1.setValid();
-            h.u.h1.a = 0xff;
+            h.u.h1.a = (bit<8>)8w0xff;
         }
     }
 }
