--- before_pass
+++ after_pass
@@ -41,19 +41,19 @@ control deparser(packet_out b, in Header
 }
 control ingress(inout Header_t h, inout Meta_t m, inout standard_metadata_t standard_meta) {
     action usat_plus() {
-        standard_meta.egress_spec = 0;
+        standard_meta.egress_spec = (bit<9>)9w0;
         h.h.res_8 = h.h.opr1_8 |+| h.h.opr2_8;
     }
     action usat_minus() {
-        standard_meta.egress_spec = 0;
+        standard_meta.egress_spec = (bit<9>)9w0;
         h.h.res_8 = h.h.opr1_8 |-| h.h.opr2_8;
     }
     action sat_plus() {
-        standard_meta.egress_spec = 0;
+        standard_meta.egress_spec = (bit<9>)9w0;
         h.h.res_16 = h.h.opr1_16 |+| h.h.opr2_16;
     }
     action sat_minus() {
-        standard_meta.egress_spec = 0;
+        standard_meta.egress_spec = (bit<9>)9w0;
         h.h.res_16 = h.h.opr1_16 |-| h.h.opr2_16;
     }
     action drop() {
@@ -74,10 +74,10 @@ control ingress(inout Header_t h, inout
         }
         default_action = drop();
         const entries = {
-                        0x1 : usat_plus();
-                        0x2 : usat_minus();
-                        0x3 : sat_plus();
-                        0x4 : sat_minus();
+                        8w0x1 : usat_plus();
+                        8w0x2 : usat_minus();
+                        8w0x3 : sat_plus();
+                        8w0x4 : sat_minus();
         }
     }
     apply {
