--- before_pass
+++ after_pass
@@ -40,26 +40,26 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Header_t h, inout Meta_t m, inout standard_metadata_t standard_meta) {
-    @name("usat_plus") action usat_plus() {
+    @name("ingress.usat_plus") action usat_plus() {
         standard_meta.egress_spec = 9w0;
         h.h.res_8 = h.h.opr1_8 |+| h.h.opr2_8;
     }
-    @name("usat_minus") action usat_minus() {
+    @name("ingress.usat_minus") action usat_minus() {
         standard_meta.egress_spec = 9w0;
         h.h.res_8 = h.h.opr1_8 |-| h.h.opr2_8;
     }
-    @name("sat_plus") action sat_plus() {
+    @name("ingress.sat_plus") action sat_plus() {
         standard_meta.egress_spec = 9w0;
         h.h.res_16 = h.h.opr1_16 |+| h.h.opr2_16;
     }
-    @name("sat_minus") action sat_minus() {
+    @name("ingress.sat_minus") action sat_minus() {
         standard_meta.egress_spec = 9w0;
         h.h.res_16 = h.h.opr1_16 |-| h.h.opr2_16;
     }
-    @name("drop") action drop() {
+    @name("ingress.drop") action drop() {
         mark_to_drop(standard_meta);
     }
-    @name("t") table t_0 {
+    @name("ingress.t") table t_0 {
         key = {
             h.h.op: exact @name("h.h.op") ;
         }
