--- before_pass
+++ after_pass
@@ -8,14 +8,14 @@ extern State {
     bit<16> get(in bit<16> index);
 }
 control c(inout bit<16> p) {
-    Virtual() cntr = {
-        State(16s1024) state;
+    @name("cntr") Virtual() cntr_0 = {
+        @name("state") State(16s1024) state_0;
         bit<16> f(in bit<16> ix) {
-            return state.get(ix);
+            return state_0.get(ix);
         }
     };
     apply {
-        cntr.run(16w6);
+        cntr_0.run(16w6);
     }
 }
 control ctr(inout bit<16> x);
