--- before_pass
+++ after_pass
@@ -20,9 +20,9 @@ control ingress(inout parsed_packet_t hd
     apply {
         if (standard_metadata.parser_error == error.PacketTooShort) {
             hdr.eth.setValid();
-            hdr.eth.type = 0;
-            hdr.eth.src = 0;
-            hdr.eth.dst = 0;
+            hdr.eth.type = (bit<16>)16w0;
+            hdr.eth.src = (bit<48>)48w0;
+            hdr.eth.dst = (bit<48>)48w0;
         }
     }
 }
