--- before_pass
+++ after_pass
@@ -22,22 +22,22 @@ parser MyParser(packet_in b, out my_pack
     }
 }
 control MyIngress(inout my_packet p, inout my_metadata meta, inout standard_metadata_t s) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name("set_data") action set_data_0() {
+    @name("set_data") action set_data() {
     }
-    @name("t") table t {
+    @name("t") table t_0 {
         actions = {
-            set_data_0();
-            @defaultonly NoAction_1();
+            set_data();
+            @defaultonly NoAction_0();
         }
         key = {
             meta.err: exact @name("meta.err") ;
         }
-        default_action = NoAction_1();
+        default_action = NoAction_0();
     }
     apply {
-        t.apply();
+        t_0.apply();
     }
 }
 control MyEgress(inout my_packet p, inout my_metadata m, inout standard_metadata_t s) {
