--- before_pass
+++ after_pass
@@ -2,7 +2,6 @@
 #include <v1model.p4>
 package Pipeline<H, M>(Parser<H, M> p, Ingress<H, M> ig, Egress<H, M> eg, Deparser<H> dp);
 package Switch<H0, M0, H1, M1>(Pipeline<H0, M0> p0, @optional Pipeline<H1, M1> p1);
-const bit<8> CONST_VAL = 8w10;
 header data_h {
     bit<24> da;
     bit<8>  db;
@@ -22,10 +21,6 @@ parser MyParser(packet_in b, out my_pack
         }
     }
 }
-control MyVerifyChecksum(inout my_packet hdr, inout my_metadata meta) {
-    apply {
-    }
-}
 control MyIngress(inout my_packet p, inout my_metadata meta, inout standard_metadata_t s) {
     action set_data() {
     }
@@ -47,10 +42,6 @@ control MyEgress(inout my_packet p, inou
     apply {
     }
 }
-control MyComputeChecksum(inout my_packet p, inout my_metadata m) {
-    apply {
-    }
-}
 control MyDeparser(packet_out b, in my_packet p) {
     apply {
     }
