--- before_pass
+++ after_pass
@@ -22,11 +22,11 @@ parser MyParser(packet_in b, out my_pack
     }
 }
 control MyIngress(inout my_packet p, inout my_metadata meta, inout standard_metadata_t s) {
-    action set_data() {
+    @name("set_data") action set_data_0() {
     }
-    table t {
+    @name("t") table t_0 {
         actions = {
-            set_data();
+            set_data_0();
             @defaultonly NoAction();
         }
         key = {
@@ -35,7 +35,7 @@ control MyIngress(inout my_packet p, ino
         default_action = NoAction();
     }
     apply {
-        t.apply();
+        t_0.apply();
     }
 }
 control MyEgress(inout my_packet p, inout my_metadata m, inout standard_metadata_t s) {
