--- before_pass
+++ after_pass
@@ -16,7 +16,7 @@ struct my_metadata {
 }
 parser MyParser(packet_in b, out my_packet p, inout my_metadata m, inout standard_metadata_t s) {
     state start {
-        b.extract(p.hdr);
+        b.extract<data_h>(p.hdr);
         transition select(p.hdr.da) {
             default: accept;
         }
@@ -55,5 +55,5 @@ control MyDeparser(packet_out b, in my_p
     apply {
     }
 }
-Pipeline(MyParser(), MyIngress(), MyEgress(), MyDeparser()) p0;
-Switch(p0) main;
+Pipeline<my_packet, my_metadata>(MyParser(), MyIngress(), MyEgress(), MyDeparser()) p0;
+Switch<my_packet, my_metadata, _, _>(p0) main;
