--- before_pass
+++ after_pass
@@ -24,9 +24,9 @@ parser MyParser(packet_in b, out my_pack
 control MyIngress(inout my_packet p, inout my_metadata meta, inout standard_metadata_t s) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("set_data") action set_data() {
+    @name("MyIngress.set_data") action set_data() {
     }
-    @name("t") table t_0 {
+    @name("MyIngress.t") table t_0 {
         actions = {
             set_data();
             @defaultonly NoAction_0();
