--- before_pass
+++ after_pass
@@ -9,10 +9,10 @@ control DeparserI(packet_out packet, in
     }
 }
 parser parserI(packet_in pkt, out H hdr, inout M meta, inout standard_metadata_t stdmeta) {
-    bit<112> tmp_0;
+    bit<112> tmp;
     state start {
-        tmp_0 = pkt.lookahead<bit<112>>();
-        transition select(tmp_0[111:96]) {
+        tmp = pkt.lookahead<bit<112>>();
+        transition select(tmp[111:96]) {
             16w0x1000 &&& 16w0x1000: accept;
         }
     }
