--- before_pass
+++ after_pass
@@ -34,7 +34,7 @@ struct m {
 }
 parser EthernetParser(packet_in b, out ethernet_t ether) {
     state start {
-        b.extract(ether);
+        b.extract<ethernet_t>(ether);
         transition accept;
     }
 }
@@ -47,13 +47,13 @@ parser L2(packet_in b, out h hdr) {
 }
 parser Ipv4Parser(packet_in b, out ipv4_t ipv4) {
     state start {
-        b.extract(ipv4);
+        b.extract<ipv4_t>(ipv4);
         transition accept;
     }
 }
 parser VlanParser(packet_in b, out vlan_t vlan) {
     state start {
-        b.extract(vlan);
+        b.extract<vlan_t>(vlan);
         transition accept;
     }
 }
@@ -107,4 +107,4 @@ control MyDeparser(packet_out b, in h hd
     apply {
     }
 }
-V1Switch(MyParser(), MyVerifyChecksum(), MyIngress(), MyEgress(), MyComputeChecksum(), MyDeparser()) main;
+V1Switch<h, m>(MyParser(), MyVerifyChecksum(), MyIngress(), MyEgress(), MyComputeChecksum(), MyDeparser()) main;
