--- before_pass
+++ after_pass
@@ -3,12 +3,12 @@ header H {
     bit<32> f;
 }
 parser p(packet_in pk) {
-    H h;
-    @name("vs") value_set<tuple<bit<32>, bit<2>>>(4) vs;
+    H h_0;
+    @name("vs") value_set<tuple<bit<32>, bit<2>>>(4) vs_0;
     state start {
-        pk.extract<H>(h);
-        transition select(h.f, 2w2) {
-            vs: next;
+        pk.extract<H>(h_0);
+        transition select(h_0.f, 2w2) {
+            vs_0: next;
             default: reject;
         }
     }
