--- before_pass
+++ after_pass
@@ -8,13 +8,23 @@ control ctrl(inout struct_t input, out h
     header_h tmp0_0;
     header_h tmp1_0;
     @name("act") action act_0() {
-        tmp0_0 = input.stack[0];
-        input.stack.pop_front(1);
-        tmp1_0 = tmp0_0;
+        {
+            tmp0_0 = input.stack[0];
+        }
+        {
+            input.stack.pop_front(1);
+        }
+        {
+            tmp1_0 = tmp0_0;
+        }
     }
     apply {
-        act_0();
-        output = tmp1_0;
+        {
+            act_0();
+        }
+        {
+            output = tmp1_0;
+        }
     }
 }
 control MyControl<S, H>(inout S data, out H output);
