--- before_pass
+++ after_pass
@@ -7,13 +7,13 @@ struct struct_t {
 control ctrl(inout struct_t input, out header_h output) {
     header_h tmp0;
     header_h tmp1;
-    @name("act") action act() {
+    @name("act") action act_0() {
         tmp0 = input.stack[0];
         input.stack.pop_front(1);
         tmp1 = tmp0;
     }
     apply {
-        act();
+        act_0();
         output = tmp1;
     }
 }
