--- before_pass
+++ after_pass
@@ -16,11 +16,11 @@ parser p(packet_in b, out headers hdr, i
     }
 }
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
-    H h_0;
+    H h_1;
     apply {
         stdmeta.egress_spec = 9w0;
-        h_0 = hdr.h;
-        if (hdr.h.v == h_0.v) {
+        h_1 = hdr.h;
+        if (hdr.h.v == h_1.v) {
             stdmeta.egress_spec = 9w1;
         }
     }
