--- before_pass
+++ after_pass
@@ -11,15 +11,21 @@ struct headers {
 }
 parser p(packet_in b, out headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
     state start {
-        b.extract<H>(hdr.h, 32w32);
+        {
+            b.extract<H>(hdr.h, 32w32);
+        }
         transition accept;
     }
 }
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
     H h_0;
     apply {
-        stdmeta.egress_spec = 9w0;
-        h_0 = hdr.h;
+        {
+            stdmeta.egress_spec = 9w0;
+        }
+        {
+            h_0 = hdr.h;
+        }
         if (hdr.h.v == h_0.v) {
             stdmeta.egress_spec = 9w1;
         }
@@ -39,7 +45,9 @@ control uc(inout headers hdr, inout meta
 }
 control deparser(packet_out packet, in headers hdr) {
     apply {
-        packet.emit<headers>(hdr);
+        {
+            packet.emit<headers>(hdr);
+        }
     }
 }
 V1Switch<headers, metadata>(p(), vc(), ingress(), egress(), uc(), deparser()) main;
