--- before_pass
+++ after_pass
@@ -27,37 +27,21 @@ control Eg(inout Headers hdrs, inout Met
     bool done_0;
     bool ok_0;
     @name("update") action update_0(inout Value val) {
-        {
-            val.field1 = 32w8;
-        }
+        val.field1 = 32w8;
     }
     @name("test") action test_0() {
-        {
-            inKey_0 = { 32w1 };
-        }
-        {
-            defaultKey_0 = { 32w0 };
-        }
-        {
-            same_0 = inKey_0 == defaultKey_0;
-        }
-        {
-            val_0 = { 32w0 };
-        }
-        {
-            done_0 = false;
-        }
-        {
-            ok_0 = !done_0 && same_0;
-        }
+        inKey_0 = { 32w1 };
+        defaultKey_0 = { 32w0 };
+        same_0 = inKey_0 == defaultKey_0;
+        val_0 = { 32w0 };
+        done_0 = false;
+        ok_0 = !done_0 && same_0;
         if (ok_0) {
             update_0(val_0);
         }
     }
     apply {
-        {
-            test_0();
-        }
+        test_0();
     }
 }
 control DP(packet_out b, in Headers p) {
