--- before_pass
+++ after_pass
@@ -26,10 +26,10 @@ control Eg(inout Headers hdrs, inout Met
         val.field1 = 32w8;
     }
     action test() {
-        Key inKey = { 1 };
-        Key defaultKey = { 0 };
+        Key inKey = { 32w1 };
+        Key defaultKey = { 32w0 };
         bool same = inKey == defaultKey;
-        Value val = { 0 };
+        Value val = { 32w0 };
         bool done = false;
         bool ok = !done && same;
         if (ok) {
