--- before_pass
+++ after_pass
@@ -217,11 +217,14 @@ parser ParserImpl(packet_in packet, out
         transition accept;
     }
 }
-action my_drop(inout standard_metadata_t smeta_0) {
-    mark_to_drop(smeta_0);
-}
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    @name("set_l2ptr") action set_l2ptr(bit<32> l2ptr) {
+    @name(".my_drop") action my_drop_2(inout standard_metadata_t smeta_0) {
+        mark_to_drop(smeta_0);
+    }
+    @name(".my_drop") action my_drop_3(inout standard_metadata_t smeta_0) {
+        mark_to_drop(smeta_0);
+    }
+    @name("set_l2ptr") action set_l2ptr_0(bit<32> l2ptr) {
         meta.fwd_metadata.l2ptr = l2ptr;
     }
     @name("ipv4_da_lpm") table ipv4_da_lpm {
@@ -229,12 +232,12 @@ control ingress(inout headers hdr, inout
             hdr.ipv4.dstAddr: lpm @name("hdr.ipv4.dstAddr") ;
         }
         actions = {
-            set_l2ptr();
-            my_drop(standard_metadata);
+            set_l2ptr_0();
+            my_drop_2(standard_metadata);
         }
-        default_action = my_drop(standard_metadata);
+        default_action = my_drop_2(standard_metadata);
     }
-    @name("set_bd_dmac_intf") action set_bd_dmac_intf(bit<24> bd, bit<48> dmac, bit<9> intf) {
+    @name("set_bd_dmac_intf") action set_bd_dmac_intf_0(bit<24> bd, bit<48> dmac, bit<9> intf) {
         meta.fwd_metadata.out_bd = bd;
         hdr.ethernet.dstAddr = dmac;
         standard_metadata.egress_spec = intf;
@@ -245,10 +248,10 @@ control ingress(inout headers hdr, inout
             meta.fwd_metadata.l2ptr: exact @name("meta.fwd_metadata.l2ptr") ;
         }
         actions = {
-            set_bd_dmac_intf();
-            my_drop(standard_metadata);
+            set_bd_dmac_intf_0();
+            my_drop_3(standard_metadata);
         }
-        default_action = my_drop(standard_metadata);
+        default_action = my_drop_3(standard_metadata);
     }
     apply {
         ipv4_da_lpm.apply();
@@ -256,7 +259,10 @@ control ingress(inout headers hdr, inout
     }
 }
 control egress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    @name("rewrite_mac") action rewrite_mac(bit<48> smac) {
+    @name(".my_drop") action my_drop_4(inout standard_metadata_t smeta_0) {
+        mark_to_drop(smeta_0);
+    }
+    @name("rewrite_mac") action rewrite_mac_0(bit<48> smac) {
         hdr.ethernet.srcAddr = smac;
     }
     @name("send_frame") table send_frame {
@@ -264,10 +270,10 @@ control egress(inout headers hdr, inout
             meta.fwd_metadata.out_bd: exact @name("meta.fwd_metadata.out_bd") ;
         }
         actions = {
-            rewrite_mac();
-            my_drop(standard_metadata);
+            rewrite_mac_0();
+            my_drop_4(standard_metadata);
         }
-        default_action = my_drop(standard_metadata);
+        default_action = my_drop_4(standard_metadata);
     }
     apply {
         send_frame.apply();
