--- before_pass
+++ after_pass
@@ -111,9 +111,6 @@ parser Tcp_option_parser(packet_in b, in
     state parse_tcp_option_end {
         b.extract<Tcp_option_end_h>(vec.next.end);
         tcp_hdr_bytes_left = tcp_hdr_bytes_left + 7w127;
-        transition consume_remaining_tcp_hdr_and_accept;
-    }
-    state consume_remaining_tcp_hdr_and_accept {
         b.extract<Tcp_option_padding_h>(padding, (bit<32>)((bit<9>)tcp_hdr_bytes_left << 3));
         transition accept;
     }
@@ -146,9 +143,6 @@ parser Tcp_option_parser(packet_in b, in
 parser ParserImpl(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     @name("Tcp_option_parser") Tcp_option_parser() Tcp_option_parser_inst;
     state start {
-        transition parse_ethernet;
-    }
-    state parse_ethernet {
         packet.extract<ethernet_t>(hdr.ethernet);
         transition select(hdr.ethernet.etherType) {
             16w0x800: parse_ipv4;
