--- before_pass
+++ after_pass
@@ -144,14 +144,14 @@ parser Tcp_option_parser(packet_in b, in
     }
 }
 parser ParserImpl(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    const bit<16> ETHERTYPE_IPV4 = 0x800;
+    const bit<16> ETHERTYPE_IPV4 = 16w0x800;
     state start {
         transition parse_ethernet;
     }
     state parse_ethernet {
         packet.extract(hdr.ethernet);
         transition select(hdr.ethernet.etherType) {
-            ETHERTYPE_IPV4: parse_ipv4;
+            16w0x800: parse_ipv4;
             default: accept;
         }
     }
