--- before_pass
+++ after_pass
@@ -145,7 +145,6 @@ parser Tcp_option_parser(packet_in b, in
 }
 parser ParserImpl(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     @name("Tcp_option_parser") Tcp_option_parser() Tcp_option_parser_inst;
-    const bit<16> ETHERTYPE_IPV4 = 16w0x800;
     state start {
         transition parse_ethernet;
     }
