--- before_pass
+++ after_pass
@@ -178,7 +178,7 @@ control ingress(inout headers hdr, inout
     }
     table ipv4_da_lpm {
         key = {
-            hdr.ipv4.dstAddr: lpm;
+            hdr.ipv4.dstAddr: lpm @name("hdr.ipv4.dstAddr") ;
         }
         actions = {
             set_l2ptr();
@@ -194,7 +194,7 @@ control ingress(inout headers hdr, inout
     }
     table mac_da {
         key = {
-            meta.fwd_metadata.l2ptr: exact;
+            meta.fwd_metadata.l2ptr: exact @name("meta.fwd_metadata.l2ptr") ;
         }
         actions = {
             set_bd_dmac_intf();
@@ -213,7 +213,7 @@ control egress(inout headers hdr, inout
     }
     table send_frame {
         key = {
-            meta.fwd_metadata.out_bd: exact;
+            meta.fwd_metadata.out_bd: exact @name("meta.fwd_metadata.out_bd") ;
         }
         actions = {
             rewrite_mac();
