--- before_pass
+++ after_pass
@@ -13,9 +13,7 @@ parser ParserI(packet_in pk, out H hdr,
 }
 control myc(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
     @name("set_eg") action set_eg_0(bit<9> eg) {
-        {
-            smeta.egress_spec = eg;
-        }
+        smeta.egress_spec = eg;
     }
     @name("myt") table myt_0 {
         key = {
@@ -33,17 +31,13 @@ control myc(inout H hdr, inout M meta, i
         default_action = NoAction();
     }
     apply {
-        {
-            myt_0.apply();
-        }
+        myt_0.apply();
     }
 }
 control IngressI(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
     @name("myc") myc() myc_inst_0;
     apply {
-        {
-            myc_inst_0.apply(hdr, meta, smeta);
-        }
+        myc_inst_0.apply(hdr, meta, smeta);
     }
 }
 control EgressI(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
