--- before_pass
+++ after_pass
@@ -14,10 +14,10 @@ parser ParserI(packet_in pk, out H hdr,
 control IngressI(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("myc.set_eg") action myc_set_eg_0(bit<9> eg) {
+    @name("IngressI.myc.set_eg") action myc_set_eg_0(bit<9> eg) {
         smeta.egress_spec = eg;
     }
-    @name("myc.myt") table myc_myt {
+    @name("IngressI.myc.myt") table myc_myt {
         key = {
             meta.f1: exact @name("meta.f1") ;
             meta.f2: exact @name("meta.f2") ;
