--- before_pass
+++ after_pass
@@ -12,7 +12,9 @@ parser ParserI(packet_in pk, out H hdr,
     }
 }
 control IngressI(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
-    @name("myc.set_eg") action myc_set_eg_0(bit<9> eg) {
+    @name(".NoAction") action NoAction_1() {
+    }
+    @name("myc.set_eg") action myc_set_eg(bit<9> eg) {
         smeta.egress_spec = eg;
     }
     @name("myc.myt") table myc_myt_0 {
@@ -21,14 +23,14 @@ control IngressI(inout H hdr, inout M me
             meta.f2: exact @name("meta.f2") ;
         }
         actions = {
-            myc_set_eg_0();
-            @defaultonly NoAction();
+            myc_set_eg();
+            @defaultonly NoAction_1();
         }
         const entries = {
-                        (32w1, 32w0xffffffff) : myc_set_eg_0(9w1);
-                        (32w2, 32w0xffffffff) : myc_set_eg_0(9w2);
+                        (32w1, 32w0xffffffff) : myc_set_eg(9w1);
+                        (32w2, 32w0xffffffff) : myc_set_eg(9w2);
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     apply {
         myc_myt_0.apply();
