--- before_pass
+++ after_pass
@@ -24,12 +24,12 @@ control MyIC(inout ethernet_t a, inout E
     }
     @name(".NoAction") action NoAction_3() {
     }
-    @name("ap") ActionProfile(32w1024) ap_0;
-    @name("a1") action a1() {
+    @name("MyIC.ap") ActionProfile(32w1024) ap_0;
+    @name("MyIC.a1") action a1() {
     }
-    @name("a2") action a2() {
+    @name("MyIC.a2") action a2() {
     }
-    @name("tbl") table tbl_0 {
+    @name("MyIC.tbl") table tbl_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
@@ -40,7 +40,7 @@ control MyIC(inout ethernet_t a, inout E
         psa_implementation = ap_0;
         default_action = NoAction_0();
     }
-    @name("tbl2") table tbl2_0 {
+    @name("MyIC.tbl2") table tbl2_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
