--- before_pass
+++ after_pass
@@ -30,20 +30,22 @@ control MyIC(inout ethernet_t a, inout E
             a.srcAddr: exact;
         }
         actions = {
-            NoAction;
-            a2;
+            NoAction();
+            a2();
         }
         psa_implementation = ap;
+        default_action = NoAction();
     }
     table tbl2 {
         key = {
             a.srcAddr: exact;
         }
         actions = {
-            NoAction;
-            a1;
+            NoAction();
+            a1();
         }
         psa_implementation = ap;
+        default_action = NoAction();
     }
     apply {
         tbl.apply();
