--- before_pass
+++ after_pass
@@ -27,7 +27,7 @@ control MyIC(inout ethernet_t a, inout E
     }
     table tbl {
         key = {
-            a.srcAddr: exact;
+            a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
             NoAction();
@@ -38,7 +38,7 @@ control MyIC(inout ethernet_t a, inout E
     }
     table tbl2 {
         key = {
-            a.srcAddr: exact;
+            a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
             NoAction();
