--- before_pass
+++ after_pass
@@ -17,8 +17,8 @@ header ethernet_t {
 struct headers {
     ethernet_t ethernet;
 }
-action my_drop(inout standard_metadata_t smeta) {
-    mark_to_drop(smeta);
+action my_drop(inout standard_metadata_t smeta_0) {
+    mark_to_drop(smeta_0);
 }
 parser ParserImpl(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     state start {
