--- before_pass
+++ after_pass
@@ -1,11 +1,16 @@
 #include <core.p4>
 #include <v1model.p4>
 bit<16> sometimes_dec(in bit<16> x) {
+    bool hasReturned = false;
+    bit<16> retval;
     if (x > 16w5) {
-        return x + 16w65535;
+        hasReturned = true;
+        retval = x + 16w65535;
     } else {
-        return x;
+        hasReturned = true;
+        retval = x;
     }
+    return retval;
 }
 struct metadata {
 }
