--- before_pass
+++ after_pass
@@ -7,18 +7,14 @@ struct Meta {
 }
 parser p(packet_in b, out Headers h, inout Meta m, inout standard_metadata_t sm) {
     state start {
-        {
-            m.b = m.b + 1w1;
-        }
+        m.b = m.b + 1w1;
         transition accept;
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
     Meta x_0;
     apply {
-        {
-            clone3<Meta>(CloneType.I2E, 32w64, x_0);
-        }
+        clone3<Meta>(CloneType.I2E, 32w64, x_0);
     }
 }
 control vrfy(inout Headers h, inout Meta m) {
