--- before_pass
+++ after_pass
@@ -30,7 +30,7 @@ struct metadata {
 }
 parser subParserImpl(packet_in pkt, inout headers hdr, out bit<8> ret_next_hdr_type) {
     state start {
-        pkt.extract(hdr.h2.next);
+        pkt.extract<h2_t>(hdr.h2.next);
         verify(hdr.h2.last.hdr_type == 8w2, error.BadHeaderType);
         ret_next_hdr_type = hdr.h2.last.next_hdr_type;
         transition accept;
@@ -40,7 +40,7 @@ parser parserI(packet_in pkt, out header
     subParserImpl() subp;
     bit<8> my_next_hdr_type;
     state start {
-        pkt.extract(hdr.h1);
+        pkt.extract<h1_t>(hdr.h1);
         verify(hdr.h1.hdr_type == 8w1, error.BadHeaderType);
         transition select(hdr.h1.next_hdr_type) {
             8w2: parse_first_h2;
@@ -57,7 +57,7 @@ parser parserI(packet_in pkt, out header
         }
     }
     state parse_other_h2 {
-        pkt.extract(hdr.h2.next);
+        pkt.extract<h2_t>(hdr.h2.next);
         verify(hdr.h2.last.hdr_type == 8w2, error.BadHeaderType);
         transition select(hdr.h2.last.next_hdr_type) {
             8w2: parse_other_h2;
@@ -66,7 +66,7 @@ parser parserI(packet_in pkt, out header
         }
     }
     state parse_h3 {
-        pkt.extract(hdr.h3);
+        pkt.extract<h3_t>(hdr.h3);
         verify(hdr.h3.hdr_type == 8w3, error.BadHeaderType);
         transition accept;
     }
@@ -105,9 +105,9 @@ control uc(inout headers hdr, inout meta
 }
 control DeparserI(packet_out packet, in headers hdr) {
     apply {
-        packet.emit(hdr.h1);
-        packet.emit(hdr.h2);
-        packet.emit(hdr.h3);
+        packet.emit<h1_t>(hdr.h1);
+        packet.emit<h2_t[5]>(hdr.h2);
+        packet.emit<h3_t>(hdr.h3);
     }
 }
-V1Switch(parserI(), vc(), cIngress(), cEgress(), uc(), DeparserI()) main;
+V1Switch<headers, metadata>(parserI(), vc(), cIngress(), cEgress(), uc(), DeparserI()) main;
