--- before_pass
+++ after_pass
@@ -86,7 +86,7 @@ control ingress(inout headers hdr, inout
     ErrorIndex_t error_idx_0;
     @name("parser_error_counts") DirectCounter<PacketCounter_t>(PSA_CounterType_t.PACKETS) parser_error_counts_0;
     @name("set_error_idx") action set_error_idx_0(ErrorIndex_t idx) {
-        error_idx_0 = idx;
+        ;
         parser_error_counts_0.count();
     }
     @name("parser_error_count_and_convert") table parser_error_count_and_convert_0 {
