--- before_pass
+++ after_pass
@@ -37,7 +37,7 @@ control deparser(packet_out b, in Header
 }
 control ingress(inout Header_t h, inout Meta_t m, inout standard_metadata_t standard_meta) {
     action a() {
-        standard_meta.egress_spec = 0;
+        standard_meta.egress_spec = (bit<9>)9w0;
     }
     action a_with_control_params(bit<9> x) {
         standard_meta.egress_spec = x;
@@ -52,10 +52,10 @@ control ingress(inout Header_t h, inout
         }
         default_action = a();
         const entries = {
-                        1 .. 8 : a_with_control_params(21);
-                        6 .. 12 : a_with_control_params(22);
-                        15 : a_with_control_params(24);
-                        default : a_with_control_params(23);
+                        8w1 .. 8w8 : a_with_control_params(9w21);
+                        8w6 .. 8w12 : a_with_control_params(9w22);
+                        8w15 : a_with_control_params(9w24);
+                        default : a_with_control_params(9w23);
         }
     }
     apply {
