--- before_pass
+++ after_pass
@@ -14,9 +14,7 @@ struct Meta_t {
 }
 parser p(packet_in b, out Header_t h, inout Meta_t m, inout standard_metadata_t sm) {
     state start {
-        {
-            b.extract<hdr>(h.h);
-        }
+        b.extract<hdr>(h.h);
         transition accept;
     }
 }
@@ -34,21 +32,15 @@ control egress(inout Header_t h, inout M
 }
 control deparser(packet_out b, in Header_t h) {
     apply {
-        {
-            b.emit<hdr>(h.h);
-        }
+        b.emit<hdr>(h.h);
     }
 }
 control ingress(inout Header_t h, inout Meta_t m, inout standard_metadata_t standard_meta) {
     @name("a") action a_0() {
-        {
-            standard_meta.egress_spec = 9w0;
-        }
+        standard_meta.egress_spec = 9w0;
     }
     @name("a_with_control_params") action a_with_control_params_0(bit<9> x) {
-        {
-            standard_meta.egress_spec = x;
-        }
+        standard_meta.egress_spec = x;
     }
     @name("t_range") table t_range_0 {
         key = {
@@ -67,9 +59,7 @@ control ingress(inout Header_t h, inout
         }
     }
     apply {
-        {
-            t_range_0.apply();
-        }
+        t_range_0.apply();
     }
 }
 V1Switch<Header_t, Meta_t>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
