--- before_pass
+++ after_pass
@@ -33,9 +33,9 @@ extern ExternCounter {
     void increment();
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    ExternCounter(5) extr;
+    @name("extr") ExternCounter(5) extr_0;
     apply {
-        extr.increment();
+        extr_0.increment();
     }
 }
 V1Switch<Headers, Meta>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
