--- before_pass
+++ after_pass
@@ -22,8 +22,10 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout EMPTY b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
+    @name(".NoAction") action NoAction_1() {
+    }
     @name("counter") Counter<bit<10>, bit<12>>(32w1024, PSA_CounterType_t.PACKETS) counter;
-    @name("execute") action execute_1() {
+    @name("execute") action execute_0() {
         counter.count(12w1024);
     }
     @name("tbl") table tbl {
@@ -31,10 +33,10 @@ control MyIC(inout ethernet_t a, inout E
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
-            NoAction();
-            execute_1();
+            NoAction_1();
+            execute_0();
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     apply {
         tbl.apply();
