--- before_pass
+++ after_pass
@@ -12,7 +12,7 @@ header ethernet_t {
 }
 parser MyIP(packet_in buffer, out ethernet_t eth, inout EMPTY b, in psa_ingress_parser_input_metadata_t c, in EMPTY_H d, in EMPTY_H e) {
     state start {
-        buffer.extract(eth);
+        buffer.extract<ethernet_t>(eth);
         transition accept;
     }
 }
@@ -52,6 +52,6 @@ control MyED(packet_out buffer, out EMPT
     apply {
     }
 }
-IngressPipeline(MyIP(), MyIC(), MyID()) ip;
-EgressPipeline(MyEP(), MyEC(), MyED()) ep;
-PSA_Switch(ip, PacketReplicationEngine(), ep, BufferingQueueingEngine()) main;
+IngressPipeline<ethernet_t, EMPTY, EMPTY, EMPTY_H, EMPTY_H, EMPTY_H>(MyIP(), MyIC(), MyID()) ip;
+EgressPipeline<EMPTY, EMPTY, EMPTY, EMPTY_H, EMPTY_H, EMPTY_H>(MyEP(), MyEC(), MyED()) ep;
+PSA_Switch<ethernet_t, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY_H, EMPTY_H, EMPTY_H, EMPTY_H>(ip, PacketReplicationEngine(), ep, BufferingQueueingEngine()) main;
