--- before_pass
+++ after_pass
@@ -12,7 +12,9 @@ header ethernet_t {
 }
 parser MyIP(packet_in buffer, out ethernet_t eth, inout EMPTY b, in psa_ingress_parser_input_metadata_t c, in EMPTY_H d, in EMPTY_H e) {
     state start {
-        buffer.extract<ethernet_t>(eth);
+        {
+            buffer.extract<ethernet_t>(eth);
+        }
         transition accept;
     }
 }
@@ -24,7 +26,9 @@ parser MyEP(packet_in buffer, out EMPTY
 control MyIC(inout ethernet_t a, inout EMPTY b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
     @name("counter") Counter<bit<10>, bit<12>>(32w1024, PSA_CounterType_t.PACKETS) counter_0;
     @name("execute") action execute_0() {
-        counter_0.count(12w1024);
+        {
+            counter_0.count(12w1024);
+        }
     }
     @name("tbl") table tbl_0 {
         key = {
@@ -37,7 +41,9 @@ control MyIC(inout ethernet_t a, inout E
         default_action = NoAction();
     }
     apply {
-        tbl_0.apply();
+        {
+            tbl_0.apply();
+        }
     }
 }
 control MyEC(inout EMPTY a, inout EMPTY b, in psa_egress_input_metadata_t c, inout psa_egress_output_metadata_t d) {
