--- before_pass
+++ after_pass
@@ -25,9 +25,9 @@ parser parserI(packet_in pkt, out Parsed
 }
 control cIngress(inout Parsed_packet hdr, inout metadata m, inout standard_metadata_t stdmeta) {
     bit<32> z_0;
-    @name("foo") action foo() {
+    @name("cIngress.foo") action foo() {
     }
-    @name("t") table t_0 {
+    @name("cIngress.t") table t_0 {
         key = {
             hdr.x.add1: exact @name("hdr.x.add1") ;
             m.x.add2  : exact @name("m.x.add2") ;
