--- before_pass
+++ after_pass
@@ -17,10 +17,10 @@ struct Headers {
     Ethernet eth;
 }
 parser prs(packet_in p, out Headers h) {
-    Ethernet e;
+    Ethernet e_0;
     state start {
-        p.extract<Ethernet>(e);
-        transition select(e.type) {
+        p.extract<Ethernet>(e_0);
+        transition select(e_0.type) {
             EthTypes.IPv4: accept;
             EthTypes.ARP: accept;
             default: reject;
