--- before_pass
+++ after_pass
@@ -48,18 +48,21 @@ control c(out bit<32> v) {
     bit<32> setByAction;
     bit<32> e;
     bool touched;
-    @name("a1") action a1() {
+    @name("a1") action a1_0() {
         setByAction = 32w1;
     }
-    @name("a2") action a2() {
+    @name("a1") action a1_1() {
+        setByAction = 32w1;
+    }
+    @name("a2") action a2_0() {
         setByAction = 32w1;
     }
     @name("t") table t {
         actions = {
-            a1();
-            a2();
+            a1_0();
+            a2_0();
         }
-        default_action = a1();
+        default_action = a1_0();
     }
     apply {
         d_2 = 32w1;
@@ -70,14 +73,14 @@ control c(out bit<32> v) {
         }
         e = e + 32w1;
         switch (t.apply().action_run) {
-            a1: {
+            a1_0: {
                 touched = true;
             }
         }
         if (e > 32w0) {
             t.apply();
         } else {
-            a1();
+            a1_1();
         }
     }
 }
