--- before_pass
+++ after_pass
@@ -48,16 +48,16 @@ control c(out bit<32> v) {
     bit<32> setByAction_0;
     bit<32> e_0;
     bool touched_0;
-    @name("a1") action a1() {
+    @name("c.a1") action a1() {
         setByAction_0 = 32w1;
     }
-    @name("a1") action a1_2() {
+    @name("c.a1") action a1_2() {
         setByAction_0 = 32w1;
     }
-    @name("a2") action a2() {
+    @name("c.a2") action a2() {
         setByAction_0 = 32w1;
     }
-    @name("t") table t_0 {
+    @name("c.t") table t_0 {
         actions = {
             a1();
             a2();
