--- before_pass
+++ after_pass
@@ -40,7 +40,7 @@ parser p1(packet_in p, out Header h) {
 }
 control c(out bit<32> v) {
     bit<32> b_1;
-    bit<32> d_1 = 32w1;
+    bit<32> d_1;
     bit<32> setByAction_0;
     bit<32> e_0;
     bit<32> f_0;
@@ -59,6 +59,7 @@ control c(out bit<32> v) {
         default_action = a1_0();
     }
     apply {
+        d_1 = 32w1;
         b_1 = b_1 + 32w1;
         d_1 = d_1 + 32w1;
         if (e_0 > 32w0) {
