--- before_pass
+++ after_pass
@@ -8,7 +8,6 @@ extern void func(in Header h);
 extern bit<32> g(inout bit<32> v, in bit<32> w);
 parser p1(packet_in p, out Header h) {
     Header[2] stack_0;
-    bool b_0;
     bool c_0;
     bool d_0;
     bit<32> tmp;
@@ -45,11 +44,9 @@ parser p1(packet_in p, out Header h) {
     }
 }
 control c(out bit<32> v) {
-    bit<32> b_1;
     bit<32> d_1;
     bit<32> setByAction_0;
     bit<32> e_0;
-    bit<32> f_0;
     bool touched_0;
     @name("a1") action a1_0() {
         setByAction_0 = 32w1;
