--- before_pass
+++ after_pass
@@ -5,7 +5,7 @@ header hdr {
 }
 control compute(inout hdr h) {
     apply {
-        hash(h.a, HashAlgorithm.crc16, 10w0, { 16w1 }, 10w4);
+        hash<bit<16>, bit<10>, tuple<bit<16>>, bit<10>>(h.a, HashAlgorithm.crc16, 10w0, { 16w1 }, 10w4);
     }
 }
 struct Headers {
@@ -15,7 +15,7 @@ struct Meta {
 }
 parser p(packet_in b, out Headers h, inout Meta m, inout standard_metadata_t sm) {
     state start {
-        b.extract(h.h);
+        b.extract<hdr>(h.h);
         transition accept;
     }
 }
@@ -33,7 +33,7 @@ control egress(inout Headers h, inout Me
 }
 control deparser(packet_out b, in Headers h) {
     apply {
-        b.emit(h.h);
+        b.emit<hdr>(h.h);
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
@@ -43,4 +43,4 @@ control ingress(inout Headers h, inout M
         sm.egress_spec = (bit<9>)9w0;
     }
 }
-V1Switch(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
+V1Switch<Headers, Meta>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
