--- before_pass
+++ after_pass
@@ -23,9 +23,11 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout user_meta_t b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
+    @name(".NoAction") action NoAction_1() {
+    }
     bit<16> tmp_0;
     @name("h") Hash<bit<16>>(PSA_HashAlgorithm_t.CRC16) h;
-    @name("a1") action a1() {
+    @name("a1") action a1_0() {
         tmp_0 = h.get_hash<ethernet_t>(a);
         b.data = tmp_0;
     }
@@ -34,10 +36,10 @@ control MyIC(inout ethernet_t a, inout u
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
-            NoAction();
-            a1();
+            NoAction_1();
+            a1_0();
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     apply {
         tbl.apply();
