--- before_pass
+++ after_pass
@@ -49,16 +49,16 @@ parser ParserImpl(packet_in packet, out
     }
 }
 control egress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    @name(".rewrite_mac") action rewrite_mac_0(bit<48> smac) {
+    @name(".rewrite_mac") action rewrite_mac(bit<48> smac) {
         hdr.ethernet.srcAddr = smac;
     }
-    @name("._drop") action _drop_0() {
+    @name("._drop") action _drop() {
         mark_to_drop(standard_metadata);
     }
-    @name("send_frame") table send_frame_0 {
+    @name("send_frame") table send_frame {
         actions = {
-            rewrite_mac_0();
-            _drop_0();
+            rewrite_mac();
+            _drop();
             @defaultonly NoAction();
         }
         key = {
@@ -68,36 +68,36 @@ control egress(inout headers hdr, inout
         default_action = NoAction();
     }
     apply {
-        send_frame_0.apply();
+        send_frame.apply();
     }
 }
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    @name(".bcast") action bcast_0() {
+    @name(".bcast") action bcast() {
         standard_metadata.mcast_grp = 16w1;
     }
-    @name(".set_dmac") action set_dmac_0(bit<48> dmac) {
+    @name(".set_dmac") action set_dmac(bit<48> dmac) {
         hdr.ethernet.dstAddr = dmac;
     }
-    @name("._drop") action _drop_1() {
+    @name("._drop") action _drop_2() {
         mark_to_drop(standard_metadata);
     }
-    @name(".set_nhop") action set_nhop_0(bit<32> nhop_ipv4, bit<9> port) {
+    @name(".set_nhop") action set_nhop(bit<32> nhop_ipv4, bit<9> port) {
         meta.routing_metadata.nhop_ipv4 = nhop_ipv4;
         standard_metadata.egress_spec = port;
         hdr.ipv4.ttl = hdr.ipv4.ttl + 8w255;
     }
-    @name("broadcast") table broadcast_0 {
+    @name("broadcast") table broadcast {
         actions = {
-            bcast_0();
+            bcast();
             @defaultonly NoAction();
         }
         size = 1;
         default_action = NoAction();
     }
-    @name("forward") table forward_0 {
+    @name("forward") table forward {
         actions = {
-            set_dmac_0();
-            _drop_1();
+            set_dmac();
+            _drop_2();
             @defaultonly NoAction();
         }
         key = {
@@ -106,10 +106,10 @@ control ingress(inout headers hdr, inout
         size = 512;
         default_action = NoAction();
     }
-    @name("ipv4_lpm") table ipv4_lpm_0 {
+    @name("ipv4_lpm") table ipv4_lpm {
         actions = {
-            set_nhop_0();
-            _drop_1();
+            set_nhop();
+            _drop_2();
             @defaultonly NoAction();
         }
         key = {
@@ -120,10 +120,10 @@ control ingress(inout headers hdr, inout
     }
     apply {
         if (hdr.ipv4.isValid()) {
-            ipv4_lpm_0.apply();
-            forward_0.apply();
+            ipv4_lpm.apply();
+            forward.apply();
         } else {
-            broadcast_0.apply();
+            broadcast.apply();
         }
     }
 }
