--- before_pass
+++ after_pass
@@ -62,7 +62,7 @@ control egress(inout headers hdr, inout
             @defaultonly NoAction();
         }
         key = {
-            standard_metadata.egress_port: exact;
+            standard_metadata.egress_port: exact @name("standard_metadata.egress_port") ;
         }
         size = 256;
         default_action = NoAction();
@@ -106,7 +106,7 @@ control ingress(inout headers hdr, inout
             @defaultonly NoAction();
         }
         key = {
-            meta.routing_metadata.nhop_ipv4: exact;
+            meta.routing_metadata.nhop_ipv4: exact @name("meta.routing_metadata.nhop_ipv4") ;
         }
         size = 512;
         default_action = NoAction();
@@ -118,7 +118,7 @@ control ingress(inout headers hdr, inout
             @defaultonly NoAction();
         }
         key = {
-            hdr.ipv4.dstAddr: lpm;
+            hdr.ipv4.dstAddr: lpm @name("hdr.ipv4.dstAddr") ;
         }
         size = 1024;
         default_action = NoAction();
