--- before_pass
+++ after_pass
@@ -57,13 +57,15 @@ control egress(inout headers hdr, inout
     }
     @name("send_frame") table send_frame {
         actions = {
-            rewrite_mac;
-            _drop;
+            rewrite_mac();
+            _drop();
+            @defaultonly NoAction();
         }
         key = {
             standard_metadata.egress_port: exact;
         }
         size = 256;
+        default_action = NoAction();
     }
     apply {
         send_frame.apply();
@@ -91,29 +93,35 @@ control ingress(inout headers hdr, inout
     }
     @name("broadcast") table broadcast {
         actions = {
-            bcast;
+            bcast();
+            @defaultonly NoAction();
         }
         size = 1;
+        default_action = NoAction();
     }
     @name("forward") table forward {
         actions = {
-            set_dmac;
-            _drop;
+            set_dmac();
+            _drop();
+            @defaultonly NoAction();
         }
         key = {
             meta.routing_metadata.nhop_ipv4: exact;
         }
         size = 512;
+        default_action = NoAction();
     }
     @name("ipv4_lpm") table ipv4_lpm {
         actions = {
-            set_nhop;
-            _drop;
+            set_nhop();
+            _drop();
+            @defaultonly NoAction();
         }
         key = {
             hdr.ipv4.dstAddr: lpm;
         }
         size = 1024;
+        default_action = NoAction();
     }
     apply {
         if (hdr.ipv4.isValid()) {
