--- before_pass
+++ after_pass
@@ -6,7 +6,7 @@ parser P(packet_in p, out H[2] h) {
     bit<32> x;
     H tmp;
     state start {
-        p.extract(tmp);
+        p.extract<H>(tmp);
         transition select(tmp.field) {
             32w0: n1;
             default: n2;
@@ -25,10 +25,10 @@ parser P(packet_in p, out H[2] h) {
         transition n4;
     }
     state n4 {
-        p.extract(h[x]);
+        p.extract<H>(h[x]);
         transition accept;
     }
 }
 parser Simple<T>(packet_in p, out T t);
 package top<T>(Simple<T> prs);
-top(P()) main;
+top<H[2]>(P()) main;
