--- before_pass
+++ after_pass
@@ -16,7 +16,7 @@ control my(out H[2] s) {
     bit<1> tmp_6;
     bit<1> tmp_8;
     bit<1> tmp_9;
-    @name("act") action act() {
+    @name("my.act") action act() {
         a_0 = 32w0;
         tmp = a_0;
         s[tmp].z = 1w1;
@@ -34,7 +34,7 @@ control my(out H[2] s) {
         tmp_9 = f(tmp_6, 1w1);
         s[tmp_5].z = tmp_6;
     }
-    @name("tbl_act") table tbl_act_0 {
+    @name("my.tbl_act") table tbl_act_0 {
         actions = {
             act();
         }
