--- before_pass
+++ after_pass
@@ -16,7 +16,7 @@ control my(out H[2] s) {
     bit<1> tmp_16;
     bit<1> tmp_17;
     bit<1> tmp_18;
-    @name("act") action act() {
+    @name("act") action act_0() {
         a = 32w0;
         tmp_7 = a;
         s[tmp_7].z = 1w1;
@@ -36,9 +36,9 @@ control my(out H[2] s) {
     }
     @name("tbl_act") table tbl_act {
         actions = {
-            act();
+            act_0();
         }
-        const default_action = act();
+        const default_action = act_0();
     }
     apply {
         tbl_act.apply();
