--- before_pass
+++ after_pass
@@ -18,53 +18,23 @@ control my(out H[2] s) {
     @name("act") action act_0() {
         bit<1> tmp_8;
         bit<1> tmp_9;
-        {
-            a_0 = 32w0;
-        }
-        {
-            tmp = a_0;
-        }
-        {
-            s[tmp].z = 1w1;
-        }
-        {
-            tmp_0 = a_0 + 32w1;
-        }
-        {
-            tmp_1 = tmp_0;
-        }
-        {
-            s[tmp_1].z = 1w0;
-        }
-        {
-            tmp_2 = a_0;
-        }
-        {
-            tmp_3 = s[tmp_2].z;
-        }
-        {
-            tmp_8 = f(tmp_3, 1w0);
-            tmp_4 = tmp_8;
-        }
-        {
-            s[tmp_2].z = tmp_3;
-        }
-        {
-            a_0 = (bit<32>)tmp_4;
-        }
-        {
-            tmp_5 = a_0;
-        }
-        {
-            tmp_6 = s[tmp_5].z;
-        }
-        {
-            tmp_9 = f(tmp_6, 1w1);
-            tmp_7 = tmp_9;
-        }
-        {
-            s[tmp_5].z = tmp_6;
-        }
+        a_0 = 32w0;
+        tmp = a_0;
+        s[tmp].z = 1w1;
+        tmp_0 = a_0 + 32w1;
+        tmp_1 = tmp_0;
+        s[tmp_1].z = 1w0;
+        tmp_2 = a_0;
+        tmp_3 = s[tmp_2].z;
+        tmp_8 = f(tmp_3, 1w0);
+        tmp_4 = tmp_8;
+        s[tmp_2].z = tmp_3;
+        a_0 = (bit<32>)tmp_4;
+        tmp_5 = a_0;
+        tmp_6 = s[tmp_5].z;
+        tmp_9 = f(tmp_6, 1w1);
+        tmp_7 = tmp_9;
+        s[tmp_5].z = tmp_6;
     }
     @name("tbl_act") table tbl_act_0 {
         actions = {
@@ -73,9 +43,7 @@ control my(out H[2] s) {
         const default_action = act_0();
     }
     apply {
-        {
-            tbl_act_0.apply();
-        }
+        tbl_act_0.apply();
     }
 }
 top(my()) main;
