--- before_pass
+++ after_pass
@@ -20,9 +20,9 @@ parser IngressParserImpl(packet_in pkt,
     }
 }
 control cIngress(inout headers_t hdr, inout metadata_t user_meta, in psa_ingress_input_metadata_t istd, inout psa_ingress_output_metadata_t ostd) {
-    Counter<bit<10>, bit<12>>(1024, PSA_CounterType_t.PACKETS) counter;
+    Counter<bit<10>, bit<12>>(32w1024, PSA_CounterType_t.PACKETS) counter;
     action execute() {
-        counter.count(256);
+        counter.count(12w256);
     }
     table tbl {
         actions = {
