--- before_pass
+++ after_pass
@@ -1,21 +1,21 @@
 control ctrl() {
     apply {
-        bit<32> a;
-        bit<32> b;
-        bit<32> c;
-        a = 32w0;
-        b = 32w1;
-        c = 32w2;
-        if (a == 32w0) {
-            b = 32w2;
+        bit<32> a_0;
+        bit<32> b_0;
+        bit<32> c_0;
+        a_0 = 32w0;
+        b_0 = 32w1;
+        c_0 = 32w2;
+        if (a_0 == 32w0) {
+            b_0 = 32w2;
             return;
-            c = 32w3;
+            c_0 = 32w3;
         } else {
-            b = 32w3;
+            b_0 = 32w3;
             return;
-            c = 32w4;
+            c_0 = 32w4;
         }
-        c = 32w5;
+        c_0 = 32w5;
     }
 }
 control noop();
