--- before_pass
+++ after_pass
@@ -4,18 +4,18 @@ control ctrl() {
     bit<32> c_0;
     apply {
         a_0 = 32w0;
-        b_0 = 32w1;
-        c_0 = 32w2;
+        ;
+        ;
         if (a_0 == 32w0) {
-            b_0 = 32w2;
+            ;
             return;
-            c_0 = 32w3;
+            ;
         } else {
-            b_0 = 32w3;
+            ;
             return;
-            c_0 = 32w4;
+            ;
         }
-        c_0 = 32w5;
+        ;
     }
 }
 control noop();
