--- before_pass
+++ after_pass
@@ -3,19 +3,35 @@ control ctrl() {
     bit<32> b_0;
     bit<32> c_0;
     apply {
-        a_0 = 32w0;
-        b_0 = 32w1;
-        c_0 = 32w2;
+        {
+            a_0 = 32w0;
+        }
+        {
+            b_0 = 32w1;
+        }
+        {
+            c_0 = 32w2;
+        }
         if (a_0 == 32w0) {
-            b_0 = 32w2;
+            {
+                b_0 = 32w2;
+            }
             return;
-            c_0 = 32w3;
+            {
+                c_0 = 32w3;
+            }
         } else {
-            b_0 = 32w3;
+            {
+                b_0 = 32w3;
+            }
             return;
-            c_0 = 32w4;
+            {
+                c_0 = 32w4;
+            }
+        }
+        {
+            c_0 = 32w5;
         }
-        c_0 = 32w5;
     }
 }
 control noop();
