--- before_pass
+++ after_pass
@@ -39,32 +39,32 @@ parser prs(packet_in p, out Headers_t he
     }
 }
 control pipe(inout Headers_t headers, out bool pass) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name("Reject") action Reject_0(IPv4Address add) {
+    @name("Reject") action Reject(IPv4Address add) {
         pass = false;
         headers.ipv4.srcAddr = add;
     }
-    @name("Check_src_ip") table Check_src_ip {
+    @name("Check_src_ip") table Check_src_ip_0 {
         key = {
             headers.ipv4.srcAddr: lpm @name("headers.ipv4.srcAddr") ;
         }
         actions = {
-            Reject_0();
-            NoAction_1();
+            Reject();
+            NoAction_0();
         }
         implementation = hash_table(32w1024);
-        const default_action = NoAction_1();
+        const default_action = NoAction_0();
     }
     apply {
-        bool hasReturned_0 = false;
+        bool hasReturned = false;
         pass = true;
         if (!headers.ipv4.isValid()) {
             pass = false;
-            hasReturned_0 = true;
+            hasReturned = true;
         }
-        if (!hasReturned_0) {
-            Check_src_ip.apply();
+        if (!hasReturned) {
+            Check_src_ip_0.apply();
         }
     }
 }
