--- before_pass
+++ after_pass
@@ -39,7 +39,9 @@ parser prs(packet_in p, out Headers_t he
     }
 }
 control pipe(inout Headers_t headers, out bool pass) {
-    @name("Reject") action Reject(IPv4Address add) {
+    @name(".NoAction") action NoAction_1() {
+    }
+    @name("Reject") action Reject_0(IPv4Address add) {
         pass = false;
         headers.ipv4.srcAddr = add;
     }
@@ -48,11 +50,11 @@ control pipe(inout Headers_t headers, ou
             headers.ipv4.srcAddr: lpm @name("headers.ipv4.srcAddr") ;
         }
         actions = {
-            Reject();
-            NoAction();
+            Reject_0();
+            NoAction_1();
         }
         implementation = hash_table(32w1024);
-        const default_action = NoAction();
+        const default_action = NoAction_1();
     }
     apply {
         bool hasReturned_0 = false;
