--- before_pass
+++ after_pass
@@ -39,30 +39,30 @@ parser prs(packet_in p, out Headers_t he
     }
 }
 control pipe(inout Headers_t headers, out bool pass) {
-    @name("Reject") action Reject_0(IPv4Address add) {
+    @name("Reject") action Reject(IPv4Address add) {
         pass = false;
         headers.ipv4.srcAddr = add;
     }
-    @name("Check_src_ip") table Check_src_ip_0 {
+    @name("Check_src_ip") table Check_src_ip {
         key = {
             headers.ipv4.srcAddr: lpm @name("headers.ipv4.srcAddr") ;
         }
         actions = {
-            Reject_0();
+            Reject();
             NoAction();
         }
         implementation = hash_table(32w1024);
         const default_action = NoAction();
     }
     apply {
-        bool hasReturned = false;
+        bool hasReturned_0 = false;
         pass = true;
         if (!headers.ipv4.isValid()) {
             pass = false;
-            hasReturned = true;
+            hasReturned_0 = true;
         }
-        if (!hasReturned) {
-            Check_src_ip_0.apply();
+        if (!hasReturned_0) {
+            Check_src_ip.apply();
         }
     }
 }
