--- before_pass
+++ after_pass
@@ -52,11 +52,13 @@ control ingress(inout headers hdr, inout
     }
     table acl_table {
         actions = {
-            assign_non_const_array_index;
+            assign_non_const_array_index();
+            @defaultonly NoAction();
         }
         key = {
             hdr.ethernet.etherType: exact;
         }
+        default_action = NoAction();
     }
     apply {
         acl_table.apply();
