--- before_pass
+++ after_pass
@@ -3,7 +3,7 @@
 typedef bit<48> EthernetAddress;
 typedef bit<32> IPv4Address;
 typedef bit<4> PortId;
-const PortId DROP_PORT = 0xf;
+const PortId DROP_PORT = (bit<4>)4w0xf;
 header ethernet_t {
     EthernetAddress dstAddr;
     EthernetAddress srcAddr;
@@ -119,7 +119,7 @@ control egress(inout headers hdr, inout
     }
 }
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    register<bit<16>>(65536) registerRound;
+    register<bit<16>>(32w65536) registerRound;
     action read_round() {
         registerRound.read(meta.local_metadata.round, hdr.myhdr.inst);
     }
