--- before_pass
+++ after_pass
@@ -97,10 +97,10 @@ control computeChecksum(inout headers hd
 control egress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("_drop") action _drop() {
+    @name("egress._drop") action _drop() {
         mark_to_drop(standard_metadata);
     }
-    @name("drop_tbl") table drop_tbl_0 {
+    @name("egress.drop_tbl") table drop_tbl_0 {
         key = {
             meta.local_metadata.set_drop: exact @name("meta.ingress_metadata.set_drop") ;
         }
@@ -116,11 +116,11 @@ control egress(inout headers hdr, inout
     }
 }
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    @name("registerRound") register<bit<16>>(32w65536) registerRound_0;
-    @name("read_round") action read_round() {
+    @name("ingress.registerRound") register<bit<16>>(32w65536) registerRound_0;
+    @name("ingress.read_round") action read_round() {
         registerRound_0.read(meta.local_metadata.round, hdr.myhdr.inst);
     }
-    @name("round_tbl") table round_tbl_0 {
+    @name("ingress.round_tbl") table round_tbl_0 {
         key = {
         }
         actions = {
