--- before_pass
+++ after_pass
@@ -108,8 +108,8 @@ control egress(inout headers hdr, inout
             meta.local_metadata.set_drop: exact;
         }
         actions = {
-            _drop;
-            NoAction;
+            _drop();
+            NoAction();
         }
         size = 2;
         default_action = NoAction();
@@ -127,10 +127,10 @@ control ingress(inout headers hdr, inout
         key = {
         }
         actions = {
-            read_round;
+            read_round();
         }
         size = 8;
-        default_action = read_round;
+        default_action = read_round();
     }
     apply {
         if (hdr.ipv4.isValid()) {
