--- before_pass
+++ after_pass
@@ -1,12 +1,12 @@
 control c() {
-    bit<32> x;
-    @name("a") action a_0(inout bit<32> arg_0) {
-        bool hasReturned_0 = false;
+    bit<32> x_0;
+    @name("a") action a(inout bit<32> arg_0) {
+        bool hasReturned = false;
         arg_0 = 32w1;
-        hasReturned_0 = true;
+        hasReturned = true;
     }
     apply {
-        a_0(x);
+        a(x_0);
     }
 }
 control proto();
