--- before_pass
+++ after_pass
@@ -43,10 +43,10 @@ parser parse(packet_in pk, out parsed_pa
 control ingress(inout parsed_packet_t h, inout local_metadata_t local_metadata, inout standard_metadata_t standard_metadata) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("do_act") action do_act() {
+    @name("ingress.do_act") action do_act() {
         h.bvh1.row.alt1.valid = 1w0;
     }
-    @name("tns") table tns_0 {
+    @name("ingress.tns") table tns_0 {
         key = {
             h.bvh1.row.alt1.valid                : exact @name("h.bvh1.row.alt1.valid") ;
             local_metadata.col.bvh.row.alt0.valid: exact @name("local_metadata.col.bvh.row.alt0.valid") ;
