--- before_pass
+++ after_pass
@@ -44,8 +44,8 @@ control ingress(inout parsed_packet_t h,
     tst_t s;
     bitvec_hdr bh;
     action do_act() {
-        h.bvh1.row.alt1.valid = (bit<1>)1w0;
-        local_metadata.col.bvh.row.alt0.valid = (bit<1>)1w0;
+        h.bvh1.row.alt1.valid = 1w0;
+        local_metadata.col.bvh.row.alt0.valid = 1w0;
     }
     table tns {
         key = {
@@ -62,12 +62,12 @@ control ingress(inout parsed_packet_t h,
         tns.apply();
         bh.row.alt0.valid = h.bvh0.row.alt0.valid;
         s.row0.alt0 = local_metadata.row1.alt1;
-        s.row1.alt0.valid = (bit<1>)1w1;
+        s.row1.alt0.valid = 1w1;
         s.row1.alt1.port = local_metadata.row0.alt1.port + 7w1;
-        s.col.bvh.row.alt0.valid = (bit<1>)1w0;
-        local_metadata.col.bvh.row.alt0.valid = (bit<1>)1w0;
+        s.col.bvh.row.alt0.valid = 1w0;
+        local_metadata.col.bvh.row.alt0.valid = 1w0;
         local_metadata.row0.alt0 = local_metadata.row1.alt1;
-        local_metadata.row1.alt0.valid = (bit<1>)1w1;
+        local_metadata.row1.alt0.valid = 1w1;
         local_metadata.row1.alt1.port = local_metadata.row0.alt1.port + 7w1;
         clone3<row_t>(CloneType.I2E, 32w0, local_metadata.row0);
     }
