--- before_pass
+++ after_pass
@@ -36,10 +36,18 @@ control Ing(inout Headers headers, inout
     bit<8> n_0;
     @name("debug") register<bit<8>>(32w2) debug_0;
     apply {
-        n_0 = 8w0b11111111;
-        n_0[7:4] = 4w0;
-        debug_0.write(32w1, n_0);
-        standard_meta.egress_spec = 9w0;
+        {
+            n_0 = 8w0b11111111;
+        }
+        {
+            n_0[7:4] = 4w0;
+        }
+        {
+            debug_0.write(32w1, n_0);
+        }
+        {
+            standard_meta.egress_spec = 9w0;
+        }
     }
 }
 control Eg(inout Headers hdrs, inout Metadata meta, inout standard_metadata_t standard_meta) {
@@ -48,8 +56,12 @@ control Eg(inout Headers hdrs, inout Met
 }
 control DP(packet_out b, in Headers p) {
     apply {
-        b.emit<Ethernet_h>(p.ethernet);
-        b.emit<Ipv4_h>(p.ip);
+        {
+            b.emit<Ethernet_h>(p.ethernet);
+        }
+        {
+            b.emit<Ipv4_h>(p.ip);
+        }
     }
 }
 control Verify(inout Headers hdrs, inout Metadata meta) {
