--- before_pass
+++ after_pass
@@ -37,8 +37,8 @@ control Ing(inout Headers headers, inout
     apply {
         bit<8> n = 8w0b11111111;
         n[7:4] = 4w0;
-        debug.write(1, n);
-        standard_meta.egress_spec = 0;
+        debug.write(32w1, n);
+        standard_meta.egress_spec = (bit<9>)9w0;
     }
 }
 control Eg(inout Headers hdrs, inout Metadata meta, inout standard_metadata_t standard_meta) {
