--- before_pass
+++ after_pass
@@ -33,12 +33,12 @@ parser P(packet_in b, out Headers p, ino
     }
 }
 control Ing(inout Headers headers, inout Metadata meta, inout standard_metadata_t standard_meta) {
-    bit<8> n;
-    @name("debug") register<bit<8>>(32w2) debug;
+    bit<8> n_0;
+    @name("debug") register<bit<8>>(32w2) debug_0;
     apply {
-        n = 8w0b11111111;
-        n[7:4] = 4w0;
-        debug.write(32w1, n);
+        n_0 = 8w0b11111111;
+        n_0[7:4] = 4w0;
+        debug_0.write(32w1, n_0);
         standard_meta.egress_spec = 9w0;
     }
 }
