--- before_pass
+++ after_pass
@@ -14,7 +14,7 @@ struct value_set_t {
     bit<32> field;
 }
 parser MyParser(packet_in b, out my_packet p, inout my_metadata m, inout standard_metadata_t s) {
-    @name("pvs") value_set<value_set_t>(4) pvs_0;
+    @name("MyParser.pvs") value_set<value_set_t>(4) pvs_0;
     state start {
         b.extract<data_h>(p.data);
         transition select(p.data.da) {
@@ -33,9 +33,9 @@ control MyVerifyChecksum(inout my_packet
 control MyIngress(inout my_packet p, inout my_metadata meta, inout standard_metadata_t s) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("set_data") action set_data() {
+    @name("MyIngress.set_data") action set_data() {
     }
-    @name("t") table t_0 {
+    @name("MyIngress.t") table t_0 {
         actions = {
             set_data();
             @defaultonly NoAction_0();
