--- before_pass
+++ after_pass
@@ -14,11 +14,11 @@ struct value_set_t {
     bit<32> field;
 }
 parser MyParser(packet_in b, out my_packet p, inout my_metadata m, inout standard_metadata_t s) {
-    @name("pvs") value_set<value_set_t>(4) pvs;
+    @name("pvs") value_set<value_set_t>(4) pvs_0;
     state start {
         b.extract<data_h>(p.data);
         transition select(p.data.da) {
-            pvs: accept;
+            pvs_0: accept;
             32w0x810: foo;
         }
     }
@@ -31,22 +31,22 @@ control MyVerifyChecksum(inout my_packet
     }
 }
 control MyIngress(inout my_packet p, inout my_metadata meta, inout standard_metadata_t s) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name("set_data") action set_data_0() {
+    @name("set_data") action set_data() {
     }
-    @name("t") table t {
+    @name("t") table t_0 {
         actions = {
-            set_data_0();
-            @defaultonly NoAction_1();
+            set_data();
+            @defaultonly NoAction_0();
         }
         key = {
             meta.data[0].da: exact @name("meta.data[0].da") ;
         }
-        default_action = NoAction_1();
+        default_action = NoAction_0();
     }
     apply {
-        t.apply();
+        t_0.apply();
     }
 }
 control MyEgress(inout my_packet p, inout my_metadata m, inout standard_metadata_t s) {
