--- before_pass
+++ after_pass
@@ -16,7 +16,7 @@ struct value_set_t {
 parser MyParser(packet_in b, out my_packet p, inout my_metadata m, inout standard_metadata_t s) {
     value_set<value_set_t>(4) pvs;
     state start {
-        b.extract(p.data);
+        b.extract<data_h>(p.data);
         transition select(p.data.da) {
             pvs: accept;
             32w0x810: foo;
@@ -59,4 +59,4 @@ control MyDeparser(packet_out b, in my_p
     apply {
     }
 }
-V1Switch(MyParser(), MyVerifyChecksum(), MyIngress(), MyEgress(), MyComputeChecksum(), MyDeparser()) main;
+V1Switch<my_packet, my_metadata>(MyParser(), MyVerifyChecksum(), MyIngress(), MyEgress(), MyComputeChecksum(), MyDeparser()) main;
