--- before_pass
+++ after_pass
@@ -1,6 +1,8 @@
 control proto(out bit<32> x);
 package top(proto _c);
 control c(out bit<32> x) {
+    bool w_0;
+    bit<8> z_0;
     apply {
         x = 32w8;
         x = 32w8;
@@ -29,7 +31,6 @@ control c(out bit<32> x) {
         x = 32w5;
         x = 32w5;
         x = 32w17;
-        bool w_0;
         w_0 = false;
         w_0 = false;
         w_0 = true;
@@ -46,7 +47,6 @@ control c(out bit<32> x) {
         w_0 = true;
         w_0 = true;
         w_0 = false;
-        bit<8> z_0;
         z_0 = 8w0;
         z_0 = 8w0;
         z_0 = 8w128;
