--- before_pass
+++ after_pass
@@ -19,11 +19,7 @@ control DeparserI(packet_out packet, in
     }
 }
 parser parserI(packet_in pkt, out Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
-    const bit<16> c0 = 16w0xdead;
-    bit<16> v0;
     state start {
-        const bit<16> c1 = 16w0xdead;
-        bit<16> v1;
         pkt.extract<Ethernet_h>(hdr.ethernet);
         transition accept;
     }
