--- before_pass
+++ after_pass
@@ -25,16 +25,18 @@ parser parserI(packet_in pkt, out Parsed
     }
 }
 control cIngress(inout Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
-    @name("foo") action foo() {
+    @name(".NoAction") action NoAction_1() {
+    }
+    @name("foo") action foo_0() {
     }
     @name("tbl1") table tbl1 {
         key = {
         }
         actions = {
-            foo();
-            NoAction();
+            foo_0();
+            NoAction_1();
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     apply {
         tbl1.apply();
