--- before_pass
+++ after_pass
@@ -15,16 +15,12 @@ struct mystruct1 {
 }
 control DeparserI(packet_out packet, in Parsed_packet hdr) {
     apply {
-        {
-            packet.emit<Ethernet_h>(hdr.ethernet);
-        }
+        packet.emit<Ethernet_h>(hdr.ethernet);
     }
 }
 parser parserI(packet_in pkt, out Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
     state start {
-        {
-            pkt.extract<Ethernet_h>(hdr.ethernet);
-        }
+        pkt.extract<Ethernet_h>(hdr.ethernet);
         transition accept;
     }
 }
@@ -41,9 +37,7 @@ control cIngress(inout Parsed_packet hdr
         default_action = NoAction();
     }
     apply {
-        {
-            tbl1_0.apply();
-        }
+        tbl1_0.apply();
     }
 }
 control cEgress(inout Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
