--- before_pass
+++ after_pass
@@ -15,7 +15,7 @@ struct mystruct1 {
 }
 control DeparserI(packet_out packet, in Parsed_packet hdr) {
     apply {
-        packet.emit(hdr.ethernet);
+        packet.emit<Ethernet_h>(hdr.ethernet);
     }
 }
 parser parserI(packet_in pkt, out Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
@@ -25,7 +25,7 @@ parser parserI(packet_in pkt, out Parsed
         const bit<16> c1 = 16w0xdead;
         bit<16> v1;
         {
-            pkt.extract(hdr.ethernet);
+            pkt.extract<Ethernet_h>(hdr.ethernet);
         }
         transition accept;
     }
