--- before_pass
+++ after_pass
@@ -6,7 +6,9 @@ control Ing() {
     bool b_0;
     bit<32> a_0;
     @name("cond") action cond_0() {
-        b_0 = true;
+        {
+            b_0 = true;
+        }
         if (b_0) {
             a_0 = 32w5;
         } else {
@@ -14,7 +16,9 @@ control Ing() {
         }
     }
     apply {
-        cond_0();
+        {
+            cond_0();
+        }
     }
 }
 top(Ing()) main;
