--- before_pass
+++ after_pass
@@ -39,16 +39,16 @@ parser prs(packet_in p, out Headers_t he
     }
 }
 control pipe(inout Headers_t headers, out bool pass) {
-    @name("Reject") action Reject_0(IPv4Address add) {
+    @name("Reject") action Reject(IPv4Address add) {
         pass = false;
         headers.ipv4.srcAddr = add;
     }
-    @name("Check_src_ip") table Check_src_ip_0 {
+    @name("Check_src_ip") table Check_src_ip {
         key = {
             headers.ipv4.srcAddr: exact @name("headers.ipv4.srcAddr") ;
         }
         actions = {
-            Reject_0();
+            Reject();
             NoAction();
         }
         implementation = hash_table(32w1024);
@@ -56,8 +56,8 @@ control pipe(inout Headers_t headers, ou
     }
     apply {
         pass = true;
-        switch (Check_src_ip_0.apply().action_run) {
-            Reject_0: {
+        switch (Check_src_ip.apply().action_run) {
+            Reject: {
                 pass = false;
             }
             NoAction: {
