--- before_pass
+++ after_pass
@@ -3,13 +3,17 @@ parser Parser();
 package Package(Parser p1, Parser p2);
 parser Parser1()(Parser p) {
     state start {
-        p.apply();
+        {
+            p.apply();
+        }
         transition accept;
     }
 }
 parser Parser2()(Parser p) {
     state start {
-        p.apply();
+        {
+            p.apply();
+        }
         transition accept;
     }
 }
