--- before_pass
+++ after_pass
@@ -14,44 +14,9 @@ control DeparserImpl(packet_out packet,
     }
 }
 control ingress(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t standard_metadata) {
-    standard_metadata_t standard_metadata_1;
     @name(".send_to_cpu") action send_to_cpu() {
         {
-            standard_metadata_1.ingress_port = standard_metadata.ingress_port;
-            standard_metadata_1.egress_spec = standard_metadata.egress_spec;
-            standard_metadata_1.egress_port = standard_metadata.egress_port;
-            standard_metadata_1.instance_type = standard_metadata.instance_type;
-            standard_metadata_1.packet_length = standard_metadata.packet_length;
-            standard_metadata_1.enq_timestamp = standard_metadata.enq_timestamp;
-            standard_metadata_1.enq_qdepth = standard_metadata.enq_qdepth;
-            standard_metadata_1.deq_timedelta = standard_metadata.deq_timedelta;
-            standard_metadata_1.deq_qdepth = standard_metadata.deq_qdepth;
-            standard_metadata_1.ingress_global_timestamp = standard_metadata.ingress_global_timestamp;
-            standard_metadata_1.egress_global_timestamp = standard_metadata.egress_global_timestamp;
-            standard_metadata_1.mcast_grp = standard_metadata.mcast_grp;
-            standard_metadata_1.egress_rid = standard_metadata.egress_rid;
-            standard_metadata_1.checksum_error = standard_metadata.checksum_error;
-            standard_metadata_1.parser_error = standard_metadata.parser_error;
-            standard_metadata_1.priority = standard_metadata.priority;
-        }
-        standard_metadata_1.egress_spec = 9w64;
-        {
-            standard_metadata.ingress_port = standard_metadata_1.ingress_port;
-            standard_metadata.egress_spec = standard_metadata_1.egress_spec;
-            standard_metadata.egress_port = standard_metadata_1.egress_port;
-            standard_metadata.instance_type = standard_metadata_1.instance_type;
-            standard_metadata.packet_length = standard_metadata_1.packet_length;
-            standard_metadata.enq_timestamp = standard_metadata_1.enq_timestamp;
-            standard_metadata.enq_qdepth = standard_metadata_1.enq_qdepth;
-            standard_metadata.deq_timedelta = standard_metadata_1.deq_timedelta;
-            standard_metadata.deq_qdepth = standard_metadata_1.deq_qdepth;
-            standard_metadata.ingress_global_timestamp = standard_metadata_1.ingress_global_timestamp;
-            standard_metadata.egress_global_timestamp = standard_metadata_1.egress_global_timestamp;
-            standard_metadata.mcast_grp = standard_metadata_1.mcast_grp;
-            standard_metadata.egress_rid = standard_metadata_1.egress_rid;
-            standard_metadata.checksum_error = standard_metadata_1.checksum_error;
-            standard_metadata.parser_error = standard_metadata_1.parser_error;
-            standard_metadata.priority = standard_metadata_1.priority;
+            standard_metadata.egress_spec = 9w64;
         }
     }
     @name(".NoAction") action NoAction_0() {
