--- before_pass
+++ after_pass
@@ -19,7 +19,7 @@ action send_to_cpu(inout standard_metada
 control ingress(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t standard_metadata) {
     table t0 {
         key = {
-            standard_metadata.ingress_port: ternary;
+            standard_metadata.ingress_port: ternary @name("standard_metadata.ingress_port") ;
         }
         actions = {
             send_to_cpu(standard_metadata);
