--- before_pass
+++ after_pass
@@ -14,9 +14,7 @@ control DeparserImpl(packet_out packet,
     }
 }
 action send_to_cpu(inout standard_metadata_t standard_metadata) {
-    {
-        standard_metadata.egress_spec = 9w64;
-    }
+    standard_metadata.egress_spec = 9w64;
 }
 control ingress(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t standard_metadata) {
     @name("t0") table t0_0 {
@@ -30,9 +28,7 @@ control ingress(inout headers_t hdr, ino
         default_action = NoAction();
     }
     apply {
-        {
-            t0_0.apply();
-        }
+        t0_0.apply();
     }
 }
 control egress(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t standard_metadata) {
