--- before_pass
+++ after_pass
@@ -14,23 +14,23 @@ control DeparserImpl(packet_out packet,
     }
 }
 control ingress(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t standard_metadata) {
-    @name(".send_to_cpu") action send_to_cpu_1(inout standard_metadata_t standard_metadata_0) {
+    @name(".send_to_cpu") action send_to_cpu(inout standard_metadata_t standard_metadata_0) {
         standard_metadata_0.egress_spec = 9w64;
     }
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name("t0") table t0 {
+    @name("t0") table t0_0 {
         key = {
             standard_metadata.ingress_port: ternary @name("standard_metadata.ingress_port") ;
         }
         actions = {
-            send_to_cpu_1(standard_metadata);
-            @defaultonly NoAction_1();
+            send_to_cpu(standard_metadata);
+            @defaultonly NoAction_0();
         }
-        default_action = NoAction_1();
+        default_action = NoAction_0();
     }
     apply {
-        t0.apply();
+        t0_0.apply();
     }
 }
 control egress(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t standard_metadata) {
