--- before_pass
+++ after_pass
@@ -52,10 +52,10 @@ control ingress(inout headers hdr, inout
     apply {
         debug_hdr.apply();
         if (hdr.u[0].short.isValid()) {
-            hdr.u[0].short.data = (bit<16>)16w0xffff;
+            hdr.u[0].short.data = 16w0xffff;
         }
         if (hdr.u[0].byte.isValid()) {
-            hdr.u[0].byte.data = (bit<8>)8w0xff;
+            hdr.u[0].byte.data = 8w0xff;
         }
     }
 }
