--- before_pass
+++ after_pass
@@ -9,12 +9,12 @@ parser prs(packet_in p, out Headers_t he
 }
 control pipe(inout Headers_t headers, out bool pass) {
     bool x;
-    @name("Reject") action Reject(bool rej_0) {
+    @name("Reject") action Reject_0(bool rej_0) {
         pass = rej_0;
     }
     apply {
         x = true;
-        Reject(x);
+        Reject_0(x);
     }
 }
 ebpfFilter<Headers_t>(prs(), pipe()) main;
