--- before_pass
+++ after_pass
@@ -8,13 +8,13 @@ parser prs(packet_in p, out Headers_t he
     }
 }
 control pipe(inout Headers_t headers, out bool pass) {
-    bool x_0;
-    @name("Reject") action Reject_0(bool rej_0) {
+    bool x;
+    @name("Reject") action Reject(bool rej_0) {
         pass = rej_0;
     }
     apply {
-        x_0 = true;
-        Reject_0(x_0);
+        x = true;
+        Reject(x);
     }
 }
 ebpfFilter<Headers_t>(prs(), pipe()) main;
