--- before_pass
+++ after_pass
@@ -102,24 +102,20 @@ control ingress(inout headers_t hdr, ino
     }
     apply {
         if (standard_metadata.instance_type == 32w6) {
-            {
-                byte0_0 = 8w10;
-                byte1_0 = 8w252;
-                byte2_0 = 8w129;
-                byte3_0 = 8w2;
-                ipv4_address_0 = byte0_0 ++ byte1_0 ++ byte2_0 ++ byte3_0;
-                hdr.ipv4.srcAddr = ipv4_address_0;
-            }
+            byte0_0 = 8w10;
+            byte1_0 = 8w252;
+            byte2_0 = 8w129;
+            byte3_0 = 8w2;
+            ipv4_address_0 = byte0_0 ++ byte1_0 ++ byte2_0 ++ byte3_0;
+            hdr.ipv4.srcAddr = ipv4_address_0;
             meta.fwd.l2ptr = 32w0xe50b;
         } else if (standard_metadata.instance_type == 32w4) {
-            {
-                byte0_0 = 8w10;
-                byte1_0 = 8w199;
-                byte2_0 = 8w86;
-                byte3_0 = 8w99;
-                ipv4_address_0 = byte0_0 ++ byte1_0 ++ byte2_0 ++ byte3_0;
-                hdr.ipv4.srcAddr = ipv4_address_0;
-            }
+            byte0_0 = 8w10;
+            byte1_0 = 8w199;
+            byte2_0 = 8w86;
+            byte3_0 = 8w99;
+            ipv4_address_0 = byte0_0 ++ byte1_0 ++ byte2_0 ++ byte3_0;
+            hdr.ipv4.srcAddr = ipv4_address_0;
             meta.fwd.l2ptr = 32w0xec1c;
         } else {
             ipv4_da_lpm_0.apply();
