--- before_pass
+++ after_pass
@@ -89,7 +89,7 @@ control ingress(inout headers_t hdr, ino
     }
     table ipv4_da_lpm {
         key = {
-            hdr.ipv4.dstAddr: lpm;
+            hdr.ipv4.dstAddr: lpm @name("hdr.ipv4.dstAddr") ;
         }
         actions = {
             set_l2ptr();
@@ -108,7 +108,7 @@ control ingress(inout headers_t hdr, ino
     }
     table mac_da {
         key = {
-            meta.fwd.l2ptr: exact;
+            meta.fwd.l2ptr: exact @name("meta.fwd.l2ptr") ;
         }
         actions = {
             set_bd_dmac_intf();
@@ -137,8 +137,8 @@ control egress(inout headers_t hdr, inou
     }
     table get_multicast_copy_out_bd {
         key = {
-            standard_metadata.mcast_grp : exact;
-            standard_metadata.egress_rid: exact;
+            standard_metadata.mcast_grp : exact @name("standard_metadata.mcast_grp") ;
+            standard_metadata.egress_rid: exact @name("standard_metadata.egress_rid") ;
         }
         actions = {
             set_out_bd();
@@ -159,7 +159,7 @@ control egress(inout headers_t hdr, inou
     }
     table send_frame {
         key = {
-            meta.fwd.out_bd: exact;
+            meta.fwd.out_bd: exact @name("meta.fwd.out_bd") ;
         }
         actions = {
             rewrite_mac();
