--- before_pass
+++ after_pass
@@ -172,12 +172,12 @@ control egress(inout headers_t hdr, inou
     apply {
         if (standard_metadata.instance_type == 32w1) {
             hdr.switch_to_cpu.setValid();
-            hdr.switch_to_cpu.word0 = (bit<32>)32w0x12e012e;
-            hdr.switch_to_cpu.word1 = (bit<32>)32w0x5a5a5a5a;
+            hdr.switch_to_cpu.word0 = 32w0x12e012e;
+            hdr.switch_to_cpu.word1 = 32w0x5a5a5a5a;
         } else if (standard_metadata.instance_type == 32w2) {
             hdr.switch_to_cpu.setValid();
-            hdr.switch_to_cpu.word0 = (bit<32>)32w0xe2e0e2e;
-            hdr.switch_to_cpu.word1 = (bit<32>)32w0x5a5a5a5a;
+            hdr.switch_to_cpu.word0 = 32w0xe2e0e2e;
+            hdr.switch_to_cpu.word1 = 32w0x5a5a5a5a;
         } else {
             if (standard_metadata.instance_type == 32w5) {
                 get_multicast_copy_out_bd.apply();
