--- before_pass
+++ after_pass
@@ -44,7 +44,7 @@ parser P(packet_in b, out Headers p, ino
 }
 control Ing(inout Headers headers, inout Metadata meta, inout standard_metadata_t standard_meta) {
     apply {
-        standard_meta.egress_spec = (bit<9>)9w0;
+        standard_meta.egress_spec = 9w0;
     }
 }
 control Eg(inout Headers hdrs, inout Metadata meta, inout standard_metadata_t standard_meta) {
