--- before_pass
+++ after_pass
@@ -6,7 +6,7 @@ header hdr {
 control compute(inout hdr h) {
     apply {
         hdr tmp;
-        tmp.f = h.f + 1;
+        tmp.f = h.f + 32w1;
         h.f = tmp.f;
     }
 }
@@ -42,7 +42,7 @@ control ingress(inout Headers h, inout M
     compute() c;
     apply {
         c.apply(h.h);
-        sm.egress_spec = 0;
+        sm.egress_spec = (bit<9>)9w0;
     }
 }
 V1Switch(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
