--- before_pass
+++ after_pass
@@ -77,14 +77,14 @@ control MyIngress(inout headers hdr, ino
     }
     @name(".NoAction") action NoAction_1() {
     }
-    @name("set_dmac") action set_dmac(macAddr_t dstAddr) {
+    @name("MyIngress.set_dmac") action set_dmac(macAddr_t dstAddr) {
         hdr.ethernet.dstAddr = dstAddr;
     }
-    @name("drop") action drop() {
+    @name("MyIngress.drop") action drop() {
     }
-    @name("drop") action drop_2() {
+    @name("MyIngress.drop") action drop_2() {
     }
-    @name("forward") table forward_0 {
+    @name("MyIngress.forward") table forward_0 {
         key = {
             hdr.ipv4.dstAddr: exact @name("hdr.ipv4.dstAddr") ;
         }
@@ -96,11 +96,11 @@ control MyIngress(inout headers hdr, ino
         size = 1024;
         default_action = NoAction_0();
     }
-    @name("set_nhop") action set_nhop(ip4Addr_t dstAddr, egressSpec_t port) {
+    @name("MyIngress.set_nhop") action set_nhop(ip4Addr_t dstAddr, egressSpec_t port) {
         hdr.ipv4.dstAddr = dstAddr;
         standard_metadata.egress_spec = port;
     }
-    @name("ipv4_lpm") table ipv4_lpm_0 {
+    @name("MyIngress.ipv4_lpm") table ipv4_lpm_0 {
         key = {
             hdr.ipv4.dstAddr: lpm @name("hdr.ipv4.dstAddr") ;
         }
@@ -112,7 +112,7 @@ control MyIngress(inout headers hdr, ino
         size = 1024;
         default_action = NoAction_1();
     }
-    @name("send_digest") action send_digest() {
+    @name("MyIngress.send_digest") action send_digest() {
         meta.test_digest.in_mac_srcAddr = hdr.ethernet.srcAddr;
         meta.test_digest.my_parser_error = error.PacketTooShort;
         meta.test_digest.pkt_type = MyPacketTypes.IPv4;
@@ -132,11 +132,11 @@ control MyIngress(inout headers hdr, ino
 control MyEgress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     @name(".NoAction") action NoAction_5() {
     }
-    @name("rewrite_mac") action rewrite_mac(macAddr_t srcAddr) {
+    @name("MyEgress.rewrite_mac") action rewrite_mac(macAddr_t srcAddr) {
         hdr.ethernet.srcAddr = srcAddr;
         hdr.ipv4.ttl = hdr.ipv4.ttl + 8w255;
     }
-    @name("send_frame") table send_frame_0 {
+    @name("MyEgress.send_frame") table send_frame_0 {
         key = {
             standard_metadata.egress_port: exact @name("standard_metadata.egress_port") ;
         }
