--- before_pass
+++ after_pass
@@ -88,7 +88,7 @@ control MyIngress(inout headers hdr, ino
     }
     table forward {
         key = {
-            hdr.ipv4.dstAddr: exact;
+            hdr.ipv4.dstAddr: exact @name("hdr.ipv4.dstAddr") ;
         }
         actions = {
             set_dmac();
@@ -104,7 +104,7 @@ control MyIngress(inout headers hdr, ino
     }
     table ipv4_lpm {
         key = {
-            hdr.ipv4.dstAddr: lpm;
+            hdr.ipv4.dstAddr: lpm @name("hdr.ipv4.dstAddr") ;
         }
         actions = {
             set_nhop();
@@ -138,7 +138,7 @@ control MyEgress(inout headers hdr, inou
     }
     table send_frame {
         key = {
-            standard_metadata.egress_port: exact;
+            standard_metadata.egress_port: exact @name("standard_metadata.egress_port") ;
         }
         actions = {
             rewrite_mac();
