--- before_pass
+++ after_pass
@@ -38,7 +38,7 @@ parser ParserImpl(packet_in packet, out
     }
 }
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    @name("debug_hdr") table debug_hdr_0 {
+    @name("debug_hdr") table debug_hdr {
         key = {
             hdr.base.t           : exact @name("hdr.base.t") ;
             hdr.u.short.isValid(): exact @name("hdr.u.short.$valid$") ;
@@ -50,7 +50,7 @@ control ingress(inout headers hdr, inout
         const default_action = NoAction();
     }
     apply {
-        debug_hdr_0.apply();
+        debug_hdr.apply();
         if (hdr.u.short.isValid()) {
             hdr.u.short.data = 16w0xffff;
         } else if (hdr.u.byte.isValid()) {
