--- before_pass
+++ after_pass
@@ -21,7 +21,7 @@ struct metadata {
 }
 parser ParserImpl(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     state start {
-        packet.extract(hdr.base);
+        packet.extract<S>(hdr.base);
         transition select(hdr.base.t) {
             8w0: parseO1;
             8w1: parseO2;
@@ -29,11 +29,11 @@ parser ParserImpl(packet_in packet, out
         }
     }
     state parseO1 {
-        packet.extract(hdr.u.byte);
+        packet.extract<O1>(hdr.u.byte);
         transition accept;
     }
     state parseO2 {
-        packet.extract(hdr.u.short);
+        packet.extract<O2>(hdr.u.short);
         transition accept;
     }
 }
@@ -64,7 +64,7 @@ control egress(inout headers hdr, inout
 }
 control DeparserImpl(packet_out packet, in headers hdr) {
     apply {
-        packet.emit(hdr);
+        packet.emit<headers>(hdr);
     }
 }
 control verifyChecksum(inout headers hdr, inout metadata meta) {
@@ -75,4 +75,4 @@ control computeChecksum(inout headers hd
     apply {
     }
 }
-V1Switch(ParserImpl(), verifyChecksum(), ingress(), egress(), computeChecksum(), DeparserImpl()) main;
+V1Switch<headers, metadata>(ParserImpl(), verifyChecksum(), ingress(), egress(), computeChecksum(), DeparserImpl()) main;
