--- before_pass
+++ after_pass
@@ -52,9 +52,9 @@ control ingress(inout headers hdr, inout
     apply {
         debug_hdr.apply();
         if (hdr.u.short.isValid()) {
-            hdr.u.short.data = (bit<16>)16w0xffff;
+            hdr.u.short.data = 16w0xffff;
         } else if (hdr.u.byte.isValid()) {
-            hdr.u.byte.data = (bit<8>)8w0xff;
+            hdr.u.byte.data = 8w0xff;
         }
     }
 }
