--- before_pass
+++ after_pass
@@ -13,7 +13,7 @@ action drop(inout standard_metadata_t sm
     mark_to_drop(smeta_0);
 }
 control IngressI(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
-    @name("forward") table forward_0 {
+    @name("forward") table forward {
         key = {
         }
         actions = {
@@ -22,7 +22,7 @@ control IngressI(inout H hdr, inout M me
         const default_action = drop(smeta);
     }
     apply {
-        forward_0.apply();
+        forward.apply();
     }
 }
 control EgressI(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
