--- before_pass
+++ after_pass
@@ -10,19 +10,19 @@ parser ParserI(packet_in pk, out H hdr,
     }
 }
 control IngressI(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
-    @name(".drop") action drop_1(inout standard_metadata_t smeta_0) {
+    @name(".drop") action drop(inout standard_metadata_t smeta_0) {
         mark_to_drop(smeta_0);
     }
-    @name("forward") table forward {
+    @name("forward") table forward_0 {
         key = {
         }
         actions = {
-            drop_1(smeta);
+            drop(smeta);
         }
-        const default_action = drop_1(smeta);
+        const default_action = drop(smeta);
     }
     apply {
-        forward.apply();
+        forward_0.apply();
     }
 }
 control EgressI(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
