--- before_pass
+++ after_pass
@@ -1,14 +1,14 @@
 typedef bit<48> mac_addr_t;
 typedef bit<128> ipv6_addr_t;
 typedef bit<9> port_t;
-const bit<16> TYPE_IPV6 = 0x86dd;
+const bit<16> TYPE_IPV6 = 16w0x86dd;
 const bit<9> TYPE_CPU = 9w192;
-const bit<8> PROTO_TCP = 6;
-const bit<8> PROTO_UDP = 17;
-const bit<8> PROTO_ICMP6 = 58;
-const bit<8> TCP_SEQ_LEN = 4;
-const bit<8> ICMP6_ECHO_REQUEST = 128;
-const bit<8> ICMP6_ECHO_REPLY = 129;
+const bit<8> PROTO_TCP = 8w6;
+const bit<8> PROTO_UDP = 8w17;
+const bit<8> PROTO_ICMP6 = 8w58;
+const bit<8> TCP_SEQ_LEN = 8w4;
+const bit<8> ICMP6_ECHO_REQUEST = 8w128;
+const bit<8> ICMP6_ECHO_REPLY = 8w129;
 header ethernet_t {
     mac_addr_t dstAddr;
     mac_addr_t srcAddr;
