--- before_pass
+++ after_pass
@@ -20,10 +20,10 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout EMPTY b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
-    Counter<bit<10>, bit<12>>(1024, PSA_CounterType_t.PACKETS) counter0;
-    Counter<bit<10>, bit<12>>(1024, PSA_CounterType_t.PACKETS) counter1;
+    Counter<bit<10>, bit<12>>(32w1024, PSA_CounterType_t.PACKETS) counter0;
+    Counter<bit<10>, bit<12>>(32w1024, PSA_CounterType_t.PACKETS) counter1;
     apply {
-        counter0.count(1024);
+        counter0.count(12w1024);
     }
 }
 control MyEC(inout EMPTY a, inout EMPTY b, in psa_egress_input_metadata_t c, inout psa_egress_output_metadata_t d) {
