--- before_pass
+++ after_pass
@@ -34,21 +34,18 @@ control cEgress(inout headers_t hdr, ino
     apply {
     }
 }
-control CommonDeparserImpl(packet_out packet, inout headers_t hdr) {
-    apply {
-        packet.emit<ethernet_t>(hdr.ethernet);
-    }
-}
 control IngressDeparserImpl(packet_out buffer, out empty_metadata_t clone_i2e_meta, out empty_metadata_t resubmit_meta, out empty_metadata_t normal_meta, inout headers_t hdr, in metadata_t meta, in psa_ingress_output_metadata_t istd) {
-    @name("cp") CommonDeparserImpl() cp_0;
     apply {
-        cp_0.apply(buffer, hdr);
+        {
+            buffer.emit<ethernet_t>(hdr.ethernet);
+        }
     }
 }
 control EgressDeparserImpl(packet_out buffer, out empty_metadata_t clone_e2e_meta, out empty_metadata_t recirculate_meta, inout headers_t hdr, in metadata_t meta, in psa_egress_output_metadata_t istd, in psa_egress_deparser_input_metadata_t edstd) {
-    @name("cp") CommonDeparserImpl() cp_1;
     apply {
-        cp_1.apply(buffer, hdr);
+        {
+            buffer.emit<ethernet_t>(hdr.ethernet);
+        }
     }
 }
 IngressPipeline<headers_t, metadata_t, empty_metadata_t, empty_metadata_t, empty_metadata_t, empty_metadata_t>(IngressParserImpl(), cIngress(), IngressDeparserImpl()) ip;
