--- before_pass
+++ after_pass
@@ -26,7 +26,7 @@ control MyIC(inout ethernet_t a, inout E
     }
     table tbl {
         key = {
-            a.srcAddr: exact;
+            a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
             NoAction();
@@ -36,7 +36,7 @@ control MyIC(inout ethernet_t a, inout E
     }
     table tbl2 {
         key = {
-            a.srcAddr: exact;
+            a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
             NoAction();
