--- before_pass
+++ after_pass
@@ -20,8 +20,12 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout EMPTY b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
+    @name(".NoAction") action NoAction_1() {
+    }
+    @name(".NoAction") action NoAction_2() {
+    }
     @name("meter0") DirectMeter(PSA_MeterType_t.PACKETS) meter0;
-    @name("execute_meter") action execute_meter() {
+    @name("execute_meter") action execute_meter_0() {
         meter0.execute();
     }
     @name("tbl") table tbl {
@@ -29,20 +33,20 @@ control MyIC(inout ethernet_t a, inout E
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
-            NoAction();
+            NoAction_1();
         }
         psa_direct_meter = meter0;
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     @name("tbl2") table tbl2 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
-            NoAction();
-            execute_meter();
+            NoAction_2();
+            execute_meter_0();
         }
-        default_action = NoAction();
+        default_action = NoAction_2();
     }
     apply {
         tbl.apply();
