--- before_pass
+++ after_pass
@@ -20,33 +20,33 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout EMPTY b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
-    DirectMeter(PSA_MeterType_t.PACKETS) meter0;
-    action execute_meter() {
-        meter0.execute();
+    @name("meter0") DirectMeter(PSA_MeterType_t.PACKETS) meter0_0;
+    @name("execute_meter") action execute_meter_0() {
+        meter0_0.execute();
     }
-    table tbl {
+    @name("tbl") table tbl_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
             NoAction();
         }
-        psa_direct_meter = meter0;
+        psa_direct_meter = meter0_0;
         default_action = NoAction();
     }
-    table tbl2 {
+    @name("tbl2") table tbl2_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
             NoAction();
-            execute_meter();
+            execute_meter_0();
         }
         default_action = NoAction();
     }
     apply {
-        tbl.apply();
-        tbl2.apply();
+        tbl_0.apply();
+        tbl2_0.apply();
     }
 }
 control MyEC(inout EMPTY a, inout EMPTY b, in psa_egress_input_metadata_t c, inout psa_egress_output_metadata_t d) {
