--- before_pass
+++ after_pass
@@ -94,31 +94,37 @@ control cIngress(inout headers hdr, inou
             hdr.tcp.dstPort: exact;
         }
         actions = {
-            foo1;
-            foo2;
+            foo1();
+            foo2();
+            @defaultonly NoAction();
         }
         size = 8;
+        default_action = NoAction();
     }
     table t1 {
         key = {
             hdr.tcp.dstPort: exact;
         }
         actions = {
-            foo1;
-            foo2;
+            foo1();
+            foo2();
+            @defaultonly NoAction();
         }
         size = 8;
+        default_action = NoAction();
     }
     table t2 {
         actions = {
-            foo1;
-            foo2;
+            foo1();
+            foo2();
+            @defaultonly NoAction();
         }
         key = {
             hdr.tcp.srcPort: exact;
             meta.hash1     : selector;
         }
         size = 16;
+        default_action = NoAction();
     }
     apply {
         t0.apply();
