--- before_pass
+++ after_pass
@@ -59,14 +59,14 @@ struct metadata {
 }
 parser parserI(packet_in pkt, out headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
     state start {
-        pkt.extract(hdr.ethernet);
+        pkt.extract<ethernet_t>(hdr.ethernet);
         transition select(hdr.ethernet.etherType) {
             16w0x800: parse_ipv4;
             default: accept;
         }
     }
     state parse_ipv4 {
-        pkt.extract(hdr.ipv4, (bit<32>)(9w8 * (9w4 * (bit<9>)(pkt.lookahead<IPv4_up_to_ihl_only_h>()).ihl - 9w20)));
+        pkt.extract<ipv4_t>(hdr.ipv4, (bit<32>)(9w8 * (9w4 * (bit<9>)(pkt.lookahead<IPv4_up_to_ihl_only_h>()).ihl - 9w20)));
         verify(hdr.ipv4.version == 4w4, error.IPv4IncorrectVersion);
         verify(hdr.ipv4.ihl >= 4w5, error.IPv4HeaderTooShort);
         transition select(hdr.ipv4.protocol) {
@@ -75,7 +75,7 @@ parser parserI(packet_in pkt, out header
         }
     }
     state parse_tcp {
-        pkt.extract(hdr.tcp);
+        pkt.extract<tcp_t>(hdr.tcp);
         transition accept;
     }
 }
@@ -147,9 +147,9 @@ control uc(inout headers hdr, inout meta
 }
 control DeparserI(packet_out packet, in headers hdr) {
     apply {
-        packet.emit(hdr.ethernet);
-        packet.emit(hdr.ipv4);
-        packet.emit(hdr.tcp);
+        packet.emit<ethernet_t>(hdr.ethernet);
+        packet.emit<ipv4_t>(hdr.ipv4);
+        packet.emit<tcp_t>(hdr.tcp);
     }
 }
 V1Switch<headers, metadata>(parserI(), vc(), cIngress(), cEgress(), uc(), DeparserI()) main;
