--- before_pass
+++ after_pass
@@ -47,33 +47,33 @@ control MyVerifyChecksum(inout headers h
     }
 }
 control MyIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name("drop") action drop_0() {
+    @name("drop") action drop() {
         mark_to_drop(standard_metadata);
     }
-    @name("ipv4_forward") action ipv4_forward_0(macAddr_t dstAddr, egressSpec_t port) {
+    @name("ipv4_forward") action ipv4_forward(macAddr_t dstAddr, egressSpec_t port) {
         hdr.ethernet.srcAddr = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = dstAddr;
         standard_metadata.egress_spec = port;
         hdr.ipv4.ttl = hdr.ipv4.ttl + 8w255;
         hdr.ipv4.hdrChecksum = 16w1;
     }
-    @name("ipv4_lpm") table ipv4_lpm {
+    @name("ipv4_lpm") table ipv4_lpm_0 {
         key = {
             hdr.ipv4.dstAddr: lpm @name("hdr.ipv4.dstAddr") ;
         }
         actions = {
-            ipv4_forward_0();
-            drop_0();
-            NoAction_1();
+            ipv4_forward();
+            drop();
+            NoAction_0();
         }
         size = 1024;
-        default_action = NoAction_1();
+        default_action = NoAction_0();
     }
     apply {
         if (hdr.ipv4.isValid()) {
-            ipv4_lpm.apply();
+            ipv4_lpm_0.apply();
         }
     }
 }
