--- before_pass
+++ after_pass
@@ -47,10 +47,10 @@ control ingress(inout Header_t h, inout
             h.h.l: lpm;
         }
         actions = {
-            a;
-            a_with_control_params;
+            a();
+            a_with_control_params();
         }
-        default_action = a;
+        default_action = a();
         const entries = {
                         0x11 &&& 0xf0 : a_with_control_params(11);
                         0x12 : a_with_control_params(12);
