--- before_pass
+++ after_pass
@@ -30,9 +30,6 @@ struct headers_t {
     ethernet_t ethernet;
     ipv4_t     ipv4;
 }
-action my_drop(inout standard_metadata_t smeta_0) {
-    mark_to_drop(smeta_0);
-}
 parser ParserImpl(packet_in packet, out headers_t hdr, inout meta_t meta, inout standard_metadata_t standard_metadata) {
     state start {
         packet.extract<ethernet_t>(hdr.ethernet);
@@ -47,7 +44,15 @@ parser ParserImpl(packet_in packet, out
     }
 }
 control ingress(inout headers_t hdr, inout meta_t meta, inout standard_metadata_t standard_metadata) {
-    @name("set_output") action set_output(bit<9> out_port) {
+    @name(".my_drop") action my_drop_1(inout standard_metadata_t smeta_0) {
+        mark_to_drop(smeta_0);
+    }
+    @name(".my_drop") action my_drop_2(inout standard_metadata_t smeta_0) {
+        mark_to_drop(smeta_0);
+    }
+    @name(".NoAction") action NoAction_1() {
+    }
+    @name("set_output") action set_output_0(bit<9> out_port) {
         standard_metadata.egress_spec = out_port;
     }
     @name("ipv4_da_lpm") table ipv4_da_lpm {
@@ -55,20 +60,20 @@ control ingress(inout headers_t hdr, ino
             hdr.ipv4.dstAddr: lpm @name("hdr.ipv4.dstAddr") ;
         }
         actions = {
-            set_output();
-            my_drop(standard_metadata);
+            set_output_0();
+            my_drop_1(standard_metadata);
         }
-        default_action = my_drop(standard_metadata);
+        default_action = my_drop_1(standard_metadata);
     }
     @name("ipv4_sa_filter") table ipv4_sa_filter {
         key = {
             hdr.ipv4.srcAddr: ternary @name("hdr.ipv4.srcAddr") ;
         }
         actions = {
-            my_drop(standard_metadata);
-            NoAction();
+            my_drop_2(standard_metadata);
+            NoAction_1();
         }
-        const default_action = NoAction();
+        const default_action = NoAction_1();
     }
     apply {
         if (hdr.ipv4.isValid()) {
