--- before_pass
+++ after_pass
@@ -52,10 +52,10 @@ control ingress(inout headers_t hdr, ino
     }
     @name(".NoAction") action NoAction_0() {
     }
-    @name("set_output") action set_output(bit<9> out_port) {
+    @name("ingress.set_output") action set_output(bit<9> out_port) {
         standard_metadata.egress_spec = out_port;
     }
-    @name("ipv4_da_lpm") table ipv4_da_lpm_0 {
+    @name("ingress.ipv4_da_lpm") table ipv4_da_lpm_0 {
         key = {
             hdr.ipv4.dstAddr: lpm @name("hdr.ipv4.dstAddr") ;
         }
@@ -65,7 +65,7 @@ control ingress(inout headers_t hdr, ino
         }
         default_action = my_drop(standard_metadata);
     }
-    @name("ipv4_sa_filter") table ipv4_sa_filter_0 {
+    @name("ingress.ipv4_sa_filter") table ipv4_sa_filter_0 {
         key = {
             hdr.ipv4.srcAddr: ternary @name("hdr.ipv4.srcAddr") ;
         }
