--- before_pass
+++ after_pass
@@ -318,8 +318,8 @@ control ingress(inout headers hdr, inout
     }
 }
 control IngressDeparserImpl(packet_out packet, inout headers hdr, in metadata meta, in psa_ingress_output_metadata_t istd, out psa_ingress_deparser_output_metadata_t ostd) {
+    clone_metadata_t clone_md_0;
     apply {
-        clone_metadata_t clone_md_0;
         clone_md_0.data.h1 = clone_1_t {data = 32w0};
         clone_md_0.type = 3w0;
         if (meta.custom_clone_id == 3w1) {
