--- before_pass
+++ after_pass
@@ -269,27 +269,27 @@ parser EgressParserImpl(packet_in buffer
     }
 }
 control egress(inout headers hdr, inout metadata user_meta, in psa_egress_input_metadata_t istd, inout psa_egress_output_metadata_t ostd) {
-    @name(".NoAction") action NoAction_2() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name("process_clone_h0") action process_clone_h0_0() {
+    @name("process_clone_h0") action process_clone_h0() {
         user_meta.fwd_metadata.outport = (bit<32>)user_meta.clone_0.data;
     }
-    @name("process_clone_h1") action process_clone_h1_0() {
+    @name("process_clone_h1") action process_clone_h1() {
         user_meta.fwd_metadata.outport = user_meta.clone_1.data;
     }
-    @name("t") table t {
+    @name("t") table t_0 {
         key = {
             user_meta.custom_clone_id: exact @name("user_meta.custom_clone_id") ;
         }
         actions = {
-            process_clone_h0_0();
-            process_clone_h1_0();
-            NoAction_2();
+            process_clone_h0();
+            process_clone_h1();
+            NoAction_0();
         }
-        default_action = NoAction_2();
+        default_action = NoAction_0();
     }
     apply {
-        t.apply();
+        t_0.apply();
     }
 }
 parser IngressParserImpl(packet_in buffer, out headers parsed_hdr, inout metadata user_meta, in psa_ingress_parser_input_metadata_t istd, out psa_parser_output_metadata_t ostd) {
@@ -314,35 +314,35 @@ parser IngressParserImpl(packet_in buffe
     }
 }
 control ingress(inout headers hdr, inout metadata user_meta, in psa_ingress_input_metadata_t istd, inout psa_ingress_output_metadata_t ostd) {
-    @name(".NoAction") action NoAction_3() {
+    @name(".NoAction") action NoAction_1() {
     }
-    @name("do_clone") action do_clone_0(PortId_t port) {
+    @name("do_clone") action do_clone(PortId_t port) {
         ostd.clone = true;
         ostd.clone_port = port;
         user_meta.custom_clone_id = 3w1;
     }
-    @name("t") table t_2 {
+    @name("t") table t_1 {
         key = {
             user_meta.fwd_metadata.outport: exact @name("user_meta.fwd_metadata.outport") ;
         }
         actions = {
-            do_clone_0();
-            NoAction_3();
+            do_clone();
+            NoAction_1();
         }
-        default_action = NoAction_3();
+        default_action = NoAction_1();
     }
     apply {
-        t_2.apply();
+        t_1.apply();
     }
 }
 control IngressDeparserImpl(packet_out packet, inout headers hdr, in metadata meta, in psa_ingress_output_metadata_t istd, out psa_ingress_deparser_output_metadata_t ostd) {
-    clone_metadata_t clone_md;
+    clone_metadata_t clone_md_0;
     apply {
-        clone_md.data.h1.setValid();
-        clone_md.data.h1 = clone_1_t {data = 32w0};
-        clone_md.type = 3w0;
+        clone_md_0.data.h1.setValid();
+        clone_md_0.data.h1 = clone_1_t {data = 32w0};
+        clone_md_0.type = 3w0;
         if (meta.custom_clone_id == 3w1) {
-            ostd.clone_metadata = clone_md;
+            ostd.clone_metadata = clone_md_0;
         }
         packet.emit<ethernet_t>(hdr.ethernet);
         packet.emit<ipv4_t>(hdr.ipv4);
