--- before_pass
+++ after_pass
@@ -220,9 +220,6 @@ struct headers {
 }
 parser CommonParser(packet_in buffer, out headers parsed_hdr, inout metadata user_meta) {
     state start {
-        transition parse_ethernet;
-    }
-    state parse_ethernet {
         buffer.extract<ethernet_t>(parsed_hdr.ethernet);
         transition select(parsed_hdr.ethernet.etherType) {
             16w0x800: parse_ipv4;
