--- before_pass
+++ after_pass
@@ -21,7 +21,7 @@ typedef bit<14> PacketLength_t;
 typedef bit<16> EgressInstance_t;
 typedef bit<48> Timestamp_t;
 typedef error ParserError_t;
-const PortId_t PORT_CPU = (bit<10>)10w255;
+const PortId_t PORT_CPU = 10w255;
 enum InstanceType_t {
     NORMAL,
     CLONE,
@@ -86,7 +86,7 @@ match_kind {
 }
 action send_to_port(inout psa_ingress_output_metadata_t meta, in PortId_t egress_port) {
     meta.drop = false;
-    meta.multicast_group = (bit<10>)10w0;
+    meta.multicast_group = 10w0;
     meta.egress_port = egress_port;
 }
 action multicast(inout psa_ingress_output_metadata_t meta, in MulticastGroup_t multicast_group) {
