--- before_pass
+++ after_pass
@@ -3,27 +3,27 @@ header h {
 }
 parser p() {
     state start {
-        h[4] stack;
-        stack[0].setInvalid();
-        stack[1].setInvalid();
-        stack[2].setInvalid();
-        stack[3].setInvalid();
-        stack[3].setValid();
-        h b = stack[3];
-        b = stack.last;
-        stack[2] = b;
-        b = stack.next;
+        h[4] stack_0;
+        stack_0[0].setInvalid();
+        stack_0[1].setInvalid();
+        stack_0[2].setInvalid();
+        stack_0[3].setInvalid();
+        stack_0[3].setValid();
+        h b_0 = stack_0[3];
+        b_0 = stack_0.last;
+        stack_0[2] = b_0;
+        b_0 = stack_0.next;
         transition accept;
     }
 }
 control c() {
     apply {
-        h[4] stack;
-        stack[3].setValid();
-        h b = stack[3];
-        stack[2] = b;
-        stack.push_front(2);
-        stack.pop_front(2);
+        h[4] stack_1;
+        stack_1[3].setValid();
+        h b_1 = stack_1[3];
+        stack_1[2] = b_1;
+        stack_1.push_front(2);
+        stack_1.pop_front(2);
     }
 }
 parser Simple();
