--- before_pass
+++ after_pass
@@ -37,7 +37,7 @@ control deparser(packet_out b, in Header
 }
 control ingress(inout Header_t h, inout Meta_t m, inout standard_metadata_t standard_meta) {
     action a() {
-        standard_meta.egress_spec = 0;
+        standard_meta.egress_spec = (bit<9>)9w0;
     }
     action a_with_control_params(bit<9> x) {
         standard_meta.egress_spec = x;
@@ -53,10 +53,10 @@ control ingress(inout Header_t h, inout
         }
         default_action = a();
         const entries = {
-                        (0x1, 0x1111 &&& 0xf) : a_with_control_params(1);
-                        (0x2, 0x1181) : a_with_control_params(2);
-                        (0x3, 0x1111 &&& 0xf000) : a_with_control_params(3);
-                        (0x4, default) : a_with_control_params(4);
+                        (8w0x1, 16w0x1111 &&& 16w0xf) : a_with_control_params(9w1);
+                        (8w0x2, 16w0x1181) : a_with_control_params(9w2);
+                        (8w0x3, 16w0x1111 &&& 16w0xf000) : a_with_control_params(9w3);
+                        (8w0x4, default) : a_with_control_params(9w4);
         }
     }
     apply {
