--- before_pass
+++ after_pass
@@ -19,10 +19,10 @@ control c(inout bit<16> p) {
                 ix_0.a = x.a;
                 ix_0.b = x.b;
             }
-            if (ix_0.a < ix_0.b) {
-                x.a = ix_0.a + 16w1;
+            if (x.a < x.b) {
+                x.a = x.a + 16w1;
             }
-            if (ix_0.a > ix_0.b) {
+            if (ix_0.a > x.b) {
                 x.a = ix_0.a + 16w65535;
             }
         }
