--- before_pass
+++ after_pass
@@ -8,25 +8,25 @@ extern Virtual {
     abstract void g(inout data ix);
 }
 control c(inout bit<16> p) {
-    bit<16> tmp_0;
-    @name("cntr") Virtual() cntr = {
+    bit<16> tmp;
+    @name("cntr") Virtual() cntr_0 = {
         bit<16> f(in bit<16> ix) {
             return ix + 16w1;
         }
         void g(inout data x) {
-            data ix_1;
-            ix_1 = x;
-            if (ix_1.a < ix_1.b) {
-                x.a = ix_1.a + 16w1;
+            data ix_0;
+            ix_0 = x;
+            if (ix_0.a < ix_0.b) {
+                x.a = ix_0.a + 16w1;
             }
-            if (ix_1.a > ix_1.b) {
-                x.a = ix_1.a + 16w65535;
+            if (ix_0.a > ix_0.b) {
+                x.a = ix_0.a + 16w65535;
             }
         }
     };
     apply {
-        tmp_0 = cntr.f(16w6);
-        p = tmp_0;
+        tmp = cntr_0.f(16w6);
+        p = tmp;
     }
 }
 control ctr(inout bit<16> x);
