--- before_pass
+++ after_pass
@@ -32,11 +32,11 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name("c") direct_counter(CounterType.packets) c_0;
-    @name("my_action") action my_action(bit<9> a) {
+    @name("ingress.c") direct_counter(CounterType.packets) c_0;
+    @name("ingress.my_action") action my_action(bit<9> a) {
         sm.egress_spec = a;
     }
-    @name("t") table t_0 {
+    @name("ingress.t") table t_0 {
         actions = {
             my_action();
         }
