--- before_pass
+++ after_pass
@@ -6,15 +6,9 @@ control c(out bool b) {
     Wide x_0;
     Narrow y_0;
     apply {
-        {
-            x_0 = (Wide)32w3;
-        }
-        {
-            y_0 = (Narrow)(Narrow_t)(Wide_t)x_0;
-        }
-        {
-            b = y_0 == (Narrow)9w10;
-        }
+        x_0 = (Wide)32w3;
+        y_0 = (Narrow)(Narrow_t)(Wide_t)x_0;
+        b = y_0 == (Narrow)9w10;
     }
 }
 control ctrl(out bool b);
