--- before_pass
+++ after_pass
@@ -15,19 +15,19 @@ struct meta_t {
     bit<8> f2;
 }
 control MyC(inout hdr_t hdr, inout meta_t meta, in intrinsic_metadata_t intr_md) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name("c2.a") table c2_a_0 {
+    @name("c2.a") table c2_a {
         key = {
             meta_t {f0 = 8w0,f1 = 8w0,f2 = 8w0}.f0: exact @name("meta.f0") ;
         }
         actions = {
-            NoAction_1();
+            NoAction_0();
         }
-        default_action = NoAction_1();
+        default_action = NoAction_0();
     }
     apply {
-        c2_a_0.apply();
+        c2_a.apply();
     }
 }
 P<hdr_t, meta_t>(MyC()) main;
