--- before_pass
+++ after_pass
@@ -12,15 +12,6 @@ header hdr {
     bit<32> f;
     row_t   row;
 }
-control compute(inout hdr h) {
-    hdr[1] tmp_0;
-    apply {
-        tmp_0[0].row.alt1.valid = 1w1;
-        tmp_0[0].f = h.f + 32w1;
-        h.f = tmp_0[0].f;
-        h.row.alt1.valid = tmp_0[0].row.alt1.valid;
-    }
-}
 struct Headers {
     hdr h;
 }
@@ -50,9 +41,14 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name("c") compute() c_0;
+    hdr[1] c_tmp;
     apply {
-        c_0.apply(h.h);
+        {
+            c_tmp[0].row.alt1.valid = 1w1;
+            c_tmp[0].f = h.h.f + 32w1;
+            h.h.f = c_tmp[0].f;
+            h.h.row.alt1.valid = c_tmp[0].row.alt1.valid;
+        }
         sm.egress_spec = 9w0;
     }
 }
