--- before_pass
+++ after_pass
@@ -14,12 +14,12 @@ header hdr {
 }
 control compute(inout hdr h) {
     apply {
-        hdr[1] tmp;
-        tmp[0].row.alt1.valid = 1w1;
-        tmp[0].f = h.f + 32w1;
-        h.f = tmp[0].f;
-        tmp[0].row.alt0.port = h.row.alt0.port + 7w1;
-        h.row.alt1.valid = tmp[0].row.alt1.valid;
+        hdr[1] tmp_0;
+        tmp_0[0].row.alt1.valid = 1w1;
+        tmp_0[0].f = h.f + 32w1;
+        h.f = tmp_0[0].f;
+        tmp_0[0].row.alt0.port = h.row.alt0.port + 7w1;
+        h.row.alt1.valid = tmp_0[0].row.alt1.valid;
     }
 }
 struct Headers {
@@ -51,9 +51,9 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    compute() c;
+    @name("c") compute() c_0;
     apply {
-        c.apply(h.h);
+        c_0.apply(h.h);
         sm.egress_spec = 9w0;
     }
 }
