--- before_pass
+++ after_pass
@@ -6,8 +6,8 @@ package top(p _p);
 control c() {
     apply {
         bit<16> var;
-        bit<32> hdr = (bit<32>)32w0;
-        hash<bit<16>, bit<16>, bit<32>, bit<16>>(var, HashAlgorithm.crc16, (Hash)0, hdr, (Hash)0xffff);
+        bit<32> hdr = 32w0;
+        hash<bit<16>, bit<16>, bit<32>, bit<16>>(var, HashAlgorithm.crc16, 16w0, hdr, 16w0xffff);
     }
 }
 top(c()) main;
