--- before_pass
+++ after_pass
@@ -12,14 +12,18 @@ struct headers_t {
 }
 parser ParserImpl(packet_in packet, out headers_t hdr, inout meta_t meta, inout standard_metadata_t standard_metadata) {
     state start {
-        packet.extract<ethernet_t>(hdr.ethernet);
+        {
+            packet.extract<ethernet_t>(hdr.ethernet);
+        }
         transition accept;
     }
 }
 control ingress(inout headers_t hdr, inout meta_t meta, inout standard_metadata_t standard_metadata) {
     bit<16> rand_int_0;
     apply {
-        random<bit<16>>(rand_int_0, 16w0, 16w49151);
+        {
+            random<bit<16>>(rand_int_0, 16w0, 16w49151);
+        }
         if (rand_int_0 < 16w32768) {
             mark_to_drop(standard_metadata);
         }
@@ -31,7 +35,9 @@ control egress(inout headers_t hdr, inou
 }
 control DeparserImpl(packet_out packet, in headers_t hdr) {
     apply {
-        packet.emit<ethernet_t>(hdr.ethernet);
+        {
+            packet.emit<ethernet_t>(hdr.ethernet);
+        }
     }
 }
 control verifyChecksum(inout headers_t hdr, inout meta_t meta) {
