--- before_pass
+++ after_pass
@@ -22,8 +22,8 @@ parser ParserImpl(packet_in packet, out
 control ingress(inout headers_t hdr, inout meta_t meta, inout standard_metadata_t standard_metadata) {
     apply {
         bit<16> rand_int;
-        random<bit<16>>(rand_int, 0, 49151);
-        if (rand_int < 32768) {
+        random<bit<16>>(rand_int, 16w0, 16w49151);
+        if (rand_int < 16w32768) {
             mark_to_drop(standard_metadata);
         }
     }
