--- before_pass
+++ after_pass
@@ -16,9 +16,9 @@ control IngressImpl(inout headers hdr, i
     bit<32> value_0;
     bit<32> tmp;
     bit<32> tmp_0;
-    @name("update_value") action update_value_0(out bit<32> value) {
+    @name("update_value") action update_value_0(out bit<32> value_1) {
         tmp = test_func();
-        value = tmp;
+        value_1 = tmp;
     }
     apply {
         tmp_0 = test_func();
