--- before_pass
+++ after_pass
@@ -13,7 +13,7 @@ control IngressImpl(inout headers hdr, i
     bit<32> value_2;
     bit<32> tmp_1;
     bit<32> tmp_2;
-    @name("update_value") action update_value(out bit<32> value_1) {
+    @name("update_value") action update_value_0(out bit<32> value_1) {
         {
             bool hasReturned_0 = false;
             bit<32> retval_0;
@@ -31,7 +31,7 @@ control IngressImpl(inout headers hdr, i
             retval_0 = 32w1;
             tmp_2 = retval_0;
         }
-        update_value(value_2);
+        update_value_0(value_2);
     }
 }
 control VerifyChecksumImpl(inout headers hdr, inout metadata meta) {
