--- before_pass
+++ after_pass
@@ -10,28 +10,28 @@ parser ParserImpl(packet_in packet, out
     }
 }
 control IngressImpl(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    bit<32> value_0;
-    bit<32> tmp;
-    bit<32> tmp_0;
-    @name("update_value") action update_value_0(out bit<32> value_1) {
+    bit<32> value_2;
+    bit<32> tmp_1;
+    bit<32> tmp_2;
+    @name("update_value") action update_value(out bit<32> value_1) {
         {
-            bool hasReturned = false;
-            bit<32> retval;
-            hasReturned = true;
-            retval = 32w1;
-            tmp = retval;
+            bool hasReturned_0 = false;
+            bit<32> retval_0;
+            hasReturned_0 = true;
+            retval_0 = 32w1;
+            tmp_1 = retval_0;
         }
-        value_1 = tmp;
+        value_1 = tmp_1;
     }
     apply {
         {
-            bool hasReturned = false;
-            bit<32> retval;
-            hasReturned = true;
-            retval = 32w1;
-            tmp_0 = retval;
+            bool hasReturned_0 = false;
+            bit<32> retval_0;
+            hasReturned_0 = true;
+            retval_0 = 32w1;
+            tmp_2 = retval_0;
         }
-        update_value_0(value_0);
+        update_value(value_2);
     }
 }
 control VerifyChecksumImpl(inout headers hdr, inout metadata meta) {
