--- before_pass
+++ after_pass
@@ -10,7 +10,13 @@ parser ParserImpl(packet_in packet, out
     }
 }
 bit<32> test_func() {
-    return 32w1;
+    bool hasReturned = false;
+    bit<32> retval;
+    {
+        hasReturned = true;
+        retval = 32w1;
+    }
+    return retval;
 }
 control IngressImpl(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     bit<32> value_0;
