--- before_pass
+++ after_pass
@@ -13,7 +13,7 @@ control IngressImpl(inout headers hdr, i
     bit<32> value_0;
     bit<32> tmp;
     bit<32> tmp_0;
-    @name("update_value") action update_value(out bit<32> value_1) {
+    @name("update_value") action update_value(out bit<32> value_2) {
         {
             bool hasReturned = false;
             bit<32> retval;
@@ -21,7 +21,7 @@ control IngressImpl(inout headers hdr, i
             retval = 32w1;
             tmp = retval;
         }
-        value_1 = tmp;
+        value_2 = tmp;
     }
     apply {
         {
