--- before_pass
+++ after_pass
@@ -14,12 +14,12 @@ bit<32> test_func() {
 }
 control IngressImpl(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     bit<32> value_0;
+    bit<32> tmp;
+    bit<32> tmp_0;
     @name("update_value") action update_value_0(out bit<32> value) {
-        bit<32> tmp;
         tmp = test_func();
         value = tmp;
     }
-    bit<32> tmp_0;
     apply {
         tmp_0 = test_func();
         value_0 = tmp_0;
