--- before_pass
+++ after_pass
@@ -13,13 +13,13 @@ bit<32> test_func() {
     return 32w1;
 }
 control IngressImpl(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    action update_value(out bit<32> value) {
+    @name("update_value") action update_value_0(out bit<32> value) {
         value = test_func();
     }
     apply {
-        bit<32> value;
-        value = test_func();
-        update_value(value);
+        bit<32> value_0;
+        value_0 = test_func();
+        update_value_0(value_0);
     }
 }
 control VerifyChecksumImpl(inout headers hdr, inout metadata meta) {
