--- before_pass
+++ after_pass
@@ -33,14 +33,14 @@ control MyDeparser(packet_out packet, in
     }
 }
 control MyComputeChecksum(inout headers hdr, inout metadata meta) {
-    h_t h_1;
-    bit<16> tmp_0;
-    @name("checksum") Checksum16() checksum;
+    h_t h_0;
+    bit<16> tmp;
+    @name("checksum") Checksum16() checksum_0;
     apply {
-        h_1.setValid();
-        h_1 = { hdr.h.src, hdr.h.dst, 16w0 };
-        tmp_0 = checksum.get<h_t>(h_1);
-        hdr.h.csum = tmp_0;
+        h_0.setValid();
+        h_0 = { hdr.h.src, hdr.h.dst, 16w0 };
+        tmp = checksum_0.get<h_t>(h_0);
+        hdr.h.csum = tmp;
     }
 }
 V1Switch<headers, metadata>(MyParser(), MyVerifyChecksum(), MyIngress(), MyEgress(), MyComputeChecksum(), MyDeparser()) main;
