--- before_pass
+++ after_pass
@@ -12,7 +12,9 @@ struct headers {
 }
 parser MyParser(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     state start {
-        packet.extract<h_t>(hdr.h);
+        {
+            packet.extract<h_t>(hdr.h);
+        }
         transition accept;
     }
 }
@@ -35,9 +37,15 @@ control MyDeparser(packet_out packet, in
 control MyComputeChecksum(inout headers hdr, inout metadata meta) {
     h_t h_0;
     @name("checksum") Checksum16() checksum_0;
+    bit<16> tmp;
     apply {
-        h_0 = { hdr.h.src, hdr.h.dst, 16w0 };
-        hdr.h.csum = checksum_0.get<h_t>(h_0);
+        {
+            h_0 = { hdr.h.src, hdr.h.dst, 16w0 };
+        }
+        {
+            tmp = checksum_0.get<h_t>(h_0);
+            hdr.h.csum = tmp;
+        }
     }
 }
 V1Switch<headers, metadata>(MyParser(), MyVerifyChecksum(), MyIngress(), MyEgress(), MyComputeChecksum(), MyDeparser()) main;
