--- before_pass
+++ after_pass
@@ -11,7 +11,7 @@ struct metadata {
 parser MyParser(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     state start {
         packet.advance(32w8);
-        packet.extract(hdr.byte);
+        packet.extract<H>(hdr.byte);
         packet.advance((bit<32>)(hdr.byte.byte * 8w8));
         transition accept;
     }
@@ -36,4 +36,4 @@ control MyDeparser(packet_out packet, in
     apply {
     }
 }
-V1Switch(MyParser(), MyVerifyChecksum(), MyIngress(), MyEgress(), MyComputeChecksum(), MyDeparser()) main;
+V1Switch<headers, metadata>(MyParser(), MyVerifyChecksum(), MyIngress(), MyEgress(), MyComputeChecksum(), MyDeparser()) main;
