--- before_pass
+++ after_pass
@@ -10,9 +10,15 @@ struct metadata {
 }
 parser MyParser(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     state start {
-        packet.advance(32w8);
-        packet.extract<H>(hdr.byte);
-        packet.advance((bit<32>)(hdr.byte.byte << 3));
+        {
+            packet.advance(32w8);
+        }
+        {
+            packet.extract<H>(hdr.byte);
+        }
+        {
+            packet.advance((bit<32>)(hdr.byte.byte << 3));
+        }
         transition accept;
     }
 }
