--- before_pass
+++ after_pass
@@ -1,7 +1,7 @@
 #include <core.p4>
 control Ing(out bit<32> a) {
     bool b;
-    @name("cond") action cond() {
+    @name("cond") action cond_0() {
         if (b) {
             a = 32w5;
         } else {
@@ -10,7 +10,7 @@ control Ing(out bit<32> a) {
     }
     apply {
         b = true;
-        cond();
+        cond_0();
     }
 }
 control s(out bit<32> a);
