--- before_pass
+++ after_pass
@@ -11,7 +11,7 @@ struct Headers_t {
 }
 parser prs(packet_in p, out Headers_t headers) {
     state start {
-        p.extract(headers.ethernet);
+        p.extract<Ethernet_h>(headers.ethernet);
         transition select(headers.ethernet.etherType) {
             16w0x800: ip;
             default: reject;
@@ -21,4 +21,4 @@ parser prs(packet_in p, out Headers_t he
         transition accept;
     }
 }
-ebpfFilter(prs()) main;
+ebpfFilter<Headers_t>(prs()) main;
