--- before_pass
+++ after_pass
@@ -20,11 +20,11 @@ control MyVerifyChecksum(inout headers h
 control MyIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("c") jnf_counter(CounterType.packets) c_0;
-    @name("a") action a() {
+    @name("MyIngress.c") jnf_counter(CounterType.packets) c_0;
+    @name("MyIngress.a") action a() {
         c_0.count();
     }
-    @name("t") table t_0 {
+    @name("MyIngress.t") table t_0 {
         actions = {
             a();
             @defaultonly NoAction_0();
