--- before_pass
+++ after_pass
@@ -60,7 +60,7 @@ parser parserI(packet_in pkt, out header
         }
     }
     state parse_ipv4 {
-        pkt.extract<ipv4_t>(hdr.ipv4, (bit<32>)(9w8 * (bit<9>)(pkt.lookahead<IPv4_up_to_ihl_only_h>()).ihl));
+        pkt.extract<ipv4_t>(hdr.ipv4, (bit<32>)((bit<9>)(pkt.lookahead<IPv4_up_to_ihl_only_h>()).ihl << 3));
         transition select(hdr.ipv4.protocol) {
             8w6: parse_tcp;
             default: accept;
