--- before_pass
+++ after_pass
@@ -71,14 +71,14 @@ parser parserI(packet_in pkt, out header
     }
 }
 control cIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
-    bit<1> eth_valid;
-    bit<1> ipv4_valid;
-    bit<1> tcp_valid;
+    bit<1> eth_valid_0;
+    bit<1> ipv4_valid_0;
+    bit<1> tcp_valid_0;
     apply {
-        eth_valid = (bit<1>)hdr.ethernet.isValid();
-        ipv4_valid = (bit<1>)hdr.ipv4.isValid();
-        tcp_valid = (bit<1>)hdr.tcp.isValid();
-        hdr.ethernet.dstAddr = 31w0 ++ eth_valid ++ 7w0 ++ ipv4_valid ++ 7w0 ++ tcp_valid;
+        eth_valid_0 = (bit<1>)hdr.ethernet.isValid();
+        ipv4_valid_0 = (bit<1>)hdr.ipv4.isValid();
+        tcp_valid_0 = (bit<1>)hdr.tcp.isValid();
+        hdr.ethernet.dstAddr = 31w0 ++ eth_valid_0 ++ 7w0 ++ ipv4_valid_0 ++ 7w0 ++ tcp_valid_0;
     }
 }
 control cEgress(inout headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
