--- before_pass
+++ after_pass
@@ -13,13 +13,17 @@ control Aux(inout M meta) {
     @name("a") action a_0() {
     }
     apply {
-        a_0();
+        {
+            a_0();
+        }
     }
 }
 control IngressI(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
     @name("aux") Aux() aux_0;
     apply {
-        aux_0.apply(meta);
+        {
+            aux_0.apply(meta);
+        }
     }
 }
 control EgressI(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
