--- before_pass
+++ after_pass
@@ -37,7 +37,7 @@ control deparser(packet_out b, in Header
 }
 control ingress(inout Header_t h, inout Meta_t m, inout standard_metadata_t standard_meta) {
     action a() {
-        standard_meta.egress_spec = 0;
+        standard_meta.egress_spec = (bit<9>)9w0;
     }
     action a_with_control_params(bit<9> x) {
         standard_meta.egress_spec = x;
@@ -52,9 +52,9 @@ control ingress(inout Header_t h, inout
         }
         default_action = a();
         const entries = {
-                        0x1111 &&& 0xf : a_with_control_params(1)@priority(3) ;
-                        0x1181 : a_with_control_params(2);
-                        0x1181 &&& 0xf00f : a_with_control_params(3)@priority(1) ;
+                        16w0x1111 &&& 16w0xf : a_with_control_params(9w1)@priority(3) ;
+                        16w0x1181 : a_with_control_params(9w2);
+                        16w0x1181 &&& 16w0xf00f : a_with_control_params(9w3)@priority(1) ;
         }
     }
     apply {
