--- before_pass
+++ after_pass
@@ -36,29 +36,29 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Header_t h, inout Meta_t m, inout standard_metadata_t standard_meta) {
-    action a() {
+    @name("a") action a_0() {
         standard_meta.egress_spec = 9w0;
     }
-    action a_with_control_params(bit<9> x) {
+    @name("a_with_control_params") action a_with_control_params_0(bit<9> x) {
         standard_meta.egress_spec = x;
     }
-    table t_ternary {
+    @name("t_ternary") table t_ternary_0 {
         key = {
             h.h.t: ternary @name("h.h.t") ;
         }
         actions = {
-            a();
-            a_with_control_params();
+            a_0();
+            a_with_control_params_0();
         }
-        default_action = a();
+        default_action = a_0();
         const entries = {
-                        16w0x1111 &&& 16w0xf : a_with_control_params(9w1)@priority(3) ;
-                        16w0x1181 : a_with_control_params(9w2);
-                        16w0x1181 &&& 16w0xf00f : a_with_control_params(9w3)@priority(1) ;
+                        16w0x1111 &&& 16w0xf : a_with_control_params_0(9w1)@priority(3) ;
+                        16w0x1181 : a_with_control_params_0(9w2);
+                        16w0x1181 &&& 16w0xf00f : a_with_control_params_0(9w3)@priority(1) ;
         }
     }
     apply {
-        t_ternary.apply();
+        t_ternary_0.apply();
     }
 }
 V1Switch<Header_t, Meta_t>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
