--- before_pass
+++ after_pass
@@ -12,16 +12,10 @@ parser p1(packet_in p, out Header h) {
     }
     state chain1 {
         p.extract<Header>(h);
-        transition next1;
-    }
-    state next1 {
         transition endchain;
     }
     state chain2 {
         p.extract<Header>(h);
-        transition next2;
-    }
-    state next2 {
         transition endchain;
     }
     state endchain {
