--- before_pass
+++ after_pass
@@ -11,11 +11,15 @@ parser p1(packet_in p, out Header h) {
         }
     }
     state chain1 {
-        p.extract<Header>(h);
+        {
+            p.extract<Header>(h);
+        }
         transition endchain;
     }
     state chain2 {
-        p.extract<Header>(h);
+        {
+            p.extract<Header>(h);
+        }
         transition endchain;
     }
     state endchain {
