--- before_pass
+++ after_pass
@@ -15,12 +15,12 @@ struct metadata_t {
 }
 control DeparserI(packet_out packet, in Parsed_packet hdr) {
     apply {
-        packet.emit(hdr.ethernet);
+        packet.emit<Ethernet_h>(hdr.ethernet);
     }
 }
 parser parserI(packet_in pkt, out Parsed_packet hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
     state start {
-        pkt.extract(hdr.ethernet);
+        pkt.extract<Ethernet_h>(hdr.ethernet);
         transition accept;
     }
 }
@@ -81,4 +81,4 @@ control uc(inout Parsed_packet hdr, inou
     apply {
     }
 }
-V1Switch(parserI(), vc(), cIngress(), cEgress(), uc(), DeparserI()) main;
+V1Switch<Parsed_packet, metadata_t>(parserI(), vc(), cIngress(), cEgress(), uc(), DeparserI()) main;
