--- before_pass
+++ after_pass
@@ -27,19 +27,19 @@ parser parserI(packet_in pkt, out Parsed
 control cIngress(inout Parsed_packet hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("a1") action a1() {
+    @name("cIngress.a1") action a1() {
         hdr.ethernet.srcAddr = 48w1;
     }
-    @name("a2") action a2() {
+    @name("cIngress.a2") action a2() {
         hdr.ethernet.srcAddr[47:40] = 8w2;
     }
-    @name("a3") action a3() {
+    @name("cIngress.a3") action a3() {
         hdr.ethernet.srcAddr[47:40] = 8w3;
     }
-    @name("a4") action a4() {
+    @name("cIngress.a4") action a4() {
         hdr.ethernet.srcAddr[47:40] = 8w4;
     }
-    @name("t1") table t1_0 {
+    @name("cIngress.t1") table t1_0 {
         key = {
             hdr.ethernet.dstAddr: exact @name("hdr.ethernet.dstAddr") ;
         }
