--- before_pass
+++ after_pass
@@ -33,22 +33,22 @@ parser P(packet_in b, out Headers p, ino
     }
 }
 control Ing(inout Headers headers, inout Metadata meta, inout standard_metadata_t standard_meta) {
-    bit<8> n;
-    @name("debug") register<bit<8>>(32w2) debug;
-    action act() {
-        n = 8w0b11111111;
-        n[7:4] = 4w0;
-        debug.write(32w1, n);
+    bit<8> n_0;
+    @name("debug") register<bit<8>>(32w2) debug_0;
+    @name("act") action act_0() {
+        n_0 = 8w0b11111111;
+        n_0[7:4] = 4w0;
+        debug_0.write(32w1, n_0);
         standard_meta.egress_spec = 9w0;
     }
-    table tbl_act {
+    @name("tbl_act") table tbl_act_0 {
         actions = {
-            act();
+            act_0();
         }
-        const default_action = act();
+        const default_action = act_0();
     }
     apply {
-        tbl_act.apply();
+        tbl_act_0.apply();
     }
 }
 control Eg(inout Headers hdrs, inout Metadata meta, inout standard_metadata_t standard_meta) {
