--- before_pass
+++ after_pass
@@ -52,16 +52,12 @@ control egress(inout headers hdr, inout
 }
 control IngressDeparserImpl(packet_out buffer, out empty_t clone_i2e_meta, out empty_t resubmit_meta, out empty_t normal_meta, inout headers hdr, in metadata meta, in psa_ingress_output_metadata_t istd) {
     apply {
-        {
-            buffer.emit<ethernet_t>(hdr.ethernet);
-        }
+        buffer.emit<ethernet_t>(hdr.ethernet);
     }
 }
 control EgressDeparserImpl(packet_out buffer, out empty_t clone_e2e_meta, out empty_t recirculate_meta, inout headers hdr, in metadata meta, in psa_egress_output_metadata_t istd, in psa_egress_deparser_input_metadata_t edstd) {
     apply {
-        {
-            buffer.emit<ethernet_t>(hdr.ethernet);
-        }
+        buffer.emit<ethernet_t>(hdr.ethernet);
     }
 }
 IngressPipeline<headers, metadata, empty_t, empty_t, empty_t, empty_t>(IngressParserImpl(), ingress(), IngressDeparserImpl()) ip;
