--- before_pass
+++ after_pass
@@ -23,16 +23,16 @@ parser CommonParser(packet_in buffer, ou
     }
 }
 parser IngressParserImpl(packet_in buffer, out headers parsed_hdr, inout metadata user_meta, in psa_ingress_parser_input_metadata_t istd, in empty_t resubmit_meta, in empty_t recirculate_meta) {
-    CommonParser() p;
+    @name("p") CommonParser() p_0;
     state start {
-        p.apply(buffer, parsed_hdr, user_meta);
+        p_0.apply(buffer, parsed_hdr, user_meta);
         transition accept;
     }
 }
 parser EgressParserImpl(packet_in buffer, out headers parsed_hdr, inout metadata user_meta, in psa_egress_parser_input_metadata_t istd, in empty_t normal_meta, in empty_t clone_i2e_meta, in empty_t clone_e2e_meta) {
-    CommonParser() p;
+    @name("p") CommonParser() p_1;
     state start {
-        p.apply(buffer, parsed_hdr, user_meta);
+        p_1.apply(buffer, parsed_hdr, user_meta);
         transition accept;
     }
 }
@@ -50,15 +50,15 @@ control CommonDeparserImpl(packet_out pa
     }
 }
 control IngressDeparserImpl(packet_out buffer, out empty_t clone_i2e_meta, out empty_t resubmit_meta, out empty_t normal_meta, inout headers hdr, in metadata meta, in psa_ingress_output_metadata_t istd) {
-    CommonDeparserImpl() cp;
+    @name("cp") CommonDeparserImpl() cp_0;
     apply {
-        cp.apply(buffer, hdr);
+        cp_0.apply(buffer, hdr);
     }
 }
 control EgressDeparserImpl(packet_out buffer, out empty_t clone_e2e_meta, out empty_t recirculate_meta, inout headers hdr, in metadata meta, in psa_egress_output_metadata_t istd, in psa_egress_deparser_input_metadata_t edstd) {
-    CommonDeparserImpl() cp;
+    @name("cp") CommonDeparserImpl() cp_1;
     apply {
-        cp.apply(buffer, hdr);
+        cp_1.apply(buffer, hdr);
     }
 }
 IngressPipeline<headers, metadata, empty_t, empty_t, empty_t, empty_t>(IngressParserImpl(), ingress(), IngressDeparserImpl()) ip;
