--- before_pass
+++ after_pass
@@ -14,22 +14,22 @@ control Ing(inout Headers headers, inout
     }
 }
 control Eg(inout Headers hdrs, inout Metadata meta, inout standard_metadata_t standard_meta) {
-    bit<32> _sub;
-    bit<64> res;
-    bit<32> tmp_0;
-    @name("update") action update_0(in bool p_0, inout bit<64> val_0) {
-        _sub = val_0[31:0];
+    bit<32> _sub_0;
+    bit<64> res_0;
+    bit<32> tmp;
+    @name("update") action update(in bool p_0, inout bit<64> val_0) {
+        _sub_0 = val_0[31:0];
         if (p_0) {
-            tmp_0 = _sub;
+            tmp = _sub_0;
         } else {
-            tmp_0 = 32w1;
+            tmp = 32w1;
         }
-        _sub = tmp_0;
-        val_0[31:0] = _sub;
+        _sub_0 = tmp;
+        val_0[31:0] = _sub_0;
     }
     apply {
-        res = 64w0;
-        update_0(true, res);
+        res_0 = 64w0;
+        update(true, res_0);
     }
 }
 control DP(packet_out b, in Headers p) {
