--- before_pass
+++ after_pass
@@ -215,8 +215,8 @@ control egress(inout headers hdr, inout
     }
     table get_multicast_copy_out_bd {
         key = {
-            standard_metadata.mcast_grp : exact;
-            standard_metadata.egress_rid: exact;
+            standard_metadata.mcast_grp : exact @name("standard_metadata.mcast_grp") ;
+            standard_metadata.egress_rid: exact @name("standard_metadata.egress_rid") ;
         }
         actions = {
             set_out_bd();
@@ -232,7 +232,7 @@ control egress(inout headers hdr, inout
     }
     table send_frame {
         key = {
-            meta.fwd.out_bd: exact;
+            meta.fwd.out_bd: exact @name("meta.fwd.out_bd") ;
         }
         actions = {
             rewrite_mac();
