--- before_pass
+++ after_pass
@@ -186,9 +186,10 @@ control ingress(inout headers hdr, inout
         standard_metadata.mcast_grp = mcast_grp;
         meta.egress_port = port;
     }
+    bool key_0;
     @name("ingress.ipv6_tbl") table ipv6_tbl_0 {
         key = {
-            hdr.ipv6.dstAddr[127:120] == 8w0xff: exact @name("mcast_key") ;
+            key_0: exact @name("mcast_key") ;
         }
         actions = {
             set_mcast_grp();
@@ -198,6 +199,7 @@ control ingress(inout headers hdr, inout
     }
     apply {
         if (hdr.ipv6.isValid()) {
+            key_0 = hdr.ipv6.dstAddr[127:120] == 8w0xff;
             ipv6_tbl_0.apply();
         }
     }
