--- before_pass
+++ after_pass
@@ -240,13 +240,13 @@ parser MyParser(packet_in packet, out he
     }
 }
 control ingress(inout headers hdr, inout metadata_t meta, inout standard_metadata_t standard_metadata) {
+    bool key_0;
     @name(".NoAction") action NoAction_0() {
     }
     @name("ingress.set_mcast_grp") action set_mcast_grp(bit<16> mcast_grp, bit<9> port) {
         standard_metadata.mcast_grp = mcast_grp;
         meta._egress_port1 = port;
     }
-    bool key_0;
     @name("ingress.ipv6_tbl") table ipv6_tbl_0 {
         key = {
             key_0: exact @name("mcast_key") ;
