--- before_pass
+++ after_pass
@@ -20,7 +20,7 @@ control DeparserI(packet_out packet, in
 }
 control cBar(inout mystruct1 meta) {
     apply {
-        meta.a = meta.a - 1;
+        meta.a = meta.a - 4w1;
     }
 }
 parser parserI(packet_in pkt, out Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
@@ -34,7 +34,7 @@ parser parserI(packet_in pkt, out Parsed
 control cIngress(inout Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
     cBar() cbar_inst2;
     action foo() {
-        meta.b = meta.b + 5;
+        meta.b = meta.b + 4w5;
     }
     table guh {
         key = {
