--- before_pass
+++ after_pass
@@ -24,14 +24,18 @@ control D() {
 }
 control C()(E d) {
     apply {
-        d.apply();
+        {
+            d.apply();
+        }
     }
 }
 control MyIngress(inout my_packet p, inout my_metadata m, inout standard_metadata_t s) {
     @name("d") D() d_0;
     @name("c") C(d_0) c_0;
     apply {
-        c_0.apply();
+        {
+            c_0.apply();
+        }
     }
 }
 control MyEgress(inout my_packet p, inout my_metadata m, inout standard_metadata_t s) {
