--- before_pass
+++ after_pass
@@ -28,10 +28,10 @@ control C()(E d) {
     }
 }
 control MyIngress(inout my_packet p, inout my_metadata m, inout standard_metadata_t s) {
-    D() d;
-    C(d) c;
+    @name("d") D() d_0;
+    @name("c") C(d_0) c_0;
     apply {
-        c.apply();
+        c_0.apply();
     }
 }
 control MyEgress(inout my_packet p, inout my_metadata m, inout standard_metadata_t s) {
