--- before_pass
+++ after_pass
@@ -1,26 +1,26 @@
 control ctrl(out bit<32> c) {
-    action e() {
+    @name("e") action e_0() {
         exit;
     }
-    table t {
+    @name("t") table t_0 {
         actions = {
-            e();
+            e_0();
         }
-        default_action = e();
+        default_action = e_0();
     }
     apply {
-        bit<32> a;
-        bit<32> b;
-        a = 32w0;
-        b = 32w1;
+        bit<32> a_0;
+        bit<32> b_0;
+        a_0 = 32w0;
+        b_0 = 32w1;
         c = 32w2;
-        if (a == 32w0) {
-            b = 32w2;
-            t.apply();
+        if (a_0 == 32w0) {
+            b_0 = 32w2;
+            t_0.apply();
             c = 32w3;
         } else {
-            b = 32w3;
-            t.apply();
+            b_0 = 32w3;
+            t_0.apply();
             c = 32w4;
         }
         c = 32w5;
