--- before_pass
+++ after_pass
@@ -1,21 +1,21 @@
 control ctrl(out bit<32> c) {
-    bit<32> a;
-    @name("e") action e_0() {
+    bit<32> a_0;
+    @name("e") action e() {
         exit;
     }
-    @name("t") table t {
+    @name("t") table t_0 {
         actions = {
-            e_0();
+            e();
         }
-        default_action = e_0();
+        default_action = e();
     }
     apply {
-        a = 32w0;
+        a_0 = 32w0;
         c = 32w2;
-        if (a == 32w0) {
-            t.apply();
+        if (a_0 == 32w0) {
+            t_0.apply();
         } else {
-            t.apply();
+            t_0.apply();
         }
         c = 32w5;
     }
