--- before_pass
+++ after_pass
@@ -11,18 +11,10 @@ struct h {
 parser MyParser(packet_in b, out h hdr, inout m meta, inout standard_metadata_t std) {
     error e_0;
     state start {
-        {
-            verify(meta.x == 8s0, error.NewError);
-        }
-        {
-            verify(true, error.NoError);
-        }
-        {
-            e_0 = error.NoError;
-        }
-        {
-            verify(true, e_0);
-        }
+        verify(meta.x == 8s0, error.NewError);
+        verify(true, error.NoError);
+        e_0 = error.NoError;
+        verify(true, e_0);
         transition accept;
     }
 }
