--- before_pass
+++ after_pass
@@ -7,7 +7,7 @@ extern Random2 {
     bit<10> read();
 }
 parser caller() {
-    @name("rand1") Random2() rand1_0;
+    @name("caller.rand1") Random2() rand1_0;
     state start {
         transition callee_start;
     }
@@ -20,7 +20,7 @@ parser caller() {
     }
 }
 control ingress() {
-    @name("rand1") Random2() rand1_1;
+    @name("ingress.rand1") Random2() rand1_1;
     apply {
         rand1_1.read();
     }
