--- before_pass
+++ after_pass
@@ -13,10 +13,10 @@ parser callee(Random2 rand) {
     }
 }
 parser caller() {
-    Random2() rand1;
-    callee() ca;
+    @name("rand1") Random2() rand1_0;
+    @name("ca") callee() ca_0;
     state start {
-        ca.apply(rand1);
+        ca_0.apply(rand1_0);
         transition accept;
     }
 }
@@ -26,10 +26,10 @@ control foo2(Random2 rand) {
     }
 }
 control ingress() {
-    Random2() rand1;
-    foo2() foo2_inst;
+    @name("rand1") Random2() rand1_1;
+    @name("foo2_inst") foo2() foo2_inst_0;
     apply {
-        foo2_inst.apply(rand1);
+        foo2_inst_0.apply(rand1_1);
     }
 }
 Top(ingress(), caller()) main;
