--- before_pass
+++ after_pass
@@ -8,29 +8,21 @@ control c(inout bit<32> y) {
         void a(inout bit<32> arg) {
             bit<32> tmp;
             bit<32> tmp_0;
-            {
-                tmp = this.b();
-                tmp_0 = arg + tmp;
-                arg = tmp_0;
-            }
+            tmp = this.b();
+            tmp_0 = arg + tmp;
+            arg = tmp_0;
         }
     };
     apply {
-        {
-            x_0.a(y);
-        }
+        x_0.a(y);
     }
 }
 control t(inout bit<32> b) {
     @name("c1") c() c1_0;
     @name("c2") c() c2_0;
     apply {
-        {
-            c1_0.apply(b);
-        }
-        {
-            c2_0.apply(b);
-        }
+        c1_0.apply(b);
+        c2_0.apply(b);
     }
 }
 control cs(inout bit<32> arg);
