--- before_pass
+++ after_pass
@@ -4,21 +4,21 @@ extern X {
     abstract void a(inout bit<32> arg);
 }
 control c(inout bit<32> y) {
-    X() x = {
+    @name("x") X() x_0 = {
         void a(inout bit<32> arg) {
             arg = arg + this.b();
         }
     };
     apply {
-        x.a(y);
+        x_0.a(y);
     }
 }
 control t(inout bit<32> b) {
-    c() c1;
-    c() c2;
+    @name("c1") c() c1_0;
+    @name("c2") c() c2_0;
     apply {
-        c1.apply(b);
-        c2.apply(b);
+        c1_0.apply(b);
+        c2_0.apply(b);
     }
 }
 control cs(inout bit<32> arg);
