--- before_pass
+++ after_pass
@@ -4,7 +4,7 @@ extern X {
     abstract void a(inout bit<32> arg);
 }
 control t(inout bit<32> b) {
-    @name("c1.x") X() c1_x = {
+    @name("t.c1.x") X() c1_x = {
         void a(inout bit<32> arg) {
             bit<32> c1_tmp;
             bit<32> c1_tmp_0;
@@ -13,7 +13,7 @@ control t(inout bit<32> b) {
             arg = c1_tmp_0;
         }
     };
-    @name("c2.x") X() c2_x = {
+    @name("t.c2.x") X() c2_x = {
         void a(inout bit<32> arg) {
             bit<32> c2_tmp;
             bit<32> c2_tmp_0;
