--- before_pass
+++ after_pass
@@ -26,12 +26,12 @@ parser parserI(packet_in pkt, out Parsed
 }
 control cIngress(inout Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
     action foo() {
-        meta.b = meta.b + 5;
-        if (meta.b > 10) {
-            meta.b = meta.b ^ 5;
+        meta.b = meta.b + 4w5;
+        if (meta.b > 4w10) {
+            meta.b = meta.b ^ 4w5;
             return;
         }
-        meta.b = meta.b + 5;
+        meta.b = meta.b + 4w5;
     }
     table guh {
         key = {
