--- before_pass
+++ after_pass
@@ -66,7 +66,7 @@ parser parserI(packet_in pkt, out header
         }
     }
     state parse_ipv4 {
-        pkt.extract<ipv4_t>(hdr.ipv4, (bit<32>)(9w8 * (9w4 * (bit<9>)(pkt.lookahead<IPv4_up_to_ihl_only_h>()).ihl - 9w20)));
+        pkt.extract<ipv4_t>(hdr.ipv4, (bit<32>)(((bit<9>)(pkt.lookahead<IPv4_up_to_ihl_only_h>()).ihl << 2) + 9w492 << 3));
         verify(hdr.ipv4.version == 4w4, error.IPv4IncorrectVersion);
         verify(hdr.ipv4.ihl >= 4w5, error.IPv4HeaderTooShort);
         transition select(hdr.ipv4.protocol) {
@@ -82,7 +82,7 @@ parser parserI(packet_in pkt, out header
 control cIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
     action foo() {
         hdr.tcp.srcPort = hdr.tcp.srcPort + 16w1;
-        hdr.ipv4.ttl = hdr.ipv4.ttl - 8w1;
+        hdr.ipv4.ttl = hdr.ipv4.ttl + 8w255;
         hdr.ipv4.dstAddr = hdr.ipv4.dstAddr + 32w4;
     }
     table guh {
