--- before_pass
+++ after_pass
@@ -46,4 +46,4 @@ control deparser_impl(packet_out packet,
     apply {
     }
 }
-V1Switch(parser_impl(), verify_checksum_impl(), ingress_impl(), egress_impl(), compute_checksum_impl(), deparser_impl()) main;
+V1Switch<my_headers_t, local_metadata_t>(parser_impl(), verify_checksum_impl(), ingress_impl(), egress_impl(), compute_checksum_impl(), deparser_impl()) main;
