--- before_pass
+++ after_pass
@@ -34,18 +34,18 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name("concat") action concat_0() {
+    @name("concat") action concat() {
         h.h.c = h.h.a ++ h.h.b;
         sm.egress_spec = 9w0;
     }
-    @name("t") table t_0 {
+    @name("t") table t {
         actions = {
-            concat_0();
+            concat();
         }
-        const default_action = concat_0();
+        const default_action = concat();
     }
     apply {
-        t_0.apply();
+        t.apply();
     }
 }
 V1Switch<Headers, Meta>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
