--- before_pass
+++ after_pass
@@ -1,6 +1,8 @@
 control c(inout bit<32> b) {
     @name("a") action a_0() {
-        b = 32w1;
+        {
+            b = 32w1;
+        }
     }
     @name("t") table t_0 {
         actions = {
@@ -11,7 +13,9 @@ control c(inout bit<32> b) {
     apply {
         switch (t_0.apply().action_run) {
             a_0: {
-                b[6:3] = 4w1;
+                {
+                    b[6:3] = 4w1;
+                }
             }
         }
     }
