--- before_pass
+++ after_pass
@@ -47,10 +47,10 @@ struct Parsed_packet {
     Ipv4_h     ip;
 }
 parser TopParser(packet_in b, out Parsed_packet p) {
-    @name("ck") Ck16() ck_0;
     bit<16> tmp;
     bool tmp_0;
     bool tmp_1;
+    @name("ck") Ck16() ck_0;
     state start {
         b.extract<Ethernet_h>(p.ethernet);
         transition select(p.ethernet.etherType) {
@@ -153,8 +153,8 @@ control TopPipe(inout Parsed_packet head
     }
 }
 control TopDeparser(inout Parsed_packet p, packet_out b) {
-    @name("ck") Ck16() ck_1;
     bit<16> tmp_2;
+    @name("ck") Ck16() ck_1;
     apply {
         b.emit<Ethernet_h>(p.ethernet);
         if (p.ip.isValid()) {
