--- before_pass
+++ after_pass
@@ -65,10 +65,10 @@ parser TopParser(packet_in b, out Parsed
     }
 }
 control TopPipe(inout Parsed_packet headers, in error parseError, in InControl inCtrl, out OutControl outCtrl) {
+    IPv4Address nextHop_0;
     @name("Drop_action") action Drop_action_0() {
         outCtrl.outputPort = 4w0xf;
     }
-    IPv4Address nextHop_0;
     @name("Set_nhop") action Set_nhop_0(IPv4Address ipv4_dest, PortId port) {
         nextHop_0 = ipv4_dest;
         headers.ip.ttl = headers.ip.ttl + 8w255;
