--- before_pass
+++ after_pass
@@ -85,11 +85,11 @@ control TopPipe(inout Parsed_packet head
             headers.ip.dstAddr: lpm;
         }
         actions = {
-            Drop_action;
-            Set_nhop;
+            Drop_action();
+            Set_nhop();
         }
         size = 1024;
-        default_action = Drop_action;
+        default_action = Drop_action();
     }
     action Send_to_cpu() {
         outCtrl.outputPort = CPU_OUT_PORT;
@@ -99,10 +99,10 @@ control TopPipe(inout Parsed_packet head
             headers.ip.ttl: exact;
         }
         actions = {
-            Send_to_cpu;
-            NoAction;
+            Send_to_cpu();
+            NoAction();
         }
-        const default_action = NoAction;
+        const default_action = NoAction();
     }
     action Set_dmac(EthernetAddress dmac) {
         headers.ethernet.dstAddr = dmac;
@@ -112,11 +112,11 @@ control TopPipe(inout Parsed_packet head
             nextHop: exact;
         }
         actions = {
-            Drop_action;
-            Set_dmac;
+            Drop_action();
+            Set_dmac();
         }
         size = 1024;
-        default_action = Drop_action;
+        default_action = Drop_action();
     }
     action Set_smac(EthernetAddress smac) {
         headers.ethernet.srcAddr = smac;
@@ -126,11 +126,11 @@ control TopPipe(inout Parsed_packet head
             outCtrl.outputPort: exact;
         }
         actions = {
-            Drop_action;
-            Set_smac;
+            Drop_action();
+            Set_smac();
         }
         size = 16;
-        default_action = Drop_action;
+        default_action = Drop_action();
     }
     apply {
         if (parseError != error.NoError) {
