--- before_pass
+++ after_pass
@@ -47,10 +47,10 @@ struct Parsed_packet {
     Ipv4_h     ip;
 }
 parser TopParser(packet_in b, out Parsed_packet p) {
-    bit<16> tmp;
-    bool tmp_0;
-    bool tmp_1;
-    @name("ck") Ck16() ck_0;
+    bit<16> tmp_3;
+    bool tmp_4;
+    bool tmp_5;
+    @name("ck") Ck16() ck;
     state start {
         b.extract<Ethernet_h>(p.ethernet);
         transition select(p.ethernet.etherType) {
@@ -61,117 +61,117 @@ parser TopParser(packet_in b, out Parsed
         b.extract<Ipv4_h>(p.ip);
         verify(p.ip.version == 4w4, error.IPv4IncorrectVersion);
         verify(p.ip.ihl == 4w5, error.IPv4OptionsNotSupported);
-        ck_0.clear();
-        ck_0.update<Ipv4_h>(p.ip);
-        tmp = ck_0.get();
-        tmp_0 = tmp == 16w0;
-        tmp_1 = tmp_0;
-        verify(tmp_1, error.IPv4ChecksumError);
+        ck.clear();
+        ck.update<Ipv4_h>(p.ip);
+        tmp_3 = ck.get();
+        tmp_4 = tmp_3 == 16w0;
+        tmp_5 = tmp_4;
+        verify(tmp_5, error.IPv4ChecksumError);
         transition accept;
     }
 }
 control TopPipe(inout Parsed_packet headers, in error parseError, in InControl inCtrl, out OutControl outCtrl) {
-    IPv4Address nextHop_0;
-    @name("Drop_action") action Drop_action_0() {
+    IPv4Address nextHop;
+    @name("Drop_action") action Drop_action() {
         outCtrl.outputPort = 4w0xf;
     }
-    @name("Set_nhop") action Set_nhop_0(IPv4Address ipv4_dest, PortId port) {
-        nextHop_0 = ipv4_dest;
+    @name("Set_nhop") action Set_nhop(IPv4Address ipv4_dest, PortId port) {
+        nextHop = ipv4_dest;
         headers.ip.ttl = headers.ip.ttl + 8w255;
         outCtrl.outputPort = port;
     }
-    @name("ipv4_match") table ipv4_match_0 {
+    @name("ipv4_match") table ipv4_match {
         key = {
             headers.ip.dstAddr: lpm @name("headers.ip.dstAddr") ;
         }
         actions = {
-            Drop_action_0();
-            Set_nhop_0();
+            Drop_action();
+            Set_nhop();
         }
         size = 1024;
-        default_action = Drop_action_0();
+        default_action = Drop_action();
     }
-    @name("Send_to_cpu") action Send_to_cpu_0() {
+    @name("Send_to_cpu") action Send_to_cpu() {
         outCtrl.outputPort = 4w0xe;
     }
-    @name("check_ttl") table check_ttl_0 {
+    @name("check_ttl") table check_ttl {
         key = {
             headers.ip.ttl: exact @name("headers.ip.ttl") ;
         }
         actions = {
-            Send_to_cpu_0();
+            Send_to_cpu();
             NoAction();
         }
         const default_action = NoAction();
     }
-    @name("Set_dmac") action Set_dmac_0(EthernetAddress dmac) {
+    @name("Set_dmac") action Set_dmac(EthernetAddress dmac) {
         headers.ethernet.dstAddr = dmac;
     }
-    @name("dmac") table dmac_0 {
+    @name("dmac") table dmac_1 {
         key = {
-            nextHop_0: exact @name("nextHop") ;
+            nextHop: exact @name("nextHop") ;
         }
         actions = {
-            Drop_action_0();
-            Set_dmac_0();
+            Drop_action();
+            Set_dmac();
         }
         size = 1024;
-        default_action = Drop_action_0();
+        default_action = Drop_action();
     }
-    @name("Set_smac") action Set_smac_0(EthernetAddress smac) {
+    @name("Set_smac") action Set_smac(EthernetAddress smac) {
         headers.ethernet.srcAddr = smac;
     }
-    @name("smac") table smac_0 {
+    @name("smac") table smac_1 {
         key = {
             outCtrl.outputPort: exact @name("outCtrl.outputPort") ;
         }
         actions = {
-            Drop_action_0();
-            Set_smac_0();
+            Drop_action();
+            Set_smac();
         }
         size = 16;
-        default_action = Drop_action_0();
+        default_action = Drop_action();
     }
     apply {
-        bool hasReturned = false;
+        bool hasReturned_0 = false;
         if (parseError != error.NoError) {
-            Drop_action_0();
-            hasReturned = true;
+            Drop_action();
+            hasReturned_0 = true;
         }
-        if (!hasReturned) {
-            ipv4_match_0.apply();
+        if (!hasReturned_0) {
+            ipv4_match.apply();
             if (outCtrl.outputPort == 4w0xf) {
-                hasReturned = true;
+                hasReturned_0 = true;
             }
         }
-        if (!hasReturned) {
-            check_ttl_0.apply();
+        if (!hasReturned_0) {
+            check_ttl.apply();
             if (outCtrl.outputPort == 4w0xe) {
-                hasReturned = true;
+                hasReturned_0 = true;
             }
         }
-        if (!hasReturned) {
-            dmac_0.apply();
+        if (!hasReturned_0) {
+            dmac_1.apply();
             if (outCtrl.outputPort == 4w0xf) {
-                hasReturned = true;
+                hasReturned_0 = true;
             }
         }
-        if (!hasReturned) {
-            smac_0.apply();
+        if (!hasReturned_0) {
+            smac_1.apply();
         }
     }
 }
 control TopDeparser(inout Parsed_packet p, packet_out b) {
-    bit<16> tmp_2;
-    @name("ck") Ck16() ck_1;
+    bit<16> tmp_6;
+    @name("ck") Ck16() ck_2;
     apply {
         b.emit<Ethernet_h>(p.ethernet);
         if (p.ip.isValid()) {
-            ck_1.clear();
+            ck_2.clear();
             p.ip.hdrChecksum = 16w0;
-            ck_1.update<Ipv4_h>(p.ip);
-            tmp_2 = ck_1.get();
-            p.ip.hdrChecksum = tmp_2;
+            ck_2.update<Ipv4_h>(p.ip);
+            tmp_6 = ck_2.get();
+            p.ip.hdrChecksum = tmp_6;
         }
         b.emit<Ipv4_h>(p.ip);
     }
