--- before_pass
+++ after_pass
@@ -82,7 +82,7 @@ control TopPipe(inout Parsed_packet head
     }
     table ipv4_match {
         key = {
-            headers.ip.dstAddr: lpm;
+            headers.ip.dstAddr: lpm @name("headers.ip.dstAddr") ;
         }
         actions = {
             Drop_action();
@@ -96,7 +96,7 @@ control TopPipe(inout Parsed_packet head
     }
     table check_ttl {
         key = {
-            headers.ip.ttl: exact;
+            headers.ip.ttl: exact @name("headers.ip.ttl") ;
         }
         actions = {
             Send_to_cpu();
@@ -109,7 +109,7 @@ control TopPipe(inout Parsed_packet head
     }
     table dmac {
         key = {
-            nextHop: exact;
+            nextHop: exact @name("nextHop") ;
         }
         actions = {
             Drop_action();
@@ -123,7 +123,7 @@ control TopPipe(inout Parsed_packet head
     }
     table smac {
         key = {
-            outCtrl.outputPort: exact;
+            outCtrl.outputPort: exact @name("outCtrl.outputPort") ;
         }
         actions = {
             Drop_action();
