--- before_pass
+++ after_pass
@@ -10,19 +10,19 @@ struct headers_t {
 struct metadata_t {
 }
 control ingressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
-    @name(".foo") action foo_1(out bit<8> x_0, in bit<8> y_0=5) {
+    @name(".foo") action foo(out bit<8> x_0, in bit<8> y_0=5) {
         x_0 = y_0 >> 2;
     }
-    @name(".foo") action foo_2(out bit<8> x_0, in bit<8> y_0=5) {
+    @name(".foo") action foo_0(out bit<8> x_0, in bit<8> y_0=5) {
         x_0 = y_0 >> 2;
     }
-    bit<8> tmp_1;
-    bit<8> tmp_2;
+    bit<8> tmp;
+    bit<8> tmp_0;
     apply {
-        tmp_2 = hdr.h1.f1;
-        foo_1(tmp_1, tmp_2);
-        hdr.h1.f1 = tmp_1;
-        foo_2(x_0 = hdr.h1.f2, y_0 = 8w5);
+        tmp_0 = hdr.h1.f1;
+        foo(tmp, tmp_0);
+        hdr.h1.f1 = tmp;
+        foo_0(x_0 = hdr.h1.f2, y_0 = 8w5);
     }
 }
 parser parserImpl(packet_in packet, out headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
