--- before_pass
+++ after_pass
@@ -9,8 +9,8 @@ struct headers_t {
 }
 struct metadata_t {
 }
-action foo(out bit<8> x, in bit<8> y=5) {
-    x = y >> 2;
+action foo(out bit<8> x_0, in bit<8> y_0=5) {
+    x_0 = y_0 >> 2;
 }
 control ingressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
     bit<8> tmp;
@@ -19,7 +19,7 @@ control ingressImpl(inout headers_t hdr,
         tmp_0 = hdr.h1.f1;
         foo(tmp, tmp_0);
         hdr.h1.f1 = tmp;
-        foo(x = hdr.h1.f2, y = 8w5);
+        foo(x_0 = hdr.h1.f2, y_0 = 8w5);
     }
 }
 parser parserImpl(packet_in packet, out headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
