--- before_pass
+++ after_pass
@@ -12,19 +12,11 @@ struct metadata_t {
 control ingressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
     bit<8> tmp;
     bit<8> tmp_0;
-    bit<8> x;
-    bit<8> y;
-    bit<8> x_1;
-    bit<8> y_1;
     @name(".foo") action foo() {
-        y = tmp_0;
-        x = y >> 2;
-        tmp = x;
+        tmp = tmp_0 >> 2;
     }
     @name(".foo") action foo_0() {
-        y_1 = 8w5;
-        x_1 = y_1 >> 2;
-        hdr.h1.f2 = x_1;
+        hdr.h1.f2 = 8w5 >> 2;
     }
     apply {
         tmp_0 = hdr.h1.f1;
