--- before_pass
+++ after_pass
@@ -10,19 +10,27 @@ struct headers_t {
 struct metadata_t {
 }
 control ingressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
-    @name(".foo") action foo(out bit<8> x, in bit<8> y=5) {
+    bit<8> tmp;
+    bit<8> tmp_0;
+    bit<8> x;
+    bit<8> y;
+    @name(".foo") action foo() {
+        y = tmp_0;
         x = y >> 2;
+        tmp = x;
     }
-    @name(".foo") action foo_0(out bit<8> x_1, in bit<8> y_1=5) {
+    bit<8> x_1;
+    bit<8> y_1;
+    @name(".foo") action foo_0() {
+        y_1 = 8w5;
         x_1 = y_1 >> 2;
+        hdr.h1.f2 = x_1;
     }
-    bit<8> tmp;
-    bit<8> tmp_0;
     apply {
         tmp_0 = hdr.h1.f1;
-        foo(tmp, tmp_0);
+        foo();
         hdr.h1.f1 = tmp;
-        foo_0(x_1 = hdr.h1.f2, y_1 = 8w5);
+        foo_0();
     }
 }
 parser parserImpl(packet_in packet, out headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
