--- before_pass
+++ after_pass
@@ -10,11 +10,11 @@ struct headers_t {
 struct metadata_t {
 }
 control ingressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
-    @name(".foo") action foo(out bit<8> x_0, in bit<8> y_0=5) {
-        x_0 = y_0 >> 2;
+    @name(".foo") action foo(out bit<8> x, in bit<8> y=5) {
+        x = y >> 2;
     }
-    @name(".foo") action foo_0(out bit<8> x_0, in bit<8> y_0=5) {
-        x_0 = y_0 >> 2;
+    @name(".foo") action foo_0(out bit<8> x_1, in bit<8> y_1=5) {
+        x_1 = y_1 >> 2;
     }
     bit<8> tmp;
     bit<8> tmp_0;
@@ -22,7 +22,7 @@ control ingressImpl(inout headers_t hdr,
         tmp_0 = hdr.h1.f1;
         foo(tmp, tmp_0);
         hdr.h1.f1 = tmp;
-        foo_0(x_0 = hdr.h1.f2, y_0 = 8w5);
+        foo_0(x_1 = hdr.h1.f2, y_1 = 8w5);
     }
 }
 parser parserImpl(packet_in packet, out headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
