--- before_pass
+++ after_pass
@@ -16,9 +16,9 @@ parser prs(packet_in p, out Headers h) {
 control c(inout Headers h, inout standard_metadata_t sm) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("do_act") action do_act() {
+    @name("c.do_act") action do_act() {
     }
-    @name("tns") table tns_0 {
+    @name("c.tns") table tns_0 {
         key = {
             h.eth.tst[13:4]: exact @name("h.eth.tst[13:4]") ;
         }
