--- before_pass
+++ after_pass
@@ -14,22 +14,22 @@ parser prs(packet_in p, out Headers h) {
     }
 }
 control c(inout Headers h, inout standard_metadata_t sm) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name("do_act") action do_act_0() {
+    @name("do_act") action do_act() {
     }
-    @name("tns") table tns {
+    @name("tns") table tns_0 {
         key = {
             h.eth.tst[13:4]: exact @name("h.eth.tst[13:4]") ;
         }
         actions = {
-            do_act_0();
-            @defaultonly NoAction_1();
+            do_act();
+            @defaultonly NoAction_0();
         }
-        default_action = NoAction_1();
+        default_action = NoAction_0();
     }
     apply {
-        tns.apply();
+        tns_0.apply();
     }
 }
 parser p<H>(packet_in _p, out H h);
