--- before_pass
+++ after_pass
@@ -7,15 +7,11 @@ struct M {
 }
 control Ingress(inout M sm);
 package V1Switch(Ingress ig);
-control Forwarding(inout M sm) {
-    apply {
-        sm.es = (PortIdUInt_t)sm.e;
-    }
-}
 control FabricIngress(inout M sm) {
-    @name("forwarding") Forwarding() forwarding_0;
     apply {
-        forwarding_0.apply(sm);
+        {
+            sm.es = (PortIdUInt_t)sm.e;
+        }
     }
 }
 V1Switch(FabricIngress()) main;
