--- before_pass
+++ after_pass
@@ -9,13 +9,17 @@ control Ingress(inout M sm);
 package V1Switch(Ingress ig);
 control Forwarding(inout M sm) {
     apply {
-        sm.es = (PortIdUInt_t)sm.e;
+        {
+            sm.es = (PortIdUInt_t)sm.e;
+        }
     }
 }
 control FabricIngress(inout M sm) {
     @name("forwarding") Forwarding() forwarding_0;
     apply {
-        forwarding_0.apply(sm);
+        {
+            forwarding_0.apply(sm);
+        }
     }
 }
 V1Switch(FabricIngress()) main;
