--- before_pass
+++ after_pass
@@ -10,11 +10,11 @@ parser foo(out bit<8> x, in bit<8> y=5)
     }
 }
 parser parserImpl(out h1_t hdr) {
-    @name("foo") foo() foo_inst;
-    @name("foo") foo() foo_inst_0;
+    @name("foo") foo() foo_inst_1;
+    @name("foo") foo() foo_inst_2;
     state start {
-        foo_inst.apply(hdr.f1, hdr.f1);
-        foo_inst_0.apply(x = hdr.f2, y = 8w5);
+        foo_inst_1.apply(hdr.f1, hdr.f1);
+        foo_inst_2.apply(x = hdr.f2, y = 8w5);
         transition accept;
     }
 }
