--- before_pass
+++ after_pass
@@ -4,18 +4,18 @@ header h1_t {
     bit<8> f2;
 }
 parser parserImpl(out h1_t hdr) {
-    bit<8> tmp;
-    bit<8> tmp_0;
+    bit<8> tmp_1;
+    bit<8> tmp_2;
     state start {
-        tmp_0 = hdr.f1;
+        tmp_2 = hdr.f1;
         transition foo_start;
     }
     state foo_start {
-        tmp = tmp_0 >> 2;
+        tmp_1 = tmp_2 >> 2;
         transition start_0;
     }
     state start_0 {
-        hdr.f1 = tmp;
+        hdr.f1 = tmp_1;
         transition foo_start_0;
     }
     state foo_start_0 {
