--- before_pass
+++ after_pass
@@ -5,16 +5,26 @@ header h1_t {
 }
 parser foo(out bit<8> x, in bit<8> y=5) {
     state start {
-        x = y >> 2;
+        {
+            x = y >> 2;
+        }
         transition accept;
     }
 }
 parser parserImpl(out h1_t hdr) {
     @name("foo") foo() foo_inst_1;
     @name("foo") foo() foo_inst_2;
+    bit<8> tmp;
+    bit<8> tmp_0;
     state start {
-        foo_inst_1.apply(hdr.f1, hdr.f1);
-        foo_inst_2.apply(x = hdr.f2, y = 8w5);
+        {
+            tmp_0 = hdr.f1;
+            foo_inst_1.apply(tmp, tmp_0);
+            hdr.f1 = tmp;
+        }
+        {
+            foo_inst_2.apply(x = hdr.f2, y = 8w5);
+        }
         transition accept;
     }
 }
