--- before_pass
+++ after_pass
@@ -5,14 +5,14 @@ struct S {
 }
 parser p() {
     state start {
-        bit<8> x = 5;
-        S s = { 0, 0 };
-        tuple<bit<8>, bit<8>> t = { 0, 0 };
+        bit<8> x = (bit<8>)8w5;
+        S s = { 8w0, 8w0 };
+        tuple<bit<8>, bit<8>> t = { 8w0, 8w0 };
         transition select(x, x, { x, x }, x) {
-            (0, 0, { 0, 0 }, 0): accept;
-            (1, 1, default, 1): accept;
-            (1, 1, s, 2): accept;
-            (1, 1, t, 2): accept;
+            (8w0, 8w0, { 8w0, 8w0 }, 8w0): accept;
+            (8w1, 8w1, default, 8w1): accept;
+            (8w1, 8w1, s, 8w2): accept;
+            (8w1, 8w1, t, 8w2): accept;
             default: reject;
         }
     }
