--- before_pass
+++ after_pass
@@ -31,13 +31,13 @@ parser LJparse(packet_in b, out Parsed_r
     }
 }
 control LjPipe(inout Parsed_rep p, in error parseError, in InControl inCtrl, out OutControl outCtrl) {
-    @name("Drop_action") action Drop_action(out PortId port_0) {
+    @name("Drop_action") action Drop_action_0(out PortId port_0) {
         port_0 = 4w0xf;
     }
-    @name("Drop_1") action Drop_0() {
+    @name("Drop_1") action Drop() {
         outCtrl.outputPort = 4w0xf;
     }
-    @name("Forward") action Forward(PortId outPort) {
+    @name("Forward") action Forward_0(PortId outPort) {
         outCtrl.outputPort = outPort;
     }
     @name("Enet_lkup") table Enet_lkup {
@@ -45,11 +45,11 @@ control LjPipe(inout Parsed_rep p, in er
             p.arpa_pak.dest: exact @name("p.arpa_pak.dest") ;
         }
         actions = {
-            Drop_action(outCtrl.outputPort);
-            Drop_0();
-            Forward();
+            Drop_action_0(outCtrl.outputPort);
+            Drop();
+            Forward_0();
         }
-        default_action = Drop_0();
+        default_action = Drop();
     }
     apply {
         outCtrl.outputPort = 4w0xf;
