--- before_pass
+++ after_pass
@@ -31,16 +31,16 @@ parser LJparse(packet_in b, out Parsed_r
     }
 }
 control LjPipe(inout Parsed_rep p, in error parseError, in InControl inCtrl, out OutControl outCtrl) {
-    @name("Drop_action") action Drop_action(out PortId port) {
+    @name("LjPipe.Drop_action") action Drop_action(out PortId port) {
         port = 4w0xf;
     }
-    @name("Drop_1") action Drop_0() {
+    @name("LjPipe.Drop_1") action Drop_0() {
         outCtrl.outputPort = 4w0xf;
     }
-    @name("Forward") action Forward(PortId outPort) {
+    @name("LjPipe.Forward") action Forward(PortId outPort) {
         outCtrl.outputPort = outPort;
     }
-    @name("Enet_lkup") table Enet_lkup_0 {
+    @name("LjPipe.Enet_lkup") table Enet_lkup_0 {
         key = {
             p.arpa_pak.dest: exact @name("p.arpa_pak.dest") ;
         }
