--- before_pass
+++ after_pass
@@ -31,8 +31,8 @@ parser LJparse(packet_in b, out Parsed_r
     }
 }
 control LjPipe(inout Parsed_rep p, in error parseError, in InControl inCtrl, out OutControl outCtrl) {
-    @name("Drop_action") action Drop_action(out PortId port_0) {
-        port_0 = 4w0xf;
+    @name("Drop_action") action Drop_action(out PortId port) {
+        port = 4w0xf;
     }
     @name("Drop_1") action Drop_0() {
         outCtrl.outputPort = 4w0xf;
