--- before_pass
+++ after_pass
@@ -24,12 +24,12 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout user_meta_t b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
-    ActionSelector(PSA_HashAlgorithm_t.CRC32, 32w1024, 32w16) as;
-    action a1() {
+    @name("as") ActionSelector(PSA_HashAlgorithm_t.CRC32, 32w1024, 32w16) as_0;
+    @name("a1") action a1_0() {
     }
-    action a2() {
+    @name("a2") action a2_0() {
     }
-    table tbl {
+    @name("tbl") table tbl_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
             b.data1  : selector @name("b.data1") ;
@@ -37,14 +37,14 @@ control MyIC(inout ethernet_t a, inout u
         }
         actions = {
             NoAction();
-            a1();
-            a2();
+            a1_0();
+            a2_0();
         }
-        psa_implementation = as;
+        psa_implementation = as_0;
         default_action = NoAction();
     }
     apply {
-        tbl.apply();
+        tbl_0.apply();
     }
 }
 control MyEC(inout EMPTY a, inout EMPTY b, in psa_egress_input_metadata_t c, inout psa_egress_output_metadata_t d) {
