--- before_pass
+++ after_pass
@@ -3,14 +3,9 @@ error {
 }
 #include <core.p4>
 #include <v1model.p4>
-const bit<16> TYPE_IPV4 = 16w0x800;
 typedef bit<9> egressSpec_t;
 typedef bit<48> macAddr_t;
 typedef bit<32> ip4Addr_t;
-typedef bit<32> switchID_t;
-typedef bit<19> qdepth_t;
-typedef bit<48> timestamp_t;
-typedef bit<32> timedelta_t;
 header ethernet_t {
     macAddr_t dstAddr;
     macAddr_t srcAddr;
