--- before_pass
+++ after_pass
@@ -72,9 +72,9 @@ control MyIngress(inout headers hdr, ino
             hdr.ipv4.dstAddr: exact;
         }
         actions = {
-            set_dmac;
-            drop;
-            NoAction;
+            set_dmac();
+            drop();
+            NoAction();
         }
         size = 1024;
         default_action = NoAction();
@@ -88,9 +88,9 @@ control MyIngress(inout headers hdr, ino
             hdr.ipv4.dstAddr: lpm;
         }
         actions = {
-            set_nhop;
-            drop;
-            NoAction;
+            set_nhop();
+            drop();
+            NoAction();
         }
         size = 1024;
         default_action = NoAction();
@@ -115,8 +115,8 @@ control MyEgress(inout headers hdr, inou
             standard_metadata.egress_port: exact;
         }
         actions = {
-            rewrite_mac;
-            NoAction;
+            rewrite_mac();
+            NoAction();
         }
         size = 1024;
         default_action = NoAction();
