--- before_pass
+++ after_pass
@@ -10,12 +10,12 @@ struct Metadata {
 }
 control DeparserI(packet_out packet, in Parsed_packet hdr) {
     apply {
-        packet.emit(hdr.h);
+        packet.emit<H>(hdr.h);
     }
 }
 parser parserI(packet_in pkt, out Parsed_packet hdr, inout Metadata meta, inout standard_metadata_t stdmeta) {
     state start {
-        pkt.extract(hdr.h);
+        pkt.extract<H>(hdr.h);
         transition accept;
     }
 }
@@ -35,4 +35,4 @@ control uc(inout Parsed_packet hdr, inou
     apply {
     }
 }
-V1Switch(parserI(), vc(), cIngress(), cEgress(), uc(), DeparserI()) main;
+V1Switch<Parsed_packet, Metadata>(parserI(), vc(), cIngress(), cEgress(), uc(), DeparserI()) main;
