--- before_pass
+++ after_pass
@@ -15,13 +15,13 @@ control IngressI(inout H hdr, inout M me
     }
     @name(".NoAction") action NoAction_3() {
     }
-    @name("drop") action drop() {
+    @name("IngressI.drop") action drop() {
         mark_to_drop(smeta);
     }
-    @name("drop") action drop_2() {
+    @name("IngressI.drop") action drop_2() {
         mark_to_drop(smeta);
     }
-    @name("indirect") table indirect_0 {
+    @name("IngressI.indirect") table indirect_0 {
         key = {
         }
         actions = {
@@ -31,7 +31,7 @@ control IngressI(inout H hdr, inout M me
         const default_action = NoAction_0();
         @name("ap") @max_group_size(200) implementation = action_profile(32w128);
     }
-    @name("indirect_ws") table indirect_ws_0 {
+    @name("IngressI.indirect_ws") table indirect_ws_0 {
         key = {
             meta.hash1: selector @name("meta.hash1") ;
         }
