--- before_pass
+++ after_pass
@@ -1,6 +1,6 @@
 #include <core.p4>
 #include <v1model.p4>
-const bit<6> TYPE_ADDR_IPV4 = 0x1;
+const bit<6> TYPE_ADDR_IPV4 = 6w0x1;
 header preamble_t {
     bit<336> data;
 }
@@ -82,7 +82,7 @@ parser PROTParser(packet_in packet, out
     }
     state parse_prot_host_addr_src_select {
         transition select(hdr.prot_common.srcType) {
-            TYPE_ADDR_IPV4: parse_prot_host_addr_src_ipv4;
+            6w0x1: parse_prot_host_addr_src_ipv4;
         }
     }
     state parse_prot_host_addr_src_ipv4 {
@@ -97,7 +97,7 @@ parser PROTParser(packet_in packet, out
         transition parse_prot_inf_0;
     }
     state parse_prot_inf_0 {
-        meta.currPos = (bit<8>)(1 + 2 + meta.addrLen / 64);
+        meta.currPos = (bit<8>)(3 + meta.addrLen / 64);
         bool currentISelected = hdr.prot_common.curri == meta.currPos;
         subParser.apply(packet, hdr.prot_inf_0, meta, currentISelected, hdr.prot_common.curri);
         transition parse_prot_h_0_pre;
