--- before_pass
+++ after_pass
@@ -91,13 +91,13 @@ parser PROTParser(packet_in packet, out
         transition parse_prot_addr_padding;
     }
     state parse_prot_addr_padding {
-        bit<9> paddingLen = (9w64 - meta.addrLen % 9w64) % 9w64;
+        bit<9> paddingLen = 9w64 - (meta.addrLen & 9w63) & 9w63;
         packet.extract<prot_host_addr_padding_t>(hdr.prot_host_addr_padding, (bit<32>)paddingLen);
         meta.addrLen = meta.addrLen + paddingLen;
         transition parse_prot_inf_0;
     }
     state parse_prot_inf_0 {
-        meta.currPos = (bit<8>)(9w3 + meta.addrLen / 9w64);
+        meta.currPos = (bit<8>)(9w3 + (meta.addrLen >> 6));
         bool currentISelected = hdr.prot_common.curri == meta.currPos;
         subParser.apply(packet, hdr.prot_inf_0, meta, currentISelected, hdr.prot_common.curri);
         transition parse_prot_h_0_pre;
@@ -110,7 +110,7 @@ parser PROTParser(packet_in packet, out
     }
     state parse_prot_h_0 {
         packet.extract<prot_h_t>(hdr.prot_h_0.next);
-        meta.hLeft = meta.hLeft - 8w1;
+        meta.hLeft = meta.hLeft + 8w255;
         meta.currPos = meta.currPos + 8w1;
         transition parse_prot_h_0_pre;
     }
