--- before_pass
+++ after_pass
@@ -36,10 +36,10 @@ control ingress(inout headers hdr, inout
         if (hdr.h == hdr.a[0]) {
             hdr.same.same = hdr.same.same | 8w4;
         }
-        H[2] tmp;
-        tmp[0] = hdr.h;
-        tmp[1] = hdr.a[0];
-        if (tmp == hdr.a) {
+        H[2] tmp_0;
+        tmp_0[0] = hdr.h;
+        tmp_0[1] = hdr.a[0];
+        if (tmp_0 == hdr.a) {
             hdr.same.same = hdr.same.same | 8w8;
         }
     }
