--- before_pass
+++ after_pass
@@ -16,9 +16,9 @@ struct headers {
 }
 parser p(packet_in b, out headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
     state start {
-        b.extract(hdr.h, 32w32);
-        b.extract(hdr.a.next, 32w32);
-        b.extract(hdr.a.next, 32w32);
+        b.extract<H>(hdr.h, 32w32);
+        b.extract<H>(hdr.a.next, 32w32);
+        b.extract<H>(hdr.a.next, 32w32);
         transition accept;
     }
 }
@@ -58,7 +58,7 @@ control uc(inout headers hdr, inout meta
 }
 control deparser(packet_out packet, in headers hdr) {
     apply {
-        packet.emit(hdr);
+        packet.emit<headers>(hdr);
     }
 }
 V1Switch<headers, metadata>(p(), vc(), ingress(), egress(), uc(), deparser()) main;
