--- before_pass
+++ after_pass
@@ -4,20 +4,14 @@ control ctrl(out bit<32> c) {
     bit<32> b_0;
     @name("e") action e_0() {
         exit;
-        ;
     }
     apply {
         a_0 = 32w0;
-        ;
         c = 32w2;
         if (a_0 == 32w0) {
-            ;
             e_0();
-            ;
         } else {
-            ;
             e_0();
-            ;
         }
         c = 32w5;
     }
