--- before_pass
+++ after_pass
@@ -1,15 +1,15 @@
 control ctrl(out bit<32> c) {
-    bit<32> a_0;
-    @name("e") action e_0() {
+    bit<32> a;
+    @name("e") action e() {
         exit;
     }
     apply {
-        a_0 = 32w0;
+        a = 32w0;
         c = 32w2;
-        if (a_0 == 32w0) {
-            e_0();
+        if (a == 32w0) {
+            e();
         } else {
-            e_0();
+            e();
         }
         c = 32w5;
     }
