--- before_pass
+++ after_pass
@@ -2,24 +2,24 @@ control ctrl(out bit<32> c) {
     bit<32> x;
     action e() {
         exit;
-        x = 1;
+        x = (bit<32>)32w1;
     }
     apply {
         bit<32> a;
         bit<32> b;
-        a = 0;
-        b = 1;
-        c = 2;
-        if (a == 0) {
-            b = 2;
+        a = (bit<32>)32w0;
+        b = (bit<32>)32w1;
+        c = (bit<32>)32w2;
+        if (a == 32w0) {
+            b = (bit<32>)32w2;
             e();
-            c = 3;
+            c = (bit<32>)32w3;
         } else {
-            b = 3;
+            b = (bit<32>)32w3;
             e();
-            c = 4;
+            c = (bit<32>)32w4;
         }
-        c = 5;
+        c = (bit<32>)32w5;
     }
 }
 control noop(out bit<32> c);
