--- before_pass
+++ after_pass
@@ -6,11 +6,11 @@ control c(out H[2] h);
 package top(c _c);
 control my(out H[2] s) {
     apply {
-        bit<32> a = 0;
-        s[a].z = 1;
-        s[a + 1].z = 0;
-        a = f(s[a].z, 0);
-        a = f(s[a].z, 1);
+        bit<32> a = (bit<32>)32w0;
+        s[a].z = (bit<1>)1w1;
+        s[a + 32w1].z = (bit<1>)1w0;
+        a = f(s[a].z, 1w0);
+        a = f(s[a].z, 1w1);
     }
 }
 top(my()) main;
