--- before_pass
+++ after_pass
@@ -9,23 +9,13 @@ control my(out H[2] s) {
     bit<32> tmp;
     bit<32> tmp_0;
     apply {
-        {
-            a_0 = 32w0;
-        }
-        {
-            s[a_0].z = 1w1;
-        }
-        {
-            s[a_0 + 32w1].z = 1w0;
-        }
-        {
-            tmp = f(s[a_0].z, 1w0);
-            a_0 = tmp;
-        }
-        {
-            tmp_0 = f(s[a_0].z, 1w1);
-            a_0 = tmp_0;
-        }
+        a_0 = 32w0;
+        s[a_0].z = 1w1;
+        s[a_0 + 32w1].z = 1w0;
+        tmp = f(s[a_0].z, 1w0);
+        a_0 = tmp;
+        tmp_0 = f(s[a_0].z, 1w1);
+        a_0 = tmp_0;
     }
 }
 top(my()) main;
