--- before_pass
+++ after_pass
@@ -19,7 +19,9 @@ parser EmptyParser(packet_in b, out head
 }
 control EmptyVerifyChecksum(inout headers_t hdr, inout metadata_t meta) {
     apply {
-        verify_checksum<tuple<bit<16>>, bit<16>>(false, { 16w0 }, meta.cksum.result, HashAlgorithm.csum16);
+        {
+            verify_checksum<tuple<bit<16>>, bit<16>>(false, { 16w0 }, meta.cksum.result, HashAlgorithm.csum16);
+        }
     }
 }
 control EmptyIngress(inout headers_t headers, inout metadata_t meta, inout standard_metadata_t standard_metadata) {
@@ -28,14 +30,22 @@ control EmptyIngress(inout headers_t hea
 }
 control EmptyEgress(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t standard_metadata) {
     apply {
-        mark_to_drop(standard_metadata);
+        {
+            mark_to_drop(standard_metadata);
+        }
     }
 }
 control EmptyComputeChecksum(inout headers_t hdr, inout metadata_t meta) {
     apply {
-        update_checksum<tuple<bit<16>>, bit<16>>(false, { 16w0 }, meta.cksum.result, HashAlgorithm.csum16);
-        update_checksum<tuple<bit<16>>, bit<16>>(hdr.e.isValid(), { 16w0 }, meta.cksum.result, HashAlgorithm.csum16);
-        update_checksum<tuple<bit<16>>, bit<16>>(meta.b == 1w0, { 16w0 }, meta.cksum.result, HashAlgorithm.csum16);
+        {
+            update_checksum<tuple<bit<16>>, bit<16>>(false, { 16w0 }, meta.cksum.result, HashAlgorithm.csum16);
+        }
+        {
+            update_checksum<tuple<bit<16>>, bit<16>>(hdr.e.isValid(), { 16w0 }, meta.cksum.result, HashAlgorithm.csum16);
+        }
+        {
+            update_checksum<tuple<bit<16>>, bit<16>>(meta.b == 1w0, { 16w0 }, meta.cksum.result, HashAlgorithm.csum16);
+        }
     }
 }
 control EmptyDeparser(packet_out b, in headers_t hdr) {
