--- before_pass
+++ after_pass
@@ -35,7 +35,7 @@ control EmptyComputeChecksum(inout heade
     apply {
         update_checksum(false, { 16w0 }, meta.cksum.result, HashAlgorithm.csum16);
         update_checksum(hdr.e.isValid(), { 16w0 }, meta.cksum.result, HashAlgorithm.csum16);
-        update_checksum(meta.b == 0, { 16w0 }, meta.cksum.result, HashAlgorithm.csum16);
+        update_checksum(meta.b == 1w0, { 16w0 }, meta.cksum.result, HashAlgorithm.csum16);
     }
 }
 control EmptyDeparser(packet_out b, in headers_t hdr) {
