--- before_pass
+++ after_pass
@@ -33,7 +33,7 @@ control ingressImpl(inout headers_t hdr,
         key = {
             hdr.ethernet.srcAddr[22:18]            : exact @name("ethernet.srcAddr.slice") ;
             hdr.ethernet.dstAddr & 48w0x10101010101: exact @name("dstAddr_lsbs") ;
-            hdr.ethernet.etherType - 16w10         : exact @name("etherType_less_10") ;
+            hdr.ethernet.etherType + 16w65526      : exact @name("etherType_less_10") ;
         }
         actions = {
             foo();
