--- before_pass
+++ after_pass
@@ -47,10 +47,10 @@ control ingress(inout Header_t h, inout
             h.h.e: exact;
         }
         actions = {
-            a;
-            a_with_control_params;
+            a();
+            a_with_control_params();
         }
-        default_action = a;
+        default_action = a();
         const entries = {
                         0x1 : a_with_control_params(1);
                         0x2 : a_with_control_params(2);
