--- before_pass
+++ after_pass
@@ -12,7 +12,7 @@ struct headers {
 }
 parser MyParser(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     state start {
-        packet.extract(hdr.stack.next);
+        packet.extract<elem>(hdr.stack.next);
         meta.temp = hdr.stack.last;
         transition accept;
     }
@@ -35,7 +35,7 @@ control MyComputeChecksum(inout headers
 }
 control MyDeparser(packet_out packet, in headers hdr) {
     apply {
-        packet.emit(hdr.stack);
+        packet.emit<elem[100]>(hdr.stack);
     }
 }
-V1Switch(MyParser(), MyVerifyChecksum(), MyIngress(), MyEgress(), MyComputeChecksum(), MyDeparser()) main;
+V1Switch<headers, metadata>(MyParser(), MyVerifyChecksum(), MyIngress(), MyEgress(), MyComputeChecksum(), MyDeparser()) main;
