--- before_pass
+++ after_pass
@@ -16,9 +16,9 @@ control ingress(inout H pkt_hdr, in Meta
     bit<32> input_traffic_bytes_tmp_0;
     bit<32> sum_rtt_Tr_tmp_0;
     bit<32> num_pkts_with_rtt_tmp_0;
-    @name("input_traffic_bytes") Register<bit<32>>(32w1) input_traffic_bytes_0;
-    @name("sum_rtt_Tr") Register<bit<32>>(32w1) sum_rtt_Tr_0;
-    @name("num_pkts_with_rtt") Register<bit<32>>(32w1) num_pkts_with_rtt_0;
+    @name("ingress.input_traffic_bytes") Register<bit<32>>(32w1) input_traffic_bytes_0;
+    @name("ingress.sum_rtt_Tr") Register<bit<32>>(32w1) sum_rtt_Tr_0;
+    @name("ingress.num_pkts_with_rtt") Register<bit<32>>(32w1) num_pkts_with_rtt_0;
     apply {
         @atomic {
             input_traffic_bytes_0.read(32w0, input_traffic_bytes_tmp_0);
