--- before_pass
+++ after_pass
@@ -76,12 +76,12 @@ action ac() {
     a = fct(a + b, a + c);
     a = fct(a, fct(b, a) + c);
     a = fct(a + b * c, a * (b + c));
-    a = (Bit)b;
-    a = (Bit)b + c;
-    a = (Bit)(b + c);
-    a = (Bit)f.z + b;
-    a = (Bit)fct((Bit)f.z + b, b + c);
+    a = b;
+    a = b + c;
+    a = b + c;
+    a = f.z + b;
+    a = fct(f.z + b, b + c);
     f = s {z = a + b,w = c};
     g = t {s1 = s {z = a + b,w = c},s2 = s {z = a + b,w = c}};
-    g = t {s1 = s {z = (Bit)(a + b) + b,w = c},s2 = s {z = a + (b + c),w = c}};
+    g = t {s1 = s {z = a + b + b,w = c},s2 = s {z = a + (b + c),w = c}};
 }
