--- before_pass
+++ after_pass
@@ -81,7 +81,7 @@ action ac() {
     a = (Bit)(b + c);
     a = (Bit)f.z + b;
     a = (Bit)fct((Bit)f.z + b, b + c);
-    f = { a + b, c };
-    g = { { a + b, c }, { a + b, c } };
-    g = { { (Bit)(a + b) + b, c }, { a + (b + c), c } };
+    f = s {z = a + b,w = c};
+    g = t {s1 = s {z = a + b,w = c},s2 = s {z = a + b,w = c}};
+    g = t {s1 = s {z = (Bit)(a + b) + b,w = c},s2 = s {z = a + (b + c),w = c}};
 }
