--- before_pass
+++ after_pass
@@ -16,10 +16,8 @@ control c(inout bit<32> x) {
             } else {
                 tmp = a_0;
             }
-            {
-                hasReturned = true;
-                retval = tmp;
-            }
+            hasReturned = true;
+            retval = tmp;
             tmp_0 = retval;
         }
         tmp_1 = tmp_0;
@@ -34,10 +32,8 @@ control c(inout bit<32> x) {
             } else {
                 tmp = a_1;
             }
-            {
-                hasReturned = true;
-                retval = tmp;
-            }
+            hasReturned = true;
+            retval = tmp;
             tmp_2 = retval;
         }
         tmp_3 = tmp_2;
@@ -52,10 +48,8 @@ control c(inout bit<32> x) {
             } else {
                 tmp = a_2;
             }
-            {
-                hasReturned = true;
-                retval = tmp;
-            }
+            hasReturned = true;
+            retval = tmp;
             tmp_4 = retval;
         }
         x = tmp_4;
