--- before_pass
+++ after_pass
@@ -1,58 +1,58 @@
 control c(inout bit<32> x) {
-    bit<32> tmp_0;
-    bit<32> tmp_1;
-    bit<32> tmp_2;
-    bit<32> tmp_3;
-    bit<32> tmp_4;
+    bit<32> tmp_5;
+    bit<32> tmp_6;
+    bit<32> tmp_7;
+    bit<32> tmp_8;
+    bit<32> tmp_9;
     apply {
         {
-            bit<32> a_0 = x;
-            bit<32> b_0 = x + 32w1;
-            bool hasReturned = false;
-            bit<32> retval;
-            bit<32> tmp;
-            if (a_0 > b_0) {
-                tmp = b_0;
+            bit<32> a = x;
+            bit<32> b = x + 32w1;
+            bool hasReturned_0 = false;
+            bit<32> retval_0;
+            bit<32> tmp_10;
+            if (a > b) {
+                tmp_10 = b;
             } else {
-                tmp = a_0;
+                tmp_10 = a;
             }
-            hasReturned = true;
-            retval = tmp;
-            tmp_0 = retval;
+            hasReturned_0 = true;
+            retval_0 = tmp_10;
+            tmp_5 = retval_0;
         }
-        tmp_1 = tmp_0;
+        tmp_6 = tmp_5;
         {
-            bit<32> a_1 = x;
-            bit<32> b_1 = x + 32w4294967295;
-            bool hasReturned = false;
-            bit<32> retval;
-            bit<32> tmp;
-            if (a_1 > b_1) {
-                tmp = b_1;
+            bit<32> a_3 = x;
+            bit<32> b_3 = x + 32w4294967295;
+            bool hasReturned_0 = false;
+            bit<32> retval_0;
+            bit<32> tmp_10;
+            if (a_3 > b_3) {
+                tmp_10 = b_3;
             } else {
-                tmp = a_1;
+                tmp_10 = a_3;
             }
-            hasReturned = true;
-            retval = tmp;
-            tmp_2 = retval;
+            hasReturned_0 = true;
+            retval_0 = tmp_10;
+            tmp_7 = retval_0;
         }
-        tmp_3 = tmp_2;
+        tmp_8 = tmp_7;
         {
-            bit<32> a_2 = tmp_1;
-            bit<32> b_2 = tmp_3;
-            bool hasReturned = false;
-            bit<32> retval;
-            bit<32> tmp;
-            if (a_2 > b_2) {
-                tmp = b_2;
+            bit<32> a_4 = tmp_6;
+            bit<32> b_4 = tmp_8;
+            bool hasReturned_0 = false;
+            bit<32> retval_0;
+            bit<32> tmp_10;
+            if (a_4 > b_4) {
+                tmp_10 = b_4;
             } else {
-                tmp = a_2;
+                tmp_10 = a_4;
             }
-            hasReturned = true;
-            retval = tmp;
-            tmp_4 = retval;
+            hasReturned_0 = true;
+            retval_0 = tmp_10;
+            tmp_9 = retval_0;
         }
-        x = tmp_4;
+        x = tmp_9;
     }
 }
 control ctr(inout bit<32> x);
