--- before_pass
+++ after_pass
@@ -19,23 +19,23 @@ control MyVerifyChecksum(inout headers h
     }
 }
 control C(inout headers hdr, inout metadata meta)(bool b) {
-    register<bit<16>>(32w8) r;
+    @name("r") register<bit<16>>(32w8) r_0;
     apply {
-        r.read(hdr.h.x, 32w0);
+        r_0.read(hdr.h.x, 32w0);
     }
 }
 control H(inout headers hdr, inout metadata meta) {
-    C(true) c1;
-    C(false) c2;
+    @name("c1") C(true) c1_0;
+    @name("c2") C(false) c2_0;
     apply {
-        c1.apply(hdr, meta);
-        c2.apply(hdr, meta);
+        c1_0.apply(hdr, meta);
+        c2_0.apply(hdr, meta);
     }
 }
 control MyIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    H() h;
+    @name("h") H() h_0;
     apply {
-        h.apply(hdr, meta);
+        h_0.apply(hdr, meta);
     }
 }
 control MyEgress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
