--- before_pass
+++ after_pass
@@ -10,7 +10,7 @@ struct metadata {
 }
 parser MyParser(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     state start {
-        hdr.h.x = 0;
+        hdr.h.x = (bit<16>)16w0;
         transition accept;
     }
 }
@@ -21,7 +21,7 @@ control MyVerifyChecksum(inout headers h
 control C(inout headers hdr, inout metadata meta)(bool b) {
     register<bit<16>>(32w8) r;
     apply {
-        r.read(hdr.h.x, 0);
+        r.read(hdr.h.x, 32w0);
     }
 }
 control H(inout headers hdr, inout metadata meta) {
