--- before_pass
+++ after_pass
@@ -22,14 +22,8 @@ control MyIngress(inout headers hdr, ino
     @name("h.c1.r") register<bit<16>>(32w8) h_c1_r;
     @name("h.c2.r") register<bit<16>>(32w8) h_c2_r;
     apply {
-        {
-            {
-                h_c1_r.read(hdr.h.x, 32w0);
-            }
-            {
-                h_c2_r.read(hdr.h.x, 32w0);
-            }
-        }
+        h_c1_r.read(hdr.h.x, 32w0);
+        h_c2_r.read(hdr.h.x, 32w0);
     }
 }
 control MyEgress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
