--- before_pass
+++ after_pass
@@ -10,7 +10,9 @@ struct metadata {
 }
 parser MyParser(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     state start {
-        hdr.h.x = 16w0;
+        {
+            hdr.h.x = 16w0;
+        }
         transition accept;
     }
 }
@@ -21,21 +23,29 @@ control MyVerifyChecksum(inout headers h
 control C(inout headers hdr, inout metadata meta)(bool b) {
     @name("r") register<bit<16>>(32w8) r_0;
     apply {
-        r_0.read(hdr.h.x, 32w0);
+        {
+            r_0.read(hdr.h.x, 32w0);
+        }
     }
 }
 control H(inout headers hdr, inout metadata meta) {
     @name("c1") C(true) c1_0;
     @name("c2") C(false) c2_0;
     apply {
-        c1_0.apply(hdr, meta);
-        c2_0.apply(hdr, meta);
+        {
+            c1_0.apply(hdr, meta);
+        }
+        {
+            c2_0.apply(hdr, meta);
+        }
     }
 }
 control MyIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     @name("h") H() h_0;
     apply {
-        h_0.apply(hdr, meta);
+        {
+            h_0.apply(hdr, meta);
+        }
     }
 }
 control MyEgress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
