--- before_pass
+++ after_pass
@@ -20,7 +20,7 @@ control ingress(inout parsed_packet_t h,
     bitvec_hdr bh;
     apply {
         bh.x = h.bvh0.x;
-        clone3(CloneType.I2E, 32w0, h);
+        clone3<parsed_packet_t>(CloneType.I2E, 32w0, h);
     }
 }
 control egress(inout parsed_packet_t hdr, inout local_metadata_t local_metadata, inout standard_metadata_t standard_metadata) {
@@ -39,4 +39,4 @@ control compute_checksum(inout parsed_pa
     apply {
     }
 }
-V1Switch(parse(), verifyChecksum(), ingress(), egress(), compute_checksum(), deparser()) main;
+V1Switch<parsed_packet_t, local_metadata_t>(parse(), verifyChecksum(), ingress(), egress(), compute_checksum(), deparser()) main;
