--- before_pass
+++ after_pass
@@ -19,8 +19,12 @@ parser parse(packet_in pk, out parsed_pa
 control ingress(inout parsed_packet_t h, inout local_metadata_t local_metadata, inout standard_metadata_t standard_metadata) {
     bitvec_hdr bh_0;
     apply {
-        bh_0.x = h.bvh0.x;
-        clone3<parsed_packet_t>(CloneType.I2E, 32w0, h);
+        {
+            bh_0.x = h.bvh0.x;
+        }
+        {
+            clone3<parsed_packet_t>(CloneType.I2E, 32w0, h);
+        }
     }
 }
 control egress(inout parsed_packet_t hdr, inout local_metadata_t local_metadata, inout standard_metadata_t standard_metadata) {
