--- before_pass
+++ after_pass
@@ -40,13 +40,11 @@ control deparser(packet_out b, in Header
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
     Choice c_c;
     apply {
-        {
-            c_c = Choice.First;
-            if (c_c == Choice.Second) {
-                h.h.c = h.h.a;
-            } else {
-                h.h.c = h.h.b;
-            }
+        c_c = Choice.First;
+        if (c_c == Choice.Second) {
+            h.h.c = h.h.a;
+        } else {
+            h.h.c = h.h.b;
         }
         sm.egress_spec = 9w0;
     }
