--- before_pass
+++ after_pass
@@ -34,18 +34,18 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    action compare() {
+    @name("compare") action compare_0() {
         h.h.c = (bit<8>)(bit<1>)(h.h.a < h.h.b);
         sm.egress_spec = 9w0;
     }
-    table t {
+    @name("t") table t_0 {
         actions = {
-            compare();
+            compare_0();
         }
-        const default_action = compare();
+        const default_action = compare_0();
     }
     apply {
-        t.apply();
+        t_0.apply();
     }
 }
 V1Switch<Headers, Meta>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
