--- before_pass
+++ after_pass
@@ -29,30 +29,20 @@ parser ProtAddrParser(packet_in packet,
         }
     }
     state ipv4 {
-        {
-            packet.extract<addr_ipv4_t>(addr.ipv4);
-        }
+        packet.extract<addr_ipv4_t>(addr.ipv4);
         transition accept;
     }
     state ipv6 {
-        {
-            packet.extract<addr_ipv6_t>(addr.ipv6);
-        }
+        packet.extract<addr_ipv6_t>(addr.ipv6);
         transition accept;
     }
 }
 parser ProtParser(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     @name("addrParser") ProtAddrParser() addrParser_0;
     state start {
-        {
-            packet.extract<addr_type_t>(hdr.addr_type);
-        }
-        {
-            addrParser_0.apply(packet, hdr.addr_type.dstType, hdr.addr_dst);
-        }
-        {
-            addrParser_0.apply(packet, hdr.addr_type.srcType, hdr.addr_src);
-        }
+        packet.extract<addr_type_t>(hdr.addr_type);
+        addrParser_0.apply(packet, hdr.addr_type.dstType, hdr.addr_dst);
+        addrParser_0.apply(packet, hdr.addr_type.srcType, hdr.addr_src);
         transition accept;
     }
 }
@@ -74,9 +64,7 @@ control ProtComputeChecksum(inout header
 }
 control ProtDeparser(packet_out packet, in headers hdr) {
     apply {
-        {
-            packet.emit<headers>(hdr);
-        }
+        packet.emit<headers>(hdr);
     }
 }
 V1Switch<headers, metadata>(ProtParser(), ProtVerifyChecksum(), ProtIngress(), ProtEgress(), ProtComputeChecksum(), ProtDeparser()) main;
