--- before_pass
+++ after_pass
@@ -20,7 +20,7 @@ parser parserI(packet_in pkt, out Parsed
     }
 }
 control cIngress(inout Parsed_packet hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
-    @name("E.c1.stats") counter(32w65536, CounterType.packets) E_c1_stats;
+    @name("cIngress.E.c1.stats") counter(32w65536, CounterType.packets) E_c1_stats;
     apply {
         hdr.ethernet.etherType = hdr.ethernet.etherType << 1;
         hdr.ethernet.etherType = hdr.ethernet.etherType + 16w1;
