--- before_pass
+++ after_pass
@@ -34,14 +34,14 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name("c.add") action c_add_0() {
+    @name("c.add") action c_add() {
         h.h.c = (bit<64>)(h.h.a + h.h.b);
     }
     @name("c.t") table c_t_0 {
         actions = {
-            c_add_0();
+            c_add();
         }
-        const default_action = c_add_0();
+        const default_action = c_add();
     }
     apply {
         c_t_0.apply();
