--- before_pass
+++ after_pass
@@ -45,26 +45,45 @@ parser Parser(packet_in packet, out head
 }
 control Ingress(inout headers hdr, out bool xout) {
     bit<48> tmp_0;
-    @name("send_back") action send_back_0(bit<32> result_0) {
-        tmp_0 = hdr.ethernet.dstAddr;
-        hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
-        hdr.ethernet.srcAddr = tmp_0;
-        hdr.p4calc.res = result_0;
-    }
     @name("operation_add") action operation_add_0() {
-        send_back_0(hdr.p4calc.operand_a + hdr.p4calc.operand_b);
+        {
+            tmp_0 = hdr.ethernet.dstAddr;
+            hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
+            hdr.ethernet.srcAddr = tmp_0;
+            hdr.p4calc.res = hdr.p4calc.operand_a + hdr.p4calc.operand_b;
+        }
     }
     @name("operation_sub") action operation_sub_0() {
-        send_back_0(hdr.p4calc.operand_a - hdr.p4calc.operand_b);
+        {
+            tmp_0 = hdr.ethernet.dstAddr;
+            hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
+            hdr.ethernet.srcAddr = tmp_0;
+            hdr.p4calc.res = hdr.p4calc.operand_a - hdr.p4calc.operand_b;
+        }
     }
     @name("operation_and") action operation_and_0() {
-        send_back_0(hdr.p4calc.operand_a & hdr.p4calc.operand_b);
+        {
+            tmp_0 = hdr.ethernet.dstAddr;
+            hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
+            hdr.ethernet.srcAddr = tmp_0;
+            hdr.p4calc.res = hdr.p4calc.operand_a & hdr.p4calc.operand_b;
+        }
     }
     @name("operation_or") action operation_or_0() {
-        send_back_0(hdr.p4calc.operand_a | hdr.p4calc.operand_b);
+        {
+            tmp_0 = hdr.ethernet.dstAddr;
+            hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
+            hdr.ethernet.srcAddr = tmp_0;
+            hdr.p4calc.res = hdr.p4calc.operand_a | hdr.p4calc.operand_b;
+        }
     }
     @name("operation_xor") action operation_xor_0() {
-        send_back_0(hdr.p4calc.operand_a ^ hdr.p4calc.operand_b);
+        {
+            tmp_0 = hdr.ethernet.dstAddr;
+            hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
+            hdr.ethernet.srcAddr = tmp_0;
+            hdr.p4calc.res = hdr.p4calc.operand_a ^ hdr.p4calc.operand_b;
+        }
     }
     @name("operation_drop") action operation_drop_0() {
         xout = false;
