--- before_pass
+++ after_pass
@@ -29,7 +29,7 @@ struct headers {
 }
 parser Parser(packet_in packet, out headers hdr) {
     state start {
-        packet.extract(hdr.ethernet);
+        packet.extract<ethernet_t>(hdr.ethernet);
         transition select(hdr.ethernet.etherType) {
             16w0x1234: check_p4calc;
             default: accept;
@@ -42,7 +42,7 @@ parser Parser(packet_in packet, out head
         }
     }
     state parse_p4calc {
-        packet.extract(hdr.p4calc);
+        packet.extract<p4calc_t>(hdr.p4calc);
         transition accept;
     }
 }
@@ -103,4 +103,4 @@ control Ingress(inout headers hdr, out b
         }
     }
 }
-ebpfFilter(Parser(), Ingress()) main;
+ebpfFilter<headers>(Parser(), Ingress()) main;
