--- before_pass
+++ after_pass
@@ -45,37 +45,40 @@ parser Parser(packet_in packet, out head
 }
 control Ingress(inout headers hdr, out bool xout) {
     bit<48> tmp_6;
-    @name("operation_add") action operation_add() {
+    @name("operation_add") action operation_add_0() {
         tmp_6 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
         hdr.ethernet.srcAddr = tmp_6;
         hdr.p4calc.res = hdr.p4calc.operand_a + hdr.p4calc.operand_b;
     }
-    @name("operation_sub") action operation_sub() {
+    @name("operation_sub") action operation_sub_0() {
         tmp_6 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
         hdr.ethernet.srcAddr = tmp_6;
         hdr.p4calc.res = hdr.p4calc.operand_a - hdr.p4calc.operand_b;
     }
-    @name("operation_and") action operation_and() {
+    @name("operation_and") action operation_and_0() {
         tmp_6 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
         hdr.ethernet.srcAddr = tmp_6;
         hdr.p4calc.res = hdr.p4calc.operand_a & hdr.p4calc.operand_b;
     }
-    @name("operation_or") action operation_or() {
+    @name("operation_or") action operation_or_0() {
         tmp_6 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
         hdr.ethernet.srcAddr = tmp_6;
         hdr.p4calc.res = hdr.p4calc.operand_a | hdr.p4calc.operand_b;
     }
-    @name("operation_xor") action operation_xor() {
+    @name("operation_xor") action operation_xor_0() {
         tmp_6 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
         hdr.ethernet.srcAddr = tmp_6;
         hdr.p4calc.res = hdr.p4calc.operand_a ^ hdr.p4calc.operand_b;
     }
-    @name("operation_drop") action operation_drop() {
+    @name("operation_drop") action operation_drop_0() {
+        xout = false;
+    }
+    @name("operation_drop") action operation_drop_1() {
         xout = false;
     }
     @name("calculate") table calculate {
@@ -83,20 +86,20 @@ control Ingress(inout headers hdr, out b
             hdr.p4calc.op: exact @name("hdr.p4calc.op") ;
         }
         actions = {
-            operation_add();
-            operation_sub();
-            operation_and();
-            operation_or();
-            operation_xor();
-            operation_drop();
+            operation_add_0();
+            operation_sub_0();
+            operation_and_0();
+            operation_or_0();
+            operation_xor_0();
+            operation_drop_0();
         }
-        const default_action = operation_drop();
+        const default_action = operation_drop_0();
         const entries = {
-                        8w0x2b : operation_add();
-                        8w0x2d : operation_sub();
-                        8w0x26 : operation_and();
-                        8w0x7c : operation_or();
-                        8w0x5e : operation_xor();
+                        8w0x2b : operation_add_0();
+                        8w0x2d : operation_sub_0();
+                        8w0x26 : operation_and_0();
+                        8w0x7c : operation_or_0();
+                        8w0x5e : operation_xor_0();
         }
         implementation = hash_table(32w8);
     }
@@ -105,7 +108,7 @@ control Ingress(inout headers hdr, out b
         if (hdr.p4calc.isValid()) {
             calculate.apply();
         } else {
-            operation_drop();
+            operation_drop_1();
         }
     }
 }
