--- before_pass
+++ after_pass
@@ -19,9 +19,9 @@ struct headers {
     p4calc_t   p4calc;
 }
 parser Parser(packet_in packet, out headers hdr) {
-    p4calc_t tmp;
-    p4calc_t tmp_1;
-    p4calc_t tmp_2;
+    p4calc_t tmp_3;
+    p4calc_t tmp_4;
+    p4calc_t tmp_5;
     state start {
         packet.extract<ethernet_t>(hdr.ethernet);
         transition select(hdr.ethernet.etherType) {
@@ -30,10 +30,10 @@ parser Parser(packet_in packet, out head
         }
     }
     state check_p4calc {
-        tmp = packet.lookahead<p4calc_t>();
-        tmp_1 = packet.lookahead<p4calc_t>();
-        tmp_2 = packet.lookahead<p4calc_t>();
-        transition select(tmp.p, tmp_1.four, tmp_2.ver) {
+        tmp_3 = packet.lookahead<p4calc_t>();
+        tmp_4 = packet.lookahead<p4calc_t>();
+        tmp_5 = packet.lookahead<p4calc_t>();
+        transition select(tmp_3.p, tmp_4.four, tmp_5.ver) {
             (8w0x50, 8w0x34, 8w0x1): parse_p4calc;
             default: accept;
         }
@@ -44,68 +44,68 @@ parser Parser(packet_in packet, out head
     }
 }
 control Ingress(inout headers hdr, out bool xout) {
-    bit<48> tmp_0;
-    @name("operation_add") action operation_add_0() {
-        tmp_0 = hdr.ethernet.dstAddr;
+    bit<48> tmp_6;
+    @name("operation_add") action operation_add() {
+        tmp_6 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
-        hdr.ethernet.srcAddr = tmp_0;
+        hdr.ethernet.srcAddr = tmp_6;
         hdr.p4calc.res = hdr.p4calc.operand_a + hdr.p4calc.operand_b;
     }
-    @name("operation_sub") action operation_sub_0() {
-        tmp_0 = hdr.ethernet.dstAddr;
+    @name("operation_sub") action operation_sub() {
+        tmp_6 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
-        hdr.ethernet.srcAddr = tmp_0;
+        hdr.ethernet.srcAddr = tmp_6;
         hdr.p4calc.res = hdr.p4calc.operand_a - hdr.p4calc.operand_b;
     }
-    @name("operation_and") action operation_and_0() {
-        tmp_0 = hdr.ethernet.dstAddr;
+    @name("operation_and") action operation_and() {
+        tmp_6 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
-        hdr.ethernet.srcAddr = tmp_0;
+        hdr.ethernet.srcAddr = tmp_6;
         hdr.p4calc.res = hdr.p4calc.operand_a & hdr.p4calc.operand_b;
     }
-    @name("operation_or") action operation_or_0() {
-        tmp_0 = hdr.ethernet.dstAddr;
+    @name("operation_or") action operation_or() {
+        tmp_6 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
-        hdr.ethernet.srcAddr = tmp_0;
+        hdr.ethernet.srcAddr = tmp_6;
         hdr.p4calc.res = hdr.p4calc.operand_a | hdr.p4calc.operand_b;
     }
-    @name("operation_xor") action operation_xor_0() {
-        tmp_0 = hdr.ethernet.dstAddr;
+    @name("operation_xor") action operation_xor() {
+        tmp_6 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
-        hdr.ethernet.srcAddr = tmp_0;
+        hdr.ethernet.srcAddr = tmp_6;
         hdr.p4calc.res = hdr.p4calc.operand_a ^ hdr.p4calc.operand_b;
     }
-    @name("operation_drop") action operation_drop_0() {
+    @name("operation_drop") action operation_drop() {
         xout = false;
     }
-    @name("calculate") table calculate_0 {
+    @name("calculate") table calculate {
         key = {
             hdr.p4calc.op: exact @name("hdr.p4calc.op") ;
         }
         actions = {
-            operation_add_0();
-            operation_sub_0();
-            operation_and_0();
-            operation_or_0();
-            operation_xor_0();
-            operation_drop_0();
+            operation_add();
+            operation_sub();
+            operation_and();
+            operation_or();
+            operation_xor();
+            operation_drop();
         }
-        const default_action = operation_drop_0();
+        const default_action = operation_drop();
         const entries = {
-                        8w0x2b : operation_add_0();
-                        8w0x2d : operation_sub_0();
-                        8w0x26 : operation_and_0();
-                        8w0x7c : operation_or_0();
-                        8w0x5e : operation_xor_0();
+                        8w0x2b : operation_add();
+                        8w0x2d : operation_sub();
+                        8w0x26 : operation_and();
+                        8w0x7c : operation_or();
+                        8w0x5e : operation_xor();
         }
         implementation = hash_table(32w8);
     }
     apply {
         xout = true;
         if (hdr.p4calc.isValid()) {
-            calculate_0.apply();
+            calculate.apply();
         } else {
-            operation_drop_0();
+            operation_drop();
         }
     }
 }
