--- before_pass
+++ after_pass
@@ -45,43 +45,43 @@ parser Parser(packet_in packet, out head
 }
 control Ingress(inout headers hdr, out bool xout) {
     bit<48> tmp_2;
-    @name("operation_add") action operation_add() {
+    @name("Ingress.operation_add") action operation_add() {
         tmp_2 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
         hdr.ethernet.srcAddr = tmp_2;
         hdr.p4calc.res = hdr.p4calc.operand_a + hdr.p4calc.operand_b;
     }
-    @name("operation_sub") action operation_sub() {
+    @name("Ingress.operation_sub") action operation_sub() {
         tmp_2 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
         hdr.ethernet.srcAddr = tmp_2;
         hdr.p4calc.res = hdr.p4calc.operand_a - hdr.p4calc.operand_b;
     }
-    @name("operation_and") action operation_and() {
+    @name("Ingress.operation_and") action operation_and() {
         tmp_2 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
         hdr.ethernet.srcAddr = tmp_2;
         hdr.p4calc.res = hdr.p4calc.operand_a & hdr.p4calc.operand_b;
     }
-    @name("operation_or") action operation_or() {
+    @name("Ingress.operation_or") action operation_or() {
         tmp_2 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
         hdr.ethernet.srcAddr = tmp_2;
         hdr.p4calc.res = hdr.p4calc.operand_a | hdr.p4calc.operand_b;
     }
-    @name("operation_xor") action operation_xor() {
+    @name("Ingress.operation_xor") action operation_xor() {
         tmp_2 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
         hdr.ethernet.srcAddr = tmp_2;
         hdr.p4calc.res = hdr.p4calc.operand_a ^ hdr.p4calc.operand_b;
     }
-    @name("operation_drop") action operation_drop() {
+    @name("Ingress.operation_drop") action operation_drop() {
         xout = false;
     }
-    @name("operation_drop") action operation_drop_2() {
+    @name("Ingress.operation_drop") action operation_drop_2() {
         xout = false;
     }
-    @name("calculate") table calculate_0 {
+    @name("Ingress.calculate") table calculate_0 {
         key = {
             hdr.p4calc.op: exact @name("hdr.p4calc.op") ;
         }
