--- before_pass
+++ after_pass
@@ -77,12 +77,12 @@ control Ingress(inout headers hdr, out b
             hdr.p4calc.op: exact;
         }
         actions = {
-            operation_add;
-            operation_sub;
-            operation_and;
-            operation_or;
-            operation_xor;
-            operation_drop;
+            operation_add();
+            operation_sub();
+            operation_and();
+            operation_or();
+            operation_xor();
+            operation_drop();
         }
         const default_action = operation_drop();
         const entries = {
