--- before_pass
+++ after_pass
@@ -42,8 +42,9 @@ parser InnerParser(packet_in pkt, out he
     }
 }
 parser OuterParser(packet_in pkt, out headers_t hdr, inout meta_t m, inout standard_metadata_t meta) {
+    @name("InnerParser") InnerParser() InnerParser_inst;
     state start {
-        InnerParser.apply(pkt, hdr);
+        InnerParser_inst.apply(pkt, hdr);
         transition accept;
     }
 }
