--- before_pass
+++ after_pass
@@ -39,9 +39,9 @@ parser InnerParser(packet_in pkt, out he
     }
 }
 parser OuterParser(packet_in pkt, out headers_t hdr, inout meta_t m, inout standard_metadata_t meta) {
-    @name("InnerParser") InnerParser() InnerParser_inst;
+    @name("InnerParser") InnerParser() InnerParser_inst_0;
     state start {
-        InnerParser_inst.apply(pkt, hdr);
+        InnerParser_inst_0.apply(pkt, hdr);
         transition accept;
     }
 }
