--- before_pass
+++ after_pass
@@ -79,22 +79,22 @@ parser parserI(packet_in pkt, out header
     }
 }
 control cIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
-    action foo() {
+    @name("foo") action foo_0() {
         hdr.tcp.srcPort = hdr.tcp.srcPort + 16w1;
         hdr.ipv4.ttl = hdr.ipv4.ttl + 8w255;
         hdr.ipv4.dstAddr = hdr.ipv4.dstAddr + 32w4;
     }
-    table guh {
+    @name("guh") table guh_0 {
         key = {
             hdr.tcp.dstPort: exact @name("hdr.tcp.dstPort") ;
         }
         actions = {
-            foo();
+            foo_0();
         }
-        default_action = foo();
+        default_action = foo_0();
     }
     apply {
-        guh.apply();
+        guh_0.apply();
     }
 }
 control cEgress(inout headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
