--- before_pass
+++ after_pass
@@ -14,9 +14,9 @@ parser ParserImpl(packet_in packet, out
 control IngressImpl(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("act") action act(test_t a) {
+    @name("IngressImpl.act") action act(test_t a) {
     }
-    @name("test_table") table test_table_0 {
+    @name("IngressImpl.test_table") table test_table_0 {
         key = {
             meta.test: exact @name("meta.test") ;
         }
