--- before_pass
+++ after_pass
@@ -19,11 +19,13 @@ control IngressImpl(inout headers hdr, i
             meta.test: exact;
         }
         actions = {
-            act;
+            act();
+            @defaultonly NoAction();
         }
         const entries = {
                         1 : act(1);
         }
+        default_action = NoAction();
     }
     apply {
         test_table.apply();
