--- before_pass
+++ after_pass
@@ -3,18 +3,18 @@ header Header {
     bit<32> data;
 }
 parser p0(packet_in p, out Header h) {
-    bool b;
+    bool b_0;
     state start {
-        b = true;
+        b_0 = true;
         p.extract<Header>(h);
-        transition select(h.data, b) {
+        transition select(h.data, b_0) {
             (default, true): next;
             (default, default): reject;
         }
     }
     state next {
         p.extract<Header>(h);
-        transition select(h.data, b) {
+        transition select(h.data, b_0) {
             (default, true): accept;
             (default, default): reject;
             default: reject;
