--- before_pass
+++ after_pass
@@ -5,14 +5,14 @@ header Header {
 parser p0(packet_in p, out Header h) {
     bool b = true;
     state start {
-        p.extract(h);
+        p.extract<Header>(h);
         transition select(h.data, b) {
             (default, true): next;
             (default, default): reject;
         }
     }
     state next {
-        p.extract(h);
+        p.extract<Header>(h);
         transition select(h.data, b) {
             (default, true): accept;
             (default, default): reject;
