--- before_pass
+++ after_pass
@@ -9,8 +9,8 @@ parser Parser(packet_in b, out Parsed_Pa
     state start {
         transition select(8w0, b.lookahead<bit<8>>()) {
             default: accept;
-            (0, 0 &&& 0): accept;
-            (0 &&& 0, 0x44): ipv4_option_NOP;
+            (8w0, 8w0 &&& 8w0): accept;
+            (8w0 &&& 8w0, 8w0x44): ipv4_option_NOP;
         }
     }
     state ipv4_option_NOP {
