--- before_pass
+++ after_pass
@@ -27,12 +27,12 @@ control ingress(inout Headers h, inout M
     }
     table t {
         actions = {
-            case0;
-            case1;
-            case2;
-            case3;
+            case0();
+            case1();
+            case2();
+            case3();
         }
-        const default_action = case0;
+        const default_action = case0();
     }
     apply {
         t.apply();
