--- before_pass
+++ after_pass
@@ -13,26 +13,26 @@ struct Meta {
     bit<8> y;
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name("case0") action case0() {
+    @name("case0") action case0_0() {
         h.h.c = ((bit<12>)h.h.a)[11:4];
     }
-    @name("case1") action case1() {
+    @name("case1") action case1_0() {
         h.h.c = (bit<8>)h.h.a[14:3][8:4];
     }
-    @name("case2") action case2() {
+    @name("case2") action case2_0() {
         h.h.c = (bit<8>)(16w0 ++ h.h.a[15:8]);
     }
-    @name("case3") action case3() {
+    @name("case3") action case3_0() {
         h.h.c = ((int<32>)(int<16>)h.h.a)[14:2][10:3];
     }
     @name("t") table t {
         actions = {
-            case0();
-            case1();
-            case2();
-            case3();
+            case0_0();
+            case1_0();
+            case2_0();
+            case3_0();
         }
-        const default_action = case0();
+        const default_action = case0_0();
     }
     apply {
         t.apply();
