--- before_pass
+++ after_pass
@@ -25,11 +25,11 @@ control egress(inout Headers h, inout Me
 }
 control deparser(packet_out b, in Headers h) {
     apply {
-        b.emit(h);
+        b.emit<Headers>(h);
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
     apply {
     }
 }
-V1Switch(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
+V1Switch<Headers, Meta>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
