--- before_pass
+++ after_pass
@@ -75,12 +75,12 @@ parser parserI(packet_in pkt, out header
 }
 control cIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
     apply {
-        stdmeta.egress_spec = (bit<9>)9w0;
+        stdmeta.egress_spec = 9w0;
         if (stdmeta.checksum_error == 1w1) {
-            hdr.ethernet.srcAddr = (bit<48>)48w0xbad;
+            hdr.ethernet.srcAddr = 48w0xbad;
         }
         if (stdmeta.parser_error != error.NoError) {
-            hdr.ethernet.dstAddr = (bit<48>)48w0xbad;
+            hdr.ethernet.dstAddr = 48w0xbad;
         }
         if (hdr.ipv4.isValid()) {
             hdr.ipv4.ttl = hdr.ipv4.ttl |-| 8w1;
