--- before_pass
+++ after_pass
@@ -1,12 +1,12 @@
 #include <core.p4>
 #include <v1model.p4>
 enum bit<8> MyEnum1B {
-    MBR1 = 0,
-    MBR2 = 0xff
+    MBR1 = 8w0,
+    MBR2 = 8w0xff
 }
 enum bit<16> MyEnum2B {
-    MBR1 = 10,
-    MBR2 = 0xab00
+    MBR1 = 16w10,
+    MBR2 = 16w0xab00
 }
 header hdr {
     MyEnum1B f1;
@@ -42,7 +42,7 @@ control deparser(packet_out b, in Header
 }
 control ingress(inout Header_t h, inout Meta_t m, inout standard_metadata_t standard_meta) {
     action a() {
-        standard_meta.egress_spec = 0;
+        standard_meta.egress_spec = (bit<9>)9w0;
     }
     action a_with_control_params(bit<9> x) {
         standard_meta.egress_spec = x;
@@ -58,9 +58,9 @@ control ingress(inout Header_t h, inout
         }
         const default_action = a();
         const entries = {
-                        (MyEnum1B.MBR1, default) : a_with_control_params(1);
-                        (MyEnum1B.MBR2, (bit<16>)MyEnum2B.MBR2 &&& 0xff00) : a_with_control_params(2);
-                        (MyEnum1B.MBR2, default) : a_with_control_params(3);
+                        (MyEnum1B.MBR1, default) : a_with_control_params(9w1);
+                        (MyEnum1B.MBR2, (bit<16>)MyEnum2B.MBR2 &&& 16w0xff00) : a_with_control_params(9w2);
+                        (MyEnum1B.MBR2, default) : a_with_control_params(9w3);
         }
     }
     apply {
