--- before_pass
+++ after_pass
@@ -19,7 +19,7 @@ struct Meta_t {
 }
 parser p(packet_in b, out Header_t h, inout Meta_t m, inout standard_metadata_t sm) {
     state start {
-        b.extract(h.h);
+        b.extract<hdr>(h.h);
         transition accept;
     }
 }
@@ -37,7 +37,7 @@ control egress(inout Header_t h, inout M
 }
 control deparser(packet_out b, in Header_t h) {
     apply {
-        b.emit(h.h);
+        b.emit<hdr>(h.h);
     }
 }
 control ingress(inout Header_t h, inout Meta_t m, inout standard_metadata_t standard_meta) {
@@ -67,4 +67,4 @@ control ingress(inout Header_t h, inout
         t_ternary.apply();
     }
 }
-V1Switch(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
+V1Switch<Header_t, Meta_t>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
