--- before_pass
+++ after_pass
@@ -41,10 +41,10 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Header_t h, inout Meta_t m, inout standard_metadata_t standard_meta) {
-    @name("a") action a() {
+    @name("a") action a_0() {
         standard_meta.egress_spec = 9w0;
     }
-    @name("a_with_control_params") action a_with_control_params(bit<9> x) {
+    @name("a_with_control_params") action a_with_control_params_0(bit<9> x) {
         standard_meta.egress_spec = x;
     }
     @name("t_ternary") table t_ternary {
@@ -53,14 +53,14 @@ control ingress(inout Header_t h, inout
             h.h.f2: ternary @name("h.h.f2") ;
         }
         actions = {
-            a();
-            a_with_control_params();
+            a_0();
+            a_with_control_params_0();
         }
-        const default_action = a();
+        const default_action = a_0();
         const entries = {
-                        (MyEnum1B.MBR1, default) : a_with_control_params(9w1);
-                        (MyEnum1B.MBR2, (bit<16>)MyEnum2B.MBR2 &&& 16w0xff00) : a_with_control_params(9w2);
-                        (MyEnum1B.MBR2, default) : a_with_control_params(9w3);
+                        (MyEnum1B.MBR1, default) : a_with_control_params_0(9w1);
+                        (MyEnum1B.MBR2, (bit<16>)MyEnum2B.MBR2 &&& 16w0xff00) : a_with_control_params_0(9w2);
+                        (MyEnum1B.MBR2, default) : a_with_control_params_0(9w3);
         }
     }
     apply {
