--- before_pass
+++ after_pass
@@ -1,6 +1,8 @@
 control c(inout bit<32> x) {
     @name("a") action a_0(in bit<32> arg) {
-        x = arg;
+        {
+            x = arg;
+        }
     }
     @name("t") table t_0 {
         actions = {
@@ -9,7 +11,9 @@ control c(inout bit<32> x) {
         default_action = a_0(32w10);
     }
     apply {
-        t_0.apply();
+        {
+            t_0.apply();
+        }
     }
 }
 control proto(inout bit<32> arg);
