--- before_pass
+++ after_pass
@@ -13,12 +13,14 @@ extern E {
 }
 control c(inout S s) {
     @name("e") E() e_0;
+    bit<32> tmp;
     apply {
         if (s.h.isValid()) {
             s.h.data3 = 32w0;
         }
         if (s.h.data2 == 32w0) {
-            s.h.data1 = e_0.get<bit<32>>(s.h.data2);
+            tmp = e_0.get<bit<32>>(s.h.data2);
+            s.h.data1 = tmp;
         }
     }
 }
