--- before_pass
+++ after_pass
@@ -12,13 +12,13 @@ extern E {
     bit<32> get<T>(in T b);
 }
 control c(inout S s) {
-    E() e;
+    @name("e") E() e_0;
     apply {
         if (s.h.isValid()) {
             s.h.data3 = 32w0;
         }
         if (s.h.data2 == 32w0) {
-            s.h.data1 = e.get<bit<32>>(s.h.data2);
+            s.h.data1 = e_0.get<bit<32>>(s.h.data2);
         }
     }
 }
