--- before_pass
+++ after_pass
@@ -12,15 +12,15 @@ extern E {
     bit<32> get<T>(in T b);
 }
 control c(inout S s) {
-    bit<32> tmp_0;
-    @name("e") E() e;
+    bit<32> tmp;
+    @name("e") E() e_0;
     apply {
         if (s.h.isValid()) {
             s.h.data3 = 32w0;
         }
         if (s.h.data2 == 32w0) {
-            tmp_0 = e.get<bit<32>>(s.h.data2);
-            s.h.data1 = tmp_0;
+            tmp = e_0.get<bit<32>>(s.h.data2);
+            s.h.data1 = tmp;
         }
     }
 }
