--- before_pass
+++ after_pass
@@ -5,17 +5,17 @@ extern Generic<T> {
 }
 extern void f<T>(in T arg);
 control c<T>()(T size) {
-    Generic<T>(size) x;
+    @name("x") Generic<T>(size) x_0;
     apply {
-        bit<32> a = x.get<bit<32>>();
-        bit<5> b = x.get1<bit<5>, bit<10>>(10w0, 5w0);
-        f<bit<5>>(b);
+        bit<32> a_0 = x_0.get<bit<32>>();
+        bit<5> b_0 = x_0.get1<bit<5>, bit<10>>(10w0, 5w0);
+        f<bit<5>>(b_0);
     }
 }
 control caller() {
-    c<bit<8>>(8w9) cinst;
+    @name("cinst") c<bit<8>>(8w9) cinst_0;
     apply {
-        cinst.apply();
+        cinst_0.apply();
     }
 }
 control s();
