--- before_pass
+++ after_pass
@@ -28,7 +28,7 @@ control ComputeChecksumI(inout H hdr, in
 }
 control IngressI(inout H hdr, inout M meta, inout std_meta_t std_meta) {
     action a() {
-        hash(meta.hash.hash, HashAlgorithm.crc16, (bit<16>)0, { meta.ipv4.lkp_ipv4_sa }, (bit<32>)65536);
+        hash(meta.hash.hash, HashAlgorithm.crc16, 16w0, { meta.ipv4.lkp_ipv4_sa }, 32w65536);
     }
     apply {
         a();
