--- before_pass
+++ after_pass
@@ -48,11 +48,11 @@ control pipe(inout Headers_t headers, ou
             headers.ipv4.srcAddr: exact;
         }
         actions = {
-            Reject;
-            NoAction;
+            Reject();
+            NoAction();
         }
         implementation = hash_table(1024);
-        const default_action = NoAction;
+        const default_action = NoAction();
     }
     apply {
         pass = true;
