--- before_pass
+++ after_pass
@@ -42,11 +42,11 @@ control pipe(inout Headers_t headers, ou
     @name(".NoAction") action NoAction_0() {
     }
     bool tmp;
-    @name("Reject") action Reject(IPv4Address add) {
+    @name("pipe.Reject") action Reject(IPv4Address add) {
         pass = false;
         headers.ipv4.srcAddr = add;
     }
-    @name("Check_src_ip") table Check_src_ip_0 {
+    @name("pipe.Check_src_ip") table Check_src_ip_0 {
         key = {
             headers.ipv4.srcAddr: exact @name("headers.ipv4.srcAddr") ;
         }
