--- before_pass
+++ after_pass
@@ -39,16 +39,16 @@ parser prs(packet_in p, out Headers_t he
     }
 }
 control pipe(inout Headers_t headers, out bool pass) {
-    action Reject(IPv4Address add) {
+    @name("Reject") action Reject_0(IPv4Address add) {
         pass = false;
         headers.ipv4.srcAddr = add;
     }
-    table Check_src_ip {
+    @name("Check_src_ip") table Check_src_ip_0 {
         key = {
             headers.ipv4.srcAddr: exact @name("headers.ipv4.srcAddr") ;
         }
         actions = {
-            Reject();
+            Reject_0();
             NoAction();
         }
         implementation = hash_table(32w1024);
@@ -60,7 +60,7 @@ control pipe(inout Headers_t headers, ou
             pass = false;
             return;
         }
-        if (Check_src_ip.apply().hit) {
+        if (Check_src_ip_0.apply().hit) {
             pass = pass;
         }
     }
