--- before_pass
+++ after_pass
@@ -46,10 +46,8 @@ control ingress(inout headers hdr, inout
             } else {
                 tmp_0 = x_0;
             }
-            {
-                hasReturned = true;
-                retval = tmp_0;
-            }
+            hasReturned = true;
+            retval = tmp_0;
             tmp = retval;
         }
         hdr.ethernet.srcAddr[15:0] = tmp;
