--- before_pass
+++ after_pass
@@ -7,9 +7,7 @@ struct metadata_t {
 }
 parser ParserImpl(packet_in packet, out headers_t hdr, inout metadata_t meta, inout standard_metadata_t standard_metadata) {
     state start {
-        transition select() {
-            default: accept;
-        }
+        transition accept;
     }
 }
 control IngressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t standard_metadata) {
