--- before_pass
+++ after_pass
@@ -17,15 +17,15 @@ control c(out bit<1> x) {
     H[2] a2;
     apply {
         if (a == b) {
-            x = (bit<1>)1w1;
+            x = 1w1;
         } else if (h1 == h2) {
-            x = (bit<1>)1w1;
+            x = 1w1;
         } else if (s1 == s2) {
-            x = (bit<1>)1w1;
+            x = 1w1;
         } else if (a1 == a2) {
-            x = (bit<1>)1w1;
+            x = 1w1;
         } else {
-            x = (bit<1>)1w0;
+            x = 1w0;
         }
     }
 }
