--- before_pass
+++ after_pass
@@ -24,7 +24,7 @@ control ComputeChecksumI(inout H hdr, in
 control IngressI(inout H hdr, inout M meta, inout std_meta_t std_meta) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("do_aux.adjust_lkp_fields") table do_aux_adjust_lkp_fields {
+    @name("IngressI.do_aux.adjust_lkp_fields") table do_aux_adjust_lkp_fields {
         key = {
             hdr.ipv4.isValid(): exact @name("hdr.ipv4.$valid$") ;
         }
