--- before_pass
+++ after_pass
@@ -22,14 +22,16 @@ control ComputeChecksumI(inout H hdr, in
     }
 }
 control IngressI(inout H hdr, inout M meta, inout std_meta_t std_meta) {
+    @name(".NoAction") action NoAction_1() {
+    }
     @name("do_aux.adjust_lkp_fields") table do_aux_adjust_lkp_fields_0 {
         key = {
             hdr.ipv4.isValid(): exact @name("hdr.ipv4.$valid$") ;
         }
         actions = {
-            NoAction();
+            NoAction_1();
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     apply {
         do_aux_adjust_lkp_fields_0.apply();
