--- before_pass
+++ after_pass
@@ -23,14 +23,15 @@ control pipe(inout Headers_t headers, ou
             headers.ethernet.protocol: exact;
         }
         actions = {
-            match;
-            NoAction;
+            match();
+            NoAction();
         }
         const entries = {
                         0x800 : match(true);
                         0xd000 : match(false);
         }
         implementation = hash_table(64);
+        default_action = NoAction();
     }
     apply {
         pass = true;
