--- before_pass
+++ after_pass
@@ -1,14 +1,26 @@
 #include <core.p4>
 #include <v1model.p4>
 bit<16> incr(in bit<16> x) {
-    return x + 16w1;
+    bool hasReturned = false;
+    bit<16> retval;
+    {
+        hasReturned = true;
+        retval = x + 16w1;
+    }
+    return retval;
 }
 bit<16> twoxplus1(in bit<16> x) {
+    bool hasReturned_0 = false;
+    bit<16> retval_0;
     bit<16> tmp;
     bit<16> tmp_0;
     tmp = incr(x);
     tmp_0 = x + tmp;
-    return tmp_0;
+    {
+        hasReturned_0 = true;
+        retval_0 = tmp_0;
+    }
+    return retval_0;
 }
 struct metadata {
     bit<16> tmp_port;
