--- before_pass
+++ after_pass
@@ -46,10 +46,10 @@ control ingress(inout headers hdr, inout
     bit<16> tmp_1;
     bit<16> tmp_2;
     bit<16> tmp_3;
-    @name("set_port") action set_port(bit<9> output_port) {
+    @name("ingress.set_port") action set_port(bit<9> output_port) {
         standard_metadata.egress_spec = output_port;
     }
-    @name("mac_da") table mac_da_0 {
+    @name("ingress.mac_da") table mac_da_0 {
         key = {
             hdr.ethernet.dstAddr: exact @name("hdr.ethernet.dstAddr") ;
         }
