--- before_pass
+++ after_pass
@@ -33,8 +33,8 @@ extern E {
     void g();
 }
 parser parserImpl(packet_in packet, out headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
-    @name("p1.ipv4_ethertypes") value_set<bit<16>>(8) p1_ipv4_ethertypes;
-    @name("p2.ipv4_ethertypes") value_set<bit<16>>(8) p2_ipv4_ethertypes;
+    @name("parserImpl.p1.ipv4_ethertypes") value_set<bit<16>>(8) p1_ipv4_ethertypes;
+    @name("parserImpl.p2.ipv4_ethertypes") value_set<bit<16>>(8) p2_ipv4_ethertypes;
     state start {
         transition subParser_start;
     }
