--- before_pass
+++ after_pass
@@ -33,15 +33,15 @@ extern E {
     void g();
 }
 parser parserImpl(packet_in packet, out headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
-    @name("p1.ipv4_ethertypes") value_set<bit<16>>(8) p1_ipv4_ethertypes_0;
-    @name("p2.ipv4_ethertypes") value_set<bit<16>>(8) p2_ipv4_ethertypes_0;
+    @name("p1.ipv4_ethertypes") value_set<bit<16>>(8) p1_ipv4_ethertypes;
+    @name("p2.ipv4_ethertypes") value_set<bit<16>>(8) p2_ipv4_ethertypes;
     state start {
         transition subParser_start;
     }
     state subParser_start {
         packet.extract<ethernet_t>(hdr.ethernet_1);
         transition select(hdr.ethernet_1.etherType) {
-            p1_ipv4_ethertypes_0: subParser_parse_ipv4;
+            p1_ipv4_ethertypes: subParser_parse_ipv4;
             default: start_0;
         }
     }
@@ -55,7 +55,7 @@ parser parserImpl(packet_in packet, out
     state subParser_start_0 {
         packet.extract<ethernet_t>(hdr.ethernet_2);
         transition select(hdr.ethernet_2.etherType) {
-            p2_ipv4_ethertypes_0: subParser_parse_ipv4_0;
+            p2_ipv4_ethertypes: subParser_parse_ipv4_0;
             default: start_1;
         }
     }
