--- before_pass
+++ after_pass
@@ -3,15 +3,15 @@ extern E {
     bit<32> f(in bit<32> x);
 }
 control c(inout bit<32> r) {
-    bit<32> tmp;
-    bit<32> tmp_0;
-    bit<32> tmp_1;
-    @name("e") E() e_0;
+    bit<32> tmp_2;
+    bit<32> tmp_3;
+    bit<32> tmp_4;
+    @name("e") E() e_1;
     apply {
-        tmp = e_0.f(32w4);
-        tmp_0 = e_0.f(32w5);
-        tmp_1 = tmp + tmp_0;
-        r = tmp_1;
+        tmp_2 = e_1.f(32w4);
+        tmp_3 = e_1.f(32w5);
+        tmp_4 = tmp_2 + tmp_3;
+        r = tmp_4;
     }
 }
 control simple(inout bit<32> r);
