--- before_pass
+++ after_pass
@@ -81,10 +81,10 @@ control ingress(inout headers hdr, inout
             hdr.ipv4.dstAddr: lpm;
         }
         actions = {
-            next_hop;
-            default_route_drop;
+            next_hop();
+            default_route_drop();
         }
-        default_action = default_route_drop;
+        default_action = default_route_drop();
         psa_direct_counter = per_prefix_pkt_byte_count;
     }
     apply {
