--- before_pass
+++ after_pass
@@ -8,7 +8,7 @@ header hdr_t {
 parser MyIP(packet_in buffer, out hdr_t a, inout EMPTY b, in psa_ingress_parser_input_metadata_t c, in EMPTY d, in EMPTY e) {
     value_set<bit<16>>(4) pvs;
     state start {
-        buffer.extract(a);
+        buffer.extract<hdr_t>(a);
         transition select(a.field) {
             pvs: accept;
             default: accept;
@@ -36,6 +36,6 @@ control MyED(packet_out buffer, out EMPT
     apply {
     }
 }
-IngressPipeline(MyIP(), MyIC(), MyID()) ip;
-EgressPipeline(MyEP(), MyEC(), MyED()) ep;
-PSA_Switch(ip, PacketReplicationEngine(), ep, BufferingQueueingEngine()) main;
+IngressPipeline<hdr_t, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY>(MyIP(), MyIC(), MyID()) ip;
+EgressPipeline<EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY>(MyEP(), MyEC(), MyED()) ep;
+PSA_Switch<hdr_t, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY, EMPTY>(ip, PacketReplicationEngine(), ep, BufferingQueueingEngine()) main;
