--- before_pass
+++ after_pass
@@ -38,11 +38,11 @@ struct digest_t {
     PortId_t port;
 }
 control MyID(packet_out buffer, out EMPTY a, out EMPTY b, out EMPTY c, inout headers hdr, in EMPTY e, in psa_ingress_output_metadata_t f) {
-    digest_t tmp_0;
-    @name("digest") Digest<digest_t>() digest;
+    digest_t tmp;
+    @name("digest") Digest<digest_t>() digest_0;
     apply {
-        tmp_0 = digest_t {h = hdr.h,port = f.egress_port};
-        digest.pack(tmp_0);
+        tmp = digest_t {h = hdr.h,port = f.egress_port};
+        digest_0.pack(tmp);
     }
 }
 control MyED(packet_out buffer, out EMPTY a, out EMPTY b, inout EMPTY c, in EMPTY d, in psa_egress_output_metadata_t e, in psa_egress_deparser_input_metadata_t f) {
