--- before_pass
+++ after_pass
@@ -35,12 +35,10 @@ control deparser(packet_out b, in Header
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
     apply {
-        {
-            if (h.h.a < h.h.b) {
-                h.h.c = 8w0;
-            } else {
-                h.h.c = 8w1;
-            }
+        if (h.h.a < h.h.b) {
+            h.h.c = 8w0;
+        } else {
+            h.h.c = 8w1;
         }
         sm.egress_spec = 9w0;
     }
