--- before_pass
+++ after_pass
@@ -8,9 +8,9 @@ header hdr {
 control compute(inout hdr h) {
     apply {
         if (h.a < h.b) {
-            h.c = (bit<8>)8w0;
+            h.c = 8w0;
         } else {
-            h.c = (bit<8>)8w1;
+            h.c = 8w1;
         }
     }
 }
@@ -46,7 +46,7 @@ control ingress(inout Headers h, inout M
     compute() c;
     apply {
         c.apply(h.h);
-        sm.egress_spec = (bit<9>)9w0;
+        sm.egress_spec = 9w0;
     }
 }
 V1Switch<Headers, Meta>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
