--- before_pass
+++ after_pass
@@ -5,15 +5,6 @@ header hdr {
     bit<32> b;
     bit<8>  c;
 }
-control compute(inout hdr h) {
-    apply {
-        if (h.a < h.b) {
-            h.c = 8w0;
-        } else {
-            h.c = 8w1;
-        }
-    }
-}
 struct Headers {
     hdr h;
 }
@@ -43,9 +34,14 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name("c") compute() c_0;
     apply {
-        c_0.apply(h.h);
+        {
+            if (h.h.a < h.h.b) {
+                h.h.c = 8w0;
+            } else {
+                h.h.c = 8w1;
+            }
+        }
         sm.egress_spec = 9w0;
     }
 }
