--- before_pass
+++ after_pass
@@ -8,12 +8,12 @@ control c(out bit<32> x)(Y y) {
     }
 }
 control d(out bit<32> x) {
-    bit<32> y;
-    c(Y(32w16)) cinst;
+    bit<32> y_0;
+    @name("cinst") c(Y(32w16)) cinst_0;
     apply {
-        y = 32w5;
-        cinst.apply(x);
-        cinst.apply(y);
+        y_0 = 32w5;
+        cinst_0.apply(x);
+        cinst_0.apply(y_0);
     }
 }
 control dproto(out bit<32> x);
