--- before_pass
+++ after_pass
@@ -3,23 +3,23 @@
 control empty();
 package top(empty e);
 control Ing() {
-    bool b;
-    bool tmp_1;
-    bool tmp_2;
-    @name("cond") action cond_0() {
-        b = true;
-        tmp_1 = tmp_1;
-        tmp_2 = tmp_2;
-        tmp_1 = tmp_1;
+    bool b_0;
+    bool tmp;
+    bool tmp_0;
+    @name("cond") action cond() {
+        b_0 = true;
+        tmp = tmp;
+        tmp_0 = tmp_0;
+        tmp = tmp;
     }
-    table tbl_cond {
+    @name("tbl_cond") table tbl_cond_0 {
         actions = {
-            cond_0();
+            cond();
         }
-        const default_action = cond_0();
+        const default_action = cond();
     }
     apply {
-        tbl_cond.apply();
+        tbl_cond_0.apply();
     }
 }
 top(Ing()) main;
