--- before_pass
+++ after_pass
@@ -4,14 +4,14 @@ control empty();
 package top(empty e);
 control Ing() {
     bool tmp_0;
-    @name("cond") action cond_0() {
+    @name("cond") action cond() {
         tmp_0 = tmp_0;
     }
     @name("tbl_cond") table tbl_cond {
         actions = {
-            cond_0();
+            cond();
         }
-        const default_action = cond_0();
+        const default_action = cond();
     }
     apply {
         tbl_cond.apply();
