--- before_pass
+++ after_pass
@@ -3,18 +3,18 @@
 control empty();
 package top(empty e);
 control Ing() {
-    bool tmp_0;
-    @name("cond") action cond() {
-        tmp_0 = tmp_0;
+    bool tmp;
+    @name("cond") action cond_0() {
+        tmp = tmp;
     }
-    @name("tbl_cond") table tbl_cond {
+    @name("tbl_cond") table tbl_cond_0 {
         actions = {
-            cond();
+            cond_0();
         }
-        const default_action = cond();
+        const default_action = cond_0();
     }
     apply {
-        tbl_cond.apply();
+        tbl_cond_0.apply();
     }
 }
 top(Ing()) main;
