--- before_pass
+++ after_pass
@@ -21,10 +21,10 @@ parser IngressParserImpl(packet_in pkt,
 }
 control cIngress(inout headers_t hdr, inout metadata_t user_meta, in psa_ingress_input_metadata_t istd, inout psa_ingress_output_metadata_t ostd) {
     apply {
-        if (hdr.ethernet.dstAddr[3:0] >= 4) {
+        if (hdr.ethernet.dstAddr[3:0] >= 4w4) {
             send_to_port(ostd, (PortId_t)(PortIdUint_t)hdr.ethernet.dstAddr);
         } else {
-            send_to_port(ostd, (PortId_t)0xfffffffa);
+            send_to_port(ostd, (PortId_t)(bit<32>)32w0xfffffffa);
         }
     }
 }
