--- before_pass
+++ after_pass
@@ -45,10 +45,10 @@ parser EgressParserImpl(packet_in buffer
     }
 }
 control cEgress(inout headers_t hdr, inout metadata_t user_meta, in psa_egress_input_metadata_t istd, inout psa_egress_output_metadata_t ostd) {
-    @name("add") action add_1() {
+    @name("cEgress.add") action add_1() {
         hdr.ethernet.dstAddr = hdr.ethernet.dstAddr + hdr.ethernet.srcAddr;
     }
-    @name("e") table e_0 {
+    @name("cEgress.e") table e_0 {
         actions = {
             add_1();
         }
