--- before_pass
+++ after_pass
@@ -15,30 +15,22 @@ struct mystruct1 {
 }
 control DeparserI(packet_out packet, in Parsed_packet hdr) {
     apply {
-        {
-            packet.emit<Ethernet_h>(hdr.ethernet);
-        }
+        packet.emit<Ethernet_h>(hdr.ethernet);
     }
 }
 parser parserI(packet_in pkt, out Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
     state start {
-        {
-            pkt.extract<Ethernet_h>(hdr.ethernet);
-        }
+        pkt.extract<Ethernet_h>(hdr.ethernet);
         transition accept;
     }
 }
 control cIngress(inout Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
     @name("foo") action foo_0(bit<16> bar) {
         if (bar == 16w0xf00d) {
-            {
-                hdr.ethernet.srcAddr = 48w0xdeadbeeff00d;
-            }
+            hdr.ethernet.srcAddr = 48w0xdeadbeeff00d;
             return;
         }
-        {
-            hdr.ethernet.srcAddr = 48w0x215241100ff2;
-        }
+        hdr.ethernet.srcAddr = 48w0x215241100ff2;
     }
     @name("tbl1") table tbl1_0 {
         key = {
@@ -50,9 +42,7 @@ control cIngress(inout Parsed_packet hdr
         default_action = NoAction();
     }
     apply {
-        {
-            tbl1_0.apply();
-        }
+        tbl1_0.apply();
         return;
     }
 }
