--- before_pass
+++ after_pass
@@ -24,14 +24,14 @@ control verifyChecksum(inout headers_t h
 control ingressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("my_drop") action my_drop() {
+    @name("ingressImpl.my_drop") action my_drop() {
         mark_to_drop(stdmeta);
     }
-    @name("foo") action foo(bit<9> out_port) {
+    @name("ingressImpl.foo") action foo(bit<9> out_port) {
         hdr.ethernet.dstAddr[22:18] = hdr.ethernet.srcAddr[5:1];
         stdmeta.egress_spec = out_port;
     }
-    @name("t1") table t1_0 {
+    @name("ingressImpl.t1") table t1_0 {
         key = {
             hdr.ethernet.srcAddr[22:18]            : exact @name("hdr.ethernet.srcAddr[22:18]") ;
             hdr.ethernet.dstAddr & 48w0x10101010101: exact @name("hdr.ethernet.dstAddr & 0x10101010101") ;
