--- before_pass
+++ after_pass
@@ -22,10 +22,12 @@ control verifyChecksum(inout headers_t h
     }
 }
 control ingressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
-    @name("my_drop") action my_drop() {
+    @name(".NoAction") action NoAction_1() {
+    }
+    @name("my_drop") action my_drop_0() {
         mark_to_drop(stdmeta);
     }
-    @name("foo") action foo(bit<9> out_port) {
+    @name("foo") action foo_0(bit<9> out_port) {
         hdr.ethernet.dstAddr[22:18] = hdr.ethernet.srcAddr[5:1];
         stdmeta.egress_spec = out_port;
     }
@@ -36,11 +38,11 @@ control ingressImpl(inout headers_t hdr,
             hdr.ethernet.etherType + 16w65526      : exact @name("etherType_less_10") ;
         }
         actions = {
-            foo();
-            my_drop();
-            NoAction();
+            foo_0();
+            my_drop_0();
+            NoAction_1();
         }
-        const default_action = NoAction();
+        const default_action = NoAction_1();
     }
     apply {
         t1.apply();
