--- before_pass
+++ after_pass
@@ -22,28 +22,28 @@ control verifyChecksum(inout headers_t h
     }
 }
 control ingressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
-    action my_drop() {
+    @name("my_drop") action my_drop_0() {
         mark_to_drop(stdmeta);
     }
-    action foo(bit<9> out_port) {
+    @name("foo") action foo_0(bit<9> out_port) {
         hdr.ethernet.dstAddr[22:18] = hdr.ethernet.srcAddr[5:1];
         stdmeta.egress_spec = out_port;
     }
-    table t1 {
+    @name("t1") table t1_0 {
         key = {
             hdr.ethernet.srcAddr[22:18]            : exact @name("hdr.ethernet.srcAddr[22:18]") ;
             hdr.ethernet.dstAddr & 48w0x10101010101: exact @name("hdr.ethernet.dstAddr & 0x10101010101") ;
             hdr.ethernet.etherType + 16w65526      : exact @name("etherType_less_10") ;
         }
         actions = {
-            foo();
-            my_drop();
+            foo_0();
+            my_drop_0();
             NoAction();
         }
         const default_action = NoAction();
     }
     apply {
-        t1.apply();
+        t1_0.apply();
     }
 }
 control egressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
