--- before_pass
+++ after_pass
@@ -31,8 +31,8 @@ control ingressImpl(inout headers_t hdr,
     }
     table t1 {
         key = {
-            hdr.ethernet.srcAddr[22:18]            : exact;
-            hdr.ethernet.dstAddr & 48w0x10101010101: exact;
+            hdr.ethernet.srcAddr[22:18]            : exact @name("hdr.ethernet.srcAddr[22:18]") ;
+            hdr.ethernet.dstAddr & 48w0x10101010101: exact @name("hdr.ethernet.dstAddr & 0x10101010101") ;
             hdr.ethernet.etherType - 16w10         : exact @name("etherType_less_10") ;
         }
         actions = {
