--- before_pass
+++ after_pass
@@ -161,63 +161,63 @@ control MyComputeChecksum(inout headers
     }
 }
 control MyIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    mac_addr_t mac_tmp_0;
-    ipv6_addr_t addr_tmp_0;
-    @name("set_egress_port") action set_egress_port_0(port_t out_port) {
+    mac_addr_t mac_tmp;
+    ipv6_addr_t addr_tmp;
+    @name("set_egress_port") action set_egress_port(port_t out_port) {
         standard_metadata.egress_spec = out_port;
     }
-    @name("controller_debug") action controller_debug_0() {
+    @name("controller_debug") action controller_debug() {
         meta.task = 16w3;
         meta.ingress_port = standard_metadata.ingress_port;
         clone3<metadata>(CloneType.I2E, 32w100, meta);
     }
-    @name("controller_reply") action controller_reply_0(task_t task) {
+    @name("controller_reply") action controller_reply(task_t task) {
         meta.task = task;
         meta.ingress_port = standard_metadata.ingress_port;
         clone3<metadata>(CloneType.I2E, 32w100, meta);
     }
-    @name("icmp6_echo_reply") action icmp6_echo_reply_0() {
-        mac_tmp_0 = hdr.ethernet.dst_addr;
+    @name("icmp6_echo_reply") action icmp6_echo_reply() {
+        mac_tmp = hdr.ethernet.dst_addr;
         hdr.ethernet.dst_addr = hdr.ethernet.src_addr;
-        hdr.ethernet.src_addr = mac_tmp_0;
-        addr_tmp_0 = hdr.ipv6.dst_addr;
+        hdr.ethernet.src_addr = mac_tmp;
+        addr_tmp = hdr.ipv6.dst_addr;
         hdr.ipv6.dst_addr = hdr.ipv6.src_addr;
-        hdr.ipv6.src_addr = addr_tmp_0;
+        hdr.ipv6.src_addr = addr_tmp;
         hdr.icmp6.type = 8w129;
         meta.do_cksum = 1w1;
     }
-    @name("v6_addresses") table v6_addresses_0 {
+    @name("v6_addresses") table v6_addresses {
         key = {
             hdr.ipv6.dst_addr: exact @name("hdr.ipv6.dst_addr") ;
         }
         actions = {
-            controller_debug_0();
-            controller_reply_0();
-            icmp6_echo_reply_0();
+            controller_debug();
+            controller_reply();
+            icmp6_echo_reply();
             NoAction();
         }
         size = 64;
         default_action = NoAction();
     }
-    @name("v6_networks") table v6_networks_0 {
+    @name("v6_networks") table v6_networks {
         key = {
             hdr.ipv6.dst_addr: lpm @name("hdr.ipv6.dst_addr") ;
         }
         actions = {
-            set_egress_port_0();
-            controller_debug_0();
-            controller_reply_0();
+            set_egress_port();
+            controller_debug();
+            controller_reply();
             NoAction();
         }
         size = 64;
         default_action = NoAction();
     }
-    @name("v4_networks") table v4_networks_0 {
+    @name("v4_networks") table v4_networks {
         key = {
             hdr.ipv4.dst_addr: lpm @name("hdr.ipv4.dst_addr") ;
         }
         actions = {
-            set_egress_port_0();
+            set_egress_port();
             NoAction();
         }
         size = 64;
@@ -225,11 +225,11 @@ control MyIngress(inout headers hdr, ino
     }
     apply {
         if (hdr.ipv6.isValid()) {
-            v6_addresses_0.apply();
-            v6_networks_0.apply();
+            v6_addresses.apply();
+            v6_networks.apply();
         }
         if (hdr.ipv4.isValid()) {
-            v4_networks_0.apply();
+            v4_networks.apply();
         }
     }
 }
