--- before_pass
+++ after_pass
@@ -220,36 +220,36 @@ control MyIngress(inout headers hdr, ino
             hdr.icmp6.type   : exact;
         }
         actions = {
-            controller_debug;
-            icmp6_neighbor_solicitation;
-            NoAction;
+            controller_debug();
+            icmp6_neighbor_solicitation();
+            NoAction();
         }
         size = 64;
-        default_action = NoAction;
+        default_action = NoAction();
     }
     table port2mcast {
         key = {
             standard_metadata.ingress_port: exact;
         }
         actions = {
-            multicast_pkg;
-            controller_debug;
-            NoAction;
+            multicast_pkg();
+            controller_debug();
+            NoAction();
         }
         size = 64;
-        default_action = NoAction;
+        default_action = NoAction();
     }
     table addr2mcast {
         key = {
             hdr.ipv6.dst_addr: exact;
         }
         actions = {
-            multicast_pkg;
-            controller_debug;
-            NoAction;
+            multicast_pkg();
+            controller_debug();
+            NoAction();
         }
         size = 64;
-        default_action = NoAction;
+        default_action = NoAction();
     }
     table ndp {
         key = {
@@ -257,12 +257,12 @@ control MyIngress(inout headers hdr, ino
             standard_metadata.ingress_port: exact;
         }
         actions = {
-            multicast_pkg;
-            controller_debug;
-            NoAction;
+            multicast_pkg();
+            controller_debug();
+            NoAction();
         }
         size = 64;
-        default_action = controller_debug;
+        default_action = controller_debug();
     }
     action icmp6_answer() {
         if (hdr.icmp6.isValid()) {
@@ -279,37 +279,37 @@ control MyIngress(inout headers hdr, ino
             hdr.ipv6.dst_addr: exact;
         }
         actions = {
-            controller_debug;
-            controller_reply;
-            icmp6_echo_reply;
-            NoAction;
+            controller_debug();
+            controller_reply();
+            icmp6_echo_reply();
+            NoAction();
         }
         size = 64;
-        default_action = NoAction;
+        default_action = NoAction();
     }
     table v6_networks {
         key = {
             hdr.ipv6.dst_addr: lpm;
         }
         actions = {
-            set_egress_port;
-            controller_debug;
-            controller_reply;
-            NoAction;
+            set_egress_port();
+            controller_debug();
+            controller_reply();
+            NoAction();
         }
         size = 64;
-        default_action = NoAction;
+        default_action = NoAction();
     }
     table v4_networks {
         key = {
             hdr.ipv4.dst_addr: lpm;
         }
         actions = {
-            set_egress_port;
-            NoAction;
+            set_egress_port();
+            NoAction();
         }
         size = 64;
-        default_action = NoAction;
+        default_action = NoAction();
     }
     apply {
         if (hdr.ipv6.isValid()) {
