--- before_pass
+++ after_pass
@@ -103,10 +103,24 @@ control ingress(inout headers hdr, inout
         default_action = unknown_source_0();
     }
     @name("do_L2_forward") action do_L2_forward_0(PortId_t egress_port) {
-        send_to_port(ostd, egress_port);
+        {
+            psa_ingress_output_metadata_t meta_1 = ostd;
+            PortId_t egress_port_1 = egress_port;
+            meta_1.drop = false;
+            meta_1.multicast_group = (MulticastGroup_t)32w0;
+            meta_1.egress_port = egress_port_1;
+            ostd = meta_1;
+        }
     }
     @name("do_tst") action do_tst_0(PortId_t egress_port, EthTypes serEnumT) {
-        send_to_port(ostd, egress_port);
+        {
+            psa_ingress_output_metadata_t meta_2 = ostd;
+            PortId_t egress_port_2 = egress_port;
+            meta_2.drop = false;
+            meta_2.multicast_group = (MulticastGroup_t)32w0;
+            meta_2.egress_port = egress_port_2;
+            ostd = meta_2;
+        }
     }
     @name("l2_tbl") table l2_tbl_0 {
         key = {
