--- before_pass
+++ after_pass
@@ -93,12 +93,12 @@ control ingress(inout headers hdr, inout
     }
     @name(".NoAction") action NoAction_5() {
     }
-    @name("unknown_source") action unknown_source() {
+    @name("ingress.unknown_source") action unknown_source() {
         meta.send_mac_learn_msg = true;
         meta.mac_learn_msg.srcAddr = hdr.ethernet.srcAddr;
         meta.mac_learn_msg.ingress_port = istd.ingress_port;
     }
-    @name("learned_sources") table learned_sources_0 {
+    @name("ingress.learned_sources") table learned_sources_0 {
         key = {
             hdr.ethernet.srcAddr: exact @name("hdr.ethernet.srcAddr") ;
         }
@@ -108,7 +108,7 @@ control ingress(inout headers hdr, inout
         }
         default_action = unknown_source();
     }
-    @name("do_L2_forward") action do_L2_forward(PortId_t egress_port) {
+    @name("ingress.do_L2_forward") action do_L2_forward(PortId_t egress_port) {
         {
             psa_ingress_output_metadata_t meta_1 = ostd;
             PortId_t egress_port_1 = egress_port;
@@ -118,7 +118,7 @@ control ingress(inout headers hdr, inout
             ostd = meta_1;
         }
     }
-    @name("do_tst") action do_tst(PortId_t egress_port, EthTypes serEnumT) {
+    @name("ingress.do_tst") action do_tst(PortId_t egress_port, EthTypes serEnumT) {
         {
             psa_ingress_output_metadata_t meta_2 = ostd;
             PortId_t egress_port_2 = egress_port;
@@ -128,7 +128,7 @@ control ingress(inout headers hdr, inout
             ostd = meta_2;
         }
     }
-    @name("l2_tbl") table l2_tbl_0 {
+    @name("ingress.l2_tbl") table l2_tbl_0 {
         key = {
             hdr.ethernet.dstAddr: exact @name("hdr.ethernet.dstAddr") ;
         }
@@ -138,7 +138,7 @@ control ingress(inout headers hdr, inout
         }
         default_action = NoAction_4();
     }
-    @name("tst_tbl") table tst_tbl_0 {
+    @name("ingress.tst_tbl") table tst_tbl_0 {
         key = {
             meta.mac_learn_msg.ingress_port: exact @name("meta.mac_learn_msg.ingress_port") ;
         }
@@ -160,7 +160,7 @@ control egress(inout headers hdr, inout
     }
 }
 control IngressDeparserImpl(packet_out packet, out empty_metadata_t clone_i2e_meta, out empty_metadata_t resubmit_meta, out empty_metadata_t normal_meta, inout headers hdr, in metadata meta, in psa_ingress_output_metadata_t istd) {
-    @name("mac_learn_digest") Digest<mac_learn_digest_t>() mac_learn_digest_0;
+    @name("IngressDeparserImpl.mac_learn_digest") Digest<mac_learn_digest_t>() mac_learn_digest_0;
     apply {
         if (meta.send_mac_learn_msg) {
             mac_learn_digest_0.pack(meta.mac_learn_msg);
