--- before_pass
+++ after_pass
@@ -91,8 +91,8 @@ control ingress(inout headers hdr, inout
             hdr.ethernet.srcAddr: exact;
         }
         actions = {
-            NoAction;
-            unknown_source;
+            NoAction();
+            unknown_source();
         }
         default_action = unknown_source();
     }
@@ -107,8 +107,8 @@ control ingress(inout headers hdr, inout
             hdr.ethernet.dstAddr: exact;
         }
         actions = {
-            do_L2_forward;
-            NoAction;
+            do_L2_forward();
+            NoAction();
         }
         default_action = NoAction();
     }
@@ -117,8 +117,8 @@ control ingress(inout headers hdr, inout
             meta.mac_learn_msg.ingress_port: exact;
         }
         actions = {
-            do_tst;
-            NoAction;
+            do_tst();
+            NoAction();
         }
         default_action = NoAction();
     }
