--- before_pass
+++ after_pass
@@ -88,7 +88,7 @@ control ingress(inout headers hdr, inout
     }
     table learned_sources {
         key = {
-            hdr.ethernet.srcAddr: exact;
+            hdr.ethernet.srcAddr: exact @name("hdr.ethernet.srcAddr") ;
         }
         actions = {
             NoAction();
@@ -104,7 +104,7 @@ control ingress(inout headers hdr, inout
     }
     table l2_tbl {
         key = {
-            hdr.ethernet.dstAddr: exact;
+            hdr.ethernet.dstAddr: exact @name("hdr.ethernet.dstAddr") ;
         }
         actions = {
             do_L2_forward();
@@ -114,7 +114,7 @@ control ingress(inout headers hdr, inout
     }
     table tst_tbl {
         key = {
-            meta.mac_learn_msg.ingress_port: exact;
+            meta.mac_learn_msg.ingress_port: exact @name("meta.mac_learn_msg.ingress_port") ;
         }
         actions = {
             do_tst();
