--- before_pass
+++ after_pass
@@ -26,43 +26,43 @@ extern bs {
 struct Packet_data {
 }
 control Q_pipe(inout TArg1 qArg1, inout TArg2 qArg2) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name("p1.thost.B_action") action p1_thost_B_action(out bit<9> barg_0, BParamType bData) {
+    @name("p1.thost.B_action") action p1_thost_B_action_0(out bit<9> barg_0, BParamType bData) {
         barg_0 = bData;
     }
-    @name("p1.thost.C_action") action p1_thost_C_action(bit<9> cData) {
+    @name("p1.thost.C_action") action p1_thost_C_action_0(bit<9> cData) {
         qArg1.field1 = cData;
     }
-    @name("p1.thost.T") table p1_thost_T_0 {
+    @name("p1.thost.T") table p1_thost_T {
         key = {
             qArg1.field1: ternary @name("tArg1.field1") ;
             qArg2.field2: exact @name("aArg2.field2") ;
         }
         actions = {
-            p1_thost_B_action(qArg1.field1);
-            p1_thost_C_action();
+            p1_thost_B_action_0(qArg1.field1);
+            p1_thost_C_action_0();
         }
         size = 32w5;
-        const default_action = p1_thost_C_action(9w5);
+        const default_action = p1_thost_C_action_0(9w5);
     }
-    @name("p1.Drop") action p1_Drop() {
+    @name("p1.Drop") action p1_Drop_0() {
         qArg1.drop = true;
     }
-    @name("p1.Tinner") table p1_Tinner_0 {
+    @name("p1.Tinner") table p1_Tinner {
         key = {
             qArg1.field1: ternary @name("pArg1.field1") ;
         }
         actions = {
-            p1_Drop();
-            NoAction_1();
+            p1_Drop_0();
+            NoAction_0();
         }
-        const default_action = NoAction_1();
+        const default_action = NoAction_0();
     }
     apply {
-        p1_thost_T_0.apply();
-        p1_thost_T_0.apply();
-        p1_Tinner_0.apply();
+        p1_thost_T.apply();
+        p1_thost_T.apply();
+        p1_Tinner.apply();
     }
 }
 parser prs(bs b, out Packet_data p);
