--- before_pass
+++ after_pass
@@ -26,8 +26,8 @@ extern bs {
 struct Packet_data {
 }
 control T_host(inout TArg1 tArg1, in TArg2 aArg2)(bit<32> t2Size) {
-    @name("B_action") action B_action_0(out bit<9> barg, BParamType bData) {
-        barg = bData;
+    @name("B_action") action B_action_0(out bit<9> barg_0, BParamType bData) {
+        barg_0 = bData;
     }
     @name("C_action") action C_action_0(bit<9> cData) {
         tArg1.field1 = cData;
