--- before_pass
+++ after_pass
@@ -27,10 +27,14 @@ struct Packet_data {
 }
 control T_host(inout TArg1 tArg1, in TArg2 aArg2)(bit<32> t2Size) {
     @name("B_action") action B_action_0(out bit<9> barg, BParamType bData) {
-        barg = bData;
+        {
+            barg = bData;
+        }
     }
     @name("C_action") action C_action_0(bit<9> cData) {
-        tArg1.field1 = cData;
+        {
+            tArg1.field1 = cData;
+        }
     }
     @name("T") table T_0 {
         key = {
@@ -45,13 +49,17 @@ control T_host(inout TArg1 tArg1, in TAr
         const default_action = C_action_0(9w5);
     }
     apply {
-        T_0.apply();
+        {
+            T_0.apply();
+        }
     }
 }
 control P_pipe(inout TArg1 pArg1, inout TArg2 pArg2)(bit<32> t2Size) {
     @name("thost") T_host(t2Size) thost_0;
     @name("Drop") action Drop_0() {
-        pArg1.drop = true;
+        {
+            pArg1.drop = true;
+        }
     }
     @name("Tinner") table Tinner_0 {
         key = {
@@ -64,15 +72,23 @@ control P_pipe(inout TArg1 pArg1, inout
         const default_action = NoAction();
     }
     apply {
-        thost_0.apply(pArg1, pArg2);
-        thost_0.apply(pArg1, pArg2);
-        Tinner_0.apply();
+        {
+            thost_0.apply(pArg1, pArg2);
+        }
+        {
+            thost_0.apply(pArg1, pArg2);
+        }
+        {
+            Tinner_0.apply();
+        }
     }
 }
 control Q_pipe(inout TArg1 qArg1, inout TArg2 qArg2) {
     @name("p1") P_pipe(32w5) p1_0;
     apply {
-        p1_0.apply(qArg1, qArg2);
+        {
+            p1_0.apply(qArg1, qArg2);
+        }
     }
 }
 parser prs(bs b, out Packet_data p);
