--- before_pass
+++ after_pass
@@ -25,7 +25,7 @@ extern bs {
 }
 struct Packet_data {
 }
-control T_host(inout TArg1 tArg1, in TArg2 aArg2)(bit<32> t2Size) {
+control T_host_0(inout TArg1 tArg1, in TArg2 aArg2) {
     @name("B_action") action B_action_0(out bit<9> barg_0, BParamType bData) {
         barg_0 = bData;
     }
@@ -41,15 +41,15 @@ control T_host(inout TArg1 tArg1, in TAr
             B_action_0(tArg1.field1);
             C_action_0();
         }
-        size = t2Size;
+        size = 32w5;
         const default_action = C_action_0(9w5);
     }
     apply {
         T_0.apply();
     }
 }
-control P_pipe(inout TArg1 pArg1, inout TArg2 pArg2)(bit<32> t2Size) {
-    @name("thost") T_host(t2Size) thost_0;
+control P_pipe_0(inout TArg1 pArg1, inout TArg2 pArg2) {
+    @name("thost") T_host_0() thost_0;
     @name("Drop") action Drop_0() {
         pArg1.drop = true;
     }
@@ -70,7 +70,7 @@ control P_pipe(inout TArg1 pArg1, inout
     }
 }
 control Q_pipe(inout TArg1 qArg1, inout TArg2 qArg2) {
-    @name("p1") P_pipe(32w5) p1_0;
+    @name("p1") P_pipe_0() p1_0;
     apply {
         p1_0.apply(qArg1, qArg2);
     }
