--- before_pass
+++ after_pass
@@ -28,13 +28,13 @@ struct Packet_data {
 control Q_pipe(inout TArg1 qArg1, inout TArg2 qArg2) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("p1.thost.B_action") action p1_thost_B_action_0(out bit<9> barg, BParamType bData) {
+    @name("Q_pipe.p1.thost.B_action") action p1_thost_B_action_0(out bit<9> barg, BParamType bData) {
         barg = bData;
     }
-    @name("p1.thost.C_action") action p1_thost_C_action_0(bit<9> cData) {
+    @name("Q_pipe.p1.thost.C_action") action p1_thost_C_action_0(bit<9> cData) {
         qArg1.field1 = cData;
     }
-    @name("p1.thost.T") table p1_thost_T {
+    @name("Q_pipe.p1.thost.T") table p1_thost_T {
         key = {
             qArg1.field1: ternary @name("tArg1.field1") ;
             qArg2.field2: exact @name("aArg2.field2") ;
@@ -46,10 +46,10 @@ control Q_pipe(inout TArg1 qArg1, inout
         size = 32w5;
         const default_action = p1_thost_C_action_0(9w5);
     }
-    @name("p1.Drop") action p1_Drop_0() {
+    @name("Q_pipe.p1.Drop") action p1_Drop_0() {
         qArg1.drop = true;
     }
-    @name("p1.Tinner") table p1_Tinner {
+    @name("Q_pipe.p1.Tinner") table p1_Tinner {
         key = {
             qArg1.field1: ternary @name("pArg1.field1") ;
         }
