--- before_pass
+++ after_pass
@@ -58,15 +58,9 @@ control Q_pipe(inout TArg1 qArg1, inout
         const default_action = NoAction();
     }
     apply {
-        {
-            {
-                p1_thost_T.apply();
-            }
-            {
-                p1_thost_T.apply();
-            }
-            p1_Tinner.apply();
-        }
+        p1_thost_T.apply();
+        p1_thost_T.apply();
+        p1_Tinner.apply();
     }
 }
 parser prs(bs b, out Packet_data p);
