--- before_pass
+++ after_pass
@@ -47,7 +47,7 @@ control ingress(inout headers hdr, inout
             NoAction();
         }
         key = {
-            meta.meta.meter_tag: exact;
+            meta.meta.meter_tag: exact @name("meta.meta.meter_tag") ;
         }
         size = 16;
         default_action = NoAction();
@@ -67,7 +67,7 @@ control ingress(inout headers hdr, inout
             NoAction();
         }
         key = {
-            hdr.ethernet.srcAddr: exact;
+            hdr.ethernet.srcAddr: exact @name("hdr.ethernet.srcAddr") ;
         }
         size = 16384;
         default_action = NoAction();
