--- before_pass
+++ after_pass
@@ -42,9 +42,9 @@ control ingress(inout headers hdr, inout
     }
     @name("m_filter") table m_filter {
         actions = {
-            _drop;
-            _nop;
-            NoAction;
+            _drop();
+            _nop();
+            NoAction();
         }
         key = {
             meta.meta.meter_tag: exact;
@@ -62,9 +62,9 @@ control ingress(inout headers hdr, inout
     }
     @name("m_table") table m_table {
         actions = {
-            m_action_0;
-            _nop_0;
-            NoAction;
+            m_action_0();
+            _nop_0();
+            NoAction();
         }
         key = {
             hdr.ethernet.srcAddr: exact;
