--- before_pass
+++ after_pass
@@ -1,5 +1,8 @@
 #include <core.p4>
 control c(out bool x) {
+    bool tmp;
+    bool tmp_0;
+    bool tmp_1;
     @name("t1") table t1_0 {
         key = {
             x: exact @name("x") ;
@@ -18,9 +21,6 @@ control c(out bool x) {
         }
         default_action = NoAction();
     }
-    bool tmp;
-    bool tmp_0;
-    bool tmp_1;
     apply {
         x = true;
         tmp = t1_0.apply().hit;
