--- before_pass
+++ after_pass
@@ -1,40 +1,40 @@
 #include <core.p4>
 control c(out bool x) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name(".NoAction") action NoAction_2() {
+    @name(".NoAction") action NoAction_3() {
     }
-    bool tmp_2;
-    bool tmp_3;
-    bool tmp_4;
-    @name("t1") table t1 {
+    bool tmp;
+    bool tmp_0;
+    bool tmp_1;
+    @name("t1") table t1_0 {
         key = {
             x: exact @name("x") ;
         }
         actions = {
-            NoAction_1();
+            NoAction_0();
         }
-        default_action = NoAction_1();
+        default_action = NoAction_0();
     }
-    @name("t2") table t2 {
+    @name("t2") table t2_0 {
         key = {
             x: exact @name("x") ;
         }
         actions = {
-            NoAction_2();
+            NoAction_3();
         }
-        default_action = NoAction_2();
+        default_action = NoAction_3();
     }
     apply {
         x = true;
-        tmp_2 = t1.apply().hit;
-        if (!tmp_2) {
-            tmp_3 = false;
+        tmp = t1_0.apply().hit;
+        if (!tmp) {
+            tmp_0 = false;
         } else {
-            tmp_4 = t2.apply().hit;
-            tmp_3 = tmp_4;
+            tmp_1 = t2_0.apply().hit;
+            tmp_0 = tmp_1;
         }
-        if (tmp_3) {
+        if (tmp_0) {
             x = false;
         }
     }
