--- before_pass
+++ after_pass
@@ -2,7 +2,7 @@
 control c(out bool x) {
     table t1 {
         key = {
-            x: exact;
+            x: exact @name("x") ;
         }
         actions = {
             NoAction();
@@ -11,7 +11,7 @@ control c(out bool x) {
     }
     table t2 {
         key = {
-            x: exact;
+            x: exact @name("x") ;
         }
         actions = {
             NoAction();
