--- before_pass
+++ after_pass
@@ -22,20 +22,16 @@ control c(out bool x) {
     bool tmp_0;
     bool tmp_1;
     apply {
-        {
-            x = true;
+        x = true;
+        tmp = t1_0.apply().hit;
+        if (!tmp) {
+            tmp_0 = false;
+        } else {
+            tmp_1 = t2_0.apply().hit;
+            tmp_0 = tmp_1;
         }
-        {
-            tmp = t1_0.apply().hit;
-            if (!tmp) {
-                tmp_0 = false;
-            } else {
-                tmp_1 = t2_0.apply().hit;
-                tmp_0 = tmp_1;
-            }
-            if (tmp_0) {
-                x = false;
-            }
+        if (tmp_0) {
+            x = false;
         }
     }
 }
