--- before_pass
+++ after_pass
@@ -7,7 +7,7 @@ control c(out bool x) {
     bool tmp;
     bool tmp_0;
     bool tmp_1;
-    @name("t1") table t1_0 {
+    @name("c.t1") table t1_0 {
         key = {
             x: exact @name("x") ;
         }
@@ -16,7 +16,7 @@ control c(out bool x) {
         }
         default_action = NoAction_0();
     }
-    @name("t2") table t2_0 {
+    @name("c.t2") table t2_0 {
         key = {
             x: exact @name("x") ;
         }
