--- before_pass
+++ after_pass
@@ -1,6 +1,6 @@
 #include <core.p4>
 control c(out bool x) {
-    table t1 {
+    @name("t1") table t1_0 {
         key = {
             x: exact @name("x") ;
         }
@@ -9,7 +9,7 @@ control c(out bool x) {
         }
         default_action = NoAction();
     }
-    table t2 {
+    @name("t2") table t2_0 {
         key = {
             x: exact @name("x") ;
         }
@@ -20,7 +20,7 @@ control c(out bool x) {
     }
     apply {
         x = true;
-        if (t1.apply().hit && t2.apply().hit) {
+        if (t1_0.apply().hit && t2_0.apply().hit) {
             x = false;
         }
     }
