--- before_pass
+++ after_pass
@@ -5,7 +5,7 @@ control c(out bool x) {
             x: exact;
         }
         actions = {
-            NoAction;
+            NoAction();
         }
         default_action = NoAction();
     }
@@ -14,7 +14,7 @@ control c(out bool x) {
             x: exact;
         }
         actions = {
-            NoAction;
+            NoAction();
         }
         default_action = NoAction();
     }
