--- before_pass
+++ after_pass
@@ -21,9 +21,7 @@ struct metadata {
 }
 parser ParserImpl(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     state start {
-        {
-            packet.extract<S>(hdr.base);
-        }
+        packet.extract<S>(hdr.base);
         transition select(hdr.base.t) {
             8w0: parseO1;
             8w1: parseO2;
@@ -31,15 +29,11 @@ parser ParserImpl(packet_in packet, out
         }
     }
     state parseO1 {
-        {
-            packet.extract<O1>(hdr.u.byte);
-        }
+        packet.extract<O1>(hdr.u.byte);
         transition accept;
     }
     state parseO2 {
-        {
-            packet.extract<O2>(hdr.u.short);
-        }
+        packet.extract<O2>(hdr.u.short);
         transition accept;
     }
     state skip {
@@ -59,29 +53,15 @@ control ingress(inout headers hdr, inout
         const default_action = NoAction();
     }
     apply {
-        {
-            debug_hdr_0.apply();
-        }
+        debug_hdr_0.apply();
         if (hdr.u.short.isValid()) {
-            {
-                hdr.u.short.data = 16w0xaaaa;
-            }
-            {
-                hdr.u.byte.setValid();
-            }
-            {
-                hdr.u.byte.data = 8w0xff;
-            }
+            hdr.u.short.data = 16w0xaaaa;
+            hdr.u.byte.setValid();
+            hdr.u.byte.data = 8w0xff;
         } else if (hdr.u.byte.isValid()) {
-            {
-                hdr.u.byte.data = 8w0xaa;
-            }
-            {
-                hdr.u.short.setValid();
-            }
-            {
-                hdr.u.short.data = 16w0xffff;
-            }
+            hdr.u.byte.data = 8w0xaa;
+            hdr.u.short.setValid();
+            hdr.u.short.data = 16w0xffff;
         }
     }
 }
@@ -91,9 +71,7 @@ control egress(inout headers hdr, inout
 }
 control DeparserImpl(packet_out packet, in headers hdr) {
     apply {
-        {
-            packet.emit<headers>(hdr);
-        }
+        packet.emit<headers>(hdr);
     }
 }
 control verifyChecksum(inout headers hdr, inout metadata meta) {
