--- before_pass
+++ after_pass
@@ -55,13 +55,13 @@ control ingress(inout headers hdr, inout
     apply {
         debug_hdr.apply();
         if (hdr.u.short.isValid()) {
-            hdr.u.short.data = (bit<16>)16w0xaaaa;
+            hdr.u.short.data = 16w0xaaaa;
             hdr.u.byte.setValid();
-            hdr.u.byte.data = (bit<8>)8w0xff;
+            hdr.u.byte.data = 8w0xff;
         } else if (hdr.u.byte.isValid()) {
-            hdr.u.byte.data = (bit<8>)8w0xaa;
+            hdr.u.byte.data = 8w0xaa;
             hdr.u.short.setValid();
-            hdr.u.short.data = (bit<16>)16w0xffff;
+            hdr.u.short.data = 16w0xffff;
         }
     }
 }
