--- before_pass
+++ after_pass
@@ -55,11 +55,11 @@ control pipe(inout Headers_t headers, ou
             headers.ethernet.srcAddr: exact;
         }
         actions = {
-            invalidate;
-            drop;
+            invalidate();
+            drop();
         }
         implementation = hash_table(10);
-        default_action = drop;
+        default_action = drop();
     }
     apply {
         bool hit;
