--- before_pass
+++ after_pass
@@ -39,15 +39,15 @@ parser prs(packet_in p, out Headers_t he
     }
 }
 control pipe(inout Headers_t headers, out bool pass) {
-    action invalidate() {
+    @name("invalidate") action invalidate_0() {
         headers.ipv4.setInvalid();
         headers.ethernet.setInvalid();
         pass = true;
     }
-    action drop() {
+    @name("drop") action drop_0() {
         pass = false;
     }
-    table t {
+    @name("t") table t_0 {
         key = {
             headers.ipv4.srcAddr + 32w1: exact @name(" headers.ipv4.srcAddr") ;
             headers.ipv4.dstAddr + 32w1: exact @name("headers.ipv4.dstAddr") ;
@@ -55,15 +55,15 @@ control pipe(inout Headers_t headers, ou
             headers.ethernet.srcAddr   : exact @name("headers.ethernet.srcAddr") ;
         }
         actions = {
-            invalidate();
-            drop();
+            invalidate_0();
+            drop_0();
         }
         implementation = hash_table(32w10);
-        default_action = drop();
+        default_action = drop_0();
     }
     apply {
-        bool hit;
-        hit = t.apply().hit;
+        bool hit_0;
+        hit_0 = t_0.apply().hit;
     }
 }
 ebpfFilter<Headers_t>(prs(), pipe()) main;
