--- before_pass
+++ after_pass
@@ -51,8 +51,8 @@ control pipe(inout Headers_t headers, ou
         key = {
             headers.ipv4.srcAddr + 32w1: exact @name(" headers.ipv4.srcAddr") ;
             headers.ipv4.dstAddr + 32w1: exact @name("headers.ipv4.dstAddr") ;
-            headers.ethernet.dstAddr   : exact;
-            headers.ethernet.srcAddr   : exact;
+            headers.ethernet.dstAddr   : exact @name("headers.ethernet.dstAddr") ;
+            headers.ethernet.srcAddr   : exact @name("headers.ethernet.srcAddr") ;
         }
         actions = {
             invalidate();
