--- before_pass
+++ after_pass
@@ -49,16 +49,16 @@ control pipe(inout Headers_t headers, ou
     }
     table t {
         key = {
-            headers.ipv4.srcAddr + 1: exact @name(" headers.ipv4.srcAddr") ;
-            headers.ipv4.dstAddr + 1: exact @name("headers.ipv4.dstAddr") ;
-            headers.ethernet.dstAddr: exact;
-            headers.ethernet.srcAddr: exact;
+            headers.ipv4.srcAddr + 32w1: exact @name(" headers.ipv4.srcAddr") ;
+            headers.ipv4.dstAddr + 32w1: exact @name("headers.ipv4.dstAddr") ;
+            headers.ethernet.dstAddr   : exact;
+            headers.ethernet.srcAddr   : exact;
         }
         actions = {
             invalidate();
             drop();
         }
-        implementation = hash_table(10);
+        implementation = hash_table(32w10);
         default_action = drop();
     }
     apply {
