--- before_pass
+++ after_pass
@@ -6,7 +6,7 @@ header hdr {
 }
 control compute(inout hdr h) {
     apply {
-        extern_func(h.a, 0xff);
+        extern_func(h.a, 32w0xff);
     }
 }
 struct Headers {
@@ -41,7 +41,7 @@ control ingress(inout Headers h, inout M
     compute() c;
     apply {
         c.apply(h.h);
-        sm.egress_spec = 0;
+        sm.egress_spec = (bit<9>)9w0;
     }
 }
 V1Switch(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
