--- before_pass
+++ after_pass
@@ -23,8 +23,10 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout user_meta_t b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
+    @name(".NoAction") action NoAction_1() {
+    }
     @name("reg") Register<bit<16>, bit<10>>(32w512) reg;
-    @name("execute_register") action execute_register(bit<10> idx) {
+    @name("execute_register") action execute_register_0(bit<10> idx) {
         reg.write(idx, b.data);
     }
     @name("tbl") table tbl {
@@ -32,10 +34,10 @@ control MyIC(inout ethernet_t a, inout u
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
-            NoAction();
-            execute_register();
+            NoAction_1();
+            execute_register_0();
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     apply {
         tbl.apply();
