--- before_pass
+++ after_pass
@@ -12,9 +12,9 @@ extern ValueSet {
     bit<8> index(in bit<16> proto);
 }
 parser TopParser(packet_in b, out Parsed_packet p) {
-    bit<8> setIndex_0;
-    bit<8> tmp;
-    @name("ethtype_kinds") ValueSet(32w5) ethtype_kinds_0;
+    bit<8> setIndex;
+    bit<8> tmp_0;
+    @name("ethtype_kinds") ValueSet(32w5) ethtype_kinds;
     state start {
         b.extract<Ethernet_h>(p.ethernet);
         transition select(p.ethernet.etherType) {
@@ -25,9 +25,9 @@ parser TopParser(packet_in b, out Parsed
         }
     }
     state dispatch_value_sets {
-        tmp = ethtype_kinds_0.index(p.ethernet.etherType);
-        setIndex_0 = tmp;
-        transition select(setIndex_0) {
+        tmp_0 = ethtype_kinds.index(p.ethernet.etherType);
+        setIndex = tmp_0;
+        transition select(setIndex) {
             8w1: parse_trill;
             8w2: parse_vlan_tag;
         }
