--- before_pass
+++ after_pass
@@ -16,9 +16,7 @@ parser TopParser(packet_in b, out Parsed
     @name("ethtype_kinds") ValueSet(32w5) ethtype_kinds_0;
     bit<8> tmp;
     state start {
-        {
-            b.extract<Ethernet_h>(p.ethernet);
-        }
+        b.extract<Ethernet_h>(p.ethernet);
         transition select(p.ethernet.etherType) {
             16w0x800: parse_ipv4;
             16w0x806: parse_arp;
@@ -27,10 +25,8 @@ parser TopParser(packet_in b, out Parsed
         }
     }
     state dispatch_value_sets {
-        {
-            tmp = ethtype_kinds_0.index(p.ethernet.etherType);
-            setIndex_0 = tmp;
-        }
+        tmp = ethtype_kinds_0.index(p.ethernet.etherType);
+        setIndex_0 = tmp;
         transition select(setIndex_0) {
             8w1: parse_trill;
             8w2: parse_vlan_tag;
