--- before_pass
+++ after_pass
@@ -3,12 +3,12 @@ control strength() {
         bit<4> x;
         bit<4> y;
         bit<4> z;
-        z = ~x ^ ~y;
-        if (!(x < y)) {
-            z = ~(x ^ ~y & ~z);
+        z = x ^ y;
+        if (x >= y) {
+            z = x ^ (y | z);
         }
-        if (!(x > y)) {
-            z = ~(x ^ (~y | ~z));
+        if (x <= y) {
+            z = x ^ y & z;
         }
     }
 }
