--- before_pass
+++ after_pass
@@ -33,7 +33,7 @@ control VerifyChecksumI(inout H hdr, ino
 }
 control ComputeChecksumI(inout H hdr, inout M meta) {
     apply {
-        update_checksum(hdr.ipv4.ihl == 5, { 1w0 }, hdr.ipv4.hdrChecksum, HashAlgorithm.csum16);
+        update_checksum(hdr.ipv4.ihl == 4w5, { 1w0 }, hdr.ipv4.hdrChecksum, HashAlgorithm.csum16);
     }
 }
 V1Switch(ParserI(), VerifyChecksumI(), IngressI(), EgressI(), ComputeChecksumI(), DeparserI()) main;
