--- before_pass
+++ after_pass
@@ -84,9 +84,9 @@ control egress(inout headers hdr, inout
     }
     @name("send_frame") table send_frame {
         actions = {
-            rewrite_mac;
-            _drop;
-            NoAction;
+            rewrite_mac();
+            _drop();
+            NoAction();
         }
         key = {
             standard_metadata.egress_port: exact;
@@ -129,9 +129,9 @@ control ingress(inout headers hdr, inout
     }
     @name("ecmp_group") table ecmp_group {
         actions = {
-            _drop;
-            set_ecmp_select;
-            NoAction;
+            _drop();
+            set_ecmp_select();
+            NoAction();
         }
         key = {
             hdr.ipv4.dstAddr: lpm;
@@ -141,9 +141,9 @@ control ingress(inout headers hdr, inout
     }
     @name("ecmp_nhop") table ecmp_nhop {
         actions = {
-            _drop;
-            set_nhop;
-            NoAction;
+            _drop();
+            set_nhop();
+            NoAction();
         }
         key = {
             meta.ingress_metadata.ecmp_offset: exact;
@@ -153,16 +153,16 @@ control ingress(inout headers hdr, inout
     }
     @name("flowlet") table flowlet {
         actions = {
-            lookup_flowlet_map;
-            NoAction;
+            lookup_flowlet_map();
+            NoAction();
         }
         default_action = NoAction();
     }
     @name("forward") table forward {
         actions = {
-            set_dmac;
-            _drop;
-            NoAction;
+            set_dmac();
+            _drop();
+            NoAction();
         }
         key = {
             meta.ingress_metadata.nhop_ipv4: exact;
@@ -172,8 +172,8 @@ control ingress(inout headers hdr, inout
     }
     @name("new_flowlet") table new_flowlet {
         actions = {
-            update_flowlet_id;
-            NoAction;
+            update_flowlet_id();
+            NoAction();
         }
         default_action = NoAction();
     }
