--- before_pass
+++ after_pass
@@ -125,7 +125,7 @@ control ingress(inout headers hdr, inout
     }
     @name("update_flowlet_id") action update_flowlet_id() {
         meta.ingress_metadata.flowlet_id = meta.ingress_metadata.flowlet_id + 16w1;
-        flowlet_id.write((bit<32>)meta.ingress_metadata.flowlet_map_index, (bit<16>)meta.ingress_metadata.flowlet_id);
+        flowlet_id.write((bit<32>)meta.ingress_metadata.flowlet_map_index, meta.ingress_metadata.flowlet_id);
     }
     @name("ecmp_group") table ecmp_group {
         actions = {
