--- before_pass
+++ after_pass
@@ -76,43 +76,61 @@ parser ParserImpl(packet_in packet, out
     }
 }
 control egress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    @name("rewrite_mac") action rewrite_mac(bit<48> smac) {
+    @name(".NoAction") action NoAction_2() {
+    }
+    @name("rewrite_mac") action rewrite_mac_0(bit<48> smac) {
         hdr.ethernet.srcAddr = smac;
     }
-    @name("_drop") action _drop() {
+    @name("_drop") action _drop_0() {
         mark_to_drop(standard_metadata);
     }
     @name("send_frame") table send_frame {
         actions = {
-            rewrite_mac();
-            _drop();
-            NoAction();
+            rewrite_mac_0();
+            _drop_0();
+            NoAction_2();
         }
         key = {
             standard_metadata.egress_port: exact @name("standard_metadata.egress_port") ;
         }
         size = 256;
-        default_action = NoAction();
+        default_action = NoAction_2();
     }
     apply {
         send_frame.apply();
     }
 }
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
+    @name(".NoAction") action NoAction_3() {
+    }
+    @name(".NoAction") action NoAction_4() {
+    }
+    @name(".NoAction") action NoAction_5() {
+    }
+    @name(".NoAction") action NoAction_6() {
+    }
+    @name(".NoAction") action NoAction_7() {
+    }
     @name("flowlet_id") register<bit<16>>(32w8192) flowlet_id_1;
     @name("flowlet_lasttime") register<bit<32>>(32w8192) flowlet_lasttime_1;
-    @name("_drop") action _drop_2() {
+    @name("_drop") action _drop_1() {
+        mark_to_drop(standard_metadata);
+    }
+    @name("_drop") action _drop_3() {
+        mark_to_drop(standard_metadata);
+    }
+    @name("_drop") action _drop_4() {
         mark_to_drop(standard_metadata);
     }
-    @name("set_ecmp_select") action set_ecmp_select(bit<8> ecmp_base, bit<8> ecmp_count) {
+    @name("set_ecmp_select") action set_ecmp_select_0(bit<8> ecmp_base, bit<8> ecmp_count) {
         hash<bit<14>, bit<10>, tuple<bit<32>, bit<32>, bit<8>, bit<16>, bit<16>, bit<16>>, bit<20>>(meta.ingress_metadata.ecmp_offset, HashAlgorithm.crc16, (bit<10>)ecmp_base, { hdr.ipv4.srcAddr, hdr.ipv4.dstAddr, hdr.ipv4.protocol, hdr.tcp.srcPort, hdr.tcp.dstPort, meta.ingress_metadata.flowlet_id }, (bit<20>)ecmp_count);
     }
-    @name("set_nhop") action set_nhop(bit<32> nhop_ipv4, bit<9> port) {
+    @name("set_nhop") action set_nhop_0(bit<32> nhop_ipv4, bit<9> port) {
         meta.ingress_metadata.nhop_ipv4 = nhop_ipv4;
         standard_metadata.egress_spec = port;
         hdr.ipv4.ttl = hdr.ipv4.ttl + 8w255;
     }
-    @name("lookup_flowlet_map") action lookup_flowlet_map() {
+    @name("lookup_flowlet_map") action lookup_flowlet_map_0() {
         hash<bit<13>, bit<13>, tuple<bit<32>, bit<32>, bit<8>, bit<16>, bit<16>>, bit<26>>(meta.ingress_metadata.flowlet_map_index, HashAlgorithm.crc16, 13w0, { hdr.ipv4.srcAddr, hdr.ipv4.dstAddr, hdr.ipv4.protocol, hdr.tcp.srcPort, hdr.tcp.dstPort }, 26w13);
         flowlet_id_1.read(meta.ingress_metadata.flowlet_id, (bit<32>)meta.ingress_metadata.flowlet_map_index);
         meta.ingress_metadata.flow_ipg = (bit<32>)standard_metadata.ingress_global_timestamp;
@@ -120,62 +138,62 @@ control ingress(inout headers hdr, inout
         meta.ingress_metadata.flow_ipg = meta.ingress_metadata.flow_ipg - meta.ingress_metadata.flowlet_lasttime;
         flowlet_lasttime_1.write((bit<32>)meta.ingress_metadata.flowlet_map_index, (bit<32>)standard_metadata.ingress_global_timestamp);
     }
-    @name("set_dmac") action set_dmac(bit<48> dmac) {
+    @name("set_dmac") action set_dmac_0(bit<48> dmac) {
         hdr.ethernet.dstAddr = dmac;
     }
-    @name("update_flowlet_id") action update_flowlet_id() {
+    @name("update_flowlet_id") action update_flowlet_id_0() {
         meta.ingress_metadata.flowlet_id = meta.ingress_metadata.flowlet_id + 16w1;
         flowlet_id_1.write((bit<32>)meta.ingress_metadata.flowlet_map_index, meta.ingress_metadata.flowlet_id);
     }
     @name("ecmp_group") table ecmp_group {
         actions = {
-            _drop_2();
-            set_ecmp_select();
-            NoAction();
+            _drop_1();
+            set_ecmp_select_0();
+            NoAction_3();
         }
         key = {
             hdr.ipv4.dstAddr: lpm @name("hdr.ipv4.dstAddr") ;
         }
         size = 1024;
-        default_action = NoAction();
+        default_action = NoAction_3();
     }
     @name("ecmp_nhop") table ecmp_nhop {
         actions = {
-            _drop_2();
-            set_nhop();
-            NoAction();
+            _drop_3();
+            set_nhop_0();
+            NoAction_4();
         }
         key = {
             meta.ingress_metadata.ecmp_offset: exact @name("meta.ingress_metadata.ecmp_offset") ;
         }
         size = 16384;
-        default_action = NoAction();
+        default_action = NoAction_4();
     }
     @name("flowlet") table flowlet {
         actions = {
-            lookup_flowlet_map();
-            NoAction();
+            lookup_flowlet_map_0();
+            NoAction_5();
         }
-        default_action = NoAction();
+        default_action = NoAction_5();
     }
     @name("forward") table forward {
         actions = {
-            set_dmac();
-            _drop_2();
-            NoAction();
+            set_dmac_0();
+            _drop_4();
+            NoAction_6();
         }
         key = {
             meta.ingress_metadata.nhop_ipv4: exact @name("meta.ingress_metadata.nhop_ipv4") ;
         }
         size = 512;
-        default_action = NoAction();
+        default_action = NoAction_6();
     }
     @name("new_flowlet") table new_flowlet {
         actions = {
-            update_flowlet_id();
-            NoAction();
+            update_flowlet_id_0();
+            NoAction_7();
         }
-        default_action = NoAction();
+        default_action = NoAction_7();
     }
     apply {
         @atomic {
