--- before_pass
+++ after_pass
@@ -89,7 +89,7 @@ control egress(inout headers hdr, inout
             NoAction();
         }
         key = {
-            standard_metadata.egress_port: exact;
+            standard_metadata.egress_port: exact @name("standard_metadata.egress_port") ;
         }
         size = 256;
         default_action = NoAction();
@@ -134,7 +134,7 @@ control ingress(inout headers hdr, inout
             NoAction();
         }
         key = {
-            hdr.ipv4.dstAddr: lpm;
+            hdr.ipv4.dstAddr: lpm @name("hdr.ipv4.dstAddr") ;
         }
         size = 1024;
         default_action = NoAction();
@@ -146,7 +146,7 @@ control ingress(inout headers hdr, inout
             NoAction();
         }
         key = {
-            meta.ingress_metadata.ecmp_offset: exact;
+            meta.ingress_metadata.ecmp_offset: exact @name("meta.ingress_metadata.ecmp_offset") ;
         }
         size = 16384;
         default_action = NoAction();
@@ -165,7 +165,7 @@ control ingress(inout headers hdr, inout
             NoAction();
         }
         key = {
-            meta.ingress_metadata.nhop_ipv4: exact;
+            meta.ingress_metadata.nhop_ipv4: exact @name("meta.ingress_metadata.nhop_ipv4") ;
         }
         size = 512;
         default_action = NoAction();
