--- before_pass
+++ after_pass
@@ -54,7 +54,7 @@ parser parserI(packet_in pkt, out Parsed
     }
 }
 control cIngress(inout Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
-    @name("foo") action foo() {
+    @name("foo") action foo_0() {
         meta.b = meta.b + 4w5;
     }
     @name("guh") table guh {
@@ -62,9 +62,9 @@ control cIngress(inout Parsed_packet hdr
             hdr.ethernet.srcAddr: exact @name("hdr.ethernet.srcAddr") ;
         }
         actions = {
-            foo();
+            foo_0();
         }
-        default_action = foo();
+        default_action = foo_0();
     }
     apply {
         guh.apply();
