--- before_pass
+++ after_pass
@@ -33,13 +33,13 @@ struct mystruct1 {
 }
 control DeparserI(packet_out packet, in Parsed_packet hdr) {
     apply {
-        packet.emit(hdr.ethernet);
+        packet.emit<Ethernet_h>(hdr.ethernet);
     }
 }
 parser parserI(packet_in pkt, out Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
     state start {
         bit<8> my_local = (bit<8>)8w1;
-        pkt.extract(hdr.ethernet);
+        pkt.extract<Ethernet_h>(hdr.ethernet);
         transition select(hdr.ethernet.etherType) {
             16w0x800: parse_ipv4;
             default: accept;
@@ -47,7 +47,7 @@ parser parserI(packet_in pkt, out Parsed
     }
     state parse_ipv4 {
         bit<16> my_local = (bit<16>)16w2;
-        pkt.extract(hdr.ipv4);
+        pkt.extract<ipv4_t>(hdr.ipv4);
         transition select(hdr.ipv4.version, hdr.ipv4.protocol) {
             (4w0x4, 8w0x6): accept;
             (4w0x4, 8w0x17): accept;
