--- before_pass
+++ after_pass
@@ -33,19 +33,25 @@ struct mystruct1 {
 }
 control DeparserI(packet_out packet, in Parsed_packet hdr) {
     apply {
-        packet.emit<Ethernet_h>(hdr.ethernet);
+        {
+            packet.emit<Ethernet_h>(hdr.ethernet);
+        }
     }
 }
 parser parserI(packet_in pkt, out Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
     state start {
-        pkt.extract<Ethernet_h>(hdr.ethernet);
+        {
+            pkt.extract<Ethernet_h>(hdr.ethernet);
+        }
         transition select(hdr.ethernet.etherType) {
             16w0x800: parse_ipv4;
             default: accept;
         }
     }
     state parse_ipv4 {
-        pkt.extract<ipv4_t>(hdr.ipv4);
+        {
+            pkt.extract<ipv4_t>(hdr.ipv4);
+        }
         transition select(hdr.ipv4.version, hdr.ipv4.protocol) {
             (4w0x4, 8w0x6): accept;
             (4w0x4, 8w0x17): accept;
@@ -55,7 +61,9 @@ parser parserI(packet_in pkt, out Parsed
 }
 control cIngress(inout Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
     @name("foo") action foo_0() {
-        meta.b = meta.b + 4w5;
+        {
+            meta.b = meta.b + 4w5;
+        }
     }
     @name("guh") table guh_0 {
         key = {
@@ -67,7 +75,9 @@ control cIngress(inout Parsed_packet hdr
         default_action = foo_0();
     }
     apply {
-        guh_0.apply();
+        {
+            guh_0.apply();
+        }
     }
 }
 control cEgress(inout Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
