--- before_pass
+++ after_pass
@@ -38,7 +38,7 @@ control DeparserI(packet_out packet, in
 }
 parser parserI(packet_in pkt, out Parsed_packet hdr, inout mystruct1 meta, inout standard_metadata_t stdmeta) {
     state start {
-        bit<8> my_local = (bit<8>)8w1;
+        bit<8> my_local = 8w1;
         pkt.extract<Ethernet_h>(hdr.ethernet);
         transition select(hdr.ethernet.etherType) {
             16w0x800: parse_ipv4;
@@ -46,7 +46,7 @@ parser parserI(packet_in pkt, out Parsed
         }
     }
     state parse_ipv4 {
-        bit<16> my_local = (bit<16>)16w2;
+        bit<16> my_local = 16w2;
         pkt.extract<ipv4_t>(hdr.ipv4);
         transition select(hdr.ipv4.version, hdr.ipv4.protocol) {
             (4w0x4, 8w0x6): accept;
