--- before_pass
+++ after_pass
@@ -61,7 +61,7 @@ control egress(inout headers hdr, inout
             @defaultonly NoAction();
         }
         key = {
-            meta.ingress_metadata.nexthop_index: exact;
+            meta.ingress_metadata.nexthop_index: exact @name("meta.ingress_metadata.nexthop_index") ;
         }
         size = 32768;
         default_action = NoAction();
@@ -92,7 +92,7 @@ control ingress(inout headers hdr, inout
             @defaultonly NoAction();
         }
         key = {
-            meta.ingress_metadata.bd: exact;
+            meta.ingress_metadata.bd: exact @name("meta.ingress_metadata.bd") ;
         }
         size = 65536;
         default_action = NoAction();
@@ -104,8 +104,8 @@ control ingress(inout headers hdr, inout
             @defaultonly NoAction();
         }
         key = {
-            meta.ingress_metadata.vrf: exact;
-            hdr.ipv4.dstAddr         : exact;
+            meta.ingress_metadata.vrf: exact @name("meta.ingress_metadata.vrf") ;
+            hdr.ipv4.dstAddr         : exact @name("hdr.ipv4.dstAddr") ;
         }
         size = 131072;
         default_action = NoAction();
@@ -117,8 +117,8 @@ control ingress(inout headers hdr, inout
             @defaultonly NoAction();
         }
         key = {
-            meta.ingress_metadata.vrf: exact;
-            hdr.ipv4.dstAddr         : lpm;
+            meta.ingress_metadata.vrf: exact @name("meta.ingress_metadata.vrf") ;
+            hdr.ipv4.dstAddr         : lpm @name("hdr.ipv4.dstAddr") ;
         }
         size = 16384;
         default_action = NoAction();
@@ -130,7 +130,7 @@ control ingress(inout headers hdr, inout
             @defaultonly NoAction();
         }
         key = {
-            meta.ingress_metadata.nexthop_index: exact;
+            meta.ingress_metadata.nexthop_index: exact @name("meta.ingress_metadata.nexthop_index") ;
         }
         size = 32768;
         default_action = NoAction();
@@ -141,7 +141,7 @@ control ingress(inout headers hdr, inout
             @defaultonly NoAction();
         }
         key = {
-            standard_metadata.ingress_port: exact;
+            standard_metadata.ingress_port: exact @name("standard_metadata.ingress_port") ;
         }
         size = 32768;
         default_action = NoAction();
