--- before_pass
+++ after_pass
@@ -47,13 +47,13 @@ parser ParserImpl(packet_in packet, out
 control egress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("on_miss") action on_miss() {
+    @name("egress.on_miss") action on_miss() {
     }
-    @name("rewrite_src_dst_mac") action rewrite_src_dst_mac(bit<48> smac, bit<48> dmac) {
+    @name("egress.rewrite_src_dst_mac") action rewrite_src_dst_mac(bit<48> smac, bit<48> dmac) {
         hdr.ethernet.srcAddr = smac;
         hdr.ethernet.dstAddr = dmac;
     }
-    @name("rewrite_mac") table rewrite_mac_0 {
+    @name("egress.rewrite_mac") table rewrite_mac_0 {
         actions = {
             on_miss();
             rewrite_src_dst_mac();
@@ -80,30 +80,30 @@ control ingress(inout headers hdr, inout
     }
     @name(".NoAction") action NoAction_11() {
     }
-    @name("set_vrf") action set_vrf(bit<12> vrf) {
+    @name("ingress.set_vrf") action set_vrf(bit<12> vrf) {
         meta.ingress_metadata.vrf = vrf;
     }
-    @name("on_miss") action on_miss_2() {
+    @name("ingress.on_miss") action on_miss_2() {
     }
-    @name("on_miss") action on_miss_5() {
+    @name("ingress.on_miss") action on_miss_5() {
     }
-    @name("on_miss") action on_miss_6() {
+    @name("ingress.on_miss") action on_miss_6() {
     }
-    @name("fib_hit_nexthop") action fib_hit_nexthop(bit<16> nexthop_index) {
+    @name("ingress.fib_hit_nexthop") action fib_hit_nexthop(bit<16> nexthop_index) {
         meta.ingress_metadata.nexthop_index = nexthop_index;
         hdr.ipv4.ttl = hdr.ipv4.ttl + 8w255;
     }
-    @name("fib_hit_nexthop") action fib_hit_nexthop_2(bit<16> nexthop_index) {
+    @name("ingress.fib_hit_nexthop") action fib_hit_nexthop_2(bit<16> nexthop_index) {
         meta.ingress_metadata.nexthop_index = nexthop_index;
         hdr.ipv4.ttl = hdr.ipv4.ttl + 8w255;
     }
-    @name("set_egress_details") action set_egress_details(bit<9> egress_spec) {
+    @name("ingress.set_egress_details") action set_egress_details(bit<9> egress_spec) {
         standard_metadata.egress_spec = egress_spec;
     }
-    @name("set_bd") action set_bd(bit<16> bd) {
+    @name("ingress.set_bd") action set_bd(bit<16> bd) {
         meta.ingress_metadata.bd = bd;
     }
-    @name("bd") table bd_0 {
+    @name("ingress.bd") table bd_0 {
         actions = {
             set_vrf();
             @defaultonly NoAction_1();
@@ -114,7 +114,7 @@ control ingress(inout headers hdr, inout
         size = 65536;
         default_action = NoAction_1();
     }
-    @name("ipv4_fib") table ipv4_fib_0 {
+    @name("ingress.ipv4_fib") table ipv4_fib_0 {
         actions = {
             on_miss_2();
             fib_hit_nexthop();
@@ -127,7 +127,7 @@ control ingress(inout headers hdr, inout
         size = 131072;
         default_action = NoAction_8();
     }
-    @name("ipv4_fib_lpm") table ipv4_fib_lpm_0 {
+    @name("ingress.ipv4_fib_lpm") table ipv4_fib_lpm_0 {
         actions = {
             on_miss_5();
             fib_hit_nexthop_2();
@@ -140,7 +140,7 @@ control ingress(inout headers hdr, inout
         size = 16384;
         default_action = NoAction_9();
     }
-    @name("nexthop") table nexthop_0 {
+    @name("ingress.nexthop") table nexthop_0 {
         actions = {
             on_miss_6();
             set_egress_details();
@@ -152,7 +152,7 @@ control ingress(inout headers hdr, inout
         size = 32768;
         default_action = NoAction_10();
     }
-    @name("port_mapping") table port_mapping_0 {
+    @name("ingress.port_mapping") table port_mapping_0 {
         actions = {
             set_bd();
             @defaultonly NoAction_11();
