--- before_pass
+++ after_pass
@@ -78,7 +78,7 @@ control ingress(inout headers hdr, inout
     }
     action fib_hit_nexthop(bit<16> nexthop_index) {
         meta.ingress_metadata.nexthop_index = nexthop_index;
-        hdr.ipv4.ttl = hdr.ipv4.ttl - 8w1;
+        hdr.ipv4.ttl = hdr.ipv4.ttl + 8w255;
     }
     action set_egress_details(bit<9> egress_spec) {
         standard_metadata.egress_spec = egress_spec;
