--- before_pass
+++ after_pass
@@ -46,9 +46,9 @@ parser IngressParserImpl(packet_in buffe
     }
 }
 control ingress(inout headers hdr, inout metadata user_meta, in psa_ingress_input_metadata_t istd, inout psa_ingress_output_metadata_t ostd) {
-    @name(".update_pkt_ip_byte_count") action update_pkt_ip_byte_count(inout PacketByteCountState_t s_0, in bit<16> ip_length_bytes_0) {
-        s_0[79:48] = s_0[79:48] + 32w1;
-        s_0[47:0] = s_0[47:0] + (bit<48>)ip_length_bytes_0;
+    @name(".update_pkt_ip_byte_count") action update_pkt_ip_byte_count(inout PacketByteCountState_t s, in bit<16> ip_length_bytes) {
+        s[79:48] = s[79:48] + 32w1;
+        s[47:0] = s[47:0] + (bit<48>)ip_length_bytes;
     }
     PacketByteCountState_t tmp;
     bit<80> tmp_0;
