--- before_pass
+++ after_pass
@@ -32,9 +32,9 @@ struct headers {
     ipv4_t     ipv4;
 }
 typedef bit<80> PacketByteCountState_t;
-action update_pkt_ip_byte_count(inout PacketByteCountState_t s, in bit<16> ip_length_bytes) {
-    s[79:48] = s[79:48] + 32w1;
-    s[47:0] = s[47:0] + (bit<48>)ip_length_bytes;
+action update_pkt_ip_byte_count(inout PacketByteCountState_t s_0, in bit<16> ip_length_bytes_0) {
+    s_0[79:48] = s_0[79:48] + 32w1;
+    s_0[47:0] = s_0[47:0] + (bit<48>)ip_length_bytes_0;
 }
 parser IngressParserImpl(packet_in buffer, out headers parsed_hdr, inout metadata user_meta, in psa_ingress_parser_input_metadata_t istd, in empty_metadata_t resubmit_meta, in empty_metadata_t recirculate_meta) {
     state start {
