--- before_pass
+++ after_pass
@@ -52,7 +52,7 @@ control ingress(inout headers hdr, inout
     }
     PacketByteCountState_t tmp;
     bit<80> tmp_0;
-    @name("port_pkt_ip_bytes_in") Register<PacketByteCountState_t, PortId_t>(32w512) port_pkt_ip_bytes_in_0;
+    @name("ingress.port_pkt_ip_bytes_in") Register<PacketByteCountState_t, PortId_t>(32w512) port_pkt_ip_bytes_in_0;
     apply {
         ostd.egress_port = (PortId_t)32w0;
         if (hdr.ipv4.isValid()) @atomic {
