--- before_pass
+++ after_pass
@@ -57,13 +57,11 @@ control ingress(inout headers hdr, inout
     Register<PacketByteCountState_t, PortId_t>(32w512) port_pkt_ip_bytes_in;
     apply {
         ostd.egress_port = (PortId_t)32w0;
-        if (hdr.ipv4.isValid()) {
-            @atomic {
-                PacketByteCountState_t tmp;
-                tmp = port_pkt_ip_bytes_in.read(istd.ingress_port);
-                update_pkt_ip_byte_count(tmp, hdr.ipv4.totalLen);
-                port_pkt_ip_bytes_in.write(istd.ingress_port, tmp);
-            }
+        if (hdr.ipv4.isValid()) @atomic {
+            PacketByteCountState_t tmp;
+            tmp = port_pkt_ip_bytes_in.read(istd.ingress_port);
+            update_pkt_ip_byte_count(tmp, hdr.ipv4.totalLen);
+            port_pkt_ip_bytes_in.write(istd.ingress_port, tmp);
         }
     }
 }
