--- before_pass
+++ after_pass
@@ -22,7 +22,7 @@ parser ParserI(packet_in b, out H parsed
         transition p0;
     }
     state p0 {
-        b.extract(parsedHdr.hstack.next);
+        b.extract<T>(parsedHdr.hstack.next);
         transition select(parsedHdr.hstack.next.y) {
             32w0: p0;
             default: accept;
@@ -51,4 +51,4 @@ control DeparserI(packet_out b, in H hdr
     apply {
     }
 }
-V1Switch(ParserI(), VerifyChecksumI(), IngressI(), EgressI(), ComputeChecksumI(), DeparserI()) main;
+V1Switch<H, M>(ParserI(), VerifyChecksumI(), IngressI(), EgressI(), ComputeChecksumI(), DeparserI()) main;
