--- before_pass
+++ after_pass
@@ -29,14 +29,10 @@ parser ParserI(packet_in b, out H parsed
         }
     }
 }
-control ctrl(inout M meta) {
-    apply {
-    }
-}
 control IngressI(inout H hdr, inout M meta, inout std_meta_t std_meta) {
-    @name("do_ctrl") ctrl() do_ctrl_0;
     apply {
-        do_ctrl_0.apply(meta);
+        {
+        }
     }
 }
 control EgressI(inout H hdr, inout M meta, inout std_meta_t std_meta) {
