--- before_pass
+++ after_pass
@@ -10,17 +10,35 @@ control ctrl(inout struct_t input, out b
     header_h tmp2_0;
     header_h tmp3_0;
     @name("act") action act_0() {
-        tmp0_0 = input.hdr.field;
-        input.hdr.setValid();
-        tmp1_0 = tmp0_0;
-        tmp2_0 = input.hdr;
-        input.hdr.setInvalid();
-        tmp3_0 = tmp2_0;
+        {
+            tmp0_0 = input.hdr.field;
+        }
+        {
+            input.hdr.setValid();
+        }
+        {
+            tmp1_0 = tmp0_0;
+        }
+        {
+            tmp2_0 = input.hdr;
+        }
+        {
+            input.hdr.setInvalid();
+        }
+        {
+            tmp3_0 = tmp2_0;
+        }
     }
     apply {
-        act_0();
-        out1 = tmp1_0;
-        out2 = tmp3_0;
+        {
+            act_0();
+        }
+        {
+            out1 = tmp1_0;
+        }
+        {
+            out2 = tmp3_0;
+        }
     }
 }
 control MyControl<S, H>(inout S i, out bit<8> o1, out H o2);
