--- before_pass
+++ after_pass
@@ -9,7 +9,7 @@ control ctrl(inout struct_t input, out b
     bit<8> tmp1;
     header_h tmp2;
     header_h tmp3;
-    @name("act") action act() {
+    @name("act") action act_0() {
         tmp0 = input.hdr.field;
         input.hdr.setValid();
         tmp1 = tmp0;
@@ -18,7 +18,7 @@ control ctrl(inout struct_t input, out b
         tmp3 = tmp2;
     }
     apply {
-        act();
+        act_0();
         out1 = tmp1;
         out2 = tmp3;
     }
