--- before_pass
+++ after_pass
@@ -9,15 +9,9 @@ extern Overloaded {
     void f(bit<16> b);
     void f(bit<32> a, bit<16> b);
 }
-control d_0(out bit<32> x) {
-    apply {
-        x = 32w4294967294;
-    }
-}
 control c() {
     bit<32> z_0;
     @name("o") Overloaded() o_0;
-    @name("dinst") d_0() dinst_0;
     apply {
         f();
         f(a = 32w2);
@@ -29,7 +23,9 @@ control c() {
         o_0.f(b = 16w1);
         o_0.f(a = 32w1, b = 16w2);
         o_0.f(b = 16w2, a = 32w1);
-        dinst_0.apply(z_0);
+        {
+            z_0 = 32w4294967294;
+        }
     }
 }
 control proto();
