--- before_pass
+++ after_pass
@@ -15,21 +15,21 @@ control d(out bit<32> x)(bit<32> b) {
     }
 }
 control c() {
-    Overloaded() o;
-    d(b = 32w2) dinst;
+    @name("o") Overloaded() o_0;
+    @name("dinst") d(b = 32w2) dinst_0;
     apply {
         f();
         f(a = 32w2);
         f(b = 16w1);
         f(a = 32w1, b = 16w2);
         f(b = 16w2, a = 32w1);
-        o.f();
-        o.f(a = 32w2);
-        o.f(b = 16w1);
-        o.f(a = 32w1, b = 16w2);
-        o.f(b = 16w2, a = 32w1);
-        bit<32> z;
-        dinst.apply(z);
+        o_0.f();
+        o_0.f(a = 32w2);
+        o_0.f(b = 16w1);
+        o_0.f(a = 32w1, b = 16w2);
+        o_0.f(b = 16w2, a = 32w1);
+        bit<32> z_0;
+        dinst_0.apply(z_0);
     }
 }
 control proto();
