--- before_pass
+++ after_pass
@@ -9,15 +9,15 @@ extern Overloaded {
     void f(bit<16> b);
     void f(bit<32> a, bit<16> b);
 }
-control d(out bit<32> x)(bit<32> b) {
+control d_0(out bit<32> x) {
     apply {
-        x = -b;
+        x = 32w4294967294;
     }
 }
 control c() {
     bit<32> z_0;
     @name("o") Overloaded() o_0;
-    @name("dinst") d(b = 32w2) dinst_0;
+    @name("dinst") d_0() dinst_0;
     apply {
         f();
         f(a = 32w2);
