--- before_pass
+++ after_pass
@@ -10,19 +10,19 @@ control c(inout bit<16> p) {
     }
     bit<16> local_0;
     bit<16> tmp;
-    @name("cntr") Virtual() cntr_0 = {
+    @name("c.cntr") Virtual() cntr_0 = {
         bit<16> f(in bit<16> ix) {
             return ix + local_0;
         }
     };
-    @name("final_ctr") action final_ctr() {
+    @name("c.final_ctr") action final_ctr() {
         tmp = cntr_0.total();
         p = tmp;
     }
-    @name("add_ctr") action add_ctr() {
+    @name("c.add_ctr") action add_ctr() {
         cntr_0.increment();
     }
-    @name("run_ctr") table run_ctr_0 {
+    @name("c.run_ctr") table run_ctr_0 {
         key = {
             p: exact @name("p") ;
         }
