--- before_pass
+++ after_pass
@@ -79,7 +79,7 @@ control MyIngress(inout headers hdr, ino
     }
     action operation_crc() {
         bit<32> nbase = hdr.p4calc.operand_b;
-        bit<64> ncount = (bit<64>)64w8589934592;
+        bit<64> ncount = 64w8589934592;
         bit<32> nselect;
         bit<32> ninput = hdr.p4calc.operand_a;
         hash<bit<32>, bit<32>, tuple<bit<32>>, bit<64>>(nselect, HashAlgorithm.crc32, nbase, { ninput }, ncount);
