--- before_pass
+++ after_pass
@@ -56,62 +56,50 @@ control MyIngress(inout headers hdr, ino
     bit<32> nselect_0;
     bit<32> ninput_0;
     @name("operation_add") action operation_add_0() {
-        {
-            hdr.p4calc.res = hdr.p4calc.operand_a + hdr.p4calc.operand_b;
-            tmp_0 = hdr.ethernet.dstAddr;
-            hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
-            hdr.ethernet.srcAddr = tmp_0;
-            standard_metadata.egress_spec = standard_metadata.ingress_port;
-        }
+        hdr.p4calc.res = hdr.p4calc.operand_a + hdr.p4calc.operand_b;
+        tmp_0 = hdr.ethernet.dstAddr;
+        hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
+        hdr.ethernet.srcAddr = tmp_0;
+        standard_metadata.egress_spec = standard_metadata.ingress_port;
     }
     @name("operation_sub") action operation_sub_0() {
-        {
-            hdr.p4calc.res = hdr.p4calc.operand_a - hdr.p4calc.operand_b;
-            tmp_0 = hdr.ethernet.dstAddr;
-            hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
-            hdr.ethernet.srcAddr = tmp_0;
-            standard_metadata.egress_spec = standard_metadata.ingress_port;
-        }
+        hdr.p4calc.res = hdr.p4calc.operand_a - hdr.p4calc.operand_b;
+        tmp_0 = hdr.ethernet.dstAddr;
+        hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
+        hdr.ethernet.srcAddr = tmp_0;
+        standard_metadata.egress_spec = standard_metadata.ingress_port;
     }
     @name("operation_and") action operation_and_0() {
-        {
-            hdr.p4calc.res = hdr.p4calc.operand_a & hdr.p4calc.operand_b;
-            tmp_0 = hdr.ethernet.dstAddr;
-            hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
-            hdr.ethernet.srcAddr = tmp_0;
-            standard_metadata.egress_spec = standard_metadata.ingress_port;
-        }
+        hdr.p4calc.res = hdr.p4calc.operand_a & hdr.p4calc.operand_b;
+        tmp_0 = hdr.ethernet.dstAddr;
+        hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
+        hdr.ethernet.srcAddr = tmp_0;
+        standard_metadata.egress_spec = standard_metadata.ingress_port;
     }
     @name("operation_or") action operation_or_0() {
-        {
-            hdr.p4calc.res = hdr.p4calc.operand_a | hdr.p4calc.operand_b;
-            tmp_0 = hdr.ethernet.dstAddr;
-            hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
-            hdr.ethernet.srcAddr = tmp_0;
-            standard_metadata.egress_spec = standard_metadata.ingress_port;
-        }
+        hdr.p4calc.res = hdr.p4calc.operand_a | hdr.p4calc.operand_b;
+        tmp_0 = hdr.ethernet.dstAddr;
+        hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
+        hdr.ethernet.srcAddr = tmp_0;
+        standard_metadata.egress_spec = standard_metadata.ingress_port;
     }
     @name("operation_xor") action operation_xor_0() {
-        {
-            hdr.p4calc.res = hdr.p4calc.operand_a ^ hdr.p4calc.operand_b;
-            tmp_0 = hdr.ethernet.dstAddr;
-            hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
-            hdr.ethernet.srcAddr = tmp_0;
-            standard_metadata.egress_spec = standard_metadata.ingress_port;
-        }
+        hdr.p4calc.res = hdr.p4calc.operand_a ^ hdr.p4calc.operand_b;
+        tmp_0 = hdr.ethernet.dstAddr;
+        hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
+        hdr.ethernet.srcAddr = tmp_0;
+        standard_metadata.egress_spec = standard_metadata.ingress_port;
     }
     @name("operation_crc") action operation_crc_0() {
         nbase_0 = hdr.p4calc.operand_b;
         ncount_0 = 64w8589934592;
         ninput_0 = hdr.p4calc.operand_a;
         hash<bit<32>, bit<32>, tuple<bit<32>>, bit<64>>(nselect_0, HashAlgorithm.crc32, nbase_0, { ninput_0 }, ncount_0);
-        {
-            hdr.p4calc.res = nselect_0;
-            tmp_0 = hdr.ethernet.dstAddr;
-            hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
-            hdr.ethernet.srcAddr = tmp_0;
-            standard_metadata.egress_spec = standard_metadata.ingress_port;
-        }
+        hdr.p4calc.res = nselect_0;
+        tmp_0 = hdr.ethernet.dstAddr;
+        hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
+        hdr.ethernet.srcAddr = tmp_0;
+        standard_metadata.egress_spec = standard_metadata.ingress_port;
     }
     @name("operation_drop") action operation_drop_0() {
         mark_to_drop(standard_metadata);
