--- before_pass
+++ after_pass
@@ -55,8 +55,8 @@ control MyIngress(inout headers hdr, ino
     bit<64> ncount_0;
     bit<32> nselect_0;
     bit<32> ninput_0;
-    @name("send_back") action send_back_0(bit<32> result) {
-        hdr.p4calc.res = result;
+    @name("send_back") action send_back_0(bit<32> result_0) {
+        hdr.p4calc.res = result_0;
         tmp_0 = hdr.ethernet.dstAddr;
         hdr.ethernet.dstAddr = hdr.ethernet.srcAddr;
         hdr.ethernet.srcAddr = tmp_0;
