--- before_pass
+++ after_pass
@@ -20,38 +20,38 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout EMPTY b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
-    ActionProfile(32w1024) ap;
-    action a1() {
+    @name("ap") ActionProfile(32w1024) ap_0;
+    @name("a1") action a1_0() {
     }
-    action a2() {
+    @name("a2") action a2_0() {
     }
-    table tbl {
+    @name("tbl") table tbl_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
             NoAction();
-            a1();
-            a2();
+            a1_0();
+            a2_0();
         }
-        psa_implementation = ap;
+        psa_implementation = ap_0;
         default_action = NoAction();
     }
-    table tbl2 {
+    @name("tbl2") table tbl2_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
             NoAction();
-            a1();
-            a2();
+            a1_0();
+            a2_0();
         }
-        psa_implementation = ap;
+        psa_implementation = ap_0;
         default_action = NoAction();
     }
     apply {
-        tbl.apply();
-        tbl2.apply();
+        tbl_0.apply();
+        tbl2_0.apply();
     }
 }
 control MyEC(inout EMPTY a, inout EMPTY b, in psa_egress_input_metadata_t c, inout psa_egress_output_metadata_t d) {
