--- before_pass
+++ after_pass
@@ -34,18 +34,18 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name("shift") action shift_0() {
+    @name("shift") action shift() {
         h.h.c = (bit<64>)(h.h.a >> h.h.b);
         sm.egress_spec = 9w0;
     }
-    @name("t") table t {
+    @name("t") table t_0 {
         actions = {
-            shift_0();
+            shift();
         }
-        const default_action = shift_0();
+        const default_action = shift();
     }
     apply {
-        t.apply();
+        t_0.apply();
     }
 }
 V1Switch<Headers, Meta>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
