--- before_pass
+++ after_pass
@@ -24,14 +24,14 @@ control MyIC(inout ethernet_t a, inout E
     }
     @name(".NoAction") action NoAction_3() {
     }
-    @name("counter0") DirectCounter<bit<12>>(PSA_CounterType_t.PACKETS) counter0_0;
-    @name("execute") action execute_1() {
+    @name("MyIC.counter0") DirectCounter<bit<12>>(PSA_CounterType_t.PACKETS) counter0_0;
+    @name("MyIC.execute") action execute_1() {
         counter0_0.count();
     }
-    @name("execute") action execute_3() {
+    @name("MyIC.execute") action execute_3() {
         counter0_0.count();
     }
-    @name("tbl") table tbl_0 {
+    @name("MyIC.tbl") table tbl_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
@@ -42,7 +42,7 @@ control MyIC(inout ethernet_t a, inout E
         psa_direct_counter = counter0_0;
         default_action = NoAction_0();
     }
-    @name("tbl2") table tbl2_0 {
+    @name("MyIC.tbl2") table tbl2_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
