--- before_pass
+++ after_pass
@@ -22,12 +22,12 @@ control IngressImpl(inout headers_t hdr,
 control EgressImpl(inout headers_t hdr, inout metadata meta, inout standard_metadata_t ostd) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("set_true") action set_true() {
+    @name("EgressImpl.set_true") action set_true() {
         if (meta.field == 8w0) {
             meta.cond = true;
         }
     }
-    @name("change_cond") table change_cond_0 {
+    @name("EgressImpl.change_cond") table change_cond_0 {
         key = {
             ostd.egress_spec: exact @name("ostd.egress_spec") ;
         }
