--- before_pass
+++ after_pass
@@ -20,23 +20,23 @@ control IngressImpl(inout headers_t hdr,
     }
 }
 control EgressImpl(inout headers_t hdr, inout metadata meta, inout standard_metadata_t ostd) {
-    action set_true() {
+    @name("set_true") action set_true_0() {
         if (meta.field == 8w0) {
             meta.cond = true;
         }
     }
-    table change_cond {
+    @name("change_cond") table change_cond_0 {
         key = {
             ostd.egress_spec: exact @name("ostd.egress_spec") ;
         }
         actions = {
-            set_true();
+            set_true_0();
             @defaultonly NoAction();
         }
         default_action = NoAction();
     }
     apply {
-        change_cond.apply();
+        change_cond_0.apply();
     }
 }
 control ComputeChecksumImpl(inout headers_t hdr, inout metadata meta) {
