--- before_pass
+++ after_pass
@@ -20,25 +20,25 @@ control IngressImpl(inout headers_t hdr,
     }
 }
 control EgressImpl(inout headers_t hdr, inout metadata meta, inout standard_metadata_t ostd) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name("set_true") action set_true_0() {
+    @name("set_true") action set_true() {
         if (meta.field == 8w0) {
             meta.cond = true;
         }
     }
-    @name("change_cond") table change_cond {
+    @name("change_cond") table change_cond_0 {
         key = {
             ostd.egress_spec: exact @name("ostd.egress_spec") ;
         }
         actions = {
-            set_true_0();
-            @defaultonly NoAction_1();
+            set_true();
+            @defaultonly NoAction_0();
         }
-        default_action = NoAction_1();
+        default_action = NoAction_0();
     }
     apply {
-        change_cond.apply();
+        change_cond_0.apply();
     }
 }
 control ComputeChecksumImpl(inout headers_t hdr, inout metadata meta) {
