--- before_pass
+++ after_pass
@@ -23,8 +23,14 @@ control EgressImpl(inout headers_t hdr,
     @name(".NoAction") action NoAction_0() {
     }
     @name("EgressImpl.set_true") action set_true() {
-        if (meta.field == 8w0) {
-            meta.cond = true;
+        {
+            bool cond_0;
+            {
+                bool pred;
+                cond_0 = meta.field == 8w0;
+                pred = cond_0;
+                meta.cond = (pred ? true : meta.cond);
+            }
         }
     }
     @name("EgressImpl.change_cond") table change_cond_0 {
