--- before_pass
+++ after_pass
@@ -15,7 +15,7 @@ struct headers_t {
 }
 parser IngressParserImpl(packet_in pkt, out headers_t hdr, inout metadata_t user_meta, in psa_ingress_parser_input_metadata_t istd, in empty_metadata_t resubmit_meta, in empty_metadata_t recirculate_meta) {
     state start {
-        pkt.extract(hdr.ethernet);
+        pkt.extract<ethernet_t>(hdr.ethernet);
         transition accept;
     }
 }
@@ -38,7 +38,7 @@ control cEgress(inout headers_t hdr, ino
 }
 control CommonDeparserImpl(packet_out packet, inout headers_t hdr) {
     apply {
-        packet.emit(hdr.ethernet);
+        packet.emit<ethernet_t>(hdr.ethernet);
     }
 }
 control IngressDeparserImpl(packet_out buffer, out empty_metadata_t clone_i2e_meta, out empty_metadata_t resubmit_meta, out empty_metadata_t normal_meta, inout headers_t hdr, in metadata_t meta, in psa_ingress_output_metadata_t istd) {
@@ -53,6 +53,6 @@ control EgressDeparserImpl(packet_out bu
         cp.apply(buffer, hdr);
     }
 }
-IngressPipeline(IngressParserImpl(), cIngress(), IngressDeparserImpl()) ip;
-EgressPipeline(EgressParserImpl(), cEgress(), EgressDeparserImpl()) ep;
-PSA_Switch(ip, PacketReplicationEngine(), ep, BufferingQueueingEngine()) main;
+IngressPipeline<headers_t, metadata_t, empty_metadata_t, empty_metadata_t, empty_metadata_t, empty_metadata_t>(IngressParserImpl(), cIngress(), IngressDeparserImpl()) ip;
+EgressPipeline<headers_t, metadata_t, empty_metadata_t, empty_metadata_t, empty_metadata_t, empty_metadata_t>(EgressParserImpl(), cEgress(), EgressDeparserImpl()) ep;
+PSA_Switch<headers_t, metadata_t, headers_t, metadata_t, empty_metadata_t, empty_metadata_t, empty_metadata_t, empty_metadata_t, empty_metadata_t>(ip, PacketReplicationEngine(), ep, BufferingQueueingEngine()) main;
