--- before_pass
+++ after_pass
@@ -20,10 +20,18 @@ parser IngressParserImpl(packet_in pkt,
     }
 }
 control cIngress(inout headers_t hdr, inout metadata_t user_meta, in psa_ingress_input_metadata_t istd, inout psa_ingress_output_metadata_t ostd) {
+    @name(".send_to_port") action send_to_port_1(inout psa_ingress_output_metadata_t meta_0, in PortId_t egress_port_0) {
+        meta_0.drop = false;
+        meta_0.multicast_group = (MulticastGroup_t)32w0;
+        meta_0.egress_port = egress_port_0;
+    }
+    @name(".ingress_drop") action ingress_drop_1(inout psa_ingress_output_metadata_t meta_1) {
+        meta_1.drop = true;
+    }
     apply {
-        send_to_port(ostd, (PortId_t)(PortIdUint_t)hdr.ethernet.dstAddr);
+        send_to_port_1(ostd, (PortId_t)(PortIdUint_t)hdr.ethernet.dstAddr);
         if (hdr.ethernet.dstAddr == 48w0) {
-            ingress_drop(ostd);
+            ingress_drop_1(ostd);
         }
     }
 }
