--- before_pass
+++ after_pass
@@ -23,22 +23,22 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout user_meta_t b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
-    Random<bit<16>>(16w200, 16w400) rand;
-    action execute_random() {
-        b.data = rand.read();
+    @name("rand") Random<bit<16>>(16w200, 16w400) rand_0;
+    @name("execute_random") action execute_random_0() {
+        b.data = rand_0.read();
     }
-    table tbl {
+    @name("tbl") table tbl_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
             NoAction();
-            execute_random();
+            execute_random_0();
         }
         default_action = NoAction();
     }
     apply {
-        tbl.apply();
+        tbl_0.apply();
     }
 }
 control MyEC(inout EMPTY a, inout EMPTY b, in psa_egress_input_metadata_t c, inout psa_egress_output_metadata_t d) {
