--- before_pass
+++ after_pass
@@ -26,12 +26,12 @@ control MyIC(inout ethernet_t a, inout u
     @name(".NoAction") action NoAction_0() {
     }
     bit<16> tmp;
-    @name("rand") Random<bit<16>>(16w200, 16w400) rand_0;
-    @name("execute_random") action execute_random() {
+    @name("MyIC.rand") Random<bit<16>>(16w200, 16w400) rand_0;
+    @name("MyIC.execute_random") action execute_random() {
         tmp = rand_0.read();
         b.data = tmp;
     }
-    @name("tbl") table tbl_0 {
+    @name("MyIC.tbl") table tbl_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
