--- before_pass
+++ after_pass
@@ -23,9 +23,11 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout user_meta_t b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
+    @name(".NoAction") action NoAction_1() {
+    }
     bit<16> tmp_0;
     @name("rand") Random<bit<16>>(16w200, 16w400) rand;
-    @name("execute_random") action execute_random() {
+    @name("execute_random") action execute_random_0() {
         tmp_0 = rand.read();
         b.data = tmp_0;
     }
@@ -34,10 +36,10 @@ control MyIC(inout ethernet_t a, inout u
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
-            NoAction();
-            execute_random();
+            NoAction_1();
+            execute_random_0();
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     apply {
         tbl.apply();
