--- before_pass
+++ after_pass
@@ -87,19 +87,19 @@ control ingress(inout headers_t hdr, ino
     }
     table custom_table {
         key = {
-            hdr.ethernet.srcAddr: exact;
-            hdr.ethernet.dstAddr: exact;
-            hdr.custom.e        : exact;
-            hdr.custom.ed       : exact;
-            hdr.custom.et       : exact;
-            hdr.custom.edd      : exact;
-            hdr.custom.eddt     : exact;
-            hdr.custom.edtd     : exact;
-            hdr.custom.edtt     : exact;
-            hdr.custom.etdd     : exact;
-            hdr.custom.etdt     : exact;
-            hdr.custom.ettd     : exact;
-            hdr.custom.ettt     : exact;
+            hdr.ethernet.srcAddr: exact @name("hdr.ethernet.srcAddr") ;
+            hdr.ethernet.dstAddr: exact @name("hdr.ethernet.dstAddr") ;
+            hdr.custom.e        : exact @name("hdr.custom.e") ;
+            hdr.custom.ed       : exact @name("hdr.custom.ed") ;
+            hdr.custom.et       : exact @name("hdr.custom.et") ;
+            hdr.custom.edd      : exact @name("hdr.custom.edd") ;
+            hdr.custom.eddt     : exact @name("hdr.custom.eddt") ;
+            hdr.custom.edtd     : exact @name("hdr.custom.edtd") ;
+            hdr.custom.edtt     : exact @name("hdr.custom.edtt") ;
+            hdr.custom.etdd     : exact @name("hdr.custom.etdd") ;
+            hdr.custom.etdt     : exact @name("hdr.custom.etdt") ;
+            hdr.custom.ettd     : exact @name("hdr.custom.ettd") ;
+            hdr.custom.ettt     : exact @name("hdr.custom.ettt") ;
         }
         actions = {
             set_output();
