--- before_pass
+++ after_pass
@@ -102,11 +102,11 @@ control ingress(inout headers_t hdr, ino
             hdr.custom.ettt     : exact;
         }
         actions = {
-            set_output;
-            set_headers;
-            my_drop;
+            set_output();
+            set_headers();
+            my_drop();
         }
-        default_action = my_drop;
+        default_action = my_drop();
     }
     apply {
         if (hdr.custom.isValid()) {
