--- before_pass
+++ after_pass
@@ -1,21 +1,21 @@
 #include <core.p4>
 #include <v1model.p4>
 typedef bit<48> EthD_t;
-@p4runtime_translation("p4.org/psa/v1/EthT_t" , 48) type bit<48> EthT_t;
+typedef bit<48> EthT_t;
 typedef bit<8> CustomD_t;
-type bit<8> CustomT_t;
+typedef bit<8> CustomT_t;
 typedef CustomD_t CustomDD_t;
-type CustomD_t CustomDT_t;
+typedef CustomD_t CustomDT_t;
 typedef CustomT_t CustomTD_t;
-type CustomT_t CustomTT_t;
+typedef CustomT_t CustomTT_t;
 typedef CustomDD_t CustomDDD_t;
-type CustomDD_t CustomDDT_t;
+typedef CustomDD_t CustomDDT_t;
 typedef CustomDT_t CustomDTD_t;
-type CustomDT_t CustomDTT_t;
+typedef CustomDT_t CustomDTT_t;
 typedef CustomTD_t CustomTDD_t;
-type CustomTD_t CustomTDT_t;
+typedef CustomTD_t CustomTDT_t;
 typedef CustomTT_t CustomTTD_t;
-type CustomTT_t CustomTTT_t;
+typedef CustomTT_t CustomTTT_t;
 header ethernet_t {
     EthD_t  dstAddr;
     EthT_t  srcAddr;
