--- before_pass
+++ after_pass
@@ -1,36 +1,36 @@
 control ctrl() {
-    action e() {
+    @name("e") action e_0() {
         exit;
     }
-    action f() {
+    @name("f") action f_0() {
     }
-    table t {
+    @name("t") table t_0 {
         actions = {
-            e();
-            f();
+            e_0();
+            f_0();
         }
-        default_action = e();
+        default_action = e_0();
     }
     apply {
-        bit<32> a;
-        bit<32> b;
-        bit<32> c;
-        a = 32w0;
-        b = 32w1;
-        c = 32w2;
-        switch (t.apply().action_run) {
-            e: {
-                b = 32w2;
-                t.apply();
-                c = 32w3;
+        bit<32> a_0;
+        bit<32> b_0;
+        bit<32> c_0;
+        a_0 = 32w0;
+        b_0 = 32w1;
+        c_0 = 32w2;
+        switch (t_0.apply().action_run) {
+            e_0: {
+                b_0 = 32w2;
+                t_0.apply();
+                c_0 = 32w3;
             }
-            f: {
-                b = 32w3;
-                t.apply();
-                c = 32w4;
+            f_0: {
+                b_0 = 32w3;
+                t_0.apply();
+                c_0 = 32w4;
             }
         }
-        c = 32w5;
+        c_0 = 32w5;
     }
 }
 control noop();
