--- before_pass
+++ after_pass
@@ -15,22 +15,42 @@ control ctrl() {
         default_action = e_0();
     }
     apply {
-        a_0 = 32w0;
-        b_0 = 32w1;
-        c_0 = 32w2;
+        {
+            a_0 = 32w0;
+        }
+        {
+            b_0 = 32w1;
+        }
+        {
+            c_0 = 32w2;
+        }
         switch (t_0.apply().action_run) {
             e_0: {
-                b_0 = 32w2;
-                t_0.apply();
-                c_0 = 32w3;
+                {
+                    b_0 = 32w2;
+                }
+                {
+                    t_0.apply();
+                }
+                {
+                    c_0 = 32w3;
+                }
             }
             f_0: {
-                b_0 = 32w3;
-                t_0.apply();
-                c_0 = 32w4;
+                {
+                    b_0 = 32w3;
+                }
+                {
+                    t_0.apply();
+                }
+                {
+                    c_0 = 32w4;
+                }
             }
         }
-        c_0 = 32w5;
+        {
+            c_0 = 32w5;
+        }
     }
 }
 control noop();
