--- before_pass
+++ after_pass
@@ -1,22 +1,22 @@
 control ctrl() {
-    @name("e") action e() {
+    @name("e") action e_0() {
         exit;
     }
-    @name("f") action f() {
+    @name("f") action f_0() {
     }
     @name("t") table t {
         actions = {
-            e();
-            f();
+            e_0();
+            f_0();
         }
-        default_action = e();
+        default_action = e_0();
     }
     apply {
         switch (t.apply().action_run) {
-            e: {
+            e_0: {
                 t.apply();
             }
-            f: {
+            f_0: {
                 t.apply();
             }
         }
