--- before_pass
+++ after_pass
@@ -1,23 +1,23 @@
 control ctrl() {
-    @name("e") action e_0() {
+    @name("e") action e() {
         exit;
     }
-    @name("f") action f_0() {
+    @name("f") action f() {
     }
-    @name("t") table t_0 {
+    @name("t") table t {
         actions = {
-            e_0();
-            f_0();
+            e();
+            f();
         }
-        default_action = e_0();
+        default_action = e();
     }
     apply {
-        switch (t_0.apply().action_run) {
-            e_0: {
-                t_0.apply();
+        switch (t.apply().action_run) {
+            e: {
+                t.apply();
             }
-            f_0: {
-                t_0.apply();
+            f: {
+                t.apply();
             }
         }
     }
