--- before_pass
+++ after_pass
@@ -72,7 +72,7 @@ control ingress(inout headers hdr, inout
             drop_with_count();
         }
         key = {
-            hdr.ipv4.dstAddr: lpm;
+            hdr.ipv4.dstAddr: lpm @name("hdr.ipv4.dstAddr") ;
         }
         default_action = drop_with_count();
         counters = ipv4_da_lpm_stats;
@@ -83,7 +83,7 @@ control ingress(inout headers hdr, inout
             my_drop(standard_metadata);
         }
         key = {
-            meta.fwd_metadata.l2ptr: exact;
+            meta.fwd_metadata.l2ptr: exact @name("meta.fwd_metadata.l2ptr") ;
         }
         default_action = my_drop(standard_metadata);
     }
@@ -102,7 +102,7 @@ control egress(inout headers hdr, inout
             my_drop(standard_metadata);
         }
         key = {
-            meta.fwd_metadata.out_bd: exact;
+            meta.fwd_metadata.out_bd: exact @name("meta.fwd_metadata.out_bd") ;
         }
         default_action = my_drop(standard_metadata);
     }
