--- before_pass
+++ after_pass
@@ -68,18 +68,18 @@ control ingress(inout headers hdr, inout
     }
     table ipv4_da_lpm {
         actions = {
-            set_l2ptr;
-            drop_with_count;
+            set_l2ptr();
+            drop_with_count();
         }
         key = {
             hdr.ipv4.dstAddr: lpm;
         }
-        default_action = drop_with_count;
+        default_action = drop_with_count();
         counters = ipv4_da_lpm_stats;
     }
     table mac_da {
         actions = {
-            set_bd_dmac_intf;
+            set_bd_dmac_intf();
             my_drop(standard_metadata);
         }
         key = {
@@ -98,7 +98,7 @@ control egress(inout headers hdr, inout
     }
     table send_frame {
         actions = {
-            rewrite_mac;
+            rewrite_mac();
             my_drop(standard_metadata);
         }
         key = {
