--- before_pass
+++ after_pass
@@ -47,22 +47,22 @@ control ingress(inout headers hdr, inout
     @name(".my_drop") action my_drop(inout standard_metadata_t smeta) {
         mark_to_drop(smeta);
     }
-    @name("ipv4_da_lpm_stats") direct_counter(CounterType.packets) ipv4_da_lpm_stats_0;
-    @name("set_l2ptr") action set_l2ptr(bit<32> l2ptr) {
+    @name("ingress.ipv4_da_lpm_stats") direct_counter(CounterType.packets) ipv4_da_lpm_stats_0;
+    @name("ingress.set_l2ptr") action set_l2ptr(bit<32> l2ptr) {
         ipv4_da_lpm_stats_0.count();
         meta.fwd_metadata.l2ptr = l2ptr;
     }
-    @name("drop_with_count") action drop_with_count() {
+    @name("ingress.drop_with_count") action drop_with_count() {
         ipv4_da_lpm_stats_0.count();
         mark_to_drop(standard_metadata);
     }
-    @name("set_bd_dmac_intf") action set_bd_dmac_intf(bit<24> bd, bit<48> dmac, bit<9> intf) {
+    @name("ingress.set_bd_dmac_intf") action set_bd_dmac_intf(bit<24> bd, bit<48> dmac, bit<9> intf) {
         meta.fwd_metadata.out_bd = bd;
         hdr.ethernet.dstAddr = dmac;
         standard_metadata.egress_spec = intf;
         hdr.ipv4.ttl = hdr.ipv4.ttl + 8w255;
     }
-    @name("ipv4_da_lpm") table ipv4_da_lpm_0 {
+    @name("ingress.ipv4_da_lpm") table ipv4_da_lpm_0 {
         actions = {
             set_l2ptr();
             drop_with_count();
@@ -73,7 +73,7 @@ control ingress(inout headers hdr, inout
         default_action = drop_with_count();
         counters = ipv4_da_lpm_stats_0;
     }
-    @name("mac_da") table mac_da_0 {
+    @name("ingress.mac_da") table mac_da_0 {
         actions = {
             set_bd_dmac_intf();
             my_drop(standard_metadata);
@@ -92,10 +92,10 @@ control egress(inout headers hdr, inout
     @name(".my_drop") action my_drop_0(inout standard_metadata_t smeta_1) {
         mark_to_drop(smeta_1);
     }
-    @name("rewrite_mac") action rewrite_mac(bit<48> smac) {
+    @name("egress.rewrite_mac") action rewrite_mac(bit<48> smac) {
         hdr.ethernet.srcAddr = smac;
     }
-    @name("send_frame") table send_frame_0 {
+    @name("egress.send_frame") table send_frame_0 {
         actions = {
             rewrite_mac();
             my_drop_0(standard_metadata);
