--- before_pass
+++ after_pass
@@ -30,9 +30,9 @@ control MyIC(inout ethernet_t a, inout u
     }
     table tbl {
         key = {
-            a.srcAddr: exact;
-            b.data1  : selector;
-            b.data2  : selector;
+            a.srcAddr: exact @name("a.srcAddr") ;
+            b.data1  : selector @name("b.data1") ;
+            b.data2  : selector @name("b.data2") ;
         }
         actions = {
             NoAction();
