--- before_pass
+++ after_pass
@@ -19,7 +19,7 @@ control ingressImpl(inout headers_t hdr,
     @name("foo") foo() foo_inst_0;
     apply {
         foo_inst.apply(hdr.h1.f1, hdr.h1.f1);
-        foo_inst_0.apply(x = hdr.h1.f2, y = 5);
+        foo_inst_0.apply(x = hdr.h1.f2, y = 8w5);
     }
 }
 parser parserImpl(packet_in packet, out headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
@@ -43,4 +43,4 @@ control deparserImpl(packet_out packet,
     apply {
     }
 }
-V1Switch(parserImpl(), verifyChecksum(), ingressImpl(), egressImpl(), updateChecksum(), deparserImpl()) main;
+V1Switch<headers_t, metadata_t>(parserImpl(), verifyChecksum(), ingressImpl(), egressImpl(), updateChecksum(), deparserImpl()) main;
