--- before_pass
+++ after_pass
@@ -10,12 +10,12 @@ struct headers_t {
 struct metadata_t {
 }
 control ingressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
-    bit<8> tmp_1;
-    bit<8> tmp_2;
+    bit<8> tmp;
+    bit<8> tmp_0;
     apply {
-        tmp_2 = hdr.h1.f1;
-        tmp_1 = tmp_2 >> 2;
-        hdr.h1.f1 = tmp_1;
+        tmp_0 = hdr.h1.f1;
+        tmp = tmp_0 >> 2;
+        hdr.h1.f1 = tmp;
         hdr.h1.f2 = 8w5 >> 2;
     }
 }
