--- before_pass
+++ after_pass
@@ -15,9 +15,11 @@ control foo(out bit<8> x, in bit<8> y=5)
     }
 }
 control ingressImpl(inout headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
+    @name("foo") foo() foo_inst;
+    @name("foo") foo() foo_inst_0;
     apply {
-        foo.apply(hdr.h1.f1, hdr.h1.f1);
-        foo.apply(hdr.h1.f2);
+        foo_inst.apply(hdr.h1.f1, hdr.h1.f1);
+        foo_inst_0.apply(hdr.h1.f2);
     }
 }
 parser parserImpl(packet_in packet, out headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
