--- before_pass
+++ after_pass
@@ -4,12 +4,12 @@ header hdr {
     bit<32> a;
 }
 control compute(inout hdr h) {
-    bit<8> n = 0;
+    bit<8> n = (bit<8>)8w0;
     apply {
         if (!h.isValid()) {
             return;
         }
-        if (n > 0) {
+        if (n > 8w0) {
             h.setValid();
         }
     }
@@ -46,7 +46,7 @@ control ingress(inout Headers h, inout M
     compute() c;
     apply {
         c.apply(h.h);
-        sm.egress_spec = 0;
+        sm.egress_spec = (bit<9>)9w0;
     }
 }
 V1Switch(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
