--- before_pass
+++ after_pass
@@ -6,9 +6,7 @@ header hdr {
 control compute(inout hdr h) {
     bit<8> n_0;
     apply {
-        {
-            n_0 = 8w0;
-        }
+        n_0 = 8w0;
         if (!h.isValid()) {
             return;
         }
@@ -24,9 +22,7 @@ struct Meta {
 }
 parser p(packet_in b, out Headers h, inout Meta m, inout standard_metadata_t sm) {
     state start {
-        {
-            b.extract<hdr>(h.h);
-        }
+        b.extract<hdr>(h.h);
         transition accept;
     }
 }
@@ -44,20 +40,14 @@ control egress(inout Headers h, inout Me
 }
 control deparser(packet_out b, in Headers h) {
     apply {
-        {
-            b.emit<hdr>(h.h);
-        }
+        b.emit<hdr>(h.h);
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
     @name("c") compute() c_0;
     apply {
-        {
-            c_0.apply(h.h);
-        }
-        {
-            sm.egress_spec = 9w0;
-        }
+        c_0.apply(h.h);
+        sm.egress_spec = 9w0;
     }
 }
 V1Switch<Headers, Meta>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
