--- before_pass
+++ after_pass
@@ -15,8 +15,12 @@ struct tuple_0 {
 extern bit<16> get<T>(in T data);
 control cc() {
     headers hdr_0;
+    headers tmp;
     apply {
-        get<headers>(headers {ipv4_option_timestamp = hdr_0.ipv4_option_timestamp});
+        {
+            tmp = headers {ipv4_option_timestamp = hdr_0.ipv4_option_timestamp};
+            get<headers>(tmp);
+        }
     }
 }
 control C();
