--- before_pass
+++ after_pass
@@ -13,22 +13,22 @@ struct Headers {
     U u;
 }
 control c(in Headers h) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name("a") action a_0() {
+    @name("a") action a() {
     }
-    @name("t") table t {
+    @name("t") table t_0 {
         key = {
             h.u.h1.x: exact @name("h.u.h1.x") ;
         }
         actions = {
-            a_0();
-            @defaultonly NoAction_1();
+            a();
+            @defaultonly NoAction_0();
         }
-        default_action = NoAction_1();
+        default_action = NoAction_0();
     }
     apply {
-        t.apply();
+        t_0.apply();
     }
 }
 control _c(in Headers h);
