--- before_pass
+++ after_pass
@@ -43,10 +43,10 @@ control MyIngress(inout headers hdr, ino
             hdr.ipv4.dstAddr: exact;
         }
         actions = {
-            actTbl;
-            drop;
+            actTbl();
+            drop();
         }
-        const default_action = drop;
+        const default_action = drop();
         const entries = {
                         8w0x20 ++ 8w0x2 ++ 8w0x4 ++ 8w0x20 : actTbl(24w42, 8w0x20 ++ 8w0x2 ++ 8w0x42 ++ 8w0x0);
         }
