--- before_pass
+++ after_pass
@@ -33,22 +33,22 @@ parser MyParser(packet_in packet, out he
     }
 }
 control MyIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    @name("drop") action drop() {
+    @name("drop") action drop_0() {
         mark_to_drop(standard_metadata);
     }
-    @name("actTbl") action actTbl(bit<24> id, bit<32> ip) {
+    @name("actTbl") action actTbl_0(bit<24> id, bit<32> ip) {
     }
     @name("ingress_tbl") table ingress_tbl {
         key = {
             hdr.ipv4.dstAddr: exact @name("hdr.ipv4.dstAddr") ;
         }
         actions = {
-            actTbl();
-            drop();
+            actTbl_0();
+            drop_0();
         }
-        const default_action = drop();
+        const default_action = drop_0();
         const entries = {
-                        32w0x20020420 : actTbl(24w42, 32w0x20024200);
+                        32w0x20020420 : actTbl_0(24w42, 32w0x20024200);
         }
     }
     apply {
