--- before_pass
+++ after_pass
@@ -42,8 +42,10 @@ parser parse(packet_in pk, out parsed_pa
     }
 }
 control ingress(inout parsed_packet_t h, inout local_metadata_t local_metadata, inout standard_metadata_t standard_metadata) {
+    @name(".NoAction") action NoAction_1() {
+    }
     bitvec_hdr bh;
-    @name("do_act") action do_act() {
+    @name("do_act") action do_act_0() {
         h.bvh1.row.alt1.valid = 1w0;
         local_metadata.row0.alt0.valid = 1w0;
     }
@@ -53,10 +55,10 @@ control ingress(inout parsed_packet_t h,
             local_metadata.row0.alt0.valid: exact @name("local_metadata.row0.alt0.valid") ;
         }
         actions = {
-            do_act();
-            @defaultonly NoAction();
+            do_act_0();
+            @defaultonly NoAction_1();
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     apply {
         tns.apply();
