--- before_pass
+++ after_pass
@@ -19,12 +19,15 @@ control c(out bool b) {
     bool b_2;
     bit<16> x_4;
     bool b_3;
-    @name("a") action a(in bit<16> bi_0, out bit<16> mb_0) {
+    @name("a") action a_0(in bit<16> bi_0, out bit<16> mb_0) {
+        mb_0 = -bi_0;
+    }
+    @name("a") action a_1(in bit<16> bi_0, out bit<16> mb_0) {
         mb_0 = -bi_0;
     }
     apply {
-        a(bi_0 = 16w3, mb_0 = xv);
-        a(mb_0 = xv, bi_0 = 16w0);
+        a_0(bi_0 = 16w3, mb_0 = xv);
+        a_1(mb_0 = xv, bi_0 = 16w0);
         x_3 = xv;
         b_2 = x_3 == 16w0;
         b = b_2;
