--- before_pass
+++ after_pass
@@ -1,28 +1,33 @@
 #include <core.p4>
-parser adder(in bit<32> y, out bit<32> x)(bit<32> add, bool ignore) {
+parser adder_0(in bit<32> y, out bit<32> x) {
     state start {
-        x = y + add;
+        x = y + 32w6;
         transition accept;
     }
 }
 parser par(out bool b) {
     bit<32> x_0;
-    @name("p") adder(ignore = false, add = 32w6) p_0;
+    @name("p") adder_0() p_0;
     state start {
         p_0.apply(x = x_0, y = 32w0);
         b = x_0 == 32w0;
         transition accept;
     }
 }
-control comp(inout bit<16> x, out bool b)(bit<16> compare, bit<2> ignore) {
+control comp_0(inout bit<16> x, out bool b) {
     apply {
-        b = x == compare;
+        b = x == 16w0;
+    }
+}
+control comp_1(inout bit<16> x, out bool b) {
+    apply {
+        b = x == 16w1;
     }
 }
 control c(out bool b) {
     bit<16> xv_0;
-    @name("c0") comp(ignore = 2w1, compare = 16w0) c0_0;
-    @name("c1") comp(ignore = 2w2, compare = 16w1) c1_0;
+    @name("c0") comp_0() c0_0;
+    @name("c1") comp_1() c1_0;
     @name("a") action a_0(in bit<16> bi_0, out bit<16> mb_0) {
         mb_0 = -bi_0;
     }
