--- before_pass
+++ after_pass
@@ -26,12 +26,12 @@ control c(out bool b) {
         mb = -bi;
     }
     apply {
-        bit<16> xv = (bit<16>)16w0;
+        bit<16> xv = 16w0;
         a(bi = 16w3, mb = xv);
         a(mb = xv, bi = 16w0);
         c0.apply(b = b, x = xv);
         c1.apply(xv, b);
-        xv = (bit<16>)16w1;
+        xv = 16w1;
         c0.apply(x = xv, b = b);
         c1.apply(b = b, x = xv);
     }
