--- before_pass
+++ after_pass
@@ -6,9 +6,9 @@ parser adder(in bit<32> y, out bit<32> x
     }
 }
 parser par(out bool b) {
+    bit<32> x_0;
     @name("p") adder(ignore = false, add = 32w6) p_0;
     state start {
-        bit<32> x_0;
         p_0.apply(x = x_0, y = 32w0);
         b = x_0 == 32w0;
         transition accept;
@@ -20,13 +20,14 @@ control comp(inout bit<16> x, out bool b
     }
 }
 control c(out bool b) {
+    bit<16> xv_0;
     @name("c0") comp(ignore = 2w1, compare = 16w0) c0_0;
     @name("c1") comp(ignore = 2w2, compare = 16w1) c1_0;
     @name("a") action a_0(in bit<16> bi, out bit<16> mb) {
         mb = -bi;
     }
     apply {
-        bit<16> xv_0 = 16w0;
+        xv_0 = 16w0;
         a_0(bi = 16w3, mb = xv_0);
         a_0(mb = xv_0, bi = 16w0);
         c0_0.apply(b = b, x = xv_0);
