--- before_pass
+++ after_pass
@@ -27,17 +27,21 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
+    @name(".NoAction") action NoAction_1() {
+    }
+    @name(".NoAction") action NoAction_2() {
+    }
     @name("t1") table t1 {
         actions = {
-            NoAction();
+            NoAction_1();
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     @name("t2") table t2 {
         actions = {
-            NoAction();
+            NoAction_2();
         }
-        default_action = NoAction();
+        default_action = NoAction_2();
     }
     apply {
         if (m.b == 1w0) {
