--- before_pass
+++ after_pass
@@ -31,13 +31,13 @@ control ingress(inout Headers h, inout M
     }
     @name(".NoAction") action NoAction_3() {
     }
-    @name("t1") table t1_0 {
+    @name("ingress.t1") table t1_0 {
         actions = {
             NoAction_0();
         }
         default_action = NoAction_0();
     }
-    @name("t2") table t2_0 {
+    @name("ingress.t2") table t2_0 {
         actions = {
             NoAction_3();
         }
