--- before_pass
+++ after_pass
@@ -64,26 +64,26 @@ control ingress(inout headers hdr, inout
         user_meta.fwd_meta.x2 = x2;
         user_meta.fwd_meta.x3 = (bit<32>)~hdr.ethernet.etherType;
         user_meta.fwd_meta.x4 = ~(bit<32>)hdr.ethernet.etherType;
-        user_meta.fwd_meta.exp_etherType = 0x800;
+        user_meta.fwd_meta.exp_etherType = (bit<16>)16w0x800;
         user_meta.fwd_meta.exp_x1 = 32w0xf7ff;
         user_meta.fwd_meta.exp_x2 = 16w0xf7ff;
         user_meta.fwd_meta.exp_x3 = 32w0xf7ff;
         user_meta.fwd_meta.exp_x4 = 32w0xfffff7ff;
-        hdr.ethernet.dstAddr = 0;
+        hdr.ethernet.dstAddr = (bit<48>)48w0;
         if (hdr.ethernet.etherType != user_meta.fwd_meta.exp_etherType) {
-            hdr.ethernet.dstAddr[47:40] = 1;
+            hdr.ethernet.dstAddr[47:40] = (bit<8>)8w1;
         }
         if (user_meta.fwd_meta.x1 != user_meta.fwd_meta.exp_x1) {
-            hdr.ethernet.dstAddr[39:32] = 1;
+            hdr.ethernet.dstAddr[39:32] = (bit<8>)8w1;
         }
         if (user_meta.fwd_meta.x2 != user_meta.fwd_meta.exp_x2) {
-            hdr.ethernet.dstAddr[31:24] = 1;
+            hdr.ethernet.dstAddr[31:24] = (bit<8>)8w1;
         }
         if (user_meta.fwd_meta.x3 != user_meta.fwd_meta.exp_x3) {
-            hdr.ethernet.dstAddr[23:16] = 1;
+            hdr.ethernet.dstAddr[23:16] = (bit<8>)8w1;
         }
         if (user_meta.fwd_meta.x4 != user_meta.fwd_meta.exp_x4) {
-            hdr.ethernet.dstAddr[15:8] = 1;
+            hdr.ethernet.dstAddr[15:8] = (bit<8>)8w1;
         }
         debug_table_cksum1.apply();
     }
