--- before_pass
+++ after_pass
@@ -26,7 +26,7 @@ struct headers {
 }
 parser IngressParserImpl(packet_in buffer, out headers hdr, inout metadata user_meta, inout standard_metadata_t standard_metadata) {
     state start {
-        buffer.extract(hdr.ethernet);
+        buffer.extract<ethernet_t>(hdr.ethernet);
         transition accept;
     }
 }
@@ -94,7 +94,7 @@ control egress(inout headers hdr, inout
 }
 control DeparserImpl(packet_out packet, in headers hdr) {
     apply {
-        packet.emit(hdr.ethernet);
+        packet.emit<ethernet_t>(hdr.ethernet);
     }
 }
 control verifyChecksum(inout headers hdr, inout metadata meta) {
@@ -105,4 +105,4 @@ control computeChecksum(inout headers hd
     apply {
     }
 }
-V1Switch(IngressParserImpl(), verifyChecksum(), ingress(), egress(), computeChecksum(), DeparserImpl()) main;
+V1Switch<headers, metadata>(IngressParserImpl(), verifyChecksum(), ingress(), egress(), computeChecksum(), DeparserImpl()) main;
