--- before_pass
+++ after_pass
@@ -31,6 +31,8 @@ parser IngressParserImpl(packet_in buffe
     }
 }
 control ingress(inout headers hdr, inout metadata user_meta, inout standard_metadata_t standard_metadata) {
+    @name(".NoAction") action NoAction_1() {
+    }
     bit<16> tmp_1;
     bit<32> x1_1;
     bit<16> x2_1;
@@ -51,9 +53,9 @@ control ingress(inout headers hdr, inout
             user_meta.fwd_meta.x4           : exact @name("user_meta.fwd_meta.x4") ;
         }
         actions = {
-            NoAction();
+            NoAction_1();
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     apply {
         tmp_1 = ~hdr.ethernet.etherType;
