--- before_pass
+++ after_pass
@@ -31,7 +31,7 @@ parser IngressParserImpl(packet_in buffe
     }
 }
 control ingress(inout headers hdr, inout metadata user_meta, inout standard_metadata_t standard_metadata) {
-    table debug_table_cksum1 {
+    @name("debug_table_cksum1") table debug_table_cksum1_0 {
         key = {
             hdr.ethernet.srcAddr            : exact @name("hdr.ethernet.srcAddr") ;
             hdr.ethernet.dstAddr            : exact @name("hdr.ethernet.dstAddr") ;
@@ -52,16 +52,16 @@ control ingress(inout headers hdr, inout
         }
         default_action = NoAction();
     }
-    bit<16> tmp;
-    bit<32> x1;
-    bit<16> x2;
+    bit<16> tmp_0;
+    bit<32> x1_0;
+    bit<16> x2_0;
     apply {
-        tmp = ~hdr.ethernet.etherType;
-        x1 = (bit<32>)tmp;
-        x2 = x1[31:16] + x1[15:0];
-        user_meta.fwd_meta.tmp = tmp;
-        user_meta.fwd_meta.x1 = x1;
-        user_meta.fwd_meta.x2 = x2;
+        tmp_0 = ~hdr.ethernet.etherType;
+        x1_0 = (bit<32>)tmp_0;
+        x2_0 = x1_0[31:16] + x1_0[15:0];
+        user_meta.fwd_meta.tmp = tmp_0;
+        user_meta.fwd_meta.x1 = x1_0;
+        user_meta.fwd_meta.x2 = x2_0;
         user_meta.fwd_meta.x3 = (bit<32>)~hdr.ethernet.etherType;
         user_meta.fwd_meta.x4 = ~(bit<32>)hdr.ethernet.etherType;
         user_meta.fwd_meta.exp_etherType = 16w0x800;
@@ -85,7 +85,7 @@ control ingress(inout headers hdr, inout
         if (user_meta.fwd_meta.x4 != user_meta.fwd_meta.exp_x4) {
             hdr.ethernet.dstAddr[15:8] = 8w1;
         }
-        debug_table_cksum1.apply();
+        debug_table_cksum1_0.apply();
     }
 }
 control egress(inout headers hdr, inout metadata user_meta, inout standard_metadata_t standard_metadata) {
