--- before_pass
+++ after_pass
@@ -41,12 +41,12 @@ parser prs(packet_in p, out Headers_t he
 control pipe(inout Headers_t headers, out bool pass) {
     @name(".NoAction") action NoAction_0() {
     }
-    @name("counters") CounterArray(32w10, true) counters_0;
-    @name("invalidate") action invalidate() {
+    @name("pipe.counters") CounterArray(32w10, true) counters_0;
+    @name("pipe.invalidate") action invalidate() {
         headers.ipv4.setInvalid();
         headers.ethernet.setInvalid();
     }
-    @name("t") table t_0 {
+    @name("pipe.t") table t_0 {
         actions = {
             invalidate();
             @defaultonly NoAction_0();
