--- before_pass
+++ after_pass
@@ -20,22 +20,22 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout EMPTY b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
-    Meter<bit<12>>(32w1024, PSA_MeterType_t.PACKETS) meter0;
-    action execute(bit<12> index, PSA_MeterColor_t color) {
-        meter0.execute(index, color);
+    @name("meter0") Meter<bit<12>>(32w1024, PSA_MeterType_t.PACKETS) meter0_0;
+    @name("execute") action execute_0(bit<12> index, PSA_MeterColor_t color) {
+        meter0_0.execute(index, color);
     }
-    table tbl {
+    @name("tbl") table tbl_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
             NoAction();
-            execute();
+            execute_0();
         }
         default_action = NoAction();
     }
     apply {
-        tbl.apply();
+        tbl_0.apply();
     }
 }
 control MyEC(inout EMPTY a, inout EMPTY b, in psa_egress_input_metadata_t c, inout psa_egress_output_metadata_t d) {
