--- before_pass
+++ after_pass
@@ -23,10 +23,10 @@ parser ParserImpl(packet_in packet, out
     state start {
         packet.extract(hdr.base);
         transition select(hdr.base.t) {
-            0: parseO1O1;
-            1: parseO1O2;
-            2: parseO2O1;
-            3: parseO2O2;
+            8w0: parseO1O1;
+            8w1: parseO1O2;
+            8w2: parseO2O1;
+            8w3: parseO2O2;
             default: accept;
         }
     }
@@ -68,16 +68,16 @@ control ingress(inout headers hdr, inout
     apply {
         debug_hdr.apply();
         if (hdr.u[0].short.isValid()) {
-            hdr.u[0].short.data = 0xffff;
+            hdr.u[0].short.data = (bit<16>)16w0xffff;
         }
         if (hdr.u[0].byte.isValid()) {
-            hdr.u[0].byte.data = 0xaa;
+            hdr.u[0].byte.data = (bit<8>)8w0xaa;
         }
         if (hdr.u[1].short.isValid()) {
-            hdr.u[1].short.data = 0xffff;
+            hdr.u[1].short.data = (bit<16>)16w0xffff;
         }
         if (hdr.u[1].byte.isValid()) {
-            hdr.u[1].byte.data = 0xaa;
+            hdr.u[1].byte.data = (bit<8>)8w0xaa;
         }
     }
 }
