--- before_pass
+++ after_pass
@@ -52,6 +52,8 @@ parser ParserImpl(packet_in packet, out
     }
 }
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
+    @name(".NoAction") action NoAction_1() {
+    }
     @name("debug_hdr") table debug_hdr {
         key = {
             hdr.base.t              : exact @name("hdr.base.t") ;
@@ -61,9 +63,9 @@ control ingress(inout headers hdr, inout
             hdr.u[1].short.isValid(): exact @name("hdr.u[1].short.$valid$") ;
         }
         actions = {
-            NoAction();
+            NoAction_1();
         }
-        const default_action = NoAction();
+        const default_action = NoAction_1();
     }
     apply {
         debug_hdr.apply();
