--- before_pass
+++ after_pass
@@ -13,11 +13,11 @@ parser P(packet_in b, out Headers p, ino
     }
 }
 control Ing(inout Headers headers, inout Metadata meta, inout standard_metadata_t standard_meta) {
-    S s_0;
-    @name("r") register<S>(32w100) r_0;
+    S s;
+    @name("r") register<S>(32w100) r;
     apply {
-        s_0 = { 32w0 };
-        r_0.write(32w0, s_0);
+        s = { 32w0 };
+        r.write(32w0, s);
     }
 }
 control Eg(inout Headers hdrs, inout Metadata meta, inout standard_metadata_t standard_meta) {
