--- before_pass
+++ after_pass
@@ -16,8 +16,12 @@ control Ing(inout Headers headers, inout
     S s_0;
     @name("r") register<S>(32w100) r_0;
     apply {
-        s_0 = { 32w0 };
-        r_0.write(32w0, s_0);
+        {
+            s_0 = { 32w0 };
+        }
+        {
+            r_0.write(32w0, s_0);
+        }
     }
 }
 control Eg(inout Headers hdrs, inout Metadata meta, inout standard_metadata_t standard_meta) {
