--- before_pass
+++ after_pass
@@ -31,21 +31,22 @@ control ingress(inout Headers h, inout M
     table t1 {
         size = 3;
         actions = {
-            NoAction;
+            NoAction();
         }
-        const default_action = NoAction;
+        const default_action = NoAction();
     }
     table t2 {
         key = {
             sm.ingress_port: exact;
         }
         actions = {
-            NoAction;
+            NoAction();
         }
         const entries = {
                         0 : NoAction();
         }
         size = 10;
+        default_action = NoAction();
     }
     apply {
         t1.apply();
