--- before_pass
+++ after_pass
@@ -28,33 +28,33 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name(".NoAction") action NoAction_2() {
+    @name(".NoAction") action NoAction_3() {
     }
-    @name("t1") table t1 {
+    @name("t1") table t1_0 {
         size = 3;
         actions = {
-            NoAction_1();
+            NoAction_0();
         }
-        const default_action = NoAction_1();
+        const default_action = NoAction_0();
     }
-    @name("t2") table t2 {
+    @name("t2") table t2_0 {
         key = {
             sm.ingress_port: exact @name("sm.ingress_port") ;
         }
         actions = {
-            NoAction_2();
+            NoAction_3();
         }
         const entries = {
-                        9w0 : NoAction_2();
+                        9w0 : NoAction_3();
         }
         size = 10;
-        default_action = NoAction_2();
+        default_action = NoAction_3();
     }
     apply {
-        t1.apply();
-        t2.apply();
+        t1_0.apply();
+        t2_0.apply();
     }
 }
 V1Switch<Headers, Meta>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
