--- before_pass
+++ after_pass
@@ -28,14 +28,14 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name("t1") table t1_0 {
+    @name("t1") table t1 {
         size = 3;
         actions = {
             NoAction();
         }
         const default_action = NoAction();
     }
-    @name("t2") table t2_0 {
+    @name("t2") table t2 {
         key = {
             sm.ingress_port: exact @name("sm.ingress_port") ;
         }
@@ -49,8 +49,8 @@ control ingress(inout Headers h, inout M
         default_action = NoAction();
     }
     apply {
-        t1_0.apply();
-        t2_0.apply();
+        t1.apply();
+        t2.apply();
     }
 }
 V1Switch<Headers, Meta>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
