--- before_pass
+++ after_pass
@@ -28,25 +28,29 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
+    @name(".NoAction") action NoAction_1() {
+    }
+    @name(".NoAction") action NoAction_2() {
+    }
     @name("t1") table t1 {
         size = 3;
         actions = {
-            NoAction();
+            NoAction_1();
         }
-        const default_action = NoAction();
+        const default_action = NoAction_1();
     }
     @name("t2") table t2 {
         key = {
             sm.ingress_port: exact @name("sm.ingress_port") ;
         }
         actions = {
-            NoAction();
+            NoAction_2();
         }
         const entries = {
-                        9w0 : NoAction();
+                        9w0 : NoAction_2();
         }
         size = 10;
-        default_action = NoAction();
+        default_action = NoAction_2();
     }
     apply {
         t1.apply();
