--- before_pass
+++ after_pass
@@ -10,12 +10,8 @@ struct headers_t {
 struct metadata_t {
 }
 parser parserImpl(packet_in packet, out headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
-    bit<8> tmp;
-    bit<8> tmp_0;
     state start {
-        tmp_0 = hdr.h1.f1;
-        tmp = tmp_0 >> 2;
-        hdr.h1.f1 = tmp;
+        hdr.h1.f1 = hdr.h1.f1 >> 2;
         hdr.h1.f2 = 8w1;
         transition accept;
     }
