--- before_pass
+++ after_pass
@@ -14,21 +14,9 @@ parser parserImpl(packet_in packet, out
     bit<8> tmp_0;
     state start {
         tmp_0 = hdr.h1.f1;
-        transition foo_start;
-    }
-    state foo_start {
         tmp = tmp_0 >> 2;
-        transition start_0;
-    }
-    state start_0 {
         hdr.h1.f1 = tmp;
-        transition foo_start_0;
-    }
-    state foo_start_0 {
         hdr.h1.f2 = 8w1;
-        transition start_1;
-    }
-    state start_1 {
         transition accept;
     }
 }
