--- before_pass
+++ after_pass
@@ -11,9 +11,7 @@ struct metadata_t {
 }
 parser foo(out bit<8> x, in bit<8> y=5) {
     state start {
-        {
-            x = y >> 2;
-        }
+        x = y >> 2;
         transition accept;
     }
 }
@@ -23,14 +21,10 @@ parser parserImpl(packet_in packet, out
     bit<8> tmp;
     bit<8> tmp_0;
     state start {
-        {
-            tmp_0 = hdr.h1.f1;
-            foo_inst_1.apply(tmp, tmp_0);
-            hdr.h1.f1 = tmp;
-        }
-        {
-            foo_inst_2.apply(x = hdr.h1.f2, y = 8w5);
-        }
+        tmp_0 = hdr.h1.f1;
+        foo_inst_1.apply(tmp, tmp_0);
+        hdr.h1.f1 = tmp;
+        foo_inst_2.apply(x = hdr.h1.f2, y = 8w5);
         transition accept;
     }
 }
