--- before_pass
+++ after_pass
@@ -9,22 +9,26 @@ struct headers_t {
 }
 struct metadata_t {
 }
-parser foo(out bit<8> x, in bit<8> y=5) {
-    state start {
-        x = y >> 2;
-        transition accept;
-    }
-}
 parser parserImpl(packet_in packet, out headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
     bit<8> tmp;
     bit<8> tmp_0;
-    @name("foo") foo() foo_inst_1;
-    @name("foo") foo() foo_inst_2;
     state start {
         tmp_0 = hdr.h1.f1;
-        foo_inst_1.apply(tmp, tmp_0);
+        transition foo_start;
+    }
+    state foo_start {
+        tmp = tmp_0 >> 2;
+        transition start_0;
+    }
+    state start_0 {
         hdr.h1.f1 = tmp;
-        foo_inst_2.apply(x = hdr.h1.f2, y = 8w5);
+        transition foo_start_0;
+    }
+    state foo_start_0 {
+        hdr.h1.f2 = 8w5 >> 2;
+        transition start_1;
+    }
+    state start_1 {
         transition accept;
     }
 }
