--- before_pass
+++ after_pass
@@ -10,18 +10,18 @@ struct headers_t {
 struct metadata_t {
 }
 parser parserImpl(packet_in packet, out headers_t hdr, inout metadata_t meta, inout standard_metadata_t stdmeta) {
-    bit<8> tmp;
-    bit<8> tmp_0;
+    bit<8> tmp_1;
+    bit<8> tmp_2;
     state start {
-        tmp_0 = hdr.h1.f1;
+        tmp_2 = hdr.h1.f1;
         transition foo_start;
     }
     state foo_start {
-        tmp = tmp_0 >> 2;
+        tmp_1 = tmp_2 >> 2;
         transition start_0;
     }
     state start_0 {
-        hdr.h1.f1 = tmp;
+        hdr.h1.f1 = tmp_1;
         transition foo_start_0;
     }
     state foo_start_0 {
