--- before_pass
+++ after_pass
@@ -20,7 +20,7 @@ parser parserImpl(packet_in packet, out
     @name("foo") foo() foo_inst_0;
     state start {
         foo_inst.apply(hdr.h1.f1, hdr.h1.f1);
-        foo_inst_0.apply(x = hdr.h1.f2, y = 5);
+        foo_inst_0.apply(x = hdr.h1.f2, y = 8w5);
         transition accept;
     }
 }
@@ -44,4 +44,4 @@ control deparserImpl(packet_out packet,
     apply {
     }
 }
-V1Switch(parserImpl(), verifyChecksum(), ingressImpl(), egressImpl(), updateChecksum(), deparserImpl()) main;
+V1Switch<headers_t, metadata_t>(parserImpl(), verifyChecksum(), ingressImpl(), egressImpl(), updateChecksum(), deparserImpl()) main;
