--- before_pass
+++ after_pass
@@ -33,7 +33,9 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name("c.a") action c_a_0() {
+    @name(".NoAction") action NoAction_1() {
+    }
+    @name("c.a") action c_a() {
         h.h.b = h.h.a;
     }
     @name("c.t") table c_t_0 {
@@ -41,10 +43,10 @@ control ingress(inout Headers h, inout M
             h.h.a + h.h.a: exact @name("e") ;
         }
         actions = {
-            c_a_0();
-            NoAction();
+            c_a();
+            NoAction_1();
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     apply {
         c_t_0.apply();
