--- before_pass
+++ after_pass
@@ -22,8 +22,8 @@ parser parserI(packet_in pkt, out Parsed
 }
 control cIngress(inout Parsed_packet hdr, inout Metadata meta, inout standard_metadata_t stdmeta) {
     apply {
-        digest(5, { hdr.ethernet.srcAddr });
-        hdr.ethernet.srcAddr = 0;
+        digest(32w5, { hdr.ethernet.srcAddr });
+        hdr.ethernet.srcAddr = (bit<48>)48w0;
     }
 }
 control cEgress(inout Parsed_packet hdr, inout Metadata meta, inout standard_metadata_t stdmeta) {
