--- before_pass
+++ after_pass
@@ -47,36 +47,36 @@ control verifyChecksum(inout headers hdr
     }
 }
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    @name("drop") action drop_0() {
+    @name("drop") action drop() {
         mark_to_drop(standard_metadata);
     }
-    @name("drop") action drop_1() {
+    @name("drop") action drop_2() {
         mark_to_drop(standard_metadata);
     }
-    @name("ipv4_forward") action ipv4_forward_0(macAddr_t dstAddr, egressSpec_t port) {
+    @name("ipv4_forward") action ipv4_forward(macAddr_t dstAddr, egressSpec_t port) {
         meta.test_bool = true;
     }
-    @name("ipv4_lpm") table ipv4_lpm {
+    @name("ipv4_lpm") table ipv4_lpm_0 {
         key = {
             hdr.ipv4.dstAddr: lpm @name("hdr.ipv4.dstAddr") ;
         }
         actions = {
-            NoAction_1();
-            ipv4_forward_0();
-            drop_0();
+            NoAction_0();
+            ipv4_forward();
+            drop();
         }
         size = 1024;
-        default_action = NoAction_1();
+        default_action = NoAction_0();
     }
     apply {
         meta.test_bool = false;
         if (hdr.ipv4.isValid()) {
-            ipv4_lpm.apply();
+            ipv4_lpm_0.apply();
         }
         if (!meta.test_bool) {
-            drop_1();
+            drop_2();
         }
     }
 }
