--- before_pass
+++ after_pass
@@ -20,9 +20,7 @@ struct value_set_t {
 parser MyParser(packet_in b, out my_packet p, inout my_metadata m, inout standard_metadata_t s) {
     @name("pvs") value_set<value_set_t>(4) pvs_0;
     state start {
-        {
-            b.extract<data_h>(p.data);
-        }
+        b.extract<data_h>(p.data);
         transition select(p.data.da, p.data.da) {
             pvs_0: accept;
             (32w0x810, 32w0x810): foo;
@@ -50,9 +48,7 @@ control MyIngress(inout my_packet p, ino
         default_action = NoAction();
     }
     apply {
-        {
-            t_0.apply();
-        }
+        t_0.apply();
     }
 }
 control MyEgress(inout my_packet p, inout my_metadata m, inout standard_metadata_t s) {
