--- before_pass
+++ after_pass
@@ -20,9 +20,7 @@ struct Headers {
 parser prs(packet_in p, out Headers h) {
     Ethernet e_0;
     state start {
-        {
-            p.extract<Ethernet>(e_0);
-        }
+        p.extract<Ethernet>(e_0);
         transition select(e_0.type) {
             EthTypes.IPv4: accept;
             EthTypes.ARP: accept;
@@ -32,9 +30,7 @@ parser prs(packet_in p, out Headers h) {
 }
 control c(inout Headers h, inout standard_metadata_t sm) {
     @name("do_act") action do_act_0(bit<32> type) {
-        {
-            sm.instance_type = type;
-        }
+        sm.instance_type = type;
     }
     @name("tns") table tns_0 {
         key = {
@@ -51,9 +47,7 @@ control c(inout Headers h, inout standar
         default_action = NoAction();
     }
     apply {
-        {
-            tns_0.apply();
-        }
+        tns_0.apply();
     }
 }
 parser p<H>(packet_in _p, out H h);
