--- before_pass
+++ after_pass
@@ -29,7 +29,9 @@ parser prs(packet_in p, out Headers h) {
     }
 }
 control c(inout Headers h, inout standard_metadata_t sm) {
-    @name("do_act") action do_act(bit<32> type) {
+    @name(".NoAction") action NoAction_1() {
+    }
+    @name("do_act") action do_act_0(bit<32> type) {
         sm.instance_type = type;
     }
     @name("tns") table tns {
@@ -37,14 +39,14 @@ control c(inout Headers h, inout standar
             h.eth.type: exact @name("h.eth.type") ;
         }
         actions = {
-            do_act();
-            @defaultonly NoAction();
+            do_act_0();
+            @defaultonly NoAction_1();
         }
         const entries = {
-                        EthTypes.IPv4 : do_act(32w0x800);
-                        EthTypes.VLAN : do_act(32w0x8100);
+                        EthTypes.IPv4 : do_act_0(32w0x800);
+                        EthTypes.VLAN : do_act_0(32w0x8100);
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     apply {
         tns.apply();
