--- before_pass
+++ after_pass
@@ -18,31 +18,31 @@ parser ParserImpl(packet_in packet, out
     state parse_ethernet {
         packet.extract(hdr.ethernet);
         transition select(hdr.ethernet.srcAddr, hdr.ethernet.dstAddr) {
-            (0x12f0000, 0x456): a1;
-            (0x12f0000 &&& 0xffff0000, 0x456): a2;
-            (0x12f0000, 0x456 &&& 0xfff): a3;
-            (0x12f0000 &&& 0xffff0000, 0x456 &&& 0xfff): a4;
+            (48w0x12f0000, 48w0x456): a1;
+            (48w0x12f0000 &&& 48w0xffff0000, 48w0x456): a2;
+            (48w0x12f0000, 48w0x456 &&& 48w0xfff): a3;
+            (48w0x12f0000 &&& 48w0xffff0000, 48w0x456 &&& 48w0xfff): a4;
             default: a5;
         }
     }
     state a1 {
-        meta.transition_taken = 1;
+        meta.transition_taken = (bit<16>)16w1;
         transition accept;
     }
     state a2 {
-        meta.transition_taken = 2;
+        meta.transition_taken = (bit<16>)16w2;
         transition accept;
     }
     state a3 {
-        meta.transition_taken = 3;
+        meta.transition_taken = (bit<16>)16w3;
         transition accept;
     }
     state a4 {
-        meta.transition_taken = 4;
+        meta.transition_taken = (bit<16>)16w4;
         transition accept;
     }
     state a5 {
-        meta.transition_taken = 5;
+        meta.transition_taken = (bit<16>)16w5;
         transition accept;
     }
 }
