--- before_pass
+++ after_pass
@@ -26,23 +26,23 @@ parser ParserImpl(packet_in packet, out
         }
     }
     state a1 {
-        meta.transition_taken = (bit<16>)16w1;
+        meta.transition_taken = 16w1;
         transition accept;
     }
     state a2 {
-        meta.transition_taken = (bit<16>)16w2;
+        meta.transition_taken = 16w2;
         transition accept;
     }
     state a3 {
-        meta.transition_taken = (bit<16>)16w3;
+        meta.transition_taken = 16w3;
         transition accept;
     }
     state a4 {
-        meta.transition_taken = (bit<16>)16w4;
+        meta.transition_taken = 16w4;
         transition accept;
     }
     state a5 {
-        meta.transition_taken = (bit<16>)16w5;
+        meta.transition_taken = 16w5;
         transition accept;
     }
 }
