--- before_pass
+++ after_pass
@@ -16,7 +16,7 @@ parser ParserImpl(packet_in packet, out
         transition parse_ethernet;
     }
     state parse_ethernet {
-        packet.extract(hdr.ethernet);
+        packet.extract<ethernet_t>(hdr.ethernet);
         transition select(hdr.ethernet.srcAddr, hdr.ethernet.dstAddr) {
             (48w0x12f0000, 48w0x456): a1;
             (48w0x12f0000 &&& 48w0xffff0000, 48w0x456): a2;
@@ -57,7 +57,7 @@ control ingress(inout headers hdr, inout
 }
 control DeparserImpl(packet_out packet, in headers hdr) {
     apply {
-        packet.emit(hdr.ethernet);
+        packet.emit<ethernet_t>(hdr.ethernet);
     }
 }
 control verifyChecksum(inout headers hdr, inout metadata meta) {
@@ -68,4 +68,4 @@ control computeChecksum(inout headers hd
     apply {
     }
 }
-V1Switch(ParserImpl(), verifyChecksum(), ingress(), egress(), computeChecksum(), DeparserImpl()) main;
+V1Switch<headers, metadata>(ParserImpl(), verifyChecksum(), ingress(), egress(), computeChecksum(), DeparserImpl()) main;
