--- before_pass
+++ after_pass
@@ -20,6 +20,8 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout EMPTY b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
+    @name(".NoAction") action NoAction_1() {
+    }
     PSA_MeterColor_t tmp_1;
     bool tmp_2;
     @name("meter0") Meter<bit<12>>(32w1024, PSA_MeterType_t.PACKETS) meter0;
@@ -28,9 +30,9 @@ control MyIC(inout ethernet_t a, inout E
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
-            NoAction();
+            NoAction_1();
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     apply {
         tmp_1 = meter0.execute(12w0);
