--- before_pass
+++ after_pass
@@ -20,8 +20,8 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout EMPTY b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
-    Meter<bit<12>>(32w1024, PSA_MeterType_t.PACKETS) meter0;
-    table tbl {
+    @name("meter0") Meter<bit<12>>(32w1024, PSA_MeterType_t.PACKETS) meter0_0;
+    @name("tbl") table tbl_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
@@ -31,8 +31,8 @@ control MyIC(inout ethernet_t a, inout E
         default_action = NoAction();
     }
     apply {
-        if (meter0.execute(12w0) == PSA_MeterColor_t.GREEN) {
-            tbl.apply();
+        if (meter0_0.execute(12w0) == PSA_MeterColor_t.GREEN) {
+            tbl_0.apply();
         }
     }
 }
