--- before_pass
+++ after_pass
@@ -20,6 +20,8 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout EMPTY b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
+    PSA_MeterColor_t tmp;
+    bool tmp_0;
     @name("meter0") Meter<bit<12>>(32w1024, PSA_MeterType_t.PACKETS) meter0_0;
     @name("tbl") table tbl_0 {
         key = {
@@ -30,8 +32,6 @@ control MyIC(inout ethernet_t a, inout E
         }
         default_action = NoAction();
     }
-    PSA_MeterColor_t tmp;
-    bool tmp_0;
     apply {
         tmp = meter0_0.execute(12w0);
         tmp_0 = tmp == PSA_MeterColor_t.GREEN;
