--- before_pass
+++ after_pass
@@ -20,25 +20,25 @@ parser MyEP(packet_in buffer, out EMPTY
     }
 }
 control MyIC(inout ethernet_t a, inout EMPTY b, in psa_ingress_input_metadata_t c, inout psa_ingress_output_metadata_t d) {
-    @name(".NoAction") action NoAction_1() {
+    @name(".NoAction") action NoAction_0() {
     }
-    PSA_MeterColor_t tmp_1;
-    bool tmp_2;
-    @name("meter0") Meter<bit<12>>(32w1024, PSA_MeterType_t.PACKETS) meter0;
-    @name("tbl") table tbl {
+    PSA_MeterColor_t tmp;
+    bool tmp_0;
+    @name("meter0") Meter<bit<12>>(32w1024, PSA_MeterType_t.PACKETS) meter0_0;
+    @name("tbl") table tbl_0 {
         key = {
             a.srcAddr: exact @name("a.srcAddr") ;
         }
         actions = {
-            NoAction_1();
+            NoAction_0();
         }
-        default_action = NoAction_1();
+        default_action = NoAction_0();
     }
     apply {
-        tmp_1 = meter0.execute(12w0);
-        tmp_2 = tmp_1 == PSA_MeterColor_t.GREEN;
-        if (tmp_2) {
-            tbl.apply();
+        tmp = meter0_0.execute(12w0);
+        tmp_0 = tmp == PSA_MeterColor_t.GREEN;
+        if (tmp_0) {
+            tbl_0.apply();
         }
     }
 }
