--- before_pass
+++ after_pass
@@ -11,9 +11,15 @@ struct Meta {
 }
 parser P(packet_in p, out Headers h, inout Meta m, inout standard_metadata_t sm) {
     state start {
-        p.extract<Hdr>(h.h1.next);
-        p.extract<Hdr>(h.h1.next);
-        h.h2 = h.h1;
+        {
+            p.extract<Hdr>(h.h1.next);
+        }
+        {
+            p.extract<Hdr>(h.h1.next);
+        }
+        {
+            h.h2 = h.h1;
+        }
         transition accept;
     }
 }
@@ -31,12 +37,16 @@ control egress(inout Headers h, inout Me
 }
 control deparser(packet_out b, in Headers h) {
     apply {
-        b.emit<Headers>(h);
+        {
+            b.emit<Headers>(h);
+        }
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
     apply {
-        sm.egress_spec = 9w0;
+        {
+            sm.egress_spec = 9w0;
+        }
     }
 }
 V1Switch<Headers, Meta>(P(), vrfy(), ingress(), egress(), update(), deparser()) main;
