--- before_pass
+++ after_pass
@@ -12,7 +12,7 @@ struct metadata {
 }
 parser parserI(packet_in pkt, out headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
     state start {
-        pkt.extract(hdr.h1);
+        pkt.extract<h1_t>(hdr.h1);
         transition accept;
     }
 }
@@ -45,7 +45,7 @@ control uc(inout headers hdr, inout meta
 }
 control DeparserI(packet_out packet, in headers hdr) {
     apply {
-        packet.emit(hdr.h1);
+        packet.emit<h1_t>(hdr.h1);
     }
 }
-V1Switch(parserI(), vc(), cIngress(), cEgress(), uc(), DeparserI()) main;
+V1Switch<headers, metadata>(parserI(), vc(), cIngress(), cEgress(), uc(), DeparserI()) main;
