--- before_pass
+++ after_pass
@@ -26,10 +26,10 @@ control cDoOneOp(inout headers hdr, in b
     }
 }
 control cIngress(inout headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
-    cDoOneOp() do_one_op;
+    @name("do_one_op") cDoOneOp() do_one_op_0;
     apply {
-        do_one_op.apply(hdr, hdr.h1.op1);
-        do_one_op.apply(hdr, hdr.h1.op2);
+        do_one_op_0.apply(hdr, hdr.h1.op1);
+        do_one_op_0.apply(hdr, hdr.h1.op2);
     }
 }
 control cEgress(inout headers hdr, inout metadata meta, inout standard_metadata_t stdmeta) {
