--- before_pass
+++ after_pass
@@ -43,9 +43,9 @@ parser ParserImpl(packet_in packet, out
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     table debug_hdr {
         key = {
-            hdr.base.t           : exact;
-            hdr.u.short.isValid(): exact;
-            hdr.u.byte.isValid() : exact;
+            hdr.base.t           : exact @name("hdr.base.t") ;
+            hdr.u.short.isValid(): exact @name("hdr.u.short.$valid$") ;
+            hdr.u.byte.isValid() : exact @name("hdr.u.byte.$valid$") ;
         }
         actions = {
             NoAction();
