--- before_pass
+++ after_pass
@@ -27,15 +27,9 @@ control ingress(inout H pkt_hdr, in Meta
     @name("num_pkts_with_rtt") ConditionalAccumulator<bit<32>>(32w1) num_pkts_with_rtt_0;
     apply {
         @atomic {
-            {
-                input_traffic_bytes_0.count();
-            }
-            {
-                sum_rtt_Tr_0.write(pkt_hdr.rtt, pkt_hdr.rtt < 32w2500);
-            }
-            {
-                num_pkts_with_rtt_0.write(32w1, pkt_hdr.rtt < 32w2500);
-            }
+            input_traffic_bytes_0.count();
+            sum_rtt_Tr_0.write(pkt_hdr.rtt, pkt_hdr.rtt < 32w2500);
+            num_pkts_with_rtt_0.write(32w1, pkt_hdr.rtt < 32w2500);
         }
     }
 }
