--- before_pass
+++ after_pass
@@ -24,13 +24,13 @@ struct Metadata {
 const bit<32> MAX_ALLOWABLE_RTT = 32w2500;
 control ingress(inout H pkt_hdr, in Metadata metadata) {
     Counter<bit<32>>(CounterType.packets_and_bytes) input_traffic_bytes;
-    ConditionalAccumulator<bit<32>>(1) sum_rtt_Tr;
-    ConditionalAccumulator<bit<32>>(1) num_pkts_with_rtt;
+    ConditionalAccumulator<bit<32>>(32w1) sum_rtt_Tr;
+    ConditionalAccumulator<bit<32>>(32w1) num_pkts_with_rtt;
     apply {
         @atomic {
             input_traffic_bytes.count();
             sum_rtt_Tr.write(pkt_hdr.rtt, pkt_hdr.rtt < 32w2500);
-            num_pkts_with_rtt.write(1, pkt_hdr.rtt < 32w2500);
+            num_pkts_with_rtt.write(32w1, pkt_hdr.rtt < 32w2500);
         }
     }
 }
