--- before_pass
+++ after_pass
@@ -12,7 +12,9 @@ parser ParserI(packet_in pk, out H hdr,
 }
 control IngressI(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
     @name("do_something") action do_something_1() {
-        mark_to_drop(smeta);
+        {
+            mark_to_drop(smeta);
+        }
     }
     @name("do_something") table do_something_2 {
         key = {
@@ -25,7 +27,9 @@ control IngressI(inout H hdr, inout M me
         const default_action = NoAction();
     }
     apply {
-        do_something_2.apply();
+        {
+            do_something_2.apply();
+        }
     }
 }
 control EgressI(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
