--- before_pass
+++ after_pass
@@ -11,21 +11,21 @@ parser ParserI(packet_in pk, out H hdr,
     }
 }
 control IngressI(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
-    @name("do_something") action do_something_1() {
+    @name("do_something") action do_something() {
         mark_to_drop(smeta);
     }
-    @name("do_something") table do_something_2 {
+    @name("do_something") table do_something_0 {
         key = {
             smeta.ingress_port: exact @name("smeta.ingress_port") ;
         }
         actions = {
-            do_something_1();
+            do_something();
             NoAction();
         }
         const default_action = NoAction();
     }
     apply {
-        do_something_2.apply();
+        do_something_0.apply();
     }
 }
 control EgressI(inout H hdr, inout M meta, inout standard_metadata_t smeta) {
