--- before_pass
+++ after_pass
@@ -56,7 +56,7 @@ control ingress(inout Headers h, inout M
 }
 parser p(packet_in b, out Headers h, inout Meta m, inout standard_metadata_t sm) {
     state start {
-        b.extract(h.h);
+        b.extract<hdr>(h.h);
         transition accept;
     }
 }
@@ -74,7 +74,7 @@ control egress(inout Headers h, inout Me
 }
 control deparser(packet_out b, in Headers h) {
     apply {
-        b.emit(h.h);
+        b.emit<hdr>(h.h);
     }
 }
-V1Switch(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
+V1Switch<Headers, Meta>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
