--- before_pass
+++ after_pass
@@ -13,31 +13,31 @@ struct Meta {
     bit<8> y;
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name("case0") action case0() {
+    @name("ingress.case0") action case0() {
         h.h.c = (bit<8>)((16w0 ++ h.h.a)[15:0] ++ 16w0);
     }
-    @name("case1") action case1() {
+    @name("ingress.case1") action case1() {
         h.h.c = (bit<8>)h.h.a[15:0];
     }
-    @name("case2") action case2() {
+    @name("ingress.case2") action case2() {
         h.h.c = 8w0;
     }
-    @name("case3") action case3() {
+    @name("ingress.case3") action case3() {
         h.h.c = h.h.a[7:0];
     }
-    @name("case4") action case4() {
+    @name("ingress.case4") action case4() {
         h.h.c = (bit<8>)(8w0 ++ h.h.a[15:0]);
     }
-    @name("case5") action case5() {
+    @name("ingress.case5") action case5() {
         h.h.c = (bit<8>)(8w0 ++ h.h.a[15:8]);
     }
-    @name("case6") action case6() {
+    @name("ingress.case6") action case6() {
         h.h.c = (bit<8>)(16w0 ++ h.h.a[15:8]);
     }
-    @name("case7") action case7() {
+    @name("ingress.case7") action case7() {
         h.h.c = (bit<8>)(16w0 ++ h.h.a >> 3)[31:8];
     }
-    @name("t") table t_0 {
+    @name("ingress.t") table t_0 {
         actions = {
             case0();
             case1();
