--- before_pass
+++ after_pass
@@ -13,45 +13,45 @@ struct Meta {
     bit<8> y;
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name("case0") action case0_0() {
+    @name("case0") action case0() {
         h.h.c = (bit<8>)((16w0 ++ h.h.a)[15:0] ++ 16w0);
     }
-    @name("case1") action case1_0() {
+    @name("case1") action case1() {
         h.h.c = (bit<8>)h.h.a[15:0];
     }
-    @name("case2") action case2_0() {
+    @name("case2") action case2() {
         h.h.c = 8w0;
     }
-    @name("case3") action case3_0() {
+    @name("case3") action case3() {
         h.h.c = h.h.a[7:0];
     }
-    @name("case4") action case4_0() {
+    @name("case4") action case4() {
         h.h.c = (bit<8>)(8w0 ++ h.h.a[15:0]);
     }
-    @name("case5") action case5_0() {
+    @name("case5") action case5() {
         h.h.c = (bit<8>)(8w0 ++ h.h.a[15:8]);
     }
-    @name("case6") action case6_0() {
+    @name("case6") action case6() {
         h.h.c = (bit<8>)(16w0 ++ h.h.a[15:8]);
     }
-    @name("case7") action case7_0() {
+    @name("case7") action case7() {
         h.h.c = (bit<8>)(16w0 ++ h.h.a >> 3)[31:8];
     }
-    @name("t") table t_0 {
+    @name("t") table t {
         actions = {
-            case0_0();
-            case1_0();
-            case2_0();
-            case3_0();
-            case4_0();
-            case5_0();
-            case6_0();
-            case7_0();
+            case0();
+            case1();
+            case2();
+            case3();
+            case4();
+            case5();
+            case6();
+            case7();
         }
-        const default_action = case0_0();
+        const default_action = case0();
     }
     apply {
-        t_0.apply();
+        t.apply();
     }
 }
 parser p(packet_in b, out Headers h, inout Meta m, inout standard_metadata_t sm) {
