--- before_pass
+++ after_pass
@@ -54,11 +54,11 @@ parser ParserImpl(packet_in packet, out
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     table debug_hdr {
         key = {
-            hdr.base.t              : exact;
-            hdr.u[0].byte.isValid() : exact;
-            hdr.u[0].short.isValid(): exact;
-            hdr.u[1].byte.isValid() : exact;
-            hdr.u[1].short.isValid(): exact;
+            hdr.base.t              : exact @name("hdr.base.t") ;
+            hdr.u[0].byte.isValid() : exact @name("hdr.u[0].byte.$valid$") ;
+            hdr.u[0].short.isValid(): exact @name("hdr.u[0].short.$valid$") ;
+            hdr.u[1].byte.isValid() : exact @name("hdr.u[1].byte.$valid$") ;
+            hdr.u[1].short.isValid(): exact @name("hdr.u[1].short.$valid$") ;
         }
         actions = {
             NoAction();
