--- before_pass
+++ after_pass
@@ -52,7 +52,7 @@ parser ParserImpl(packet_in packet, out
     }
 }
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    @name("debug_hdr") table debug_hdr_0 {
+    @name("debug_hdr") table debug_hdr {
         key = {
             hdr.base.t              : exact @name("hdr.base.t") ;
             hdr.u[0].byte.isValid() : exact @name("hdr.u[0].byte.$valid$") ;
@@ -66,7 +66,7 @@ control ingress(inout headers hdr, inout
         const default_action = NoAction();
     }
     apply {
-        debug_hdr_0.apply();
+        debug_hdr.apply();
         if (hdr.u[0].short.isValid()) {
             hdr.u[0].short.data = 16w0xffff;
         }
