--- before_pass
+++ after_pass
@@ -21,9 +21,7 @@ struct metadata {
 }
 parser ParserImpl(packet_in packet, out headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
     state start {
-        {
-            packet.extract<S>(hdr.base);
-        }
+        packet.extract<S>(hdr.base);
         transition select(hdr.base.t) {
             8w0: parseO1O1;
             8w1: parseO1O2;
@@ -33,39 +31,23 @@ parser ParserImpl(packet_in packet, out
         }
     }
     state parseO1O1 {
-        {
-            packet.extract<O1>(hdr.u[0].byte);
-        }
-        {
-            packet.extract<O1>(hdr.u[1].byte);
-        }
+        packet.extract<O1>(hdr.u[0].byte);
+        packet.extract<O1>(hdr.u[1].byte);
         transition accept;
     }
     state parseO1O2 {
-        {
-            packet.extract<O1>(hdr.u[0].byte);
-        }
-        {
-            packet.extract<O2>(hdr.u[1].short);
-        }
+        packet.extract<O1>(hdr.u[0].byte);
+        packet.extract<O2>(hdr.u[1].short);
         transition accept;
     }
     state parseO2O1 {
-        {
-            packet.extract<O2>(hdr.u[0].short);
-        }
-        {
-            packet.extract<O1>(hdr.u[1].byte);
-        }
+        packet.extract<O2>(hdr.u[0].short);
+        packet.extract<O1>(hdr.u[1].byte);
         transition accept;
     }
     state parseO2O2 {
-        {
-            packet.extract<O2>(hdr.u[0].short);
-        }
-        {
-            packet.extract<O2>(hdr.u[1].short);
-        }
+        packet.extract<O2>(hdr.u[0].short);
+        packet.extract<O2>(hdr.u[1].short);
         transition accept;
     }
 }
@@ -84,9 +66,7 @@ control ingress(inout headers hdr, inout
         const default_action = NoAction();
     }
     apply {
-        {
-            debug_hdr_0.apply();
-        }
+        debug_hdr_0.apply();
         if (hdr.u[0].short.isValid()) {
             hdr.u[0].short.data = 16w0xffff;
         }
@@ -107,9 +87,7 @@ control egress(inout headers hdr, inout
 }
 control DeparserImpl(packet_out packet, in headers hdr) {
     apply {
-        {
-            packet.emit<headers>(hdr);
-        }
+        packet.emit<headers>(hdr);
     }
 }
 control verifyChecksum(inout headers hdr, inout metadata meta) {
