--- before_pass
+++ after_pass
@@ -4,13 +4,13 @@ extern E {
     E(bit<32> x=0);
     void f(in bit<16> z=2);
 }
-control c()(bit<32> binit=4) {
+control ctrl();
+package top(ctrl _c);
+control c_0() {
     @name("e") E(x = 32w0) e_0;
     apply {
-        f(a = 32w0, b = binit);
+        f(a = 32w0, b = 32w4);
         e_0.f(z = 16w2);
     }
 }
-control ctrl();
-package top(ctrl _c);
-top(c(binit = 32w4)) main;
+top(c_0()) main;
