--- before_pass
+++ after_pass
@@ -14,8 +14,8 @@ control IngressImpl(inout headers hdr, i
     @name(".NoAction") action NoAction_0() {
     }
     bit<1> registerData_0;
-    @name("testRegister") register<bit<1>>(32w1) testRegister_0;
-    @name("debug_table") table debug_table_0 {
+    @name("IngressImpl.testRegister") register<bit<1>>(32w1) testRegister_0;
+    @name("IngressImpl.debug_table") table debug_table_0 {
         key = {
             meta.test: exact @name("meta.test") ;
         }
