--- before_pass
+++ after_pass
@@ -23,15 +23,9 @@ control IngressImpl(inout headers hdr, i
         default_action = NoAction();
     }
     apply {
-        {
-            testRegister_0.read(registerData_0, 32w0);
-        }
-        {
-            meta.test = (bool)registerData_0;
-        }
-        {
-            debug_table_0.apply();
-        }
+        testRegister_0.read(registerData_0, 32w0);
+        meta.test = (bool)registerData_0;
+        debug_table_0.apply();
     }
 }
 control VerifyChecksumImpl(inout headers hdr, inout metadata meta) {
