--- before_pass
+++ after_pass
@@ -38,22 +38,22 @@ control egress(inout headers hdr, inout
     }
 }
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    bool tmp;
-    @name("setDest") action setDest_0() {
+    bool tmp_0;
+    @name("setDest") action setDest() {
         hdr.ethernet.dstAddr = 48w0x6af3400426d3;
     }
-    @name("someTable") table someTable_0 {
+    @name("someTable") table someTable {
         key = {
             hdr.ethernet.srcAddr: exact @name("hdr.ethernet.srcAddr") ;
         }
         actions = {
-            setDest_0();
+            setDest();
             @defaultonly NoAction();
         }
         default_action = NoAction();
     }
     apply {
-        tmp = someTable_0.apply().hit;
+        tmp_0 = someTable.apply().hit;
     }
 }
 control DeparserImpl(packet_out packet, in headers hdr) {
