--- before_pass
+++ after_pass
@@ -38,8 +38,10 @@ control egress(inout headers hdr, inout
     }
 }
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
+    @name(".NoAction") action NoAction_1() {
+    }
     bool tmp_0;
-    @name("setDest") action setDest() {
+    @name("setDest") action setDest_0() {
         hdr.ethernet.dstAddr = 48w0x6af3400426d3;
     }
     @name("someTable") table someTable {
@@ -47,10 +49,10 @@ control ingress(inout headers hdr, inout
             hdr.ethernet.srcAddr: exact @name("hdr.ethernet.srcAddr") ;
         }
         actions = {
-            setDest();
-            @defaultonly NoAction();
+            setDest_0();
+            @defaultonly NoAction_1();
         }
-        default_action = NoAction();
+        default_action = NoAction_1();
     }
     apply {
         tmp_0 = someTable.apply().hit;
