--- before_pass
+++ after_pass
@@ -38,21 +38,21 @@ control egress(inout headers hdr, inout
     }
 }
 control ingress(inout headers hdr, inout metadata meta, inout standard_metadata_t standard_metadata) {
-    action setDest() {
+    @name("setDest") action setDest_0() {
         hdr.ethernet.dstAddr = 48w0x6af3400426d3;
     }
-    table someTable {
+    @name("someTable") table someTable_0 {
         key = {
             hdr.ethernet.srcAddr: exact @name("hdr.ethernet.srcAddr") ;
         }
         actions = {
-            setDest();
+            setDest_0();
             @defaultonly NoAction();
         }
         default_action = NoAction();
     }
     apply {
-        bool didHit = someTable.apply().hit;
+        bool didHit_0 = someTable_0.apply().hit;
     }
 }
 control DeparserImpl(packet_out packet, in headers hdr) {
