--- before_pass
+++ after_pass
@@ -32,20 +32,20 @@ control deparser(packet_out b, in Header
     }
 }
 control ingress(inout Headers h, inout Meta m, inout standard_metadata_t sm) {
-    @name("c") direct_counter(CounterType.packets) c;
-    @name("c1") direct_counter(CounterType.packets) c1;
-    @name("my_action") action my_action_0(bit<9> a) {
+    @name("c") direct_counter(CounterType.packets) c_0;
+    @name("c1") direct_counter(CounterType.packets) c1_0;
+    @name("my_action") action my_action(bit<9> a) {
         sm.egress_spec = a;
     }
-    @name("t") table t {
+    @name("t") table t_0 {
         actions = {
-            my_action_0();
+            my_action();
         }
-        const default_action = my_action_0(9w0);
-        counters = c;
+        const default_action = my_action(9w0);
+        counters = c_0;
     }
     apply {
-        t.apply();
+        t_0.apply();
     }
 }
 V1Switch<Headers, Meta>(p(), vrfy(), ingress(), egress(), update(), deparser()) main;
