--- before_pass
+++ after_pass
@@ -1,15 +1,15 @@
 control c(inout bit<32> x) {
-    @name("a") action a_0(inout bit<32> b_0, bit<32> d) {
+    @name("a") action a(inout bit<32> b_0, bit<32> d) {
         b_0 = d;
     }
-    @name("t") table t_0 {
+    @name("t") table t {
         actions = {
-            a_0(x);
+            a(x);
         }
-        default_action = a_0(x, 32w0);
+        default_action = a(x, 32w0);
     }
     apply {
-        t_0.apply();
+        t.apply();
     }
 }
 control proto(inout bit<32> x);
